--- kdebase-3.1.1/kcontrol/screensaver/scrnsave.cpp~	Fri Jan  3 05:41:58 2003
+++ kdebase-3.1.1/kcontrol/screensaver/scrnsave.cpp	Tue Apr  1 00:48:24 2003
@@ -147,23 +147,9 @@
     mMonitor = 0;
     mTesting = false;
 
-    // Add non-KDE path
     KGlobal::dirs()->addResourceType("scrsav",
                                      KGlobal::dirs()->kde_default("apps") +
-                                     "apps/ScreenSavers/");
-
-    // Add KDE specific screensaver path
-    QString relPath="System/ScreenSavers/";
-    KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers" );
-    if (servGroup)
-    {
-      relPath=servGroup->relPath();
-      kdDebug() << "relPath=" << relPath << endl;
-    }
-
-    KGlobal::dirs()->addResourceType("scrsav",
-                                     KGlobal::dirs()->kde_default("apps") +
-                                     relPath);
+                                     "screensavers/");
 
     readSettings();
 
--- kdebase-3.1.1/kscreensaver/Makefile.am~	Sun Oct 27 00:07:04 2002
+++ kdebase-3.1.1/kscreensaver/Makefile.am	Tue Apr  1 00:28:59 2003
@@ -23,6 +23,6 @@
 ########## Meta objects ##########
 
 desktop_DATA = KRandom.desktop KBlankscreen.desktop
-desktopdir = $(kde_appsdir)/System/ScreenSavers
+desktopdir = $(kde_datadir)/screensavers
 
 EXTRA_DIST = $(desktop_DATA)
--- kdebase-3.1.1/kscreensaver/Makefile.in~	Tue Mar 11 06:48:15 2003
+++ kdebase-3.1.1/kscreensaver/Makefile.in	Tue Apr  1 00:30:03 2003
@@ -307,7 +307,7 @@
 
 ########## Meta objects ##########
 desktop_DATA = KRandom.desktop KBlankscreen.desktop
-desktopdir = $(kde_appsdir)/System/ScreenSavers
+desktopdir = $(kde_datadir)/screensavers
 
 EXTRA_DIST = $(desktop_DATA)
 subdir = kscreensaver
