--- kdebase/kcontrol/kio/useragentdlg.cpp    2001/02/17 20:06:10     1.24
+++ kdebase/kcontrol/kio/useragentdlg.cpp    2001/02/23 07:00:18     1.26
@@ -246,7 +246,7 @@ void UserAgentOptions::textChanged( cons
   else
     resetPB->setEnabled( false );
 
-  if ( !server.isEmpty() && server[0].isLetterOrNumber() )
+  if ( !server.isEmpty() && !server[0].isLetterOrNumber() && server[0] != '.' )
     return;
 
   if( !login.isEmpty() && !server.isEmpty() )
