diff -urN kdebase-2.2beta1.orig/nsplugins/control/nsconfig.cpp kdebase-2.2beta1/nsplugins/control/nsconfig.cpp
--- kdebase-2.2beta1.orig/nsplugins/control/nsconfig.cpp	Mon May  7 11:11:53 2001
+++ kdebase-2.2beta1/nsplugins/control/nsconfig.cpp	Thu Jul 12 13:38:30 2001
@@ -251,6 +251,8 @@
         paths = config->readListEntry( "scanPaths" );
     else {
         paths.append("$HOME/.netscape/plugins");
+        paths.append("/usr/X11R6/lib/mozilla/plugins");
+        paths.append("/usr/X11R6/lib/netscape/plugins");
         paths.append("/usr/local/netscape/plugins");
         paths.append("/opt/netscape/plugins");
         paths.append("/opt/netscape/communicator/plugins");
diff -urN kdebase-2.2beta1.orig/nsplugins/pluginscan.cpp kdebase-2.2beta1/nsplugins/pluginscan.cpp
--- kdebase-2.2beta1.orig/nsplugins/pluginscan.cpp	Thu Apr 26 17:24:22 2001
+++ kdebase-2.2beta1/nsplugins/pluginscan.cpp	Thu Jul 12 13:38:08 2001
@@ -303,6 +303,8 @@
     if ( !config->hasKey("scanPaths") ) {
         QStringList paths;
         paths.append("$HOME/.netscape/plugins");
+        paths.append("/usr/X11R6/lib/mozilla/plugins");
+        paths.append("/usr/X11R6/lib/netscape/plugins");
         paths.append("/usr/local/netscape/plugins");
         paths.append("/opt/netscape/plugins");
         paths.append("/opt/netscape/communicator/plugins");
