--- kdebase-trinity-3.5.13.2/tdeioslave/settings/tdeio_settings.cc~	2014-05-27 14:47:44.000000000 +0300
+++ kdebase-trinity-3.5.13.2/tdeioslave/settings/tdeio_settings.cc	2014-05-27 14:48:11.446790682 +0300
@@ -122,7 +122,7 @@
 KServiceGroup::Ptr SettingsProtocol::findGroup(const TQString &relPath)
 {
 	TQString nextPart;
-	TQString alreadyFound("Settings/");
+	TQString alreadyFound("KDE-Settings/");
 	TQStringList rest = TQStringList::split('/', relPath);
 
 	kdDebug() << "Trying harder to find group " << relPath << endl;
@@ -183,7 +183,7 @@
 	servicePath.remove(0, 1); // remove starting '/'
 
 	if ( m_runMode == SettingsMode)
-		servicePath = "Settings/" + servicePath;
+		servicePath = "KDE-Settings/" + servicePath;
 
 	KServiceGroup::Ptr grp = KServiceGroup::group(servicePath);
 
@@ -212,7 +212,7 @@
 	groupPath.remove(0, 1); // remove starting '/'
 
 	if ( m_runMode == SettingsMode)
-		groupPath.prepend("Settings/");
+		groupPath.prepend("KDE-Settings/");
 
 	KServiceGroup::Ptr grp = KServiceGroup::group(groupPath);
 
@@ -249,7 +249,7 @@
 			QString relPath = g->relPath();
 
 			// Do not display the "Settings" menu group in Programs Mode.
-			if( (m_runMode == ProgramsMode) && relPath.startsWith( "Settings" ) )
+			if( (m_runMode == ProgramsMode) && relPath.startsWith( "KDE-Settings" ) )
 			{
 				kdDebug() << "SettingsProtocol: SKIPPING entry programs:/" << relPath << endl;
 				continue;
@@ -258,7 +258,7 @@
 			switch( m_runMode )
 			{
 			  case( SettingsMode ):
-				relPath.remove(0, 9); // length("Settings/") ==9
+				relPath.remove(0, 13); // length("KDE-Settings/") ==13
 				kdDebug() << "SettingsProtocol: adding entry settings:/" << relPath << endl;
 				createDirEntry(entry, groupCaption, "settings:/"+relPath, "inode/directory",g->icon());
 				break;
