--- kdebase-3.0.0/konsole/konsole/konsole.cpp.foo	Mon Apr  8 18:19:22 2002
+++ kdebase-3.0.0/konsole/konsole/konsole.cpp	Mon Apr  8 18:24:40 2002
@@ -540,7 +540,6 @@
    m_options->installEventFilter( this );
 
    //help menu
-   m_help->setAccel(QKeySequence(),m_help->idAt(0));
    m_help->insertSeparator(1);
    m_help->insertItem(i18n("&Tip of the Day..."), this, SLOT(showTip()), 0, -1, 2);
 
