--- kdebase-3.1.3/kcontrol/konsole/sessioneditor.cpp.orig	Wed Jul 30 21:28:05 2003
+++ kdebase-3.1.3/kcontrol/konsole/sessioneditor.cpp	Wed Jul 30 21:28:55 2003
@@ -216,7 +216,7 @@
         i = co->readUnsignedNumEntry("Font",(unsigned int)-1);
         fontCombo->setCurrentItem(i+1);
 
-        str = co->readPathEntry("Term","xterm");
+        str = co->readPathEntry("Term","xterm-color");
         termLine->setText(str);
 
         str = co->readEntry("KeyTab");
--- kdebase-3.1.3/konsole/konsole.desktop.orig	Wed Jul 30 21:29:55 2003
+++ kdebase-3.1.3/konsole/konsole.desktop	Wed Jul 30 21:30:16 2003
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Type=Application
-Exec=konsole
+Exec=konsole --ls
 Icon=konsole
 DocPath=konsole/index.html
 Terminal=0
--- kdebase-3.1.3/konsole/konsole/konsole.cpp.orig	Wed Jul 30 21:30:27 2003
+++ kdebase-3.1.3/konsole/konsole/konsole.cpp	Wed Jul 30 21:31:45 2003
@@ -1113,7 +1113,7 @@
    {
       n_defaultKeytab=KeyTrans::find(config->readEntry("keytab","default"))->numb(); // act. the keytab for this session
       b_fullscreen = config->readBoolEntry("Fullscreen",false);
-      n_defaultFont = n_font = QMIN(config->readUnsignedNumEntry("font",3),TOPFONT);
+      n_defaultFont = n_font = QMIN(config->readUnsignedNumEntry("font",6),TOPFONT);
       n_scroll   = QMIN(config->readUnsignedNumEntry("scrollbar",TEWidget::SCRRIGHT),2);
       n_bell = QMIN(config->readUnsignedNumEntry("bellmode",TEWidget::BELLSYSTEM),2);
 
@@ -1930,7 +1930,7 @@
 
 QString Konsole::newSession(KSimpleConfig *co, QString program, const QStrList &args, const QString &_term,const QString &_icon,const QString &_title, const QString &_cwd)
 {
-  QString emu = "xterm";
+  QString emu = "xterm-color";
   QString icon = "openterm";
   QString key;
   QString sch = s_kconfigSchema;
--- kdebase-3.1.3/konsole/konsole/konsole_part.cpp.orig	Wed Jul 30 21:32:41 2003
+++ kdebase-3.1.3/konsole/konsole/konsole_part.cpp	Wed Jul 30 21:33:26 2003
@@ -178,7 +178,7 @@
   te->setMinimumSize(150,70);    // allow resizing, cause resize in TEWidget
 
   setWidget(te);
-  se = new TESession(te,shell,eargs,"xterm");
+  se = new TESession(te,shell,eargs,"xterm-color");
   connect( se,SIGNAL(done(TESession*)),
            this,SLOT(doneSession(TESession*)) );
   connect( se,SIGNAL(openURLRequest(const QString &)),
@@ -524,7 +524,7 @@
   b_framevis = config->readBoolEntry("has frame",false);
   b_histEnabled = config->readBoolEntry("historyenabled",true);
   n_bell = QMIN(config->readUnsignedNumEntry("bellmode",TEWidget::BELLSYSTEM),2);
-  n_font = QMIN(config->readUnsignedNumEntry("font",3),TOPFONT);
+  n_font = QMIN(config->readUnsignedNumEntry("font",6),TOPFONT);
   n_keytab=config->readNumEntry("keytab",0); // act. the keytab for this session
   n_scroll = QMIN(config->readUnsignedNumEntry("scrollbar",TEWidget::SCRRIGHT),2);
   m_histSize = config->readNumEntry("history",DEFAULT_HISTORY_SIZE);
--- kdebase-3.1.3/konsole/konsole/main.cpp.orig	Wed Jul 30 21:31:51 2003
+++ kdebase-3.1.3/konsole/konsole/main.cpp	Wed Jul 30 21:33:51 2003
@@ -40,7 +40,7 @@
    { "name <name>",     I18N_NOOP("Set window class"), 0 },
    { "ls",              I18N_NOOP("Start login shell"), 0 },
    { "T <title>",       I18N_NOOP("Set the window title"), 0 },
-   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm" },
+   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm-color" },
    { "xwin",            I18N_NOOP("ignored"), 0 },
    { "noclose",         I18N_NOOP("Do not close Konsole when command exits"), 0 },
    { "nohist",          I18N_NOOP("Do not save lines in history"), 0 },
