--- kdebase-030317/kcontrol/kdm/background.cpp~	Sun Mar 16 00:35:44 2003
+++ kdebase-030317/kcontrol/kdm/background.cpp	Tue Mar 18 19:31:12 2003
@@ -53,7 +53,7 @@
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
     config->setGroup( "X-*-Greeter" );
-    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg",KDE_CONFDIR "/kdm/backgroundrc" ) );
+    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg","/etc/X11/kdm/backgroundrc" ) );
     m_background = new BGDialog( this, m_simpleConf, false );
 
     connect(m_background, SIGNAL(changed(bool)), SIGNAL(changed(bool)));
@@ -105,7 +105,7 @@
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
-    m_background->defaults();
+    m_background->kdmdefaults();
     emit changed(true);
 }
 
-void KBackground::defaults()
+void KBackground::kdmdefaults()
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
--- kdebase-030317/kcontrol/kdm/kdm-appear.cpp~	Tue Mar 18 11:15:18 2003
+++ kdebase-030317/kcontrol/kdm/kdm-appear.cpp	Tue Mar 18 11:15:43 2003
@@ -194,6 +194,7 @@
   QStringList list = QStyleFactory::keys();
   list = KDMAppearanceWidget::sortStringList(list);
   guicombo->insertStringList(list);
+  guicombo->insertItem("KDE");
 
   label = new QLabel(guicombo, i18n("GUI s&tyle:"), hlp);
   connect(guicombo, SIGNAL(activated(int)), SLOT(changed()));
@@ -322,7 +323,7 @@
 bool KDMAppearanceWidget::setLogo(QString logo)
 {
     QString flogo = logo.isEmpty() ?
-                    locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") ) :
+                    QString::fromLatin1("/etc/X11/kdm/pics/pldlogo.png") :
                     logo;
     QPixmap p(flogo);
     if (p.isNull())
@@ -339,7 +340,7 @@
 {
     KImageIO::registerFormats();
     QString fileName = KFileDialog::getOpenFileName(
-	locate("data", QString::fromLatin1("kdm/pics/") ),
+	QString::fromLatin1("/etc/X11/kdm/pics/"),
 	KImageIO::pattern());
     if (!fileName.isEmpty())
 	if (setLogo(fileName))
@@ -396,9 +397,7 @@
 
 	// we gotta check if it is a non-local file and make a tmp copy at the hd.
 	if(!url->isLocalFile()) {
-	    pixurl = "file:" +
-		     KGlobal::dirs()->resourceDirs("data").last() +
-		     "kdm/pics/" + url->fileName();
+	    pixurl = "file:/etc/X11/kdm/pics/" + url->fileName();
 	    KIO::NetAccess::copy(*url, pixurl);
 	    istmp = true;
 	} else {
@@ -453,10 +452,10 @@
   config->setGroup("X-*-Greeter");
 
   // Read the greeting string
-  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to %s at %n")));
+  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to PLD Linux at %n")));
 
   // Regular logo or clock
-  QString logoArea = config->readEntry("LogoArea", "Logo" );
+  QString logoArea = config->readEntry("LogoArea", "None" );
   if (logoArea == "Clock") {
     clockRadio->setChecked(true);
     slotAreaRadioClicked(KdmClock);
@@ -472,7 +471,7 @@
   setLogo( config->readEntry("LogoPixmap", ""));
 
   // Check the GUI type
-  guicombo->setCurrentItem (config->readEntry("GUIStyle", "Default"), true, 0);
+  guicombo->setCurrentItem (config->readEntry("GUIStyle", "KDE"), true, 0);
 
   // Check the Color Scheme
   colcombo->setCurrentItem (config->readEntry("ColorScheme", "Default"), true, 0);
@@ -503,13 +502,13 @@
 
 void KDMAppearanceWidget::defaults()
 {
-  greetstr_lined->setText( i18n("Welcome to %s at %n") );
-  logoRadio->setChecked( true );
-  slotAreaRadioClicked( KdmLogo );
+  greetstr_lined->setText( i18n("Welcome to PLD Linux at %n") );
+  noneRadio->setChecked( true );
+  slotAreaRadioClicked( KdmNone );
   posCenterRadio->setChecked( true );
   slotPosRadioClicked( 0 );
   setLogo( "" );
-  guicombo->setCurrentItem (QString::fromLatin1("Default"), true, 0);
+  guicombo->setCurrentItem (QString::fromLatin1("KDE"), true, 0);
 
   colcombo->setCurrentItem( QString::fromLatin1("Default"), true, 0 );
   echocombo->setCurrentItem( 1 );
--- kdebase-3.1.1/kcontrol/kdm/kdm-font.cpp~	Sun Sep 15 23:08:39 2002
+++ kdebase-3.1.1/kcontrol/kdm/kdm-font.cpp	Tue Mar  4 18:51:26 2003
@@ -112,8 +112,8 @@
 
 void KDMFontWidget::set_def()
 {
-  stdfont = QFont("helvetica", 12);
-  failfont = QFont("helvetica", 12, QFont::Bold);
+  stdfont = QFont("helvetica", 11);
+  failfont = QFont("helvetica", 11, QFont::Bold);
   greetfont = QFont("charter", 24);
 }
 
--- kdebase-3.1/kcontrol/kdm/kdm-sess.cpp.orig	Sun Sep 15 23:08:39 2002
+++ kdebase-3.1/kcontrol/kdm/kdm-sess.cpp	Thu Nov 21 23:18:34 2002
@@ -400,11 +400,6 @@
   sdlcombo->setCurrentItem(SdAll);
   sdrcombo->setCurrentItem(SdRoot);
 
-  sessionslb->clear();
-  sessionslb->insertItem("default");
-  sessionslb->insertItem("kde");
-  sessionslb->insertItem("failsafe");
-
 #if defined(__linux__) && defined(__i386__)
     lilo_check->setChecked(false);
     slotLiloCheckToggled(false);
--- kdebase-3.1/kcontrol/kdm/kdm-users.cpp.fixed	Fri Nov 29 03:55:12 2002
+++ kdebase-3.1/kcontrol/kdm/kdm-users.cpp	Fri Nov 29 18:53:19 2002
@@ -66,7 +66,7 @@
 
     // We assume that $kde_datadir/kdm exists, but better check for pics/ and pics/users,
     // and create them if necessary.
-    m_userPixDir = KGlobal::dirs()->resourceDirs("data").last() + "kdm/pics/";
+    m_userPixDir = "/etc/X11/kdm/pics/";
     m_notFirst = false;
     QDir testDir( m_userPixDir );
     if ( !testDir.exists() && !testDir.mkdir( testDir.absPath() )) {
--- kdebase-3.1/kcontrol/kdm/main.cpp.orig	Sun Oct 27 00:05:07 2002
+++ kdebase-3.1/kcontrol/kdm/main.cpp	Thu Nov 21 02:51:19 2002
@@ -92,7 +92,7 @@
     usermap.insert( QString::fromLocal8Bit( ps->pw_name ), ps->pw_uid );
   endpwent();
 
-  config = new KSimpleConfig( QString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));
+  config = new KSimpleConfig( QString::fromLatin1( "/etc/X11/kdm/kdmrc" ));
 
   QVBoxLayout *top = new QVBoxLayout(this);
   tab = new QTabWidget(this);
--- kdebase-3.1/kdm/kfrontend/Makefile.am.orig	Sun Oct 27 00:05:49 2002
+++ kdebase-3.1/kdm/kfrontend/Makefile.am	Wed Nov 20 23:26:39 2002
@@ -30,8 +30,8 @@
 genkdmconf_LDADD = $(LIB_X11)
 
 install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
+	@$(mkinstalldirs) $(DESTDIR)/etc/X11/kdm
+	@./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase-3.1/kdm/kfrontend/Makefile.in.orig	Mon Nov 18 22:22:07 2002
+++ kdebase-3.1/kdm/kfrontend/Makefile.in	Wed Nov 20 23:47:04 2002
@@ -786,8 +786,8 @@
 
 
 install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
+	@$(mkinstalldirs) $(DESTDIR)/etc/X11/kdm
+	@./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase-3.1.1/kdm/kfrontend/genkdmconf.c~	Sun Mar  2 19:35:05 2003
+++ kdebase-3.1.1/kdm/kfrontend/genkdmconf.c	Tue Mar  4 15:53:10 2003
@@ -56,7 +56,7 @@
 
 #define as(ar) ((int)(sizeof(ar)/sizeof(ar[0])))
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF  "/etc/X11/kdm"
 
 #define RCVERMAJOR 2
 #define RCVERMINOR 0
@@ -757,7 +757,7 @@
 "Program=\n"
 "ReverseBlending=false\n"
 "UseSHM=false\n"
-"Wallpaper=default_blue.jpg\n"
+"Wallpaper=/etc/X11/kdm/pics/pldwallpaper.png\n"
 "WallpaperList=\n"
 "WallpaperMode=Scaled\n";
 
@@ -1831,18 +1831,18 @@
 "# this opens a security hole: a secured display lock can be circumvented\n"
 "# (unless you use KDE's built-in screen lock). Default is false\n" },
 { "AllowRootLogin",	0, 0, 
-"# Allow root logins? Default is true\n" },
+"# Allow root logins? Default is false\n" },
 { "AllowNullPasswd",	0, 0, 
-"# Allow to log in, when user has set an empty password? Default is true\n" },
+"# Allow to log in, when user has set an empty password? Default is false\n" },
 { "AllowShutdown",	0, 0, 
 "# Who is allowed to shut down the system. This applies both to the\n"
 "# greeter and to the command FiFo. Default is All\n"
 "# \"None\" - no \"Shutdown...\" button is shown at all\n"
-"# \"Root\" - the root password must be entered to shut down\n"
-"# \"All\" - everybody can shut down the machine (Default)\n" },
+"# \"Root\" - the root password must be entered to shut down (Default)\n"
+"# \"All\" - everybody can shut down the machine\n" },
 { "AllowSdForceNow",	0, 0, 
 "# Who is allowed to abort all still running sessions when shutting down.\n"
-"# Same options as for AllowShutdown. Default is All\n" },
+"# Same options as for AllowShutdown. Default is Root\n" },
 { "DefaultSdMode",	0, 0, 
 "# The default choice for the shutdown condition/timing.\n"
 "# \"Schedule\" - shutdown after all sessions exit (possibly at once) (Default)\n"
@@ -1857,7 +1857,7 @@
 "# option!), i.e., KDM will behave exactly as before KDE 3.0.\n" }, 
 { "SessSaveFile",	0, 0, 
 "# Where (relatively to the user's home directory) to store the last\n"
-"# selected session. Default is .wmrc\n" },
+"# selected session. Default is .kdmrc\n" },
 { "ServerAttempts",	0, 0, 
 "# How often to try to run the X-server. Running includes executing it and\n"
 "# waiting for it to come up. Default is 1\n" },
@@ -1894,17 +1894,17 @@
 { "GUIStyle",		0, upd_guistyle, 
 "# Widget Style of the greeter:\n"
 "# Default, Windows, Platinum, Motif, MotifPlus, CDE, SGI, and any styles you\n"
-"# may have installed, e.g., HighColor; Default is \"Default\"\n" },
+"# may have installed, e.g., HighColor; Default is \"KDE\"\n" },
 { "ColorScheme",		0, 0, 
 "# Widget Colors of the greeter:\n"
 "# Default, and any color schemes you may have installed. Default is \"Default\"\n" },
 { "LogoArea",		0, upd_logoarea, 
 "# What should be shown righthand of the input lines:\n"
-"# \"Logo\" - the image specified by LogoPixmap (Default)\n"
+"# \"Logo\" - the image specified by LogoPixmap \n"
 "# \"Clock\" - a neat analog clock\n"
-"# \"None\" - nothing\n" },
+"# \"None\" - nothing (Default)\n" },
 { "LogoPixmap",		0, 0, 
-"# The image to show when LogoArea=Logo. Default is kdelogo.png\n" },
+"# The image to show when LogoArea=Logo. Default is pldlogo.png\n" },
 { "GreeterPosFixed",	0, 0, 
 "# Normally, the greeter is centered on the screen. Use this, if you want\n"
 "# it to appear elsewhere on the screen. Default is false\n" },
@@ -1925,13 +1925,13 @@
 "# - %r -> the operating system's version\n"
 "# - %m -> the machine (hardware) type\n"
 "# - %% -> a single %\n"
-"# Default is \"Welcome to %s at %n\"\n" },
+"# Default is \"Welcome to PLD Linux at %n\"\n" },
 { "GreetFont",		0, 0, 
 "# The font for the headline. Default is charter,24,bold\n" },
 { "StdFont",		0, 0, 
-"# The normal font used in the greeter. Default is helvetica,12\n" },
+"# The normal font used in the greeter. Default is helvetica,11\n" },
 { "FailFont",		0, 0, 
-"# The font used for the \"Login Failed\" message. Default is helvetica,12,bold\n" },
+"# The font used for the \"Login Failed\" message. Default is helvetica,11,bold\n" },
 { "AntiAliasing",	0, 0,
 "# Whether the fonts shown in the greeter should be antialiased. Default is false\n" },
 { "NumLock",		0, 0,
@@ -2121,10 +2121,10 @@
 { "UserAuthDir",	"", 0 },
 { "Chooser",		"", 0 },	/* XXX kill */
 { "AutoReLogin",	"true", 0 },
-{ "AllowRootLogin",	"false", 1 },
-{ "AllowNullPasswd",	"false", 1 },
-{ "AllowShutdown",	"Root", 1 },
-{ "AllowSdForceNow",	"Root", 0 },
+{ "AllowRootLogin",	"true", 0 },
+{ "AllowNullPasswd",	"true", 0 },
+{ "AllowShutdown",	"All", 0 },
+{ "AllowSdForceNow",	"All", 0 },
 { "DefaultSdMode",	"ForceNow", 0}, 
 { "InteractiveSd",	"false", 0}, 
 { "SessSaveFile",	"", 0 },	/* XXX .dmrc */
@@ -2134,7 +2134,7 @@
 { "SessionTypes",	"default,kde,failsafe", 1 },
 { "GUIStyle",		"Windows", 0 },
 { "ColorScheme",	"Pumpkin", 0 },
-{ "LogoArea",		"None", 0 },
+{ "LogoArea",		"Logo", 0 },
 { "LogoPixmap",		"", 0 },
 { "GreeterPosFixed",	"true", 0 },
 { "GreeterPosX",	"200", 0 },
@@ -2167,8 +2167,8 @@
 { "ServerAttempts",	"", 0 },
 { "ServerTimeout",	"", 0 },
 { "AllowShutdown",	"All", 1 },
-{ "AllowRootLogin",	"true", 1 },
-{ "AllowNullPasswd",	"true", 1 },
+{ "AllowRootLogin",	"true", 0 },
+{ "AllowNullPasswd",	"true", 0 },
 { "NoPassEnable",	"true", 0 },
 { "NoPassUsers",	"fred,ethel", 0 },
 };
--- kdebase-3.1.1/kdm/kfrontend/kdm_config.c~	Thu Dec  5 21:40:21 2002
+++ kdebase-3.1.1/kdm/kfrontend/kdm_config.c	Tue Mar  4 15:10:48 2003
@@ -38,7 +38,7 @@
 #include <netdb.h>
 #include <grp.h>
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 
 
 /*
@@ -519,11 +519,11 @@
 { "AutoLoginSession",	C_autoString,	(void *)PautoLoginX,	"" },
 { "AutoLogin1st",	C_autoLogin1st | C_BOOL, 0,	"true" },
 { "AutoReLogin",	C_autoReLogin | C_BOOL,	0,	"false" },
-{ "AllowNullPasswd",	C_allowNullPasswd | C_BOOL, 0,	"true" },
-{ "AllowRootLogin",	C_allowRootLogin | C_BOOL, 0,	"true" },
-{ "SessSaveFile",	C_sessSaveFile,		0,	".wmrc" },
-{ "AllowShutdown",	C_allowShutdown | C_ENUM, sd_who, "All" },
-{ "AllowSdForceNow",	C_allowNuke | C_ENUM, sd_who,	"All" },
+{ "AllowNullPasswd",	C_allowNullPasswd | C_BOOL, 0,	"false" },
+{ "AllowRootLogin",	C_allowRootLogin | C_BOOL, 0,	"false" },
+{ "SessSaveFile",	C_sessSaveFile,		0,	".kdmrc" },
+{ "AllowShutdown",	C_allowShutdown | C_ENUM, sd_who, "Root" },
+{ "AllowSdForceNow",	C_allowNuke | C_ENUM, sd_who,	"Root" },
 { "DefaultSdMode",	C_defSdMode | C_ENUM, sd_mode,	"Schedule" },
 { "InteractiveSd",	C_interactiveSd | C_BOOL, 0,	"true" },
 };
@@ -532,15 +532,15 @@
 { "SessionTypes",	C_SessionTypes,		0,	"default,failsafe" },
 { "GUIStyle",		C_GUIStyle, 		0,	"KDE" },
 { "ColorScheme",	C_ColorScheme,		0,	"Default" },
-{ "LogoArea",		C_LogoArea | C_ENUM, logoarea,	"Logo" },
+{ "LogoArea",		C_LogoArea | C_ENUM, logoarea,	"None" },
 { "LogoPixmap",		C_LogoPixmap,		0,	"" },
 { "GreeterPosFixed",	C_GreeterPosFixed | C_BOOL, 0,	"false" },
 { "GreeterPosX",	C_GreeterPosX,		0,	"0" },
 { "GreeterPosY",	C_GreeterPosY,		0,	"0" },
 { "GreeterScreen",	C_GreeterScreen,	0,	"0" },
-{ "StdFont",		C_StdFont,		0,	"helvetica,12,5,0,50,0" },
-{ "FailFont",		C_FailFont,		0,	"helvetica,12,5,0,75,0" },
-{ "GreetString",	C_GreetString,		0,	"Welcome to %s at %n" },
+{ "StdFont",		C_StdFont,		0,	"helvetica,11,5,0,50,0" },
+{ "FailFont",		C_FailFont,		0,	"helvetica,11,5,0,75,0" },
+{ "GreetString",	C_GreetString,		0,	"Welcome to PLD Linux at %n" },
 { "GreetFont",		C_GreetFont,		0,	"charter,24,5,0,50,0" },
 { "AntiAliasing",	C_AntiAliasing | C_BOOL,0,	"false" },
 { "NumLock",		C_NumLock | C_ENUM, numlock,	"Keep" },
--- kdebase-3.1/kdm/kfrontend/kdmconfig.cpp.orig	Mon Oct 28 10:55:24 2002
+++ kdebase-3.1/kdm/kfrontend/kdmconfig.cpp	Fri Nov 29 18:27:33 2002
@@ -153,7 +153,7 @@
 
     _logo = GetCfgQStr (C_LogoPixmap);
     if( _logo.isEmpty())
-	_logo = locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") );
+	_logo = QString::fromLatin1("/etc/X11/kdm/pics/pldlogo.png");
 
     _showUsers = GetCfgInt (C_ShowUsers);
     _users = GetCfgQStrList (C_SelectedUsers);
--- kdebase-030317/kdm/kfrontend/kgreeter.cpp~	Sat Mar 15 00:39:49 2003
+++ kdebase-030317/kdm/kfrontend/kgreeter.cpp	Tue Mar 18 11:02:15 2003
@@ -115,8 +115,7 @@
   , capslocked( -1 )
   , loginfailed( false )
 {
-    user_pic_dir = KGlobal::dirs()->resourceDirs( "data" ).last() +
-				      QString::fromLatin1("kdm/pics/users/");	/* XXX standardize */
+    user_pic_dir = QString::fromLatin1("/etc/X11/kdm/pics/users/");	/* XXX standardize */
 
     QGridLayout* main_grid = new QGridLayout( winFrame, 4, 2, 10 );
     QBoxLayout* hbox1 = new QHBoxLayout( 10 );
@@ -283,7 +282,7 @@
 
     UpdateLock();
 
-    stsfile = new KSimpleConfig( QString::fromLatin1(KDE_CONFDIR "/kdm/kdmsts") ); // XXX get from kdm_config
+    stsfile = new KSimpleConfig( QString::fromLatin1("/etc/X11/kdm/kdmsts") ); // XXX get from kdm_config
     stsfile->setGroup( "PrevUser" );
     enam = QString::fromLocal8Bit( dname );
     if (kdmcfg->_preselUser != PRESEL_PREV)
--- kdebase-3.1/kdm/kfrontend/pics/Makefile.am.orig	Sun Feb 20 17:34:48 2000
+++ kdebase-3.1/kdm/kfrontend/pics/Makefile.am	Fri Nov 29 18:38:41 2002
@@ -1,5 +1,5 @@
 
-picsdir = $(kde_datadir)/kdm/pics
+picsdir = /etc/X11/kdm/pics
 pics_DATA = kdelogo.png kdeflower.png
 
 usersdir = $(picsdir)/users
--- kdebase-3.1/kdm/kfrontend/pics/Makefile.in.orig	Mon Nov 25 04:46:12 2002
+++ kdebase-3.1/kdm/kfrontend/pics/Makefile.in	Fri Nov 29 18:38:16 2002
@@ -286,7 +286,7 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 
-picsdir = $(kde_datadir)/kdm/pics
+picsdir = /etc/X11/kdm/pics
 pics_DATA = kdelogo.png kdeflower.png
 
 usersdir = $(picsdir)/users
