--- kdebase-3.1/kcontrol/background/backgnd.cpp.orig	Mon Nov 18 14:14:20 2002
+++ kdebase-3.1/kcontrol/background/backgnd.cpp	Fri Nov 22 21:29:41 2002
@@ -602,6 +602,32 @@
 }
 
 
+void Backgnd::kdmdefaults()
+{
+    m_pGlobals->setCommonBackground(_defCommon);
+    m_pGlobals->setLimitCache(_defLimitCache);
+    m_pGlobals->setCacheSize(_defCacheSize);
+
+    m_eDesk = _defCommon ? 0 : m_Desk;
+
+    KBackgroundRenderer *r = m_Renderer[m_eDesk];
+
+    if (r->isActive())
+        r->stop();
+
+    r->setBackgroundMode(KBackgroundSettings::Flat);
+    r->setColorA(QColor("#0000C8"));
+    r->setColorB(_defColorB);
+    r->setWallpaperMode(_defWallpaperMode);
+    r->setMultiWallpaperMode(_defMultiMode);
+    r->setBlendMode(_defBlendMode);
+    r->setBlendBalance(_defBlendBalance);
+    r->setReverseBlending(_defReverseBlending);
+    apply();
+    emit changed(true);
+}
+
+
 void Backgnd::slotSelectDesk(int desk)
 {
     if (desk == m_Desk)
--- kdebase-3.1/kcontrol/background/backgnd.h.orig	Sun Oct 27 00:04:51 2002
+++ kdebase-3.1/kcontrol/background/backgnd.h	Fri Nov 22 15:40:37 2002
@@ -74,7 +74,8 @@
     virtual void load();
     virtual void save();
     virtual void defaults();
-
+    virtual void kdmdefaults();
+    
     QString quickHelp() const;
     void makeReadOnly();
 
