--- info.cc	Wed Nov 20 19:16:03 2002
+++ info.cc.orig	Mon Nov 18 14:15:12 2002
@@ -234,7 +234,7 @@
     // "* infopage:" at the start of a line
     QRegExp regex( "^\\*  *[^: ][^:]*:", false );
 
-    QFile f( "/usr/share/info/dir" );
+    QFile f( "/usr/info/dir" );
 
     if ( f.open(IO_ReadOnly) ) {
         QTextStream t( &f );
@@ -275,7 +275,7 @@
         finished();
     }
     else {
-        kdError(7108) << "cannot open file '/usr/share/info/dir'" << endl;
+        kdError(7108) << "cannot open file '/usr/info/dir'" << endl;
     }
     kdDebug( 7108 ) << "InfoProtocol::listDir - done" << endl;
 }*/
