--- kdebase-031021/startkde.orig	2003-10-14 22:48:48.000000000 +0200
+++ kdebase-031021/startkde	2003-10-21 13:58:14.000000000 +0200
@@ -142,18 +142,20 @@
 
 # run KPersonalizer before the session, if this is the first login
 if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
-    # start only dcopserver, don't start whole kdeinit (takes too long)
-    echo 'startkde: Running kpersonalizer...'  1>&2
-    dcopserver
-    kwin --lock &
-    kpersonalizer --before-session
-    # handle kpersonalizer restarts (language change)
-    while test $? -eq 1; do
-        kpersonalizer --r --before-session
-    done
-    dcopserver_shutdown
-    # shutdown will also make kwin quit, give it time to do so
-    sleep 1
+    if test `/usr/bin/which kpersonalizer 2>/dev/null`; then
+        # start only dcopserver, don't start whole kdeinit (takes too long)
+	echo 'startkde: Running kpersonalizer...'  1>&2
+	dcopserver
+	kwin --lock &
+	kpersonalizer --before-session
+	# handle kpersonalizer restarts (language change)
+	while test $? -eq 1; do
+    	    kpersonalizer --r --before-session
+	done
+	dcopserver_shutdown
+	# shutdown will also make kwin quit, give it time to do so
+	sleep 1
+    fi	
 fi
 
 # the splashscreen and progress indicator
