--- kdebase-3.0.3/konqueror/konq_mainwindow.cc--	2002-09-04 10:50:51.000000000 +0200
+++ kdebase-3.0.3/konqueror/konq_mainwindow.cc	2002-09-04 10:51:52.000000000 +0200
@@ -2077,6 +2077,11 @@ void KonqMainWindow::slotUpAboutToShow()
 
 void KonqMainWindow::slotUp()
 {
+  QTimer::singleShot( 0, this, SLOT( slotUpDelayed() ) );
+}
+
+void KonqMainWindow::slotUpDelayed()
+{
   openURL( 0L, m_currentView->upURL() );
 }
 
--- kdebase-3.0.3/konqueror/konq_mainwindow.h--	2002-09-04 10:51:10.000000000 +0200
+++ kdebase-3.0.3/konqueror/konq_mainwindow.h	2002-09-04 10:52:12.000000000 +0200
@@ -277,6 +277,7 @@ public slots:
 
   // Go menu
   void slotUp();
+  void slotUpDelayed();
   void slotBack();
   void slotForward();
   void slotHome();
