--- kdebase-3.0.8/kicker/core/default-apps.orig	Sun Sep 15 23:08:48 2002
+++ kdebase-3.0.8/kicker/core/default-apps	Mon Sep 30 01:43:39 2002
@@ -1,7 +1,7 @@
-System/konsole.desktop
-Settingsmenu/KControl.desktop
-Help.desktop
+Terminals/konsole.desktop
+Settings/KControl.desktop
+Help/Help.desktop
 Home.desktop
-Internet/konqbrowser.desktop
-Internet/KMail.desktop
+Network/WWW//konqbrowser.desktop
+Network/WWW/KMail.desktop
 Office/kword.desktop
--- kdebase-3.0.8/kicker/core/containerarea.cpp.orig	Sun Sep 15 23:08:48 2002
+++ kdebase-3.0.8/kicker/core/containerarea.cpp	Mon Sep 30 01:50:37 2002
@@ -134,12 +134,12 @@
 
         f.close();
     } else {
-        buttons << "System/konsole.desktop";
-        buttons << "KControl.desktop";
+        buttons << "Terminals/konsole.desktop";
+        buttons << "Settings/KControl.desktop";
         buttons << "Help.desktop";
         buttons << "Home.desktop";
-        buttons << "Internet/konqbrowser.desktop";
-        buttons << "Internet/KMail.desktop";
+        buttons << "Network/WWW/konqbrowser.desktop";
+        buttons << "Network/WWW/KMail.desktop";
         buttons << "Office/kword.desktop";
     }
 
--- kdebase-3.0.8/kicker/applets/launcher/quicklauncher.cpp.orig	Thu Jan  3 21:29:36 2002
+++ kdebase-3.0.8/kicker/applets/launcher/quicklauncher.cpp	Mon Sep 30 01:55:12 2002
@@ -66,10 +66,10 @@
    }
    else {
       buttonList.append(new QuickButton(locate("apps", "Home.desktop"), this));
-      buttonList.append(new QuickButton(locate("apps", "System/konsole.desktop"), this));
-      buttonList.append(new QuickButton(locate("apps", "KControl.desktop"), this));
-      buttonList.append(new QuickButton(locate("apps", "Help.desktop"), this));
-      buttonList.append(new QuickButton(locate("apps", "Editors/kwrite.desktop"), this));
+      buttonList.append(new QuickButton(locate("apps", "Terminals/konsole.desktop"), this));
+      buttonList.append(new QuickButton(locate("apps", "Settings/KControl.desktop"), this));
+      buttonList.append(new QuickButton(locate("apps", "Help/Help.desktop"), this));
+      buttonList.append(new QuickButton(locate("apps", "Office/Editors/kwrite.desktop"), this));
       saveConfig();
    }
 
