--- kdebase/konsole/konsole/TEPty.cpp.utmp	Sun Feb  4 19:11:41 2001
+++ kdebase/konsole/konsole/TEPty.cpp	Sun Feb  4 19:13:34 2001
@@ -349,7 +349,7 @@
       if (ptsn) {
           strncpy(ttynam, ptsname(ptyfd), 50);
           grantpt(ptyfd);
-          needGrantPty = FALSE;
+//          needGrantPty = FALSE;
       } else {
       	  perror("ptsname");
 	  close(ptyfd);
@@ -659,6 +659,7 @@
 */
 TEPty::~TEPty()
 {
+  donePty();
 }
 
 int TEPty::setupCommunication(Communication comm)
