--- kdebase-030901/kicker/core/default-apps~	2003-08-31 15:37:34.000000000 +0200
+++ kdebase-030901/kicker/core/default-apps	2003-09-02 10:00:06.000000000 +0200
@@ -1,7 +1,6 @@
 menuext/prefmenu.desktop
-System/konsole.desktop
+Terminals/konsole.desktop
 Home.desktop
-Internet/konqbrowser.desktop
-Internet/KMail.desktop
-Office/kword.desktop
-Help.desktop
+Network/WWW/konqbrowser.desktop
+Network/Mail/KMail.desktop
+Help/Help.desktop
