--- kdebase-050625/konsole/fonts/Makefile.am.orig	2005-06-24 22:46:36.000000000 +0200
+++ kdebase-050625/konsole/fonts/Makefile.am	2005-06-25 07:14:05.973746664 +0200
@@ -1,5 +1,5 @@
-fontdir = $(kde_datadir)/konsole/fonts
-font_DATA =console8x16.pcf.gz 9x15.pcf.gz
+fontdir = /usr/share/fonts/misc
+font_DATA = console8x16.pcf.gz
 
 EXTRA_DIST = $(font_DATA)
 
@@ -10,7 +10,3 @@
 console8x16.pcf.gz: $(srcdir)/console8x16.bdf
 	bdftopcf -o console8x16.pcf $(srcdir)/console8x16.bdf
 	gzip --best console8x16.pcf
-
-9x15.pcf.gz: $(srcdir)/9x15.bdf
-	bdftopcf -o 9x15.pcf $(srcdir)/9x15.bdf
-	gzip --best 9x15.pcf
