--- kdebase/kioslave/settings/kio_settings.cc.orig	2004-02-19 12:52:34.000000000 +0100
+++ kdebase/kioslave/settings/kio_settings.cc	2004-04-02 20:01:53.496036792 +0200
@@ -89,8 +89,8 @@
 	QString nextPart="";;
 	QStringList rest;
 	kdDebug()<<"Trying harder to find group "<<relPath<<endl;
-	if (relPath.startsWith("Settings/")) {
-		alreadyFound="Settings/";
+	if (relPath.startsWith("KDE-Settings/")) {
+		alreadyFound="KDE-Settings/";
 		rest=QStringList::split("/",relPath.right(relPath.length()-9));
 		kdDebug()<<"Supported root Settings detected"<<endl;
 		for (int i=0;i<rest.count();i++)
@@ -148,7 +148,7 @@
 	switch( m_runMode )
 	{
 		case( SettingsMode ):
-			if (!relPath.startsWith("/Settings")) relPath="Settings"+relPath;
+			if (!relPath.startsWith("/KDE-Settings")) relPath="KDE-Settings"+relPath;
 			else relPath=relPath.right(relPath.length()-1);
 			break;
 
@@ -172,7 +172,7 @@
 	switch( m_runMode )
 	{
 		case( SettingsMode ):
-			createDirEntry(entry, i18n("Settings"), url.url(), "inode/directory",grp->icon());
+			createDirEntry(entry, i18n("KDE-Settings"), url.url(), "inode/directory",grp->icon());
 			break;
 
 		case( ProgramsMode ):
@@ -199,7 +199,7 @@
 	switch( m_runMode )
 	{
 		case( SettingsMode ):
-			if (!relPath.startsWith("/Settings")) relPath="Settings"+relPath;
+			if (!relPath.startsWith("/KDE-Settings")) relPath="KDE-Settings"+relPath;
 			else relPath=relPath.right(relPath.length()-1);
 			break;
 
@@ -250,7 +250,7 @@
                     QString relPath=g->relPath();
 
 		    // Do not display the "Settings" menu group in Programs Mode.
-		    if( (m_runMode == ProgramsMode) && relPath.startsWith( "Settings" ) )
+		    if( (m_runMode == ProgramsMode) && relPath.startsWith( "KDE-Settings" ) )
 		    {
 			kdDebug() << "SettingsProtocol: SKIPPING entry programs:/" << relPath << endl;
 			continue;
@@ -259,7 +259,7 @@
 		    switch( m_runMode )
 		    {
 			case( SettingsMode ):
-			    relPath=relPath.right(relPath.length()-9); //Settings/ ==9
+			    relPath=relPath.right(relPath.length()-13); //KDE-Settings/ ==13
 			    kdDebug() << "SettingsProtocol: adding entry settings:/" << relPath << endl;
 			    createDirEntry(entry, groupCaption, "settings:/"+relPath, "inode/directory",g->icon());
 			    break;
