--- kdebase-3.0.8/konsole/konsole/main.cpp.orig	Sun Sep 15 23:08:53 2002
+++ kdebase-3.0.8/konsole/konsole/main.cpp	Tue Oct 15 02:26:19 2002
@@ -40,7 +40,7 @@
    { "name <name>",     I18N_NOOP("Set window class"), 0 },
    { "ls",              I18N_NOOP("Start login shell"), 0 },
    { "T <title>",       I18N_NOOP("Set the window title"), 0 },
-   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm" },
+   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm-color" },
    { "xwin",            I18N_NOOP("ignored"), 0 },
    { "noclose",         I18N_NOOP("Do not close Konsole when command exits"), 0 },
    { "nohist",          I18N_NOOP("Do not save lines in history"), 0 },
--- kdebase-3.0.8/konsole/konsole/TEPty.cpp.orig	Sun Sep 15 23:08:53 2002
+++ kdebase-3.0.8/konsole/konsole/TEPty.cpp	Tue Oct 15 02:36:30 2002
@@ -70,6 +70,7 @@
 #endif
 
 #if defined(HAVE_GRANTPT) && defined(HAVE_PTSNAME) && defined(HAVE_UNLOCKPT) && !defined(_XOPEN_SOURCE) && !defined(__svr4__)
+#define __USE_XOPEN
 #define _XOPEN_SOURCE // make stdlib.h offer the above fcts
 #endif
 
--- kdebase-3.0.8/konsole/konsole/TEWidget.cpp.orig	Sun Sep 15 23:08:53 2002
+++ kdebase-3.0.8/konsole/konsole/TEWidget.cpp	Tue Oct 15 03:00:30 2002
@@ -797,6 +821,7 @@
 
   line_selection_mode = false;
   word_selection_mode = false;
+  preserve_line_breaks = TRUE;
 
   QPoint pos = QPoint((ev->x()-tLx-bX+(font_w/2))/font_w,(ev->y()-tLy-bY)/font_h);
 
--- kdebase-3.1/konsole/konsole/konsole.cpp.orig	Sat Nov 23 19:06:02 2002
+++ kdebase-3.1/konsole/konsole/konsole.cpp	Fri Nov 29 03:40:43 2002
@@ -1864,7 +1864,7 @@
 
 QString Konsole::newSession(KSimpleConfig *co, QString program, const QStrList &args, const QString &_term,const QString &_icon,const QString &_title, const QString &_cwd)
 {
-  QString emu = "xterm";
+  QString emu = "xterm-color";
   QString icon = "openterm";
   QString key;
   QString sch = s_kconfigSchema;
--- kdebase-3.0.99/konsole/konsole/konsole_part.cpp.orig	Sun Oct 27 00:06:56 2002
+++ kdebase-3.0.99/konsole/konsole/konsole_part.cpp	Wed Nov 20 11:10:03 2002
@@ -167,7 +167,7 @@
   te->setMinimumSize(150,70);    // allow resizing, cause resize in TEWidget
 
   setWidget(te);
-  se = new TESession(te,shell,eargs,"xterm");
+  se = new TESession(te,shell,eargs,"xterm-color");
   connect( se,SIGNAL(done(TESession*)),
            this,SLOT(doneSession(TESession*)) );
   connect( se,SIGNAL(openURLRequest(const QString &)),
