--- ./kdm/server.c.orig	Wed Mar 24 21:50:03 1999
+++ ./kdm/server.c	Thu Mar 25 00:42:57 1999
@@ -89,6 +90,11 @@
     char	arg[1024];
     char	**parseArgs ();
     int		pid;
+    int         result = TRUE;
+    int         i;
+    Display     *disp;
+    pid_t       _pid;
+    int         status;
 
     Debug ("StartServer for %s\n", d->name);
     receivedUsr1 = 0;
@@ -129,8 +135,44 @@
     Debug ("Server Started %d\n", pid);
     d->serverPid = pid;
     if (serverPause ((unsigned) d->openDelay, pid))
-	return FALSE;
-    return TRUE;
+	result = FALSE;
+
+    if (result) {
+      /* fprintf(stderr, "Server started, checking\n"); */
+
+      setenv("XAUTHORITY", d->authFile, 1);
+
+      for (i=0; i<10; i++) {
+
+	if ( (_pid = fork()) == 0) {
+	  /* fprintf(stderr, "checking (%s) ", d->name); fflush(stderr); */
+	  disp = XOpenDisplay(d->name); 
+	  
+	  if (disp)
+	    exit(0);
+
+	  exit(1);
+	}
+	else if (pid == -1)
+	  return FALSE;
+
+	waitpid(_pid, &status, 0);
+	
+	if (WEXITSTATUS(status) == 0) {
+	  /* fprintf(stderr, "succes!!!!!!!\n"); */
+	  return TRUE;
+	}
+
+	sleep (1);
+      }
+
+    }
+
+    fprintf(stderr, "switching back to runlevel 3\n"); fflush(stderr); 
+    system("/sbin/telinit 3");
+    exit(-1);
+
+    return result;
 }
 
 int StartServer (d)
