--- kdebase-3.1.1/kcontrol/krdb/krdb.cpp~	Thu Dec  5 21:40:05 2002
+++ kdebase-3.1.1/kcontrol/krdb/krdb.cpp	Thu Mar 13 12:18:56 2003
@@ -52,7 +52,8 @@
    QStringList list = QStringList::split(':', QFile::decodeName(gtkrc));
    if (list.count() == 0)
    {
-      list.append(QString::fromLatin1("/etc/gtk/gtkrc"));
+      // Keep in sync with gtk+ prefix
+      list.append("/usr/X11R6/share/gtk/gtkrc." + QString::fromLatin1(getenv("LANG")).left(2));
       list.append(QDir::homeDirPath()+"/.gtkrc");
    }
    list.remove(gtkkde);
