diff -Nur --exclude *~ kdebase-2.0/acinclude.m4 kdebase-2.0.new/acinclude.m4
--- kdebase-2.0/acinclude.m4	Tue Oct 17 16:44:58 2000
+++ kdebase-2.0.new/acinclude.m4	Wed Oct 25 01:19:36 2000
@@ -1045,7 +1045,7 @@
 
 if test -z "$1"; then
 
-kde_incdirs="/usr/lib/kde/include /usr/local/kde/include /usr/kde/include /usr/include/kde /usr/include /opt/kde/include $x_includes $qt_includes"
+kde_incdirs="/usr/lib/kde/include /usr/local/kde/include /usr/kde/include /usr/include/kde /usr/include /opt/kde/include $x_includes $qt_includes $x_includes/kde"
 test -n "$KDEDIR" && kde_incdirs="$KDEDIR/include $KDEDIR/include/kde $KDEDIR $kde_incdirs"
 kde_incdirs="$ac_kde_includes $kde_incdirs"
 AC_FIND_FILE(ksock.h, $kde_incdirs, kde_incdir)
diff -Nur --exclude *~ kdebase-2.0/aclocal.m4 kdebase-2.0.new/aclocal.m4
--- kdebase-2.0/aclocal.m4	Tue Oct 17 16:45:08 2000
+++ kdebase-2.0.new/aclocal.m4	Wed Oct 25 01:19:27 2000
@@ -1056,7 +1056,7 @@
 
 if test -z "$1"; then
 
-kde_incdirs="/usr/lib/kde/include /usr/local/kde/include /usr/kde/include /usr/include/kde /usr/include /opt/kde/include $x_includes $qt_includes"
+kde_incdirs="/usr/lib/kde/include /usr/local/kde/include /usr/kde/include /usr/include/kde /usr/include /opt/kde/include $x_includes $qt_includes $x_includes/kde"
 test -n "$KDEDIR" && kde_incdirs="$KDEDIR/include $KDEDIR/include/kde $KDEDIR $kde_incdirs"
 kde_incdirs="$ac_kde_includes $kde_incdirs"
 AC_FIND_FILE(ksock.h, $kde_incdirs, kde_incdir)
