--- kdebase-3.5.5/ksysguard/ksysguardd/Linux/logfile.c	2006/10/16 07:09:47	595928
+++ kdebase-3.5.5/ksysguard/ksysguardd/Linux/logfile.c	2006/10/22 11:21:31	598023
@@ -59,9 +59,8 @@
     if ( ( fp = fopen( entry->path, "r" ) ) != NULL ) {
       snprintf( monitor, 1024, "logfiles/%s", entry->name );
       registerMonitor( monitor, "logfile", printLogFile, printLogFileInfo, sm );
+      fclose( fp );
     }
-
-    fclose( fp );
   }
 
   LogFiles = new_ctnr();

