--- kdebase/kcontrol/konqhtml/pluginopts.cpp.orig	2004-03-23 14:55:09.000000000 +0100
+++ kdebase/kcontrol/konqhtml/pluginopts.cpp	2004-03-23 15:17:25.182487976 +0100
@@ -352,21 +352,7 @@
     if ( config->hasKey( "scanPaths" ) )
         paths = config->readListEntry( "scanPaths" );
     else {//keep sync with kdebase/nsplugins/pluginscan
-        paths.append("$HOME/.netscape/plugins");
-        paths.append("/usr/lib64/browser-plugins");
-        paths.append("/usr/lib/browser-plugins");
-        paths.append("/usr/local/netscape/plugins");
-        paths.append("/opt/mozilla/plugins");
-	paths.append("/opt/mozilla/lib/plugins");
-        paths.append("/opt/netscape/plugins");
-        paths.append("/opt/netscape/communicator/plugins");
-        paths.append("/usr/lib/netscape/plugins");
-        paths.append("/usr/lib/netscape/plugins-libc5");
-        paths.append("/usr/lib/netscape/plugins-libc6");
-        paths.append("/usr/lib/mozilla/plugins");
-	paths.append("/usr/lib64/netscape/plugins");
-	paths.append("/usr/lib64/mozilla/plugins");
-        paths.append("$MOZILLA_HOME/plugins");
+        paths.append("/usr/lib/kde3/plugins/konqueror");
     }
 
     // fill list
--- kdebase/nsplugins/pluginscan.cpp.orig	2004-03-23 14:57:02.000000000 +0100
+++ kdebase/nsplugins/pluginscan.cpp	2004-03-23 15:15:28.117284592 +0100
@@ -441,20 +441,7 @@
     // setup default paths
     if ( !config->hasKey("scanPaths") ) {
         QStringList paths;
-        paths.append("$HOME/.netscape/plugins");
-        paths.append("/usr/lib64/browser-plugins");
-        paths.append("/usr/lib/browser-plugins");
-        paths.append("/usr/local/netscape/plugins");
-        paths.append("/opt/mozilla/plugins");
-	paths.append("/opt/mozilla/lib/plugins");
-        paths.append("/opt/netscape/plugins");
-        paths.append("/opt/netscape/communicator/plugins");
-        paths.append("/usr/lib/netscape/plugins");
-        paths.append("/usr/lib/netscape/plugins-libc5");
-        paths.append("/usr/lib/netscape/plugins-libc6");
-        paths.append("/usr/lib/mozilla/plugins");
-	paths.append("/usr/lib64/netscape/plugins");
-	paths.append("/usr/lib64/mozilla/plugins");
+        paths.append("/usr/lib/kde3/plugins/konqueror");
         paths.append("$MOZILLA_HOME/plugins");
         config->writeEntry( "scanPaths", paths );
     }
