--- kdebase-3.1.1/kcontrol/screensaver/scrnsave.cpp~	Fri Jan  3 05:41:58 2003
+++ kdebase-3.1.1/kcontrol/screensaver/scrnsave.cpp	Mon Mar 31 23:46:42 2003
@@ -147,23 +147,9 @@
     mMonitor = 0;
     mTesting = false;
 
-    // Add non-KDE path
     KGlobal::dirs()->addResourceType("scrsav",
-                                     KGlobal::dirs()->kde_default("apps") +
-                                     "apps/ScreenSavers/");
-
-    // Add KDE specific screensaver path
-    QString relPath="System/ScreenSavers/";
-    KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers" );
-    if (servGroup)
-    {
-      relPath=servGroup->relPath();
-      kdDebug() << "relPath=" << relPath << endl;
-    }
-
-    KGlobal::dirs()->addResourceType("scrsav",
-                                     KGlobal::dirs()->kde_default("apps") +
-                                     relPath);
+                                     KGlobal::dirs()->resourceDirs("apps") +
+                                     "screensavers/");
 
     readSettings();
 
