--- kdebase-3.0beta2.orig/konsole/konsole/TEWidget.cpp	Sat Feb  9 10:53:37 2002
+++ kdebase-3.0beta2/konsole/konsole/TEWidget.cpp	Thu Feb 14 04:00:29 2002
@@ -318,7 +318,7 @@
   connect(blinkCursorT, SIGNAL(timeout()), this, SLOT(blinkCursorEvent()));
 
   setMouseMarks(TRUE);
-  setVTFont( QFont("fixed") );
+  setVTFont( KGlobalSettings::fixedFont() );
   setColorTable(base_color_table); // init color table
 
   qApp->installEventFilter( this ); //FIXME: see below
--- kdebase-3.0beta2.orig/konsole/konsole/konsole.cpp	Sun Jan 27 01:04:54 2002
+++ kdebase-3.0beta2/konsole/konsole/konsole.cpp	Thu Feb 14 04:01:33 2002
@@ -87,6 +87,7 @@
 
 #include <kfontdialog.h>
 #include <kstandarddirs.h>
+#include <kglobalsettings.h>
 #include <qpainter.h>
 #include <kmenubar.h>
 #include <kmessagebox.h>
@@ -851,7 +852,7 @@
    {
       // Options that should be applied to all sessions /////////////
       // (1) set menu items and Konsole members
-      QFont tmpFont("fixed");
+     QFont tmpFont( KGlobalSettings::fixedFont() );
       defaultFont = config->readFontEntry("defaultfont", &tmpFont);
       setFont(QMIN(config->readUnsignedNumEntry("font",3),TOPFONT));
 
