diff -urN kdebase-2.0.1.orig/konsole/src/konsole.C kdebase-2.0.1/konsole/src/konsole.C
--- kdebase-2.0.1.orig/konsole/src/konsole.C	Mon Oct  9 22:55:29 2000
+++ kdebase-2.0.1/konsole/src/konsole.C	Tue Dec 26 19:12:11 2000
@@ -218,7 +218,7 @@
 //FIXME: call newSession here, somehow, instead the stuff below.
   // Please do it ! It forces to duplicate code... (David)
 
-  TESession* initial = new TESession(this,te,pgm,args,"xterm");
+  TESession* initial = new TESession(this,te,pgm,args,"xterm-color");
 
   connect( initial,SIGNAL(done(TESession*,int)),
            this,SLOT(doneSession(TESession*,int)) );
diff -urN kdebase-2.0.1.orig/konsole/src/konsole_part.C kdebase-2.0.1/konsole/src/konsole_part.C
--- kdebase-2.0.1.orig/konsole/src/konsole_part.C	Mon Oct  9 13:49:06 2000
+++ kdebase-2.0.1/konsole/src/konsole_part.C	Tue Dec 26 19:11:41 2000
@@ -116,7 +116,7 @@
   setWidget(te);
   // faking a KMainwindow - TESession assumes that (wrong design!)
   initial = new TESession((KMainWindow*)parentWidget,
-                          te,shell,eargs,"xterm");
+                          te,shell,eargs,"xterm-color");
   connect( initial,SIGNAL(done(TESession*,int)),
            this,SLOT(doneSession(TESession*,int)) );
   connect( te,SIGNAL(configureRequest(TEWidget*,int,int,int)),
