--- kdebase-031020/kcontrol/background/bgdefaults.h.orig	2003-03-26 15:38:26.000000000 +0100
+++ kdebase-031020/kcontrol/background/bgdefaults.h	2003-10-21 13:35:49.000000000 +0200
@@ -26,8 +26,8 @@
 // Per desktop defaults
 // Before you change this get in touch with me (torsten@kde.org)
 // Thanks!!
-#define _defColorA  QColor("#1E72A0")
-#define _defColorB  QColor("#C0C0C0")
+#define _defColorA  QColor("#39374B")
+#define _defColorB  QColor("#42528F")
 #define _defBackgroundMode KBackgroundSettings::VerticalGradient
 #define _defWallpaperMode KBackgroundSettings::NoWallpaper
 #define _defMultiMode KBackgroundSettings::NoMulti
