--- kdebase-R14.0.0/applnk/tde-essential.menu~	2012-03-23 11:06:16.000000000 +0200
+++ kdebase-R14.0.0/applnk/tde-essential.menu	2015-03-23 07:11:17.792635136 +0200
@@ -4,7 +4,7 @@
 <Menu>
         <!-- The following menus are hidden by default -->
 	<Menu>
-		<Name>Settings</Name>
+		<Name>KDE-Settings</Name>
 		<Directory>tde-settings.directory</Directory>
                 <MergeFile>../tde-settings.menu</MergeFile>
 	</Menu>
@@ -16,12 +16,4 @@
         <Move>
            <Old>Settings/Information</Old><New>Information</New>
         </Move>
-	<Menu>
-		<Name>System</Name>
-		<Menu>
-			<Name>ScreenSavers</Name>
-			<Directory>tde-system-screensavers.directory</Directory>
-	                <MergeFile>../tde-screensavers.menu</MergeFile>
-		</Menu>
-	</Menu>
 </Menu>
--- kdebase-031030/applnk/tde-settings.menu.orig	2003-04-02 22:47:54.000000000 +0200
+++ kdebase-031030/applnk/tde-settings.menu	2003-10-30 11:21:33.000000000 +0100
@@ -2,7 +2,7 @@
   "http://www.freedesktop.org/standards/menu-spec/1.0/menu.dtd">
 
 <Menu>
-	<Name>Settings</Name>
+	<Name>KDE-Settings</Name>
 	<Menu>
 		<Name>Accessibility</Name>
 		<Directory>kde-settings-accessibility.directory</Directory>
