--- kdebase-030317/kcontrol/kdm/background.cpp~	Sun Mar 16 00:35:44 2003
+++ kdebase-030317/kcontrol/kdm/background.cpp	Tue Mar 18 19:31:12 2003
@@ -53,7 +53,7 @@
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
     config->setGroup( "X-*-Greeter" );
-    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg",KDE_CONFDIR "/kdm/backgroundrc" ) );
+    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg","/etc/X11/kdm/backgroundrc" ) );
     m_background = new BGDialog( this, m_simpleConf, false );
 
     connect(m_background, SIGNAL(changed(bool)), SIGNAL(changed(bool)));
--- kdebase-trinity-3.5.13.2/kcontrol/kdm/kdm-appear.cpp~	2012-09-16 15:52:02.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kcontrol/kdm/kdm-appear.cpp	2014-05-27 13:58:32.057608488 +0300
@@ -355,7 +355,7 @@
 bool KDMAppearanceWidget::setLogo(TQString logo)
 {
     TQString flogo = logo.isEmpty() ?
-                    locate("data", TQString::fromLatin1("kdm/pics/kdelogo.png") ) :
+                    locate("data", TQString::fromLatin1("kdm/pics/pldlogo.png") ) :
                     logo;
     TQImage p(flogo);
     if (p.isNull())
@@ -502,7 +502,7 @@
   config->setGroup("X-*-Greeter");
 
   // Read the greeting string
-  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to %n")));
+  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to PLD Linux at %n")));
 
   // Regular logo or clock
   TQString logoArea = config->readEntry("LogoArea", "Logo" );
@@ -556,7 +556,7 @@
 
 void KDMAppearanceWidget::defaults()
 {
-  greetstr_lined->setText( i18n("Welcome to %n") );
+  greetstr_lined->setText( i18n("Welcome to PLD Linux at %n") );
   logoRadio->setChecked( true );
   slotAreaRadioClicked( KdmLogo );
   setLogo( "" );
--- kdebase-030924/kcontrol/kdm/kdm-users.cpp~	2003-09-23 21:46:11.000000000 +0200
+++ kdebase-030924/kcontrol/kdm/kdm-users.cpp	2003-09-24 16:39:25.000000000 +0200
@@ -65,7 +65,7 @@
     // We assume that $kde_datadir/kdm exists, but better check for pics/ and pics/users,
     // and create them if necessary.
     config->setGroup( "X-*-Greeter" );
-    m_userPixDir = config->readEntry( "FaceDir", KGlobal::dirs()->resourceDirs("data").last() + "kdm/faces" ) + '/';
+    m_userPixDir = config->readEntry( "FaceDir", "/etc/X11/kdm/faces" ) + '/';
     m_notFirst = false;
     QDir testDir( m_userPixDir );
     if ( !testDir.exists() && !testDir.mkdir( testDir.absPath() ))
--- kdebase-trinity-3.5.13.2/kcontrol/kdm/main.cpp~	2012-09-16 15:52:02.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kcontrol/kdm/main.cpp	2014-05-27 14:03:23.136893424 +0300
@@ -174,11 +174,11 @@
 	<< "' have unknown GID " << tgmapci.key() << endl;
 
   struct stat st;
-  if( stat( KDE_CONFDIR "/kdm/kdmdistrc" ,&st ) == 0) {
-    config = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/kdm/kdmdistrc" ));
+  if( stat( "/etc/X11/kdm/kdmdistrc" ,&st ) == 0) {
+    config = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmdistrc" ));
   }
   else {
-    config = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));
+    config = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmrc" ));
   }
 
   TQVBoxLayout *top = new TQVBoxLayout(this);
--- kdebase-trinity-3.5.13.2/kcontrol/screensaver/scrnsave.cpp~	2014-06-03 23:19:33.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kcontrol/screensaver/scrnsave.cpp	2014-06-04 11:12:35.914596433 +0300
@@ -51,11 +51,11 @@
 #include <fixx11h.h>
 
 #define OPEN_TDMCONFIG_AND_SET_GROUP									\
-if( stat( KDE_CONFDIR "/kdm/kdmdistrc" , &st ) == 0) {							\
-	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/kdm/kdmdistrc" ));		\
+if( stat( "/etc/X11/kdm/kdmdistrc" , &st ) == 0) {							\
+	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmdistrc" ));		\
 }													\
 else {													\
-	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));		\
+	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmrc" ));		\
 }													\
 mTDMConfig->setGroup("X-*-Greeter");
 
--- kdebase-trinity-3.5.13.2/kdesktop/lock/main.cc~	2013-05-06 11:08:22.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kdesktop/lock/main.cc	2014-06-04 11:13:06.705992318 +0300
@@ -52,11 +52,11 @@
 #endif
 
 #define OPEN_TDMCONFIG_AND_SET_GROUP									\
-if( stat( KDE_CONFDIR "/kdm/kdmdistrc" , &st ) == 0) {							\
-	tdmconfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/kdm/kdmdistrc" ));		\
+if( stat( "/etc/X11/kdm/kdmdistrc" , &st ) == 0) {							\
+	tdmconfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmdistrc" ));		\
 }													\
 else {													\
-	tdmconfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));		\
+	tdmconfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmrc" ));		\
 }													\
 tdmconfig->setGroup("X-*-Greeter");
 
--- kdebase-trinity-3.5.13.2/kdm/kfrontend/kdmctl.c~	2013-03-04 19:36:45.000000000 +0200
+++ kdebase-trinity-3.5.13.2/kdm/kfrontend/kdmctl.c	2014-06-04 11:13:21.883347041 +0300
@@ -146,7 +146,7 @@
 {
 	char *dpy = getenv( "DISPLAY" );
 	const char *ctl = getenv( "DM_CONTROL" );
-	const char *cfg = KDE_CONFDIR "/kdm/kdmrc";
+	const char *cfg = "/etc/X11/kdm/kdmrc";
 	char *ptr;
 	int fd;
 
--- kdebase-030930/kdm/kfrontend/Makefile.am~	2003-09-30 00:47:50.000000000 +0200
+++ kdebase-030930/kdm/kfrontend/Makefile.am	2003-09-30 10:06:29.000000000 +0200
@@ -38,7 +38,7 @@
 genkdmconf_LDADD = $(LIB_X11)
 
 install-data-local: genkdmconf
-	./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm --no-in-notice --face-src $(srcdir)/pics $(GENKDMCONF_FLAGS)
+	./genkdmconf --in $(DESTDIR)/etc/X11/kdm --no-in-notice --face-src $(srcdir)/pics $(GENKDMCONF_FLAGS)
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase-trinity-3.5.13.2/kdm/kfrontend/CMakeLists.txt~	2012-09-16 15:52:03.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kdm/kfrontend/CMakeLists.txt	2014-06-04 11:10:30.672252448 +0300
@@ -35,7 +35,7 @@
   set( GENKDMCONF_FLAGS "--no-old" )
 endif( )
 
-install( CODE "execute_process( COMMAND ${CMAKE_CURRENT_BINARY_DIR}/genkdmconf --in \$ENV{DESTDIR}${CONFIG_INSTALL_DIR}/kdm --no-in-notice --face-src ${CMAKE_CURRENT_SOURCE_DIR}/pics ${GENKDMCONF_FLAGS} )" )
+install( CODE "execute_process( COMMAND ${CMAKE_CURRENT_BINARY_DIR}/genkdmconf --in \$ENV{DESTDIR}/etc/X11/kdm --no-in-notice --face-src ${CMAKE_CURRENT_SOURCE_DIR}/pics ${GENKDMCONF_FLAGS} )" )
 
 
 ##### config.ci (generated) #####################
--- kdebase-trinity-3.5.13.2/kdm/config.def~	2014-05-27 13:55:54.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kdm/config.def	2014-05-27 14:35:25.710243084 +0300
@@ -29,7 +29,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 3
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #ifdef _AIX
@@ -1164,7 +1164,7 @@
 
 Key: Willing
 Type: string
-Default: ""
+Default: KDMCONF "/Xwilling"
 User: core
 # will be overwritten
 Instance: #
@@ -1581,7 +1581,7 @@
 
 Key: Setup
 Type: string
-Default: ""
+Default: KDMCONF "/Xsetup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1602,7 +1602,7 @@
 
 Key: Startup
 Type: string
-Default: ""
+Default: KDMCONF "/Xstartup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1620,7 +1620,7 @@
 
 Key: Reset
 Type: string
-Default: ""
+Default: KDMCONF "/Xreset"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1638,7 +1638,7 @@
 
 Key: Session
 Type: string
-Default: XBINDIR "/xterm -ls -T"
+Default: KDMCONF "/Xsession"
 #Merge: xdm - incompatible!
 User: core
 # will be overwritten
@@ -1746,10 +1746,10 @@
 
 Key: AllowRootLogin
 Type: bool
-Default: true
+Default: false
 User: core
 User: greeter(showRoot)
-Instance: */false
+Instance: #*/true
 Merge: xdm
 Comment:
  Allow root logins?
@@ -1759,11 +1759,11 @@
 
 Key: AllowNullPasswd
 Type: bool
-Default: true
+Default: false
 User: core
 # sensible?
-Instance: */false
-Instance: :*/true
+Instance: #*/true
+Instance: #:*/true
 Merge: xdm
 Comment:
  Allow to log in, when user has set an empty password?
@@ -1931,7 +1931,7 @@
 Default: "/usr/share/xsessions,/var/lib/menu-xdg/xsessions," KDMDATA "/sessions"
 User: core
 User: greeter-c
-Instance: */"/usr/share/xsessions,/var/lib/menu-xdg/xsessions," KDMDATA "/sessions"
+Instance: */"/etc/X11/sessions,/usr/share/xsessions,/var/lib/menu-xdg/xsessions," KDMDATA "/sessions"
 Comment:
  The directories containing session type definitions in .desktop format.
 Description:
@@ -1933,10 +1933,10 @@
 
 Key: ClientLogFile
 Type: string
-Default: ".xsession-errors"
+Default: ""
 User: core
-Instance: */".xsession-errors-%s"
-Instance: :0/".xsession-errors"
+Instance: #*/".xsession-errors-%s"
+Instance: #:0/".xsession-errors"
 Comment:
  The file (relative to $HOME) to redirect the session output to. This is
  a printf format string; one %s will be replaced with the display name.
@@ -1997,9 +1997,9 @@
  None/LOGO_NONE: nothing
  Logo/LOGO_LOGO: the image specified by <option>LogoPixmap</option>
  Clock/LOGO_CLOCK: a neat analog clock
-Default: Clock
+Default: Logo
 User: greeter
-Instance: */Logo
+Instance: #*/Clock
 Comment:
  What should be shown in the greeter's logo are:
 Description:
@@ -2009,9 +2009,9 @@
 
 Key: LogoPixmap
 Type: string
-Default: ""
+Default: KDMDATA "/pics/pldlogo.png"
 User: greeter(logo)
-Instance: */KDMDATA "/pics/kdelogo.png"
+Instance: #*/KDMDATA "/pics/kdelogo.png"
 Comment:
  The image to show when LogoArea=Logo.
 Description:
@@ -2092,9 +2092,9 @@
 
 Key: GreetString
 Type: string
-Default: "Welcome to Trinity at %n"
+Default: "Welcome to PLD Linux at %n"
 User: greeter
-Instance: #*/"Welcome to Trinity at %n"
+Instance: #*/"Welcome to PLD Linux at %n"
 Comment:
  The headline in the greeter. The following character pairs are replaced:
  - %d -> current display
@@ -2227,7 +2227,7 @@
 
 Key: MinShowUID
 Type: int
-Default: 0
+Default: 500
 User: greeter(lowUserId)
 # will be overwritten
 Instance: #*/
@@ -2288,7 +2288,7 @@
 
 Key: FaceDir
 Type: string
-Default: *KDMDATA "/faces"
+Default: *KDMCONF "/faces"
 User: greeter
 Instance: #*/"/usr/share/faces"
 Update: upd_facedir
