--- kdebase-3.1/kcontrol/kdm/kdm-users.cpp.orig	Mon Nov 18 14:14:28 2002
+++ kdebase-3.1/kcontrol/kdm/kdm-users.cpp	Sat Nov 23 08:34:11 2002
@@ -442,11 +442,11 @@
 	rballusr->setChecked(true);
 
     QString ps = config->readEntry( "FaceSource" );
-    if (su == QString::fromLatin1("UserOnly"))
+    if (ps == QString::fromLatin1("UserOnly"))
 	rbusronly->setChecked(true);
-    else if (su == QString::fromLatin1("PrefUser"))
+    else if (ps == QString::fromLatin1("PrefUser"))
 	rbprefusr->setChecked(true);
-    else if (su == QString::fromLatin1("PrefAdmin"))
+    else if (ps == QString::fromLatin1("PrefAdmin"))
 	rbprefadm->setChecked(true);
     else
 	rbadmonly->setChecked(true);
