--- kdebase-3.1.3/kcontrol/kdm/background.cpp.orig	Wed Jul 30 14:48:56 2003
+++ kdebase-3.1.3/kcontrol/kdm/background.cpp	Wed Jul 30 14:50:03 2003
@@ -54,7 +54,7 @@
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
     config->setGroup( "X-*-Greeter" );
-    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg",KDE_CONFDIR "/kdm/backgroundrc" ) );
+    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg","/etc/X11/kdm/backgroundrc" ) );
     m_background = new Backgnd( this, m_simpleConf, false );
     connect(m_background, SIGNAL(changed(bool)), SIGNAL(changed(bool)));
 
@@ -106,7 +106,7 @@
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
-    m_background->defaults();
+    m_background->kdmdefaults();
     emit changed(true);
 }
 
--- kdebase-3.1.3/kcontrol/kdm/kdm-appear.cpp.orig	Wed Jul 30 14:50:18 2003
+++ kdebase-3.1.3/kcontrol/kdm/kdm-appear.cpp	Wed Jul 30 14:55:45 2003
@@ -234,6 +234,7 @@
 
   guicombo = new KBackedComboBox(hlp);
   loadGuiStyles(guicombo);
+  //guicombo->insertItem("KDE");
   label = new QLabel(guicombo, i18n("GUI s&tyle:"), hlp);
   connect(guicombo, SIGNAL(activated(int)), SLOT(changed()));
   hglay->addWidget(label, 0, 0);
@@ -380,7 +381,7 @@
 bool KDMAppearanceWidget::setLogo(QString logo)
 {
     QString flogo = logo.isEmpty() ?
-                    locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") ) :
+		    QString::fromLatin1("/etc/X11/kdm/pics/pldlogo.png") :
                     logo;
     QPixmap p(flogo);
     if (p.isNull())
@@ -397,7 +398,7 @@
 {
     KImageIO::registerFormats();
     QString fileName = KFileDialog::getOpenFileName(
-	locate("data", QString::fromLatin1("kdm/pics/") ),
+	QString::fromLatin1("/etc/X11/kdm/pics/"),
 	KImageIO::pattern());
     if (!fileName.isEmpty())
 	if (setLogo(fileName))
@@ -454,9 +455,7 @@
 
 	// we gotta check if it is a non-local file and make a tmp copy at the hd.
 	if(!url->isLocalFile()) {
-	    pixurl = "file:" +
-		     KGlobal::dirs()->resourceDirs("data").last() +
-		     "kdm/pics/" + url->fileName();
+            pixurl = "file:/etc/X11/kdm/pics/" + url->fileName();
 	    KIO::NetAccess::copy(*url, pixurl);
 	    istmp = true;
 	} else {
@@ -511,10 +510,10 @@
   config->setGroup("X-*-Greeter");
 
   // Read the greeting string
-  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to %s at %n")));
+  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to PLD Linux at %n")));
 
   // Regular logo or clock
-  QString logoArea = config->readEntry("LogoArea", "Logo" );
+  QString logoArea = config->readEntry("LogoArea", "None" );
   if (logoArea == "Clock") {
     clockRadio->setChecked(true);
     slotAreaRadioClicked(KdmClock);
@@ -561,9 +560,9 @@
 
 void KDMAppearanceWidget::defaults()
 {
-  greetstr_lined->setText( i18n("Welcome to %s at %n") );
-  logoRadio->setChecked( true );
-  slotAreaRadioClicked( KdmLogo );
+  greetstr_lined->setText( i18n("Welcome to PLD Linux at %n") );
+  noneRadio->setChecked( true );
+  slotAreaRadioClicked( KdmNone );
   posCenterRadio->setChecked( true );
   slotPosRadioClicked( 0 );
   setLogo( "" );
--- kdebase-3.1.3/kcontrol/kdm/kdm-font.cpp.orig	Wed Jul 30 14:55:54 2003
+++ kdebase-3.1.3/kcontrol/kdm/kdm-font.cpp	Wed Jul 30 14:56:26 2003
@@ -112,8 +112,8 @@
 
 void KDMFontWidget::set_def()
 {
-  stdfont = QFont("helvetica", 12);
-  failfont = QFont("helvetica", 12, QFont::Bold);
+  stdfont = QFont("helvetica", 11);
+  failfont = QFont("helvetica", 11, QFont::Bold);
   greetfont = QFont("charter", 24);
 }
 
--- kdebase-3.1.3/kcontrol/kdm/kdm-sess.cpp.orig	Wed Jul 30 14:56:38 2003
+++ kdebase-3.1.3/kcontrol/kdm/kdm-sess.cpp	Wed Jul 30 14:58:03 2003
@@ -400,11 +400,6 @@
   sdlcombo->setCurrentItem(SdAll);
   sdrcombo->setCurrentItem(SdRoot);
 
-  sessionslb->clear();
-  sessionslb->insertItem("default");
-  sessionslb->insertItem("kde");
-  sessionslb->insertItem("failsafe");
-
 #if defined(__linux__) && defined(__i386__)
     lilo_check->setChecked(false);
     slotLiloCheckToggled(false);
--- kdebase-3.1.3/kcontrol/kdm/kdm-users.cpp.orig	Wed Jul 30 14:58:12 2003
+++ kdebase-3.1.3/kcontrol/kdm/kdm-users.cpp	Wed Jul 30 14:59:05 2003
@@ -66,7 +66,7 @@
 
     // We assume that $kde_datadir/kdm exists, but better check for pics/ and pics/users,
     // and create them if necessary.
-    m_userPixDir = KGlobal::dirs()->resourceDirs("data").last() + "kdm/pics/";
+    m_userPixDir = "/etc/X11/kdm/pics/";
     m_notFirst = false;
     QDir testDir( m_userPixDir );
     if ( !testDir.exists() && !testDir.mkdir( testDir.absPath() )) {
--- kdebase-3.1.3/kcontrol/kdm/main.cpp.orig	Wed Jul 30 14:59:11 2003
+++ kdebase-3.1.3/kcontrol/kdm/main.cpp	Wed Jul 30 14:59:56 2003
@@ -92,7 +92,7 @@
     usermap.insert( QString::fromLocal8Bit( ps->pw_name ), ps->pw_uid );
   endpwent();
 
-  config = new KSimpleConfig( QString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));
+  config = new KSimpleConfig( QString::fromLatin1( "/etc/X11/kdm/kdmrc" ));
 
   QVBoxLayout *top = new QVBoxLayout(this);
   tab = new QTabWidget(this);
--- kdebase-3.1.3/kdm/kfrontend/kdmconfig.cpp.orig	Wed Jul 30 15:03:15 2003
+++ kdebase-3.1.3/kdm/kfrontend/kdmconfig.cpp	Wed Jul 30 15:03:52 2003
@@ -148,7 +148,7 @@
 
     _logo = GetCfgQStr (C_LogoPixmap);
     if( _logo.isEmpty())
-	_logo = locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") );
+       _logo = QString::fromLatin1("/etc/X11/kdm/pics/pldlogo.png");
 
     _showUsers = GetCfgInt (C_ShowUsers);
     _users = GetCfgQStrList (C_SelectedUsers);
--- kdebase-3.1.3/kdm/kfrontend/Makefile.am.orig	Wed Jul 30 15:01:16 2003
+++ kdebase-3.1.3/kdm/kfrontend/Makefile.am	Wed Jul 30 15:02:10 2003
@@ -30,8 +30,8 @@
 genkdmconf_LDADD = $(LIB_X11)
 
 install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
+	@$(mkinstalldirs) $(DESTDIR)/etc/X11/kdm
+	@./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase-3.1.3/kdm/kfrontend/Makefile.in.orig	Wed Jul 30 15:02:16 2003
+++ kdebase-3.1.3/kdm/kfrontend/Makefile.in	Wed Jul 30 15:03:08 2003
@@ -982,8 +982,8 @@
 
 
 install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
+       @$(mkinstalldirs) $(DESTDIR)/etc/X11/kdm
+       @./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase-3.1.3/kdm/kfrontend/pics/Makefile.am.orig	Wed Jul 30 15:04:11 2003
+++ kdebase-3.1.3/kdm/kfrontend/pics/Makefile.am	Wed Jul 30 15:04:43 2003
@@ -1,5 +1,5 @@
 
-picsdir = $(kde_datadir)/kdm/pics
+picsdir = /etc/X11/kdm/pics
 pics_DATA = kdelogo.png kdeflower.png
 
 usersdir = $(picsdir)/users
--- kdebase-3.1.3/kdm/kfrontend/pics/Makefile.in.orig	Wed Jul 30 15:04:50 2003
+++ kdebase-3.1.3/kdm/kfrontend/pics/Makefile.in	Wed Jul 30 15:05:11 2003
@@ -422,7 +422,7 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 
-picsdir = $(kde_datadir)/kdm/pics
+picsdir = /etc/X11/kdm/pics
 pics_DATA = kdelogo.png kdeflower.png
 
 usersdir = $(picsdir)/users
--- kdebase-3.1.3/kdm/kfrontend/genkdmconf.c.orig	Wed Jul 30 15:05:38 2003
+++ kdebase-3.1.3/kdm/kfrontend/genkdmconf.c	Wed Jul 30 15:12:57 2003
@@ -56,7 +56,7 @@
 
 #define as(ar) ((int)(sizeof(ar)/sizeof(ar[0])))
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 
 #define RCVERMAJOR 2
 #define RCVERMINOR 0
@@ -757,7 +757,7 @@
 "Program=\n"
 "ReverseBlending=false\n"
 "UseSHM=false\n"
-"Wallpaper=default_blue.jpg\n"
+"Wallpaper=/etc/X11/kdm/pics/pldwallpaper.png\n"
 "WallpaperList=\n"
 "WallpaperMode=Scaled\n";
 
@@ -1831,18 +1831,18 @@
 "# this opens a security hole: a secured display lock can be circumvented\n"
 "# (unless you use KDE's built-in screen lock). Default is false\n" },
 { "AllowRootLogin",	0, 0, 
-"# Allow root logins? Default is true\n" },
+"# Allow root logins? Default is false\n" },
 { "AllowNullPasswd",	0, 0, 
-"# Allow to log in, when user has set an empty password? Default is true\n" },
+"# Allow to log in, when user has set an empty password? Default is false\n" },
 { "AllowShutdown",	0, 0, 
 "# Who is allowed to shut down the system. This applies both to the\n"
 "# greeter and to the command FiFo. Default is All\n"
 "# \"None\" - no \"Shutdown...\" button is shown at all\n"
-"# \"Root\" - the root password must be entered to shut down\n"
-"# \"All\" - everybody can shut down the machine (Default)\n" },
+"# \"Root\" - the root password must be entered to shut down (Default)\n"
+"# \"All\" - everybody can shut down the machine\n" },
 { "AllowSdForceNow",	0, 0, 
 "# Who is allowed to abort all still running sessions when shutting down.\n"
-"# Same options as for AllowShutdown. Default is All\n" },
+"# Same options as for AllowShutdown. Default is Root\n" },
 { "DefaultSdMode",	0, 0, 
 "# The default choice for the shutdown condition/timing.\n"
 "# \"Schedule\" - shutdown after all sessions exit (possibly at once) (Default)\n"
@@ -1857,7 +1857,7 @@
 "# option!), i.e., KDM will behave exactly as before KDE 3.0.\n" }, 
 { "SessSaveFile",	0, 0, 
 "# Where (relatively to the user's home directory) to store the last\n"
-"# selected session. Default is .wmrc\n" },
+"# selected session. Default is .kdmrc\n" },
 { "ServerAttempts",	0, 0, 
 "# How often to try to run the X-server. Running includes executing it and\n"
 "# waiting for it to come up. Default is 1\n" },
@@ -1899,11 +1899,11 @@
 "# currently is quite greyish. Default is \"\"\n" },
 { "LogoArea",		0, upd_logoarea, 
 "# What should be shown righthand of the input lines:\n"
-"# \"Logo\" - the image specified by LogoPixmap (Default)\n"
+"# \"Logo\" - the image specified by LogoPixmap\n"
 "# \"Clock\" - a neat analog clock\n"
-"# \"None\" - nothing\n" },
+"# \"None\" - nothing (Default)\n" },
 { "LogoPixmap",		0, 0, 
-"# The image to show when LogoArea=Logo. Default is kdelogo.png\n" },
+"# The image to show when LogoArea=Logo. Default is pldlogo.png\n" },
 { "GreeterPosFixed",	0, 0, 
 "# Normally, the greeter is centered on the screen. Use this, if you want\n"
 "# it to appear elsewhere on the screen. Default is false\n" },
@@ -1924,11 +1924,11 @@
 "# - %r -> the operating system's version\n"
 "# - %m -> the machine (hardware) type\n"
 "# - %% -> a single %\n"
-"# Default is \"Welcome to %s at %n\"\n" },
+"# Default is \"Welcome to PLD Linux at %s at %n\"\n" },
 { "GreetFont",		0, 0, 
 "# The font for the headline. Default is charter,24,bold\n" },
 { "StdFont",		0, 0, 
-"# The normal font used in the greeter. Default is helvetica,12\n" },
+"# The normal font used in the greeter. Default is helvetica,11\n" },
 { "FailFont",		0, 0, 
 "# The font used for the \"Login Failed\" message. Default is helvetica,12,bold\n" },
 { "AntiAliasing",	0, 0,
@@ -2120,10 +2120,10 @@
 { "UserAuthDir",	"", 0 },
 { "Chooser",		"", 0 },	/* XXX kill */
 { "AutoReLogin",	"true", 0 },
-{ "AllowRootLogin",	"false", 1 },
-{ "AllowNullPasswd",	"false", 1 },
-{ "AllowShutdown",	"Root", 1 },
-{ "AllowSdForceNow",	"Root", 0 },
+{ "AllowRootLogin",    "true", 0 },
+{ "AllowNullPasswd",   "true", 0 },
+{ "AllowShutdown",     "All", 0 },
+{ "AllowSdForceNow",   "All", 0 },
 { "DefaultSdMode",	"ForceNow", 0}, 
 { "InteractiveSd",	"false", 0}, 
 { "SessSaveFile",	"", 0 },	/* XXX .dmrc */
@@ -2133,7 +2133,7 @@
 { "SessionTypes",	"default,kde,failsafe", 1 },
 { "GUIStyle",		"Windows", 0 },
 { "ColorScheme",	"Pumpkin", 0 },
-{ "LogoArea",		"None", 0 },
+{ "LogoArea",          "Logo", 0 },
 { "LogoPixmap",		"", 0 },
 { "GreeterPosFixed",	"true", 0 },
 { "GreeterPosX",	"200", 0 },
@@ -2166,8 +2166,8 @@
 { "ServerAttempts",	"", 0 },
 { "ServerTimeout",	"", 0 },
 { "AllowShutdown",	"All", 1 },
-{ "AllowRootLogin",	"true", 1 },
-{ "AllowNullPasswd",	"true", 1 },
+{ "AllowRootLogin",    "true", 0 },
+{ "AllowNullPasswd",   "true", 0 },
 { "NoPassEnable",	"true", 0 },
 { "NoPassUsers",	"fred,ethel", 0 },
 };
--- kdebase-3.1.3/kdm/kfrontend/kdm_config.c.orig	Wed Jul 30 15:13:07 2003
+++ kdebase-3.1.3/kdm/kfrontend/kdm_config.c	Wed Jul 30 15:15:25 2003
@@ -38,7 +38,7 @@
 #include <netdb.h>
 #include <grp.h>
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF KDE_CONFDIR "/etc/X11/kdm"
 
 
 /*
@@ -519,11 +519,11 @@
 { "AutoLoginSession",	C_autoString,	(void *)PautoLoginX,	"" },
 { "AutoLogin1st",	C_autoLogin1st | C_BOOL, 0,	"true" },
 { "AutoReLogin",	C_autoReLogin | C_BOOL,	0,	"false" },
-{ "AllowNullPasswd",	C_allowNullPasswd | C_BOOL, 0,	"true" },
-{ "AllowRootLogin",	C_allowRootLogin | C_BOOL, 0,	"true" },
-{ "SessSaveFile",	C_sessSaveFile,		0,	".wmrc" },
-{ "AllowShutdown",	C_allowShutdown | C_ENUM, sd_who, "All" },
-{ "AllowSdForceNow",	C_allowNuke | C_ENUM, sd_who,	"All" },
+{ "AllowNullPasswd",   C_allowNullPasswd | C_BOOL, 0,  "false" },
+{ "AllowRootLogin",    C_allowRootLogin | C_BOOL, 0,   "false" },
+{ "SessSaveFile",      C_sessSaveFile,         0,      ".kdmrc" },
+{ "AllowShutdown",     C_allowShutdown | C_ENUM, sd_who, "Root" },
+{ "AllowSdForceNow",   C_allowNuke | C_ENUM, sd_who,   "Root" },
 { "DefaultSdMode",	C_defSdMode | C_ENUM, sd_mode,	"Schedule" },
 { "InteractiveSd",	C_interactiveSd | C_BOOL, 0,	"true" },
 };
@@ -532,15 +532,15 @@
 { "SessionTypes",	C_SessionTypes,		0,	"default,failsafe" },
 { "GUIStyle",		C_GUIStyle, 		0,	"" },
 { "ColorScheme",	C_ColorScheme,		0,	"" },
-{ "LogoArea",		C_LogoArea | C_ENUM, logoarea,	"Logo" },
+{ "LogoArea",          C_LogoArea | C_ENUM, logoarea,  "None" },
 { "LogoPixmap",		C_LogoPixmap,		0,	"" },
 { "GreeterPosFixed",	C_GreeterPosFixed | C_BOOL, 0,	"false" },
 { "GreeterPosX",	C_GreeterPosX,		0,	"0" },
 { "GreeterPosY",	C_GreeterPosY,		0,	"0" },
 { "GreeterScreen",	C_GreeterScreen,	0,	"0" },
-{ "StdFont",		C_StdFont,		0,	"helvetica,12,5,0,50,0" },
-{ "FailFont",		C_FailFont,		0,	"helvetica,12,5,0,75,0" },
-{ "GreetString",	C_GreetString,		0,	"Welcome to %s at %n" },
+{ "StdFont",           C_StdFont,              0,      "helvetica,11,5,0,50,0" },
+{ "FailFont",          C_FailFont,             0,      "helvetica,11,5,0,75,0" },
+{ "GreetString",       C_GreetString,          0,      "Welcome to PLD Linux at %n" },
 { "GreetFont",		C_GreetFont,		0,	"charter,24,5,0,50,0" },
 { "AntiAliasing",	C_AntiAliasing | C_BOOL,0,	"false" },
 { "NumLock",		C_NumLock | C_ENUM, numlock,	"Keep" },
