--- kdebase-R14.0.0/tdm/kfrontend/sessions/CMakeLists.txt~	2013-08-10 09:15:18.000000000 +0300
+++ kdebase-R14.0.0/tdm/kfrontend/sessions/CMakeLists.txt	2015-03-23 06:53:43.207998906 +0200
@@ -12,18 +12,5 @@
 configure_file( tde.desktop.cmake tde.desktop @ONLY )
 
 install( FILES
-    admin.desktop ${CMAKE_CURRENT_BINARY_DIR}/tde.desktop
-    gnome.desktop 9wm.desktop aewm++.desktop aewm.desktop
-    afterstep.desktop amaterus.desktop amiwm.desktop
-    asclassic.desktop blackbox.desktop cde.desktop
-    ctwm.desktop cwwm.desktop enlightenment.desktop
-    evilwm.desktop fluxbox.desktop flwm.desktop fvwm.desktop
-    fvwm95.desktop golem.desktop icewm.desktop ion.desktop
-    larswm.desktop lwm.desktop matchbox.desktop metacity.desktop
-    mwm.desktop olvwm.desktop olwm.desktop openbox.desktop
-    oroborus.desktop phluid.desktop pwm.desktop qvwm.desktop
-    ratpoison.desktop sapphire.desktop sawfish.desktop
-    twm.desktop ude.desktop vtwm.desktop w9wm.desktop
-    waimea.desktop wm2.desktop wmaker.desktop xfce.desktop
-    xfce4.desktop kde-plasma-safe.desktop kde-plasma.desktop
-  DESTINATION ${DATA_INSTALL_DIR}/tdm/sessions )
+  ${CMAKE_CURRENT_BINARY_DIR}/tde.desktop
+  DESTINATION ${SHARE_INSTALL_PREFIX}/xsessions )
