--- kdebase-3.0.7/nsplugins/pluginscan.cpp.orig	Mon Aug  5 17:30:51 2002
+++ kdebase-3.0.7/nsplugins/pluginscan.cpp	Fri Sep 27 11:54:05 2002
@@ -439,11 +439,13 @@
         paths.append("/usr/local/netscape/plugins");
         paths.append("/opt/netscape/plugins");
         paths.append("/opt/netscape/communicator/plugins");
-        paths.append("/usr/lib/netscape/plugins");
+        paths.append("/usr/X11R6/lib/netscape/plugins");
+	paths.append("/usr/lib/netscape/plugins");
         paths.append("/usr/lib/netscape/plugins-libc5");
         paths.append("/usr/lib/netscape/plugins-libc6");
+        paths.append("/usr/X11R6/lib/mozilla/plugins");
         paths.append("/usr/lib/mozilla/plugins");
-        paths.append("$MOZILLA_HOME/plugins");
+	paths.append("$MOZILLA_HOME/plugins");
         config->writeEntry( "scanPaths", paths );
     }
 
