--- kdebase/kcontrol/konq/rootopts.cpp.than	Fri Mar  1 15:36:50 2002
+++ kdebase/kcontrol/konq/rootopts.cpp	Mon Mar 11 16:04:04 2002
@@ -283,7 +283,7 @@
     bool bMenuBar = g_pConfig->readBoolEntry("ShowMenubar", false);
     menuBarBox->setChecked(bMenuBar);
     g_pConfig->setGroup( "General" );
-    vrootBox->setChecked( g_pConfig->readBoolEntry( "SetVRoot", false ) );
+    vrootBox->setChecked( g_pConfig->readBoolEntry( "SetVRoot", true ) );
     iconsEnabledBox->setChecked( g_pConfig->readBoolEntry( "Enabled", true ) );
 
     //
--- kdebase/kdesktop/desktop.cc.than	Wed Feb 27 16:17:36 2002
+++ kdebase/kdesktop/desktop.cc	Mon Mar 11 16:04:59 2002
@@ -342,7 +342,7 @@
 
     KConfig * config = KGlobal::config();
     config->setGroup( "General" );
-    set_vroot = config->readBoolEntry( "SetVRoot", false );
+    set_vroot = config->readBoolEntry( "SetVRoot", true );
     slotSetVRoot(); // start timer
 }
 
