--- kdebase-3.0.3/kioslave/audiocd/kcmaudiocd/kcmaudiocd.h--	2002-08-25 14:56:41.000000000 +0200
+++ kdebase-3.0.3/kioslave/audiocd/kcmaudiocd/kcmaudiocd.h	2002-08-25 14:57:01.000000000 +0200
@@ -37,7 +37,7 @@ class KAudiocdModule : public KCModule
 public:
 
   KAudiocdModule(QWidget *parent=0, const char *name=0);
-
+  ~KAudiocdModule();
   QString quickHelp() const;
   const KAboutData* aboutData() const;    
 
--- kdebase-3.0.3/kioslave/audiocd/kcmaudiocd/kcmaudiocd.cpp--	2002-08-25 14:56:37.000000000 +0200
+++ kdebase-3.0.3/kioslave/audiocd/kcmaudiocd/kcmaudiocd.cpp	2002-08-25 14:58:14.000000000 +0200
@@ -191,6 +191,11 @@ KAudiocdModule::KAudiocdModule(QWidget *
     slotServerSelectionChanged();
 };
 
+KAudiocdModule::~KAudiocdModule()
+{
+  delete config;
+}
+
 void KAudiocdModule::slotServerSelectionChanged()
 {
     cddbserver_del_push->setEnabled(cddb_server_listbox->currentItem()!=-1);
