Index: kfind/kquery.cpp
===================================================================
--- kfind/kquery.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kfind/kquery.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -72,9 +72,10 @@
   m_fileItems.clear();
   if(m_useLocate) //use "locate" instead of the internal search method
   {
+    m_url.cleanPath();
     processLocate->clearArguments();
     *processLocate << "locate";
-    *processLocate << m_url.path().latin1();
+    *processLocate << m_url.path(1).latin1();
     bufferLocate=NULL;
     bufferLocateLength=0;
     processLocate->start(KProcess::NotifyOnExit,KProcess::AllOutput);
Index: kcontrol/taskbar/kcmtaskbar.desktop
===================================================================
--- kcontrol/taskbar/kcmtaskbar.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kcontrol/taskbar/kcmtaskbar.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -159,7 +159,7 @@
 Keywords[es]=kicker,panel,kpanel,barra de tareas,barra de inicio,barra de lanzamiento,ventanas
 Keywords[et]=kicker,paneel,kpanel,tegumiriba,käivitusriba,aknad
 Keywords[eu]=kicker,panela,kpanela,ataza-barra,hasiera-barra,abiatze-barra,leihoak
-Keywords[fa]=kicker، تابلو، kpanel، میله تکلیف، میله آغاز، میله راه اندازی، پنجره‌ها
+Keywords[fa]=kicker، تابلو، kpanel، میله تکلیف، میله آغاز، میله راه‌اندازی، پنجره‌ها
 Keywords[fi]=kicker,paneeli,kpanel,tehtäväpalkki,käynnistys,käynnistyspalkki,ikkunat
 Keywords[fr]=kicker,tableau de bord,kpanel,barre des tâches,barre de démarrage,barre de lancement,fenêtres
 Keywords[fy]=kicker,panel,paniel,kpanel,taakbalke,startbalk,Begjinbalke,startmenu,finsters
Index: kcontrol/info/memory.desktop
===================================================================
--- kcontrol/info/memory.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kcontrol/info/memory.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -179,7 +179,7 @@
 Keywords[es]=Memoria,RAM,Memoria virtual,Memoria física,Memoria compartida,Swap,Intercambio,Información del sistema
 Keywords[et]=Mälu,RAM,Virtuaalmälu,Füüsiline mälu,Jagatud mälu,Swap,Saaleala,Süsteemi info
 Keywords[eu]=Memoria,RAM,Memoria birtuala,Memoria fisikoa,Partekatutako memoria, Swap,Sistemaren informazioa
-Keywords[fa]=حافظه ،RAM،حافظه مجازی، حافظه فیزیکی، حافظه مشترک، مبادله کردن ، اطلاعات سیستم
+Keywords[fa]=حافظه ،حافظه با دستیابی تصادفی،حافظه مجازی، حافظه فیزیکی، حافظه مشترک، مبادله کردن ، اطلاعات سیستم
 Keywords[fi]=Muisti,RAM,Virtuaalimuisti,Fyysinen muisti,Jaettu muisti,Järjestelmätiedot
 Keywords[fr]=mémoire,RAM,mémoire virtuelle,mémoire physique,mémoire partagée,swap,partition d'échange,informations système,ram
 Keywords[fy]=memory,RAM,virtual memory,physical memory,shared memory,swap,systeemynformaasje,ûnthâld,wikselûnthâld,,virtueel ûnthâld,fysiek ûnthâld,dield ûnthâld,
Index: kcontrol/info/cdinfo.desktop
===================================================================
--- kcontrol/info/cdinfo.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kcontrol/info/cdinfo.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -87,7 +87,7 @@
 Keywords[el]=Πληροφορίες CD-ROM,CD-ROM,CD,Οδηγός CD,Δυνατότητες εγγραφέα
 Keywords[es]=Información CD-ROM,CD-ROM,CD,Dispositivo CD,Capacidades de escritura
 Keywords[et]=CD-seadme info,CD-ROM,CD,CD-seade,kirjuti omadused
-Keywords[fa]=اطلاعات CD-ROM ،CD-ROM ،CD، گردانندۀ CD، قابلیتهای رایتر
+Keywords[fa]=اطلاعات دیسک فشرده ،دیسک فشرده ،دیسک فشرده، گردانندۀ دیسک فشرده، قابلیتهای رایتر
 Keywords[fi]=CD-ROM tiedot,CD-ROM,CD,CD-asema,Kirjoitusominaisuudet
 Keywords[fr]=CD-ROM Informations,CD-ROM,CD,lecteur CD,capacités,graveur,fonctionnalités
 Keywords[he]=CD-ROM Information,CD-ROM,CD,CD Drive,Writer Capabilities, מידע אודות תקליטורים, תקליטור, תקליטור, צורב, סידי, סידי־רום, סידי-רום
Index: kcontrol/joystick/joystick.desktop
===================================================================
--- kcontrol/joystick/joystick.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kcontrol/joystick/joystick.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -15,7 +15,7 @@
 Comment[es]=joystick - un módulo de kcontrol para probar joysticks
 Comment[et]=juhtpult - KDE juhtimiskeskuse moodul juhtpultide testimiseks
 Comment[eu]=joystick - joystickak aztertzeko kontrol modulua
-Comment[fa]=اهرم کنترل - یک پیمانه kcontrolبرای آزمایش اهرمهای کنترل
+Comment[fa]=اهرم کنترل - یک پیمانه kcontrol برای آزمایش اهرمهای کنترل
 Comment[fi]=peliohjain - KDE:n ohjauskeskuksen moduuli peliohjainta varten
 Comment[fr]=Manettes de jeu - un module pour tester les manettes de jeu
 Comment[fy]=joystick - in Konfiguraasjemodule foar it teste fan joysticks
Index: kcontrol/launch/kcmlaunch.desktop
===================================================================
--- kcontrol/launch/kcmlaunch.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kcontrol/launch/kcmlaunch.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -94,7 +94,7 @@
 Comment[es]=Elija el estilo de notificación de lanzamiento de las aplicaciones
 Comment[et]=Rakenduste käivitamisel kasutajale antava tagasiside seadistamine
 Comment[eu]=Aukeratu aplikazioen abiatze-jakinarazpenen estiloa
-Comment[fa]=انتخاب سبک بازخورد راه اندازی کاربرد
+Comment[fa]=انتخاب سبک بازخورد راه‌اندازی کاربرد
 Comment[fi]=Valitse ohjelmien käynnistymisestä kertovan tiedon tyyli
 Comment[fr]=Définit le style du témoin de démarrage des applications
 Comment[fy]=Hjir kinne jo bepale wat fisueel te sjen is by it begjinnen fan in tapassing.
@@ -157,7 +157,7 @@
 Keywords[es]=aplicación,iniciar,lanzar,ocupado,cursor,notificación,ratón,puntero,rotación,giro,disco,arrancar,programa,informar
 Keywords[et]=rakendus,käivitamine,hõivatud,kursor,tagasiside,hiir,hiirekursor
 Keywords[eu]=aplikazioa,abiatu,lanpetuta,kurtsorea,jakinarazpena,sagua,gezia,errotazioa,itzulbira,diskoa,programa,jakinarazi
-Keywords[fa]=کاربرد، آغاز، راه انداختن، مشغول، مکان‌نما، بازخورد، موشی، اشاره‌گر، چرخش، دوار، دیسک، راه اندازی، برنامه، گزارش
+Keywords[fa]=کاربرد، آغاز، راه انداختن، مشغول، مکان‌نما، بازخورد، موشی، اشاره‌گر، چرخش، دوار، دیسک، راه‌اندازی، برنامه، گزارش
 Keywords[fi]=sovellus,käynnistyminen,käynnistäminen,varattu,osoitin,palaute,hiiri, pyörivä,levy,ohjelma
 Keywords[fr]=application,démarrage,occupé,curseur,souris,pointeur,rotation,disque,programme,rapport
 Keywords[fy]=applikaasje,start,Begjinne,besich,rinnerke,feedback,mûs,wizer,rotaasje,rûndraaie,skiif,begjinne,programma,rapport
Index: pics/wallpapers/aurora.svgz.desktop
===================================================================
--- pics/wallpapers/aurora.svgz.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ pics/wallpapers/aurora.svgz.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -6,7 +6,7 @@
 Name[br]=Sklerijenn an deiz
 Name[cs]=Jitřenka
 Name[el]=Αυγή
-Name[fa]=سپیده دم
+Name[fa]=اورورا
 Name[fr]=Aurore
 Name[he]=אאורורה
 Name[is]=Norðurljós
Index: klipper/klipperrc.desktop
===================================================================
--- klipper/klipperrc.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ klipper/klipperrc.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -739,7 +739,7 @@
 Description[es]=Iniciar &mutt
 Description[et]=&Mutt'i käivitamine
 Description[eu]=Abiarazi &mutt 
-Description[fa]=راه اندازی &mutt
+Description[fa]=راه‌اندازی &mutt
 Description[fi]=Käynistä &Mutt
 Description[fr]=Démarrer &Mutt
 Description[fy]=&Mutt útfiere
@@ -966,7 +966,7 @@
 Description[es]=Inicar K&Write
 Description[et]=K&Write'i käivitamine
 Description[eu]=Abiarazi K&Write 
-Description[fa]=راه اندازی K&Write‌
+Description[fa]=راه‌اندازی K&Write‌
 Description[fi]=Käynistä K&Write
 Description[fr]=Démarrer K&Write
 Description[fy]=K&Write útfiere
Index: kicker/kicker/core/container_button.cpp
===================================================================
--- kicker/kicker/core/container_button.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kicker/kicker/core/container_button.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -318,7 +318,6 @@
                 if (_button)
                 {
                     _button->properties();
-                    clearOpMenu();
                 }
                 break;
             default:
@@ -327,6 +326,7 @@
             KickerTip::enableTipping(true);
 
             Kicker::the()->setInsertionPoint(QPoint());
+            clearOpMenu();
             sentinal = false;
             return true;
         }
Index: kicker/kicker/core/container_extension.cpp
===================================================================
--- kicker/kicker/core/container_extension.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kicker/kicker/core/container_extension.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -142,11 +142,11 @@
     _layout->setColStretch(1,10);
 
     // instantiate the autohide timer
-    _autohideTimer = new QTimer(this);
+    _autohideTimer = new QTimer(this, "_autohideTimer");
     connect(_autohideTimer, SIGNAL(timeout()), SLOT(autoHideTimeout()));
 
     // instantiate the updateLayout event compressor timer
-    _updateLayoutTimer = new QTimer(this);
+    _updateLayoutTimer = new QTimer(this, "_updateLayoutTimer");
     connect(_updateLayoutTimer, SIGNAL(timeout()), SLOT(actuallyUpdateLayout()));
 
     installEventFilter(this); // for mouse event handling
@@ -190,6 +190,7 @@
         connect(m_extension, SIGNAL(updateLayout()), SLOT(updateLayout()));
         connect(m_extension, SIGNAL(maintainFocus(bool)),
                 SLOT(maintainFocus(bool)));
+
         _layout->addWidget(m_extension, 1, 1);
     }
 
@@ -541,8 +542,6 @@
 //    kdDebug(1210) << "PanelContainer::updateLayout()" << endl;
     resetLayout();
     updateWindowManager();
-
-    emit layoutUpdated(this);
 }
 
 void ExtensionContainer::enableMouseOverEffects()
Index: kicker/kicker/core/container_applet.cpp
===================================================================
--- kicker/kicker/core/container_applet.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kicker/kicker/core/container_applet.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -249,6 +249,7 @@
 	}
 
     Kicker::the()->setInsertionPoint(QPoint());
+    clearOpMenu();
 }
 
 void AppletContainer::slotRemoved(KConfig* config)
Index: kicker/kicker/core/container_extension.h
===================================================================
--- kicker/kicker/core/container_extension.h	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kicker/kicker/core/container_extension.h	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -114,7 +114,6 @@
 
 signals:
     void removeme(ExtensionContainer*);
-    void layoutUpdated(ExtensionContainer*);
 
 protected slots:
     virtual void showPanelMenu( const QPoint& pos );
Index: kicker/kicker/ui/browser_mnu.cpp
===================================================================
--- kicker/kicker/ui/browser_mnu.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kicker/kicker/ui/browser_mnu.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -472,6 +472,8 @@
     // no mime types left to check -> stop timer
     if(it == _mimemap.end()) {
         _mimecheckTimer->stop();
+        delete _mimecheckTimer;
+        _mimecheckTimer = 0;
         return;
     }
 
Index: kicker/applets/launcher/quicklauncher.desktop
===================================================================
--- kicker/applets/launcher/quicklauncher.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kicker/applets/launcher/quicklauncher.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -19,7 +19,7 @@
 Name[es]=Lanzador rápido
 Name[et]=Kiirkäivitaja
 Name[eu]=Abiarazle bizkorra
-Name[fa]=راه انداز سریع
+Name[fa]=راه‌انداز سریع
 Name[fi]=Sovellusten pikakäynnistin
 Name[fr]=Lanceur d'applications
 Name[fy]=Snel útfierder
Index: kicker/applets/swallow/swallowapplet.desktop
===================================================================
--- kicker/applets/swallow/swallowapplet.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kicker/applets/swallow/swallowapplet.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -17,7 +17,7 @@
 Name[es]=Applet contenedor
 Name[et]=Põimimise aplett
 Name[eu]=Swallow appleta
-Name[fa]=برنامک پرستو
+Name[fa]=برنامک Swallow
 Name[fi]=Upotussovelma
 Name[fr]=Applet englobante
 Name[fy]=Ynslúte applet
@@ -79,7 +79,7 @@
 Comment[es]=El applet contenedor del panel
 Comment[et]=Paneelil töötav põimimise aplett
 Comment[eu]=Swallow paneleko appleta
-Comment[fa]=برنامک تابلوی پرستو
+Comment[fa]=برنامک تابلوی Swallow
 Comment[fi]=Paneelin upotussovelma
 Comment[fr]=L'applet englobante du tableau de bord
 Comment[fy]=Applet foar it ynlúte fan X-toepassingen
Index: kioslave/media/mediamanager/halbackend.cpp
===================================================================
--- kioslave/media/mediamanager/halbackend.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kioslave/media/mediamanager/halbackend.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -215,11 +215,6 @@
              !libhal_device_get_property_bool(m_halContext, udi, "volume.disc.is_blank", NULL) )
             return;
 
-        if ( ( libhal_device_get_property_QString(m_halContext, udi, "volume.fsusage") == "filesystem" &&
-               !libhal_device_get_property_bool(m_halContext, udi, "volume.is_mounted", NULL ) ) &&
-             ( libhal_device_get_property_bool(m_halContext, udi, "volume.ignore", NULL ) ) )
-            return;
-
         /* Query drive udi */
         QString driveUdi = libhal_device_get_property_QString(m_halContext, udi, "block.storage_device");
         /* We don't list floppy volumes because we list floppy drives */
@@ -240,7 +235,19 @@
 
         /* Create medium */
         Medium* medium = new Medium(udi, "");
+        setVolumeProperties(medium);
 
+        if ( isInFstab( medium ).isNull() )
+        {
+            // if it's not mountable by user and not by HAL, don't show it at all
+            if ( ( libhal_device_get_property_QString(m_halContext, udi, "volume.fsusage") == "filesystem" &&
+                   !libhal_device_get_property_bool(m_halContext, udi, "volume.is_mounted", NULL ) ) &&
+                 ( libhal_device_get_property_bool(m_halContext, udi, "volume.ignore", NULL ) ) )
+            {
+                delete medium;
+                return;
+            }
+        }
         QMap<QString,QString> options = MediaManagerUtils::splitOptions(mountoptions(udi));
         kdDebug() << "automount " << options["automount"] << endl;
         if (options["automount"] == "true" && allowNotification ) {
@@ -248,7 +255,6 @@
             if (!error.isEmpty())
                 kdDebug() << "error " << error << endl;
         }
-        setVolumeProperties(medium);
         m_mediaList.addMedium(medium, allowNotification);
 
         return;
@@ -992,7 +998,10 @@
 
     for (; it!=end; ++it)
     {
-        kdDebug() << "isInFstab -" << medium->deviceNode() << "- -" << (*it)->realDeviceName() << "- -" << (*it)->mountedFrom() << endl;
+        QString reald = (*it)->realDeviceName();
+        if ( reald.endsWith( "/" ) )
+            reald = reald.left( reald.length() - 1 );
+        kdDebug() << "isInFstab -" << medium->deviceNode() << "- -" << reald << "- -" << (*it)->mountedFrom() << "-" << endl;
         if ((*it)->mountedFrom() == medium->deviceNode() || ( !medium->deviceNode().isEmpty() && (*it)->realDeviceName() == medium->deviceNode() ) )
 	{
             QStringList opts = (*it)->mountOptions();
Index: kdesktop/kdiconview.h
===================================================================
--- kdesktop/kdiconview.h	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdesktop/kdiconview.h	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -71,7 +71,6 @@
 
     void setAutoAlign( bool b );
 
-    void refreshIcons();
     QStringList selectedURLs();
 
     /**
@@ -105,6 +104,7 @@
     void startDirLister();
 
 protected slots:
+
     // slots connected to the icon view
     void slotReturnPressed( QIconViewItem *item );
     void slotExecuted( QIconViewItem *item );
@@ -150,7 +150,9 @@
     void lineupIcons();
     void slotPaste(); // for krootwm
     void slotClear();
+    void refreshIcons();
 
+
 protected:
     void createActions();
     void setupSortKeys();
Index: kdesktop/lock/lockprocess.cc
===================================================================
--- kdesktop/lock/lockprocess.cc	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdesktop/lock/lockprocess.cc	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -681,7 +681,7 @@
 void LockProcess::stopSaver()
 {
     kdDebug(1204) << "LockProcess: stopping saver" << endl;
-    resume();
+    resume( true );
     stopHack();
     hideSaverWindow();
     mVisibility = false;
@@ -852,12 +852,10 @@
     mSuspended = true;
 }
 
-void LockProcess::resume()
+void LockProcess::resume( bool force )
 {
-    if (!mDialogs.isEmpty())
-        return; // no resuming with dialog visible
-    if(!mVisibility)
-        return; // no need to resume, not visible
+    if( !force && (!mDialogs.isEmpty() || !mVisibility ))
+        return; // no resuming with dialog visible or when not visible
     if(mSuspended)
     {
         bitBlt( this, 0, 0, &mSavedScreen );
@@ -931,7 +929,7 @@
     if( mDialogs.isEmpty() ) {
         XChangeActivePointerGrab( qt_xdisplay(), GRABEVENTS,
                 blankCursor.handle(), CurrentTime);
-        resume();
+        resume( false );
     } else
         fakeFocusIn( mDialogs.first()->winId());
     return rt;
@@ -988,7 +986,7 @@
                 else
                 {
                     mSuspendTimer.stop();
-                    resume();
+                    resume( false );
                 }
                 if (event->xvisibility.state != VisibilityUnobscured)
                     stayOnTop();
Index: kdesktop/lock/lockprocess.h
===================================================================
--- kdesktop/lock/lockprocess.h	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdesktop/lock/lockprocess.h	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -68,7 +68,6 @@
     void signalPipeSignal();
     bool startLock();
     void suspend();
-    void resume();
     void checkDPMSActive();
     void slotDeadTimePassed();
 
@@ -94,6 +93,7 @@
     void stayOnTop();
     void lockXF86();
     void unlockXF86();
+    void resume( bool force );
     static QVariant getConf(void *ctx, const char *key, const QVariant &dflt);
 
     bool        mLocked;
Index: kdesktop/patterns/stonewall2.desktop
===================================================================
--- kdesktop/patterns/stonewall2.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdesktop/patterns/stonewall2.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -15,7 +15,7 @@
 Comment[es]=Fuerte 2 por Tigert
 Comment[et]=Stonewall 2 (Tigert)
 Comment[eu]=Stonewall 2, Tigertek egina
-Comment[fa]=Stonewall توسط تیگرت
+Comment[fa]=Stonewall ۲ توسط تیگرت
 Comment[fi]=Kiviseinä 2, tehnyt Tigert
 Comment[fr]=Stonewall 2 par Tigert
 Comment[fy]=Stienen muorre 2, troch Tigert
Index: kdesktop/minicli.cpp
===================================================================
--- kdesktop/minicli.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdesktop/minicli.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -868,9 +868,11 @@
    FILE *fs = popen(QFile::encodeName(cmd).data(), "r");
    if (fs)
    {
-      QTextStream ts(fs, IO_ReadOnly);
-      result = ts.read().stripWhiteSpace();
-      pclose(fs);
+       { // scope for QTextStream
+           QTextStream ts(fs, IO_ReadOnly);
+           result = ts.read().stripWhiteSpace();
+       }
+       pclose(fs);
    }
    return result;
 }
Index: kdesktop/kdiconview.cc
===================================================================
--- kdesktop/kdiconview.cc	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdesktop/kdiconview.cc	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -431,6 +431,9 @@
         KAction *pasteTo = KStdAction::paste( this, SLOT( slotPopupPasteTo() ), &m_actionCollection, "pasteto" );
         pasteTo->setEnabled( false ); // only enabled during popupMenu()
 
+        KShortcut reloadShortcut = KStdAccel::shortcut(KStdAccel::Reload);
+        new KAction( i18n( "&Reload" ), "reload", reloadShortcut, this, SLOT( refreshIcons() ), &m_actionCollection, "reload" );
+
         (void) new KAction( i18n( "&Rename" ), /*"editrename",*/ Key_F2, this, SLOT( renameSelectedItem() ), &m_actionCollection, "rename" );
         (void) new KAction( i18n( "&Properties" ), ALT+Key_Return, this, SLOT( slotProperties() ), &m_actionCollection, "properties" );
         KAction* trash = new KAction( i18n( "&Move to Trash" ), "edittrash", Key_Delete, &m_actionCollection, "trash" );
Index: applnk/kde-games-kidsgames.directory
===================================================================
--- applnk/kde-games-kidsgames.directory	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ applnk/kde-games-kidsgames.directory	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -1,81 +0,0 @@
-[Desktop Entry]
-Encoding=UTF-8
-Type=Directory
-Name=Games for Kids
-Name[af]=Kinder Speletjies
-Name[ar]= العاب للأطفال
-Name[az]=Uşaxlar Üçün Oyunlar
-Name[be]=Дзіцячыя гульні
-Name[bg]=Игри за деца
-Name[bn]=বাচ্চাদের খেলা
-Name[br]=C'hoarioù evit ar Bugale
-Name[bs]=Igre za djecu
-Name[ca]=Jocs per a nens
-Name[cs]=Hry pro děti
-Name[cy]=Gêmau ar gyfer Plant
-Name[da]=Spil for børn
-Name[de]=Spiele für Kinder
-Name[el]=Παιχνίδια για παιδιά
-Name[eo]=Ludoj por infanoj
-Name[es]=Juegos para niños
-Name[et]=Laste mängud
-Name[eu]=Haur jokoak
-Name[fa]=بازیهایی برای بچه‌ها
-Name[fi]=Lasten pelit
-Name[fo]=Spøl fyri børn
-Name[fr]=Jeux pour les enfants
-Name[fy]=Spultsjes foar bêrn
-Name[ga]=Cluichí do Pháistí
-Name[gl]=Xogos para Nenos
-Name[he]=משחקים לילדים
-Name[hi]=बच्चों के लिए खेल
-Name[hr]=Igre za djecu
-Name[hsb]=Hry za dźěći
-Name[hu]=Játékok kicsiknek
-Name[is]=Leikir fyrir krakka
-Name[it]=Giochi per bambini
-Name[ja]=キッズ向けゲーム
-Name[km]=ល្បែង​សម្រាប់​កូន​ក្មេង
-Name[ko]=어린이 놀이
-Name[lo]=ເກມສຳລັບເດັກ
-Name[lt]=Žaidimai vaikams
-Name[lv]=Spēles bērniem
-Name[mk]=Игри за деца
-Name[mn]=Хүүхдийн тоглоом
-Name[ms]=Permainan Kanak-kanak
-Name[mt]=Logħob għat-tfal
-Name[nb]=Spill for barn
-Name[nds]=Spelen för Kinners
-Name[nl]=spelletjes voor kinderen
-Name[nn]=Barnespel
-Name[nso]=Dipapadi tsa Bana
-Name[pa]=ਬੱਚਿਆਂ ਲਈ ਖੇਡਾਂ
-Name[pl]=Gry dla dzieci
-Name[pt]=Jogos para Crianças
-Name[pt_BR]=Jogos para Crianças
-Name[ro]=Jocuri pentru copii
-Name[ru]=Детские игры
-Name[rw]=Imikino y'Abana
-Name[se]=Mánáidspealut
-Name[sk]=Hry pre deti
-Name[sl]=Igre za otroke
-Name[sr]=Игре за децу
-Name[sr@Latn]=Igre za decu
-Name[ss]=Imidlalo yebantfwana
-Name[sv]=Spel för barn
-Name[ta]=குழந்தைகளுக்கான விளையாட்டுகள்
-Name[tg]=Бозиҳо барои бачагон
-Name[th]=เกมส์สำหรับเด็ก
-Name[tr]=Çocuklar için Oyunlar
-Name[tt]=Bala Uyını
-Name[uk]=Ігри для дітей
-Name[uz]=Кичкинтойлар учун ўйинлар
-Name[ven]=Mitambo ya vhana
-Name[vi]=Trò chơi cho Trẻ
-Name[wa]=Djeus po ls efants
-Name[xh]=Imidlalo Yabantwana
-Name[zh_CN]=儿童游戏
-Name[zh_TW]=小孩的遊戲
-Name[zu]=Imidlalo yabantwana
-Icon=package_games_kids
-X-KDE-SuppressGenericNames=Game,Game for Children
Index: applnk/kde-games-kids.directory
===================================================================
--- applnk/kde-games-kids.directory	(.../tags/KDE/3.5.5/kdebase)	(revision 0)
+++ applnk/kde-games-kids.directory	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -0,0 +1,81 @@
+[Desktop Entry]
+Encoding=UTF-8
+Type=Directory
+Name=Games for Kids
+Name[af]=Kinder Speletjies
+Name[ar]= العاب للأطفال
+Name[az]=Uşaxlar Üçün Oyunlar
+Name[be]=Дзіцячыя гульні
+Name[bg]=Игри за деца
+Name[bn]=বাচ্চাদের খেলা
+Name[br]=C'hoarioù evit ar Bugale
+Name[bs]=Igre za djecu
+Name[ca]=Jocs per a nens
+Name[cs]=Hry pro děti
+Name[cy]=Gêmau ar gyfer Plant
+Name[da]=Spil for børn
+Name[de]=Spiele für Kinder
+Name[el]=Παιχνίδια για παιδιά
+Name[eo]=Ludoj por infanoj
+Name[es]=Juegos para niños
+Name[et]=Laste mängud
+Name[eu]=Haur jokoak
+Name[fa]=بازیهایی برای بچه‌ها
+Name[fi]=Lasten pelit
+Name[fo]=Spøl fyri børn
+Name[fr]=Jeux pour les enfants
+Name[fy]=Spultsjes foar bêrn
+Name[ga]=Cluichí do Pháistí
+Name[gl]=Xogos para Nenos
+Name[he]=משחקים לילדים
+Name[hi]=बच्चों के लिए खेल
+Name[hr]=Igre za djecu
+Name[hsb]=Hry za dźěći
+Name[hu]=Játékok kicsiknek
+Name[is]=Leikir fyrir krakka
+Name[it]=Giochi per bambini
+Name[ja]=キッズ向けゲーム
+Name[km]=ល្បែង​សម្រាប់​កូន​ក្មេង
+Name[ko]=어린이 놀이
+Name[lo]=ເກມສຳລັບເດັກ
+Name[lt]=Žaidimai vaikams
+Name[lv]=Spēles bērniem
+Name[mk]=Игри за деца
+Name[mn]=Хүүхдийн тоглоом
+Name[ms]=Permainan Kanak-kanak
+Name[mt]=Logħob għat-tfal
+Name[nb]=Spill for barn
+Name[nds]=Spelen för Kinners
+Name[nl]=spelletjes voor kinderen
+Name[nn]=Barnespel
+Name[nso]=Dipapadi tsa Bana
+Name[pa]=ਬੱਚਿਆਂ ਲਈ ਖੇਡਾਂ
+Name[pl]=Gry dla dzieci
+Name[pt]=Jogos para Crianças
+Name[pt_BR]=Jogos para Crianças
+Name[ro]=Jocuri pentru copii
+Name[ru]=Детские игры
+Name[rw]=Imikino y'Abana
+Name[se]=Mánáidspealut
+Name[sk]=Hry pre deti
+Name[sl]=Igre za otroke
+Name[sr]=Игре за децу
+Name[sr@Latn]=Igre za decu
+Name[ss]=Imidlalo yebantfwana
+Name[sv]=Spel för barn
+Name[ta]=குழந்தைகளுக்கான விளையாட்டுகள்
+Name[tg]=Бозиҳо барои бачагон
+Name[th]=เกมส์สำหรับเด็ก
+Name[tr]=Çocuklar için Oyunlar
+Name[tt]=Bala Uyını
+Name[uk]=Ігри для дітей
+Name[uz]=Кичкинтойлар учун ўйинлар
+Name[ven]=Mitambo ya vhana
+Name[vi]=Trò chơi cho Trẻ
+Name[wa]=Djeus po ls efants
+Name[xh]=Imidlalo Yabantwana
+Name[zh_CN]=儿童游戏
+Name[zh_TW]=小孩的遊戲
+Name[zu]=Imidlalo yabantwana
+Icon=package_games_kids
+X-KDE-SuppressGenericNames=Game,Game for Children

Property changes on: applnk/kde-games-kids.directory
___________________________________________________________________
Name: svn:keywords
   + Author Date Id Revision

Index: applnk/kde-utilities-accessibility.directory
===================================================================
--- applnk/kde-utilities-accessibility.directory	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ applnk/kde-utilities-accessibility.directory	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -98,6 +98,7 @@
 Comment[fy]=Tagonklikens
 Comment[ga]=Inrochtaineacht
 Comment[gl]=Acesibilidade
+Comment[he]=נגישות
 Comment[hr]=Pristupačnost
 Comment[hu]=Kezelési segítség
 Comment[is]=Aðgengi
Index: applnk/compat/email.desktop
===================================================================
--- applnk/compat/email.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ applnk/compat/email.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -162,7 +162,7 @@
 Keywords[cy]=EBost,Post,E-Bost,Cyfeiriad,Gweinydd ebost,IMAP,POP3,Blwch e-bost lleol,blwch e-bost,Gwybodaeth Defnyddiwr,Cyfeiriad Ateb,Gwybodaeth gweinydd,Enw Defnyddiwr,Cyfrineiriau,Gwesteiwr,Proffil Cyffredol,Proffil,Corff,Gwesteiwr Mewn,Gwesteiwr Allan,Dibynnydd Ebost
 Keywords[da]=E-mail,Post,Adresse,Postserver,IMAP,POP3,Lokal postkasse,postkasse,Brugerinformation,Svaradresse,Serverinformation,Brugernavn,Kodeord,Vært, Denne profil,Profil,Organisation,Indkommende vært,Udgående vært, E-mail-klient
 Keywords[de]=Email,Mail,E-Mail,Adresse,E-Mail-Server,IMAP,POP3,Lokale Mailbox,Mailbox,Postfach,Benutzerinformationen,Antwortadresse,Serverinformation,Benutzername,Passwörter,Rechner,Profil,Organisation,Posteingang,Postausgang,E-Mail-Programm
-Keywords[el]=EMail,Mail,E-Mail,Διεύθυνση,Διακομιστής email,IMAP,POP3,Τοπική θυρίδα,θυρίδα,Πληροφορίες χρήστη,Διεύθυνση απάντησης,Πληροφορίες διακομιστή,Όνομα χρήστη,Κωδικοί πρόσβασης,Υπολογιστής,Τρέχων προφίλ,Προφίλ,Οργανισμός,Υπολογιστής εισερχομένων,Υπολογιστής εξερχομένων,Πελάτης email
+Keywords[el]=EMail,Mail,E-Mail,Διεύθυνση,Διακομιστής email,IMAP,POP3,Τοπική θυρίδα,θυρίδα,Πληροφορίες χρήστη,Διεύθυνση απάντησης,Πληροφορίες διακομιστή,Όνομα χρήστη,Κωδικοί πρόσβασης,Υπολογιστής,Τρέχον προφίλ,Προφίλ,Οργανισμός,Υπολογιστής εισερχομένων,Υπολογιστής εξερχομένων,Πελάτης email
 Keywords[en_GB]=EMail,Mail,E-Mail,email,e-mail,Address,Email server,IMAP,POP3,Local mailbox,mailbox,User Information,Reply Address,Server information,User Name,Passwords,Host,Current Profile,Profile,Organisation,Organization,Incoming Host,Outgoing Host,Email Client
 Keywords[eo]=retpoŝto,poŝto,adreso,servilo,IMAP,POP,poŝtkesto,uzanto, respondadreso,servilo,pasvorto,agordo,organizaĵo,ricevo,sendo
 Keywords[es]=Correo electrónico,Correo,E-Mail,Dirección,Servidor de correo,IMAP,POP3,Buzón local,buzón,Información del usuario,Dirección de respuesta,Información del servidor,Nombre de usuario,Contraseñas,Máquina,Perfil actual,Perfil,Organización,Máquina entrante,Máquina saliente,Cliente de correo
Index: applnk/Makefile.am
===================================================================
--- applnk/Makefile.am	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ applnk/Makefile.am	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -13,7 +13,7 @@
 kde-games-arcade.directory \
 kde-games-board.directory \
 kde-games-card.directory \
-kde-games-kidsgames.directory \
+kde-games-kids.directory \
 kde-games-roguelikes.directory \
 kde-games-strategy.directory \
 kde-games.directory \
Index: konqueror/preloader/konqy_preloader.desktop
===================================================================
--- konqueror/preloader/konqy_preloader.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ konqueror/preloader/konqy_preloader.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -76,7 +76,7 @@
 Comment[es]=Reduce el tiempo de inicio de Konqueror
 Comment[et]=Kahandab Konquerori käivitamise aega
 Comment[eu]=Konquerorren abiatze denbora laburtzen du
-Comment[fa]=زمان راه اندازی Konqueror را کاهش می‌دهد
+Comment[fa]=زمان راه‌اندازی Konqueror را کاهش می‌دهد
 Comment[fi]=Vähentää Konquerorin käynnistysaikaa
 Comment[fr]=Réduit le temps de démarrage de Konqueror
 Comment[fy]=Redusearret de begjintiid fan Konqueror
Index: ksplashml/kcmksplash/ksplashthememgr.desktop
===================================================================
--- ksplashml/kcmksplash/ksplashthememgr.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ ksplashml/kcmksplash/ksplashthememgr.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -157,7 +157,7 @@
 Keywords[es]=pantalla de inicio,tema de inicio,inicio
 Keywords[et]=käivitusekraan,käivutusteema,käivitamine
 Keywords[eu]=splash pantaila,splash gaia,abioa
-Keywords[fa]=رنگ‌پاشی پرده، چهره لکه، راه اندازی
+Keywords[fa]=رنگ‌پاشی پرده، چهره لکه، راه‌اندازی
 Keywords[fi]=aloitusruutu,aloitusruututeema,käynnistys
 Keywords[fr]=écran de démarrage,thème de démarrage,démarrage,splash screen
 Keywords[fy]=opstartscherm,begjinskerm,splash screen
Index: libkonq/konq_popupmenu.cc
===================================================================
--- libkonq/konq_popupmenu.cc	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ libkonq/konq_popupmenu.cc	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -851,9 +851,25 @@
                 KTrader::OfferList::ConstIterator it = offers.begin();
                 for( ; it != offers.end(); it++ )
                 {
-                    if ((*it)->noDisplay())
-                        continue;
+                    KService::Ptr service = (*it);
 
+                    // Skip OnlyShowIn=Foo and NotShowIn=KDE entries,
+                    // but still offer NoDisplay=true entries, that's the
+                    // whole point of such desktop files. This is why we don't
+                    // use service->noDisplay() here.
+                    const QString onlyShowIn = service->property("OnlyShowIn", QVariant::String).toString();
+                    if ( !onlyShowIn.isEmpty() ) {
+                        const QStringList aList = QStringList::split(';', onlyShowIn);
+                        if (!aList.contains("KDE"))
+                            continue;
+                    }
+                    const QString notShowIn = service->property("NotShowIn", QVariant::String).toString();
+                    if ( !notShowIn.isEmpty() ) {
+                        const QStringList aList = QStringList::split(';', notShowIn);
+                        if (aList.contains("KDE"))
+                            continue;
+                    }
+
                     QCString nam;
                     nam.setNum( id );
 
Index: libkonq/favicons/favicons.desktop
===================================================================
--- libkonq/favicons/favicons.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ libkonq/favicons/favicons.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -18,7 +18,7 @@
 Name[es]=Módulo Favicon de KDED
 Name[et]=KDED favicon moodul
 Name[eu]=KDED Favicon modulua
-Name[fa]=پیمانه شمایل پسندان KDED 
+Name[fa]=پیمانه شمایل پسندان KDED
 Name[fi]=KDED-sivustokuvakemoduuli
 Name[fr]=Icône préférée
 Name[gl]=Módulo de Faviconas de KDED
Index: drkonqi/presets/enduserrc
===================================================================
--- drkonqi/presets/enduserrc	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ drkonqi/presets/enduserrc	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -273,7 +273,7 @@
 Name[ms]=Anda mungkin mahu menghantar laporan ralat untuk aplikasi ini. Periksa jika ia tercatat di  http://bugs.kde.org, atau terus melkan kepada penulisnya. Sertakan maklumat sebanyak mungkim, seperti dokumen asal. Sertakan juga cara untuk mendapatkan ralat, jika boleh.
 Name[mt]=Tista' tibgħat rapport dwar problema għal dan il-programm. Iċċekkja jekk hux elenkat fuq http://bugs.kde.org, inkella ibgħat imejl lill-awtur. Inkludi kemm tista' informazzjoni, u forsi d-dokumenti oriġinali. Jekk hemm mod kif terġa tikkawża din il-problema, inkludi dawn id-dettalji ukoll.
 Name[nb]=Du vil kanskje sende en feilrapport til forfatteren. Se etter om du finner igjen feilen i lista på http://bugs.kde.org. Hvis ikke, så send en e-post til forfatteren og forklar hva som er galt. Ta med så mye informasjon som mulig. Utviklere får ofte ikke nok informasjon til å finne igjen feilen, men sjelden for mye.  Tenk etter om du kan sende med dokumenter og annet du jobbet med når problemene dukket opp. Hvis du vet hvordan du kan gjenskape feilen, så ta også med en beskrivelse av hvordan det gjøres.
-Name[nds]=Villicht wullt Du en Fehlerbericht för dit Programm sennen. Kiek na, wat dat op http://bugs.kde.org list is, wenn nich, schick den Autor en Nettbreef. Bitte föög so veel Informatschonen as mööglich to, villicht ok de Orginaal-Dokmenten. Wenn Du den Fehler toverlässig wedderhalen kannst, bitte ok angeven, woans dat geiht.
+Name[nds]=Wenn Du en Fehlerbericht för dit Programm sennen wullt, kiek na, wat dat op http://bugs.kde.org list is, un wenn nich, schick den Autor en Nettbreef. Bitte föög so veel Informatschonen as mööglich to, villicht ok de Orginaal-Dokmenten. Wenn Du den Fehler toverlässig wedderhalen kannst, bitte ok angeven, woans dat geiht.
 Name[nl]=U wilt misschien een bugrapport voor dit programma versturen. Controleer of de fout die u hebt gevonden al bekend is bij http://bugs.kde.org. Zo neen, neem dan contact op met de auteur van het programma. Sluit zoveel mogelijk informatie toe, eventueel inclusief de originele documenten. Als u een manier weet waarop u de fout kunt reproduceren, sluit die dan ook in.
 Name[nn]=Du vil kanskje senda ein feilrapport. Sjå etter om programmet finst på http://bugs.kde.org. Viss ikkje, kan du senda e-post til utviklaren. Ta med så mykje informasjon du kan, kanskje dei opphavlege dokumenta. Dersom du kan reprodusera feilen, ta med dette òg.
 Name[pa]=ਤੁਹਾਨੂੰ ਇਸ ਕਾਰਜ ਲਈ ਬੱਗ ਜਾਣਕਾਰੀ ਭੇਜਣੀ ਚਾਹੀਦੀ ਹੈ, ਇਹ ਜਾਂਚ ਕਰੋ ਕਿ ਇਹ listed on http://bugs.kde.org ਤੇ ਉਪਲੱਬਧ ਹੈ, ਨਹੀਂ ਤਾਂ ਲੇਖਕ ਨੂੰ ਜਾਣਕਾਰੀ ਦਿਓ। ਕਿਰਪਾ ਕਰਕੇ ਵੱਧ ਤੋਂ ਵੱਧ ਜਾਣਕਾਰੀ ਉਪਲੱਬਧ ਕਰਵਾਉ, ਸੰਭਵ ਹੋਵੇ ਤਾਂ ਅਸਲੀ ਦਸਤਾਵੇਜ਼। ਜੇਕਰ ਗਲਤੀ ਪੈਦਾ ਕਰਨ ਦਾ ਤਰੀਕਾ ਪਤਾ ਹੋਵੇ ਤਾਂ ਇਸ ਨੂੰ ਜਰੂਰ ਲਿਖੋ।
Index: kwin/kcmkwin/kwindecoration/kwindecoration.desktop
===================================================================
--- kwin/kcmkwin/kwindecoration/kwindecoration.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kwin/kcmkwin/kwindecoration/kwindecoration.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -161,7 +161,7 @@
 Keywords[es]=kwin,ventana,administrador,borde,estilo,tema,aspecto,comportamiento,disposición,botón,asa,esquina,kwm,decoración
 Keywords[et]=kwin,aken,haldur,piire,stiil,teema,välimus,kasutamine,nupud,serv,kwm,dekoratsioon
 Keywords[eu]=kwin,leihoa,kudeatzailea,ertza,estiloa,gaia,itxura,antolaketa,botoia, maneiatzailea,ertzea,kwm,dekorazioa
-Keywords[fa]=kwin ، پنجره ، مدیر ، کناره ، سبک ، چهره ، ظاهر ، احساس ، طرح بندی ، دکمه ، گرداندن ، کناره، kwm ، تزئین
+Keywords[fa]=kwin ، پنجره ، مدیر ، لبه ، سبک ، چهره ، ظاهر ، احساس ، طرح بندی ، دکمه ، گرداندن ، لبه، kwm ، تزئین
 Keywords[fi]=kwin,ikkuna,ikkunaohjelma,ikkunoinnin hallintaohjelma,tausta,tyyli,teema,ulkonäkö,tuntuma,ulkoasu,painike,kahva,kulma,kwm,kehys
 Keywords[fr]=kwin,fenêtre,gestionnaire,bordure,style,thème,apparence,ergonomie,disposition,bouton,poignée,bord,kwm,décoration
 Keywords[fy]=kwin,window,manager,rand,stijl,theme,tema,look,uiterlijk,gedrag,feel,layout,opmaak,button,knoppen,handle,rand,kwm,decoratie,windowmanager,venster,vensterbeheer,finster,râne,kader,styltema,uterlik,gedrach,finsterbehear
Index: kwin/kcmkwin/kwinrules/kwinrules.desktop
===================================================================
--- kwin/kcmkwin/kwinrules/kwinrules.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kwin/kcmkwin/kwinrules/kwinrules.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -81,7 +81,7 @@
 Comment[es]=Configurar parámetros específicos de una ventana
 Comment[et]=Spetsiaalselt akendega seotud seadistused
 Comment[eu]=Konfiguratu leiho jakin baten ezarpenak
-Comment[fa]=پیکربندی تنظیمات به ‌ویژه برای یک پنجره
+Comment[fa]=پیکربندی تنظیمات به طور ‌ویژه برای یک پنجره
 Comment[fi]=Muokkaa ikkunakohtaisia asetuksia
 Comment[fr]=Configuration de paramètres spécifiques à une fenêtre
 Comment[fy]=Hjir kinne jo ynstellings kieze spesifyk foar ien finster
Index: kwin/kcmkwin/kwinoptions/kwinmoving.desktop
===================================================================
--- kwin/kcmkwin/kwinoptions/kwinmoving.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kwin/kcmkwin/kwinoptions/kwinmoving.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -161,7 +161,7 @@
 Keywords[es]=moviendo,inteligente,cascada,maximizar,zona de adhesión,adhesión,adherir,borde 
 Keywords[et]=liigutamine,tark,kaskaad,maksimeerimine,haaramise tsoon,haaramine,piire
 Keywords[eu]=mugitzen,fina,kaskadan,maximizatu,itsasketa eremua, itsasketa,ertza
-Keywords[fa]=حرکت ، هوشمند ، آبشاری ، بیشینه سازی ، بیشینه سازی، منطقه پرش ، پرش ، کناره ،
+Keywords[fa]=حرکت ، هوشمند ، آبشاری ، بیشینه سازی ، بیشینه سازی، منطقه پرش ، پرش ، لبه ،
 Keywords[fi]=liikkuva,älykäs,sarjoita,suurenna,kehys
 Keywords[fr]=déplacement,cascade,maximiser,minimiser,coller,zone d'attraction,bordure
 Keywords[fy]=verplaatsing,ferpleatse,intelligen,yntelligint,tûk,plaatsing,pleatsing,trapsgewijs,trepfoarm,maximaliseren,maksimalisearje,randen magnetisch,magnetisch veld,rand,râne,magnetische randen
Index: kwin/kcmkwin/kwinoptions/kwinadvanced.desktop
===================================================================
--- kwin/kcmkwin/kwinoptions/kwinadvanced.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kwin/kcmkwin/kwinoptions/kwinadvanced.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -166,7 +166,7 @@
 Keywords[es]=recogiendo,borde,ratón sobre,bordes activos
 Keywords[et]=varjamine,piire,hover,aktiivsed piirded
 Keywords[eu]=biltzen,ertza,gainean,ertz aktiboak
-Keywords[fa]=سایه زنی ، کناره ، پلکیدن ، کناره‌های فعال
+Keywords[fa]=سایه زنی ، لبه، پلکیدن ، لبه‌های فعال
 Keywords[fi]=varjostus,kehys,kellunta,aktiiviset kehykset
 Keywords[fr]=bordures,cacher,couvrir,bordures actives
 Keywords[fy]=oprollen,oprôlje,afrollen,ôfrôlje,hoover,actieve randen,aktive rânen,randen magnetisch,magnetische randen
Index: kwin/client.cpp
===================================================================
--- kwin/client.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kwin/client.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -477,17 +477,30 @@
     // calls it or when the decoration is created/destroyed
 
     if( Shape::version() >= 0x11 ) // 1.1, has input shape support
-        { // there appears to be no way to find out if a window has input
-          // shape set or not, so always set propagate the input shape
-          // (it's the same like the bounding shape by default)
-        XShapeCombineShape( qt_xdisplay(), frameId(), ShapeInput, 0, 0,
+        { // There appears to be no way to find out if a window has input
+          // shape set or not, so always propagate the input shape
+          // (it's the same like the bounding shape by default).
+          // Also, build the shape using a helper window, not directly
+          // in the frame window, because the sequence set-shape-to-frame,
+          // remove-shape-of-client, add-input-shape-of-client has the problem
+          // that after the second step there's a hole in the input shape
+          // until the real shape of the client is added and that can make
+          // the window lose focus (which is a problem with mouse focus policies)
+        static Window helper_window = None;
+        if( helper_window == None )
+            helper_window = XCreateSimpleWindow( qt_xdisplay(), qt_xrootwin(),
+                0, 0, 1, 1, 0, 0, 0 );
+        XResizeWindow( qt_xdisplay(), helper_window, width(), height());
+        XShapeCombineShape( qt_xdisplay(), helper_window, ShapeInput, 0, 0,
                            frameId(), ShapeBounding, ShapeSet );
-        XShapeCombineShape( qt_xdisplay(), frameId(), ShapeInput,
+        XShapeCombineShape( qt_xdisplay(), helper_window, ShapeInput,
                            clientPos().x(), clientPos().y(),
                            window(), ShapeBounding, ShapeSubtract );
-        XShapeCombineShape( qt_xdisplay(), frameId(), ShapeInput,
+        XShapeCombineShape( qt_xdisplay(), helper_window, ShapeInput,
                            clientPos().x(), clientPos().y(),
                            window(), ShapeInput, ShapeUnion );
+        XShapeCombineShape( qt_xdisplay(), frameId(), ShapeInput, 0, 0,
+                           helper_window, ShapeInput, ShapeSet );
         }
     }
 
Index: kwin/eventsrc
===================================================================
--- kwin/eventsrc	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kwin/eventsrc	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -4413,7 +4413,7 @@
 Comment[es]=Desaparece una ventana transitoria (un diálogo)
 Comment[et]=Ajutine aken (dialoog) on eemaldatud
 Comment[eu]=Leiho elkarrizketa kentzen da
-Comment[fa]=پنجره گذرا (یک محاوره ) حذف می‌شود
+Comment[fa]=پنجره گذرا )یک محاوره( حذف می‌شود
 Comment[fi]=Lyhytaikainen ikkuna (dialogi) poistetaan
 Comment[fr]=Une boîte de dialogue disparaît
 Comment[fy]=In dialooch is slúten
Index: konsole/eventsrc
===================================================================
--- konsole/eventsrc	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ konsole/eventsrc	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -609,7 +609,7 @@
 Comment[es]=Ha terminado una sesión con estado no cero
 Comment[et]=Seanss väljus nullist erineva staatusega
 Comment[eu]=Saio bat amaitu egin da ez-zero egoerarekin
-Comment[fa]=نشست توسط وضعیت غیرصفر خارج شده‌است
+Comment[fa]=نشستی توسط وضعیت غیرصفر خارج شده‌است
 Comment[fi]=Istunto lopetettiin paluuarvolla, joka ei ollut nolla
 Comment[fr]=Une session s'est terminée anormalement
 Comment[fy]=In sesje is einige mei in non-zero tastân
Index: l10n/ch/entry.desktop
===================================================================
--- l10n/ch/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/ch/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -18,7 +18,7 @@
 Name[es]=Suiza
 Name[et]=Šveits
 Name[eu]=Suitza
-Name[fa]=سویتزرلند
+Name[fa]=سوئیس
 Name[fi]=Sveitsi
 Name[fr]=Suisse
 Name[fy]=Switserlân
Index: l10n/sh/entry.desktop
===================================================================
--- l10n/sh/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/sh/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -15,7 +15,7 @@
 Name[el]=Αγία Ελένη
 Name[eo]=Sent-Heleno
 Name[es]=Santa Helena
-Name[fa]=سنت هلنا
+Name[fa]=سنت هلن
 Name[fr]=Sainte-Hélène
 Name[fy]=Sint Helena
 Name[ga]=San Héilin
Index: l10n/sv/entry.desktop
===================================================================
--- l10n/sv/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/sv/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -10,7 +10,7 @@
 Name[el]=Ελ Σαλβαντόρ
 Name[eo]=Salvadoro
 Name[et]=Salvador
-Name[fa]=ال سالوادور
+Name[fa]=السالوادور
 Name[fr]=Salvador
 Name[ga]=An tSalvadóir
 Name[gl]=O Salvador
Index: l10n/sz/entry.desktop
===================================================================
--- l10n/sz/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/sz/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -15,7 +15,7 @@
 Name[es]=Swazilandia
 Name[et]=Svaasimaa
 Name[eu]=Swazilandia
-Name[fa]=سوئیس
+Name[fa]=سوآزیلند
 Name[fi]=Swazimaa
 Name[ga]=An tSuasalainn
 Name[gl]=Suacilándia
Index: l10n/lc/entry.desktop
===================================================================
--- l10n/lc/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/lc/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -10,7 +10,7 @@
 Name[el]=Σάντα Λουτσία
 Name[eo]=Sent-Lucio
 Name[es]=Santa Lucía
-Name[fa]=سنت لوشا
+Name[fa]=سنت لوسیا
 Name[fo]=Sankta Lusia
 Name[fr]=Sainte Lucie
 Name[ga]=San Lúisia
Index: l10n/tk/entry.desktop
===================================================================
--- l10n/tk/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/tk/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -8,7 +8,7 @@
 Name[cy]=Tocelaw
 Name[el]=Τοκελάου
 Name[eo]=Tokelao
-Name[fa]=توکیلاو
+Name[fa]=توکلائو
 Name[ga]=Na hOileáin Tócala
 Name[he]=טוקלאו
 Name[hi]=तोकेलाऊ
Index: l10n/lt/entry.desktop
===================================================================
--- l10n/lt/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/lt/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -18,7 +18,7 @@
 Name[es]=Lituania
 Name[et]=Leedu
 Name[eu]=Lituania
-Name[fa]=ایتوانی
+Name[fa]=لیتوانی
 Name[fi]=Liettua
 Name[fr]=Lituanie
 Name[fy]=Litouwen
Index: l10n/tz/entry.desktop
===================================================================
--- l10n/tz/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/tz/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -17,7 +17,7 @@
 Name[es]=Tanzania, Republica de
 Name[et]=Tansaania
 Name[eu]=Tanaziar Errepublika Batua
-Name[fa]=جمهوری متحده تانزانیاِِ
+Name[fa]=جمهوری متحده تانزانیا
 Name[fi]=Tansanian yhdistäytynyt tasavalta
 Name[fr]=Tanzanie, République unie de
 Name[fy]=Tanzanië, Ferienigd republyk fan
Index: l10n/mo/entry.desktop
===================================================================
--- l10n/mo/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/mo/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -12,7 +12,7 @@
 Name[el]=Μακάο SAR (Κίνα)
 Name[et]=Macau
 Name[eu]=Macau SAR(Txina)
-Name[fa]=منطقه صنعتی ویژه ماکائو (چین)
+Name[fa]=منطقه صنعتی ویژه ماکائو )چین(
 Name[fi]=Makao SAR(Kiina)
 Name[fr]=Macao SAR (Chine)
 Name[ga]=Macao SAR(An tSín)
Index: l10n/nc/entry.desktop
===================================================================
--- l10n/nc/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/nc/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -17,7 +17,7 @@
 Name[es]=Nueva Caledonia
 Name[et]=Uus-Kaledoonia
 Name[eu]=Kaledonia Berria
-Name[fa]=کالیدونیای نو
+Name[fa]=کالدونی جدید
 Name[fi]=Uusi-Kaledonia
 Name[fr]=Nouvelle Calédonie
 Name[fy]=Nij Caledonië
Index: l10n/nf/entry.desktop
===================================================================
--- l10n/nf/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/nf/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -16,7 +16,7 @@
 Name[es]=Isla Norfolk
 Name[et]=Norfolki saar
 Name[eu]=Norfok Irla
-Name[fa]=جزیره نورفوک
+Name[fa]=جزیره نورفولک
 Name[fi]=Norfolkinsaari
 Name[fr]=Île Norfolk
 Name[fy]=Norfolk Eilân
Index: l10n/va/entry.desktop
===================================================================
--- l10n/va/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/va/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -18,7 +18,7 @@
 Name[es]=Vaticano
 Name[et]=Vatikan
 Name[eu]=Batikano Hiria
-Name[fa]=شهر واتیکان
+Name[fa]=واتیکان سیتی
 Name[fi]=Vatikaani
 Name[fr]=Vatican
 Name[fy]=Vaticaanstad
Index: l10n/vg/entry.desktop
===================================================================
--- l10n/vg/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/vg/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -16,7 +16,7 @@
 Name[es]=Islas Vírgenes Británicas
 Name[et]=Briti Neitsisaared
 Name[eu]=Britaniar Irla Birjinak
-Name[fa]=جزایر دست نخوردۀ بریتانیا
+Name[fa]=جزایر ویرجین انگلیس
 Name[fi]=Brittien Neitsytsaaret
 Name[fr]=Îles Vierges britanniques
 Name[fy]=Virgin Eilannen, Britse
Index: l10n/vi/entry.desktop
===================================================================
--- l10n/vi/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/vi/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -17,7 +17,7 @@
 Name[es]=Islas Vírgenes Americanas
 Name[et]=USA Neitsisaared
 Name[eu]=Estatu Batuar Irla Birjinak
-Name[fa]=جزایر دست نخورده ایالات متحده
+Name[fa]=جزایر ویرجین آمریکا
 Name[fi]=USA:n Neitsytsaaret
 Name[fr]=Îles Vierges américaines
 Name[fy]=Virgin Eilannen, U.S.
Index: l10n/gp/entry.desktop
===================================================================
--- l10n/gp/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/gp/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -12,7 +12,7 @@
 Name[eo]=Gvadelupo
 Name[es]=Guadalupe
 Name[eu]=Guadalupe
-Name[fa]=گوادالوپ
+Name[fa]=گوادلوپ
 Name[ga]=Guadalúip
 Name[gl]=Guadalupe
 Name[he]=גאודלופה
Index: l10n/wf/entry.desktop
===================================================================
--- l10n/wf/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/wf/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -18,7 +18,7 @@
 Name[es]=Wallis y Futuna
 Name[et]=Wallis ja Futuna
 Name[eu]=Wallis eta Futuna
-Name[fa]=جزایر والیس و فوتونا
+Name[fa]=والیس و فوتونا
 Name[fi]=Wallis ja Futuna
 Name[fr]=Wallis et Futuna
 Name[fy]=Wallis en Futuna
Index: l10n/hk/entry.desktop
===================================================================
--- l10n/hk/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/hk/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -11,7 +11,7 @@
 Name[el]=Χονγκ Κονγκ SAR (Κίνα)
 Name[et]=Hongkong
 Name[eu]=Hong Kong SAR(Txina)
-Name[fa]=منطقه صنعتی ویژه هنگ کنگ (چین)
+Name[fa]=منطقه صنعتی ویژه هنگ کنگ )چین(
 Name[fi]=Hong Kong SAR (Kiina)
 Name[fr]=Hong Kong SAR (Chine)
 Name[ga]=Hong Cong SAR(An tSín)
Index: l10n/pg/entry.desktop
===================================================================
--- l10n/pg/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/pg/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -16,7 +16,7 @@
 Name[es]=Papua Nueva Guinea
 Name[et]=Paapua Uus-Guinea
 Name[eu]=Papua Ginea Berria
-Name[fa]=پاپوا گینه نو
+Name[fa]=گینه نو پاپوا
 Name[fi]=Papua-Uusi-Guinea
 Name[fr]=Papouasie-Nouvelle-Guinée
 Name[fy]=Papua Nij Guinea
Index: l10n/pm/entry.desktop
===================================================================
--- l10n/pm/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/pm/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -18,7 +18,7 @@
 Name[es]=Saint Pierre y Miquelon
 Name[et]=Saint Pierre ja Miquelon
 Name[eu]=Saint Pierre eta Miquelon
-Name[fa]=سنت پییر و میکولون
+Name[fa]=سن پیر و میکلون
 Name[fi]=Saint-Pierre ja Miquelon
 Name[fr]=Saint-Pierre-et-Miquelon
 Name[fy]=Saint Pierre en Miquelon
Index: l10n/pn/entry.desktop
===================================================================
--- l10n/pn/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/pn/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -8,7 +8,7 @@
 Name[cy]=Ynys Pitcairn
 Name[el]=Πίτκαιρν
 Name[eo]=Pitkarna Insulo
-Name[fa]=پیتکیرن
+Name[fa]=پیتکرن
 Name[he]=פיטקרן
 Name[hi]=पिटकैर्न
 Name[ja]=英領ピトケアン諸島
Index: l10n/pw/entry.desktop
===================================================================
--- l10n/pw/entry.desktop	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ l10n/pw/entry.desktop	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -7,7 +7,7 @@
 Name[cy]=Palaw
 Name[el]=Παλάου
 Name[eo]=Palao
-Name[fa]=پالاو
+Name[fa]=پالائو
 Name[he]=פלאו
 Name[hi]=पलाऊ
 Name[is]=Palá
Index: kdm/backend/dm.c
===================================================================
--- kdm/backend/dm.c	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdm/backend/dm.c	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -689,7 +689,7 @@
 #ifdef XDMCP
 	if (di->status == remoteLogin) {
 		GSendStr( "" );
-		GSendStr( d->remoteHost );
+		GSendStr( di->remoteHost );
 	} else
 #endif
 	{
@@ -1060,7 +1060,7 @@
 {
 	int olderrno = errno;
 	char buf = (char)n;
-	Debug( "caught signal %d\n", n );
+	/* Debug( "caught signal %d\n", n ); this hangs in syslog() */
 	write( signalFds[1], &buf, 1 );
 #ifdef __EMX__
 	(void)Signal( n, SigHandler );
Index: kdm/backend/ctrl.c
===================================================================
--- kdm/backend/ctrl.c	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdm/backend/ctrl.c	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -824,6 +824,7 @@
 						goto bust;
 					}
 					if (ar[6]) {
+						free( args );
 					  exce:
 						fLog( d, fd, "bad", "excess argument(s)" );
 						goto bust;
Index: kdm/backend/socket.c
===================================================================
--- kdm/backend/socket.c	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdm/backend/socket.c	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -359,6 +359,7 @@
 		return;
 
 	ChangeMcastMembership( s, g, JOIN_MCAST_GROUP );
+	free( g );
 }
 
 /* Open or close listening sockets to match the current settings read in
Index: kdm/kfrontend/genkdmconf.c
===================================================================
--- kdm/kfrontend/genkdmconf.c	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdm/kfrontend/genkdmconf.c	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -1684,7 +1684,7 @@
 		linkfile( ce );
 	else {
 		tmpf = locate( "mktemp" ) ?
-		           "`mktemp`" :
+		           "`mktemp /tmp/xsess-env-XXXXXX`" :
 		           locate( "tempfile" ) ?
 		               "`tempfile`" :
 		               "$HOME/.xsession-env-$DISPLAY";
Index: kdm/kfrontend/kgverify.cpp
===================================================================
--- kdm/kfrontend/kgverify.cpp	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ kdm/kfrontend/kgverify.cpp	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -273,7 +273,7 @@
 {
 	authTok = (func == KGreeterPlugin::ChAuthTok);
 	cont = false;
-	if (func == KGreeterPlugin::Authenticate) {
+	if (func == KGreeterPlugin::Authenticate && ctx == KGreeterPlugin::Login) {
 		if (scheduleAutoLogin( true )) {
 			if (!_autoLoginAgain)
 				_autoLoginDelay = 0, timeable = false;
Index: ksysguard/example/eventsrc
===================================================================
--- ksysguard/example/eventsrc	(.../tags/KDE/3.5.5/kdebase)	(revision 594925)
+++ ksysguard/example/eventsrc	(.../branches/KDE/3.5/kdebase)	(revision 594925)
@@ -225,7 +225,7 @@
 Comment[es]=El sensor excedió el límite crítico
 Comment[et]=Sensor ületas kriitilise piiri
 Comment[eu]=Sensoreak bere muga kritikoa gainditu du
-Comment[fa]=حد بحرانی فراتر از حسگر
+Comment[fa]=حسگر فراتر از حد بحرانی شد
 Comment[fi]=Tunnistin ylitti kriittisen rajan
 Comment[fr]=Le capteur a dépassé sa limite critique
 Comment[fy]=Sensor limyt giet oer de kritike grins

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


