--- kdebase-3.0.3/kcontrol/konsole/sessioneditor.cpp--	2002-08-24 08:44:23.000000000 +0200
+++ kdebase-3.0.3/kcontrol/konsole/sessioneditor.cpp	2002-08-24 08:45:47.000000000 +0200
@@ -201,7 +201,7 @@ void SessionEditor::loadAllSession()
       sessionList->insertItem(i18n("Unnamed"),i);
     else
       sessionList->insertItem(sesname,i);
-
+    delete co;
     i++;
   }
 }
@@ -210,7 +210,7 @@ void SessionEditor::readSession(int num)
 {
     int i,counter;
     QString str;
-    KSimpleConfig* co;
+    KSimpleConfig* co=0L;
 
     if(sesMod) {
         disconnect(sessionList, SIGNAL(highlighted(int)), this, SLOT(readSession(int)));
@@ -264,6 +264,7 @@ void SessionEditor::readSession(int num)
             counter++;
         }
         schemaCombo->setCurrentItem(i);
+	delete co;
     }
     sesMod=false;
     oldSession=num;
@@ -327,6 +328,7 @@ void SessionEditor::saveCurrent()
   readSession(0);
   sessionList->setCurrentItem(0);
   removeButton->setEnabled(sessionList->count()>1);
+  delete co;
 }
 
 void SessionEditor::removeCurrent()
