--- kdebase-trinity-3.5.13.2/kdm/kfrontend/sessions/Makefile.am~	2012-09-16 15:52:03.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kdm/kfrontend/sessions/Makefile.am	2014-05-27 14:47:14.574323666 +0300
@@ -1,49 +1,3 @@
-sessionsdir = $(kde_datadir)/kdm/sessions
-sessions_DATA = \
-    admin.desktop tde.desktop gnome.desktop \
-    9wm.desktop \
-    aewm++.desktop \
-    aewm.desktop \
-    afterstep.desktop \
-    amaterus.desktop \
-    amiwm.desktop \
-    asclassic.desktop \
-    blackbox.desktop \
-    cde.desktop \
-    ctwm.desktop \
-    cwwm.desktop \
-    enlightenment.desktop \
-    evilwm.desktop \
-    fluxbox.desktop \
-    flwm.desktop \
-    fvwm.desktop \
-    fvwm95.desktop \
-    golem.desktop \
-    icewm.desktop \
-    ion.desktop \
-    larswm.desktop \
-    lwm.desktop \
-    matchbox.desktop \
-    metacity.desktop \
-    mwm.desktop \
-    olvwm.desktop \
-    olwm.desktop \
-    openbox.desktop \
-    oroborus.desktop \
-    phluid.desktop \
-    pwm.desktop \
-    qvwm.desktop \
-    ratpoison.desktop \
-    sapphire.desktop \
-    sawfish.desktop \
-    twm.desktop \
-    ude.desktop \
-    vtwm.desktop \
-    w9wm.desktop \
-    waimea.desktop \
-    wm2.desktop \
-    wmaker.desktop \
-    xfce.desktop \
-    xfce4.desktop
-
+sessionsdir = /usr/share/xsessions
+sessions_DATA = kde.desktop
 EXTRA_DIST = $(sessions_DATA)
--- kdebase-trinity-3.5.13.2/kdm/kfrontend/sessions/CMakeLists.txt~	2012-09-16 15:52:03.000000000 +0300
+++ kdebase-trinity-3.5.13.2/kdm/kfrontend/sessions/CMakeLists.txt	2014-06-04 11:22:28.208119436 +0300
@@ -12,18 +12,5 @@
 configure_file( tde.desktop.cmake tde.desktop @ONLY )
 
 install( FILES
-    admin.desktop ${CMAKE_CURRENT_BINARY_DIR}/tde.desktop
-    gnome.desktop 9wm.desktop aewm++.desktop aewm.desktop
-    afterstep.desktop amaterus.desktop amiwm.desktop
-    asclassic.desktop blackbox.desktop cde.desktop
-    ctwm.desktop cwwm.desktop enlightenment.desktop
-    evilwm.desktop fluxbox.desktop flwm.desktop fvwm.desktop
-    fvwm95.desktop golem.desktop icewm.desktop ion.desktop
-    larswm.desktop lwm.desktop matchbox.desktop metacity.desktop
-    mwm.desktop olvwm.desktop olwm.desktop openbox.desktop
-    oroborus.desktop phluid.desktop pwm.desktop qvwm.desktop
-    ratpoison.desktop sapphire.desktop sawfish.desktop
-    twm.desktop ude.desktop vtwm.desktop w9wm.desktop
-    waimea.desktop wm2.desktop wmaker.desktop xfce.desktop
-    xfce4.desktop
-  DESTINATION ${DATA_INSTALL_DIR}/kdm/sessions )
+  ${CMAKE_CURRENT_BINARY_DIR}/tde.desktop RENAME kde.desktop
+  DESTINATION ${DATA_INSTALL_DIR}/xsessions )
