--- kdebase-3.0.3/konsole/konsole/schema.cpp--	2002-09-02 08:16:24.000000000 +0200
+++ kdebase-3.0.3/konsole/konsole/schema.cpp	2002-09-02 08:16:38.000000000 +0200
@@ -150,6 +150,8 @@ ColorSchema::ColorSchema(const QString& 
   m_numb = serial++;
 }
 
+
+
 ColorSchema::ColorSchema()
 :m_fileRead(false)
 ,fPath(QString::null)
@@ -160,6 +162,11 @@ ColorSchema::ColorSchema()
   m_numb = 0;
 }
 
+ColorSchema::~ColorSchema()
+{
+    delete lastRead;
+}
+
 ColorSchema::ColorSchema(KConfig& c)
 :m_fileRead(false)
 ,fPath(QString::null)
--- kdebase-3.0.3/konsole/konsole/schema.h--	2002-09-02 08:15:25.000000000 +0200
+++ kdebase-3.0.3/konsole/konsole/schema.h	2002-09-02 08:16:45.000000000 +0200
@@ -68,7 +68,7 @@ public:
 	* with the same settings as the default schema is returned.
 	*/
 	ColorSchema(const QString& pathname);
-
+	~ColorSchema();
 	/**
 	* Construct a color schema from the given config file.
 	* (This is different from the constructor with a pathname
