--- kdebase-3.1.1/kcontrol/screensaver/scrnsave.cpp~	2005-01-14 16:55:19.000000000 +0100
+++ kdebase-3.1.1/kcontrol/screensaver/scrnsave.cpp	2005-01-14 17:00:15.238121600 +0100
@@ -84,10 +84,9 @@
     mMonitor = 0;
     mTesting = false;
 
-    // Add non-KDE path
     KGlobal::dirs()->addResourceType("scrsav",
-                                     KGlobal::dirs()->kde_default("apps") +
-                                     "apps/ScreenSavers/");
+                                     KGlobal::dirs()->kde_default("data") +
+                                     "kscreensaver/");
 
     setQuickHelp( i18n("<h1>Screen Saver</h1> This module allows you to enable and"
        " configure a screen saver. Note that you can enable a screen saver"
@@ -101,19 +100,6 @@
 
     setButtons( KCModule::Help | KCModule::Default | KCModule::Apply );
 
-    // Add KDE specific screensaver path
-    QString relPath="System/ScreenSavers/";
-    KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers" );
-    if (servGroup)
-    {
-      relPath=servGroup->relPath();
-      kdDebug() << "relPath=" << relPath << endl;
-    }
-
-    KGlobal::dirs()->addResourceType("scrsav",
-                                     KGlobal::dirs()->kde_default("apps") +
-                                     relPath);
-
     readSettings( false );
 
     mSetupProc = new KProcess;
--- kdebase-3.1.1/kscreensaver/Makefile.am~	Sun Oct 27 00:07:04 2002
+++ kdebase-3.1.1/kscreensaver/Makefile.am	Tue Apr  1 00:28:59 2003
@@ -23,6 +23,6 @@
 ########## Meta objects ##########
 
 desktop_DATA = KRandom.desktop KBlankscreen.desktop
-desktopdir = $(kde_appsdir)/System/ScreenSavers
+desktopdir = $(kde_datadir)/kscreensaver
 
 EXTRA_DIST = $(desktop_DATA)
--- kdebase-030502/kdesktop/lock/lockprocess.cc~	2003-05-02 00:36:02.000000000 +0200
+++ kdebase-030502/kdesktop/lock/lockprocess.cc	2003-05-02 15:41:23.000000000 +0200
@@ -91,22 +91,9 @@
     XSelectInput( qt_xdisplay(), qt_xrootwin(),
         SubstructureNotifyMask | rootAttr.your_event_mask );
 
-    // Add non-KDE path
     KGlobal::dirs()->addResourceType("scrsav",
-                                    KGlobal::dirs()->kde_default("apps") +
-                                    "System/ScreenSavers/");
-
-    // Add KDE specific screensaver path
-    QString relPath="System/ScreenSavers/";
-    KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers");
-    if (servGroup)
-    {
-      relPath=servGroup->relPath();
-      kdDebug(1204) << "relPath=" << relPath << endl;
-    }
-    KGlobal::dirs()->addResourceType("scrsav",
-                                     KGlobal::dirs()->kde_default("apps") +
-                                     relPath);
+                                     KGlobal::dirs()->kde_default("data") +
+                                     "kscreensaver/");
 
     mLockOnce = false;
 
