diff -urN kdebase-3.0beta2.orig/startkde kdebase-3.0beta2/startkde
--- kdebase-3.0beta2.orig/startkde	Sat Feb  9 02:59:15 2002
+++ kdebase-3.0beta2/startkde	Wed Feb 20 20:15:36 2002
@@ -20,6 +20,19 @@
 #   2) everything in $KDEDIR/share/autostart (kdesktop, kicker, etc.)
 #   3) the rest of the session.
 
+# Help message.
+if [ $# -eq 1 ] && [ $1 = "--help" ]; then
+    echo "This script starts KDE."
+    echo "Usage: startkde [window_manager]"
+    exit 1
+fi
+
+# Prevent an ever growing .ICEauthority file, since that will kill application
+# startup performance.
+rm -f $HOME/.ICEauthority*
+rm -f $HOME/.DCOPserver*
+rm -f $HOME/.dcop*
+
 # Set a left cursor instead of the standard X11 "X" cursor, since I've heard
 # from some users that they're confused and don't know what to do. This is
 # especially necessary on slow machines, where starting KDE takes one or two
@@ -39,70 +52,6 @@
 kdehome=$HOME/.kde
 test -n "$KDEHOME" && eval kdehome=$KDEHOME
 
-# Activate the kde font directories.
-#
-# There are 4 directories that may be used for supplying fonts for KDE.
-#
-# There are two system directories. These belong to the administrator.
-# There are two user directories, where the user may add her own fonts.
-#
-# The 'override' versions are for fonts that should come first in the list,
-# i.e. if you have a font in your 'override' directory, it will be used in
-# preference to any other.
-#
-# The preference order looks like this:
-# user override, system override, X, user, system
-#
-# Where X is the original font database that was set up before this script
-# runs.
-
-usr_odir=$kdehome/share/fonts/override
-usr_fdir=$kdehome/share/fonts
-if test -n "$KDEDIRS"; then
-  kdedirs_first=`echo $KDEDIRS|sed -e 's/:.*//'`
-  sys_odir=$kdedirs_first/share/fonts/override
-  sys_fdir=$kdedirs_first/share/fonts
-else
-  sys_odir=$KDEDIR/share/fonts/override
-  sys_fdir=$KDEDIR/share/fonts
-fi
-
-# We run mkfontdir on the user's font dirs (if we have permission) to pick
-# up any new fonts they may have installed. If mkfontdir fails, we still
-# add the user's dirs to the font path, as they might simply have been made
-# read-only by the administrator, for whatever reason.
-
-test -d $usr_odir && (mkfontdir $usr_odir ; xset +fp $usr_odir)
-test -d $sys_odir && xset +fp $sys_odir
-test -d $usr_fdir && (mkfontdir $usr_fdir ; xset fp+ $usr_fdir)
-test -d $sys_fdir && xset fp+ $sys_fdir
-
-#
-# Add any user-installed font directories to the X font path
-kde_fontsdir=$kdehome/share/fonts
-kde_fontpaths=$kde_fontsdir/fontpaths
- 
-if test -r $kde_fontpaths ; then
-    for fpath in `cat $kde_fontpaths | grep -v '#'` ; do
-        if test -s $fpath/fonts.dir ; then
-            xset fp+ $fpath
-        fi
-    done
-fi
-
-# Ask X11 to rebuild its font list.
-xset fp rehash
-
-#
-# Get Ghostscript to look into user's KDE fonts dir for additional Fontmap
-if test -n "$GS_LIB" ; then
-    GS_LIB=$kde_fontsdir:$GS_LIB
-    export GS_LIB
-else
-    GS_LIB=$kde_fontsdir
-    export GS_LIB
-fi
-
 # Link "tmp" resource to directory in /tmp
 # Creates a directory /tmp/kde-$USER and links $KDEHOME/tmp-$HOSTNAME to it.
 lnusertemp tmp >/dev/null
@@ -142,8 +91,11 @@
 # if no windowmanager is specified, ksmserver will ensure kwin is started.
 # [--restore] should be controlled by kdm
 # kwrapper is used to reduce startup time and memory usage
-kwrapper ksmserver --restore || \
-  xmessage -geometry 500x100 "Couldn't start ksmserver. Check your installation."
+if [ $# -eq 1 ]; then
+	kwrapper ksmserver --windowmanager $1
+else
+	kwrapper ksmserver --restore
+fi
 
 # Clean up
 kdeinit_shutdown
