--- kdebase-3.1.3/startkde.orig	Wed Jul 30 22:39:06 2003
+++ kdebase-3.1.3/startkde	Wed Jul 30 22:41:09 2003
@@ -49,24 +49,14 @@
 
 usr_odir=$HOME/.fonts/kde-override
 usr_fdir=$HOME/.fonts
-if test -n "$KDEDIRS"; then
-  kdedirs_first=`echo "$KDEDIRS"|sed -e 's/:.*//'`
-  sys_odir=$kdedirs_first/share/fonts/override
-  sys_fdir=$kdedirs_first/share/fonts
-else
-  sys_odir=$KDEDIR/share/fonts/override
-  sys_fdir=$KDEDIR/share/fonts
-fi
 
 # We run mkfontdir on the user's font dirs (if we have permission) to pick
 # up any new fonts they may have installed. If mkfontdir fails, we still
 # add the user's dirs to the font path, as they might simply have been made
 # read-only by the administrator, for whatever reason.
 
-test -d "$sys_odir" && xset +fp "$sys_odir"
 test -d "$usr_odir" && (mkfontdir "$usr_odir" ; xset +fp "$usr_odir")
 test -d "$usr_fdir" && (mkfontdir "$usr_fdir" ; xset fp+ "$usr_fdir")
-test -d "$sys_fdir" && xset fp+ "$sys_fdir"
 
 #
 # Add any user-installed font directories to the X font path
