diff -ruN kdebase-1.93.orig/applnk/Makefile.am kdebase-1.93/applnk/Makefile.am
--- kdebase-1.93.orig/applnk/Makefile.am	Thu Aug 17 10:01:24 2000
+++ kdebase-1.93/applnk/Makefile.am	Fri Aug 25 14:48:57 2000
@@ -8,60 +8,60 @@
 
 
 install-data-local:
-	$(mkinstalldirs) $(kde_appsdir)
-	$(INSTALL_DATA) $(srcdir)/.directory $(kde_appsdir)/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Applications
-	$(INSTALL_DATA) $(srcdir)/applications.desktop $(kde_appsdir)/Applications/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Games
-	$(INSTALL_DATA) $(srcdir)/games.desktop $(kde_appsdir)/Games/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Graphics
-	$(INSTALL_DATA) $(srcdir)/graphics.desktop $(kde_appsdir)/Graphics/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Multimedia
-	$(INSTALL_DATA) $(srcdir)/multimedia.desktop $(kde_appsdir)/Multimedia/.directory
-	$(mkinstalldirs) $(kde_appsdir)/System
-	$(INSTALL_DATA) $(srcdir)/system.desktop $(kde_appsdir)/System/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Internet
-	$(INSTALL_DATA) $(srcdir)/internet.desktop $(kde_appsdir)/Internet/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Utilities
-	$(INSTALL_DATA) $(srcdir)/utilities.desktop $(kde_appsdir)/Utilities/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings
-	$(INSTALL_DATA) $(srcdir)/settings.desktop $(kde_appsdir)/Settings/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)
+	$(INSTALL_DATA) $(srcdir)/.directory $(DESTDIR)$(kde_appsdir)/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Applications
+	$(INSTALL_DATA) $(srcdir)/applications.desktop $(DESTDIR)$(kde_appsdir)/Applications/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Games
+	$(INSTALL_DATA) $(srcdir)/games.desktop $(DESTDIR)$(kde_appsdir)/Games/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Graphics
+	$(INSTALL_DATA) $(srcdir)/graphics.desktop $(DESTDIR)$(kde_appsdir)/Graphics/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Multimedia
+	$(INSTALL_DATA) $(srcdir)/multimedia.desktop $(DESTDIR)$(kde_appsdir)/Multimedia/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/System
+	$(INSTALL_DATA) $(srcdir)/system.desktop $(DESTDIR)$(kde_appsdir)/System/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Internet
+	$(INSTALL_DATA) $(srcdir)/internet.desktop $(DESTDIR)$(kde_appsdir)/Internet/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Utilities
+	$(INSTALL_DATA) $(srcdir)/utilities.desktop $(DESTDIR)$(kde_appsdir)/Utilities/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings
+	$(INSTALL_DATA) $(srcdir)/settings.desktop $(DESTDIR)$(kde_appsdir)/Settings/.directory
 
-	$(mkinstalldirs) $(kde_appsdir)/System/ScreenSavers
-	$(INSTALL_DATA) $(srcdir)/system-screensavers.desktop $(kde_appsdir)/System/ScreenSavers/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/System/ScreenSavers
+	$(INSTALL_DATA) $(srcdir)/system-screensavers.desktop $(DESTDIR)$(kde_appsdir)/System/ScreenSavers/.directory
 
-	$(mkinstalldirs) $(kde_appsdir)/Settings/LookNFeel
-	$(INSTALL_DATA) $(srcdir)/settings-looknfeel.desktop $(kde_appsdir)/Settings/LookNFeel/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/Information
-	$(INSTALL_DATA) $(srcdir)/settings-information.desktop $(kde_appsdir)/Settings/Information/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/Sound
-	$(INSTALL_DATA) $(srcdir)/settings-sound.desktop $(kde_appsdir)/Settings/Sound/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/Peripherals
-	$(INSTALL_DATA) $(srcdir)/settings-peripherals.desktop $(kde_appsdir)/Settings/Peripherals/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel
+	$(INSTALL_DATA) $(srcdir)/settings-looknfeel.desktop $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/Information
+	$(INSTALL_DATA) $(srcdir)/settings-information.desktop $(DESTDIR)$(kde_appsdir)/Settings/Information/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/Sound
+	$(INSTALL_DATA) $(srcdir)/settings-sound.desktop $(DESTDIR)$(kde_appsdir)/Settings/Sound/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/Peripherals
+	$(INSTALL_DATA) $(srcdir)/settings-peripherals.desktop $(DESTDIR)$(kde_appsdir)/Settings/Peripherals/.directory
 
-	$(mkinstalldirs) $(kde_appsdir)/Settings/FileBrowsing
-	$(INSTALL_DATA) $(srcdir)/settings-filebrowsing.desktop $(kde_appsdir)/Settings/FileBrowsing/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/WebBrowsing
-	$(INSTALL_DATA) $(srcdir)/settings-webbrowsing.desktop $(kde_appsdir)/Settings/WebBrowsing/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/FileBrowsing
+	$(INSTALL_DATA) $(srcdir)/settings-filebrowsing.desktop $(DESTDIR)$(kde_appsdir)/Settings/FileBrowsing/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/WebBrowsing
+	$(INSTALL_DATA) $(srcdir)/settings-webbrowsing.desktop $(DESTDIR)$(kde_appsdir)/Settings/WebBrowsing/.directory
 
-	$(mkinstalldirs) $(kde_appsdir)/Settings/PowerControl
-	$(INSTALL_DATA) $(srcdir)/settings-power.desktop $(kde_appsdir)/Settings/PowerControl/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/Personalization
-	$(INSTALL_DATA) $(srcdir)/settings-person.desktop $(kde_appsdir)/Settings/Personalization/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/PowerControl
+	$(INSTALL_DATA) $(srcdir)/settings-power.desktop $(DESTDIR)$(kde_appsdir)/Settings/PowerControl/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/Personalization
+	$(INSTALL_DATA) $(srcdir)/settings-person.desktop $(DESTDIR)$(kde_appsdir)/Settings/Personalization/.directory
 
-	$(mkinstalldirs) $(kde_appsdir)/Settings/LookNFeel/Windows
-	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-windows.desktop $(kde_appsdir)/Settings/LookNFeel/Windows/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/LookNFeel/Desktop
-	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-desktop.desktop $(kde_appsdir)/Settings/LookNFeel/Desktop/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/LookNFeel/Themes
-	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-themes.desktop $(kde_appsdir)/Settings/LookNFeel/Themes/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/LookNFeel/Panel
-	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-panel.desktop $(kde_appsdir)/Settings/LookNFeel/Panel/.directory
-	$(mkinstalldirs) $(kde_appsdir)/Settings/Help
-	$(INSTALL_DATA) $(srcdir)/settings-help.desktop $(kde_appsdir)/Settings/Help/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Windows
+	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-windows.desktop $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Windows/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Desktop
+	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-desktop.desktop $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Desktop/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Themes
+	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-themes.desktop $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Themes/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Panel
+	$(INSTALL_DATA) $(srcdir)/settings-looknfeel-panel.desktop $(DESTDIR)$(kde_appsdir)/Settings/LookNFeel/Panel/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/Help
+	$(INSTALL_DATA) $(srcdir)/settings-help.desktop $(DESTDIR)$(kde_appsdir)/Settings/Help/.directory
 	
-	$(mkinstalldirs) $(kde_appsdir)/Settings/Network
-	$(INSTALL_DATA) $(srcdir)/settings-network.desktop $(kde_appsdir)/Settings/Network/.directory
+	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Settings/Network
+	$(INSTALL_DATA) $(srcdir)/settings-network.desktop $(DESTDIR)$(kde_appsdir)/Settings/Network/.directory
 
 
 uninstall-local:
