--- kdebase-3.0.3/kicker/ui/browser_mnu.h.vfolder	2002-06-24 07:56:46.000000000 -0400
+++ kdebase-3.0.3/kicker/ui/browser_mnu.h	2002-09-02 14:30:06.000000000 -0400
@@ -50,6 +50,7 @@
     void slotMimeCheck();
     void slotClearIfNeeded(const QString&);
     void slotClear();
+    void slotGnomeMenuChanged();
 
 protected:
     void mousePressEvent(QMouseEvent *);
@@ -63,6 +64,8 @@
     QMap<int, bool>    _mimemap;
     QTimer            *_mimecheckTimer;
     KDirWatch          _dirWatch;
+    KDirWatch          _gnomeDirWatch;
+    KDirWatch          _oldDirWatch;
     QPtrList<PanelBrowserMenu> _subMenus;
 
     int                _startid;
--- kdebase-3.0.3/kicker/ui/browser_mnu.cpp.vfolder	2002-08-05 10:56:20.000000000 -0400
+++ kdebase-3.0.3/kicker/ui/browser_mnu.cpp	2002-09-02 14:37:27.000000000 -0400
@@ -69,6 +69,22 @@
     connect( &_dirWatch, SIGNAL(deleted(const QString&)),
              this, SLOT(slotClear()) );
 
+    // Watch the GNOME 2.x destkop directory for changes we may need
+    connect( &_gnomeDirWatch, SIGNAL(dirty(const QString&)),
+             this, SLOT(slotGnomeMenuChanged()));
+    connect( &_gnomeDirWatch, SIGNAL(created(const QString&)),
+             this, SLOT(slotGnomeMenuChanged()));
+    connect( &_gnomeDirWatch, SIGNAL(deleted(const QString&)),
+             this, SLOT(slotGnomeMenuChanged()));
+
+    // Watch the old kde destkop directory
+    connect( &_oldDirWatch, SIGNAL(dirty(const QString&)),
+             this, SLOT(slotGnomeMenuChanged()));
+    connect( &_oldDirWatch, SIGNAL(created(const QString&)),
+             this, SLOT(slotGnomeMenuChanged()));
+    connect( &_oldDirWatch, SIGNAL(deleted(const QString&)),
+             this, SLOT(slotGnomeMenuChanged()));
+
     kdDebug() << "PanelBrowserMenu Constructor " << path << endl;
 }
 
@@ -105,6 +121,10 @@
     // start watching if not already done
     if (!_dirWatch.contains(path()))
         _dirWatch.addDir( path() );
+    if (!_gnomeDirWatch.contains("/usr/share/applications"))
+        _gnomeDirWatch.addDir("/usr/share/applications");
+    if (!_oldDirWatch.contains("/usr/share/applnk"))
+        _oldDirWatch.addDir("/usr/share/applnk");
 
     // setup icon map
     initIconMap();
@@ -477,6 +500,14 @@
     _subMenus.clear(); // deletes submenus
 }
 
+void PanelBrowserMenu::slotGnomeMenuChanged()
+{
+    KProcess *menu=new KProcess;
+    *menu << "desktop-create-kmenu";
+    connect(menu, SIGNAL(processExited(KProcess*)), this, SLOT(slotClear()));
+    menu->start();
+}
+
 void PanelBrowserMenu::initIconMap()
 {
     if(_icons) return;
