- set default word separators to catch urls.
glen, 2005-07-20

--- kdebase-3.4.1/konsole/konsole/TEWidget.cpp~	2005-05-23 15:14:58.000000000 +0300
+++ kdebase-3.4.1/konsole/konsole/TEWidget.cpp	2005-07-20 15:31:24.000000000 +0300
@@ -339,7 +339,7 @@
 ,preserve_line_breaks(true)
 ,column_selection_mode(false)
 ,scrollLoc(SCRNONE)
-,word_characters(":@-./_~")
+,word_characters(":@-./_~?=&%+#")
 ,m_bellMode(BELLSYSTEM)
 ,blinking(false)
 ,cursorBlinking(false)
--- kdebase-3.4.1/konsole/konsole/konsole.cpp~	2005-07-20 15:29:27.000000000 +0300
+++ kdebase-3.4.1/konsole/konsole/konsole.cpp	2005-07-20 15:31:18.000000000 +0300
@@ -1506,7 +1506,7 @@
      b_warnQuit=config->readBoolEntry( "WarnQuit", true );
      b_allowResize=config->readBoolEntry( "AllowResize", false);
      b_bidiEnabled = config->readBoolEntry("EnableBidi",false);
-     s_word_seps= config->readEntry("wordseps",":@-./_~");
+     s_word_seps= config->readEntry("wordseps",":@-./_~?=&%+#");
      b_framevis = config->readBoolEntry("has frame",true);
      QPtrList<TEWidget> tes = activeTEs();
      for (TEWidget *_te = tes.first(); _te; _te = tes.next()) {
--- kdebase-3.4.1/konsole/konsole/konsole_part.cpp~	2005-07-20 15:29:27.000000000 +0300
+++ kdebase-3.4.1/konsole/konsole/konsole_part.cpp	2005-07-20 15:31:36.000000000 +0300
@@ -557,7 +557,7 @@
   n_keytab=config->readNumEntry("keytab",0); // act. the keytab for this session
   n_scroll = QMIN(config->readUnsignedNumEntry("scrollbar",TEWidget::SCRRIGHT),2);
   m_histSize = config->readNumEntry("history",DEFAULT_HISTORY_SIZE);
-  s_word_seps= config->readEntry("wordseps",":@-./_~");
+  s_word_seps= config->readEntry("wordseps",":@-./_~?=&%+#");
 
   n_encoding = config->readNumEntry("encoding",0);
 
