--- kdebase-3.1/kicker/core/container_button.cpp.kicker	2002-11-20 17:40:57.000000000 -0500
+++ kdebase-3.1/kicker/core/container_button.cpp	2002-11-20 17:40:20.000000000 -0500
@@ -211,16 +211,22 @@
                                                 QPopupMenu* opMenu, QWidget* parent)
   : ButtonContainer(opMenu, parent)
 {
-    embedButton( new ServiceButton( desktopFile, this ) );
-    _actions = KPanelApplet::Preferences;
+    ServiceButton *b = new ServiceButton( desktopFile, this );
+    if (b->desktopExist()) {
+        _actions = KPanelApplet::Preferences;
+        embedButton(b);
+    } else delete b;
 }
 
 ServiceButtonContainer::ServiceButtonContainer( const KConfigGroup& config,
                                                 QPopupMenu* opMenu, QWidget* parent)
   : ButtonContainer(opMenu, parent)
 {
-    embedButton( new ServiceButton( config, this ) );
-    _actions = KPanelApplet::Preferences;
+    ServiceButton *b = new ServiceButton( config, this );
+    if (b->desktopExist()) {
+        _actions = KPanelApplet::Preferences;
+        embedButton(b);
+    } else delete b;
 }
 
 // URLButton container
--- kdebase-3.1/kicker/buttons/servicebutton.cpp.kicker	2002-11-20 17:41:24.000000000 -0500
+++ kdebase-3.1/kicker/buttons/servicebutton.cpp	2002-11-20 17:28:45.000000000 -0500
@@ -23,6 +23,7 @@
 
 #include <qtooltip.h>
 #include <qdragobject.h>
+#include <qfile.h>
 
 #include <kdesktopfile.h>
 #include <kpropertiesdialog.h>
@@ -53,6 +54,7 @@
   : PanelButton( parent, "ServiceButton" )
   , _desktopFile( desktopFile )
   , is_lmb_down(false)
+  , is_desktop_here(true)
 {
     initialize();
 }
@@ -61,6 +63,7 @@
   : PanelButton( parent, "ServiceButton" )
   , _desktopFile( config.readEntry("DesktopFile") )
   , is_lmb_down(false)
+  , is_desktop_here(true)
 {
     initialize();
 }
@@ -85,6 +88,10 @@
     } else {
         QToolTip::add( this, _desktopFile.readName() + " - " + _desktopFile.readComment() );
     }
+
+    // don't show button if desktop file not found
+	  is_desktop_here = QFile::exists( locate( "apps", _desktopFile.fileName() ) );
+
     setTitle( _desktopFile.readName() );
     setIcon( _desktopFile.readIcon() );
 }
--- kdebase-3.1.orig/kicker/buttons/servicebutton.h.orig	2002-03-07 20:16:47.000000000 -0500
+++ kdebase-3.1.orig/kicker/buttons/servicebutton.h	2002-11-20 17:28:45.000000000 -0500
@@ -38,6 +38,8 @@
 
     void saveConfig( KConfigGroup& config ) const;
 
+    bool desktopExist() const { return is_desktop_here;};
+
     virtual void properties();
 
 protected slots:
@@ -58,6 +60,7 @@
     KDesktopFile   _desktopFile;
     QPoint         last_lmb_press;
     bool           is_lmb_down;
+    bool           is_desktop_here;
 };
 
 #endif
