--- kdebase-030317/kcontrol/kdm/background.cpp~	Sun Mar 16 00:35:44 2003
+++ kdebase-030317/kcontrol/kdm/background.cpp	Tue Mar 18 19:31:12 2003
@@ -53,7 +53,7 @@
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
     config->setGroup( "X-*-Greeter" );
-    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg",KDE_CONFDIR "/kdm/backgroundrc" ) );
+    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg","/etc/X11/kdm/backgroundrc" ) );
     m_background = new BGDialog( this, m_simpleConf, false );
 
     connect(m_background, SIGNAL(changed(bool)), SIGNAL(changed(bool)));
@@ -105,7 +105,7 @@
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
-    m_background->defaults();
+    m_background->kdmdefaults();
     emit changed(true);
 }
 
-void KBackground::defaults()
+void KBackground::kdmdefaults()
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
--- kdebase-030512/kcontrol/kdm/kdm-appear.cpp~	2003-05-12 00:34:51.000000000 +0200
+++ kdebase-030512/kcontrol/kdm/kdm-appear.cpp	2003-05-12 10:47:10.000000000 +0200
@@ -379,7 +379,7 @@
 bool KDMAppearanceWidget::setLogo(QString logo)
 {
     QString flogo = logo.isEmpty() ?
-                    locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") ) :
+                    QString::fromLatin1("/etc/X11/kdm/pics/pldlogo.png") :
                     logo;
     QPixmap p(flogo);
     if (p.isNull())
@@ -396,7 +396,7 @@
 {
     KImageIO::registerFormats();
     QString fileName = KFileDialog::getOpenFileName(
-	locate("data", QString::fromLatin1("kdm/pics/") ),
+	QString::fromLatin1("/etc/X11/kdm/pics/"),
 	KImageIO::pattern());
     if (!fileName.isEmpty())
 	if (setLogo(fileName))
@@ -453,9 +453,7 @@
 
 	// we gotta check if it is a non-local file and make a tmp copy at the hd.
 	if(!url->isLocalFile()) {
-	    pixurl = "file:" +
-		     KGlobal::dirs()->resourceDirs("data").last() +
-		     "kdm/pics/" + url->fileName();
+	    pixurl = "file:/etc/X11/kdm/pics/" + url->fileName();
 	    KIO::NetAccess::copy(*url, pixurl);
 	    istmp = true;
 	} else {
@@ -510,10 +508,10 @@
   config->setGroup("X-*-Greeter");
 
   // Read the greeting string
-  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to %s at %n")));
+  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to PLD Linux at %n")));
 
   // Regular logo or clock
-  QString logoArea = config->readEntry("LogoArea", "Logo" );
+  QString logoArea = config->readEntry("LogoArea", "None" );
   if (logoArea == "Clock") {
     clockRadio->setChecked(true);
     slotAreaRadioClicked(KdmClock);
@@ -560,9 +558,9 @@
 
 void KDMAppearanceWidget::defaults()
 {
-  greetstr_lined->setText( i18n("Welcome to %s at %n") );
+  greetstr_lined->setText( i18n("Welcome to PLD Linux at %n") );
   logoRadio->setChecked( true );
-  slotAreaRadioClicked( KdmLogo );
+  slotAreaRadioClicked( KdmNone );
   posCenterRadio->setChecked( true );
   slotPosRadioClicked( 0 );
   setLogo( "" );
--- kdebase-030527/kcontrol/kdm/kdm-font.cpp~	2003-05-20 00:37:03.000000000 +0200
+++ kdebase-030527/kcontrol/kdm/kdm-font.cpp	2003-05-27 12:17:59.000000000 +0200
@@ -110,8 +110,8 @@
 
 void KDMFontWidget::set_def()
 {
-  stdfont = QFont("helvetica", 10);
-  failfont = QFont("helvetica", 10, QFont::Bold);
+  stdfont = QFont("helvetica", 11);
+  failfont = QFont("helvetica", 11, QFont::Bold);
   greetfont = QFont("charter", 20);
 }
 
--- kdebase-030418/kcontrol/kdm/kdm-sess.cpp~	2003-04-04 00:37:58.000000000 +0200
+++ kdebase-030418/kcontrol/kdm/kdm-sess.cpp	2003-04-19 19:26:08.000000000 +0200
@@ -364,7 +364,7 @@
   QString str;
 
   config->setGroup("X-:*-Core");
-  readSD(sdlcombo, "All");
+  readSD(sdlcombo, "Root");
 
   config->setGroup("X-*-Core");
   readSD(sdrcombo, "Root");
--- kdebase-3.1/kcontrol/kdm/kdm-users.cpp.fixed	Fri Nov 29 03:55:12 2002
+++ kdebase-3.1/kcontrol/kdm/kdm-users.cpp	Fri Nov 29 18:53:19 2002
@@ -66,7 +66,7 @@
 
     // We assume that $kde_datadir/kdm exists, but better check for pics/ and pics/users,
     // and create them if necessary.
-    m_userPixDir = KGlobal::dirs()->resourceDirs("data").last() + "kdm/pics/";
+    m_userPixDir = "/etc/X11/kdm/pics/";
     m_notFirst = false;
     QDir testDir( m_userPixDir );
     if ( !testDir.exists() && !testDir.mkdir( testDir.absPath() )) {
--- kdebase-3.1/kcontrol/kdm/main.cpp.orig	Sun Oct 27 00:05:07 2002
+++ kdebase-3.1/kcontrol/kdm/main.cpp	Thu Nov 21 02:51:19 2002
@@ -92,7 +92,7 @@
     usermap.insert( QString::fromLocal8Bit( ps->pw_name ), ps->pw_uid );
   endpwent();
 
-  config = new KSimpleConfig( QString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));
+  config = new KSimpleConfig( QString::fromLatin1( "/etc/X11/kdm/kdmrc" ));
 
   QVBoxLayout *top = new QVBoxLayout(this);
   tab = new QTabWidget(this);
--- kdebase-3.1/kdm/kfrontend/Makefile.am.orig	Sun Oct 27 00:05:49 2002
+++ kdebase-3.1/kdm/kfrontend/Makefile.am	Wed Nov 20 23:26:39 2002
@@ -30,8 +30,8 @@
 genkdmconf_LDADD = $(LIB_X11)
 
 install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
+	@$(mkinstalldirs) $(DESTDIR)/etc/X11/kdm
+	@./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase-3.1/kdm/kfrontend/Makefile.in.orig	Mon Nov 18 22:22:07 2002
+++ kdebase-3.1/kdm/kfrontend/Makefile.in	Wed Nov 20 23:47:04 2002
@@ -786,8 +786,8 @@
 
 
 install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
+	@$(mkinstalldirs) $(DESTDIR)/etc/X11/kdm
+	@./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase-030527/kdm/kfrontend/genkdmconf.c~	2003-05-19 00:41:26.000000000 +0200
+++ kdebase-030527/kdm/kfrontend/genkdmconf.c	2003-05-27 12:35:11.000000000 +0200
@@ -56,7 +56,7 @@
 
 #define as(ar) ((int)(sizeof(ar)/sizeof(ar[0])))
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF  "/etc/X11/kdm"
 
 #define RCVERMAJOR 2
 #define RCVERMINOR 0
@@ -757,7 +757,7 @@
 "Program=\n"
 "ReverseBlending=false\n"
 "UseSHM=false\n"
-"Wallpaper=default_blue.jpg\n"
+"Wallpaper=/etc/X11/kdm/pics/pldwallpaper.png\n"
 "WallpaperList=\n"
 "WallpaperMode=Scaled\n";
 
@@ -1823,18 +1823,20 @@
 "# this opens a security hole: a secured display lock can be circumvented\n"
 "# (unless you use KDE's built-in screen lock). Default is false\n" },
 { "AllowRootLogin",	0, 0, 
-"# Allow root logins? Default is true\n" },
+"# Allow root logins? Default is false\n" },
 { "AllowNullPasswd",	0, 0, 
-"# Allow to log in, when user has set an empty password? Default is true\n" },
+"# Allow to log in, when user has set an empty password? Default is false\n" },
 { "AllowShutdown",	0, 0, 
 "# Who is allowed to shut down the system. This applies both to the\n"
-"# greeter and to the command FiFo. Default is All\n"
+"# greeter and to the command FiFo.\n"
 "# \"None\" - no \"Shutdown...\" button is shown at all\n"
 "# \"Root\" - the root password must be entered to shut down\n"
-"# \"All\" - everybody can shut down the machine (Default)\n" },
+"# \"All\" - everybody can shut down the machine\n"
+"# Default value is \"Root\". For local displays we set this option to \"All\"\n"
+"# (see the \"[X-:*-Core]\" section below).\n" },
 { "AllowSdForceNow",	0, 0, 
 "# Who is allowed to abort all still running sessions when shutting down.\n"
-"# Same options as for AllowShutdown. Default is All\n" },
+"# Same options as for AllowShutdown. Default is Root\n" },
 { "DefaultSdMode",	0, 0, 
 "# The default choice for the shutdown condition/timing.\n"
 "# \"Schedule\" - shutdown after all sessions exit (possibly at once) (Default)\n"
@@ -1849,7 +1851,7 @@
 "# option!), i.e., KDM will behave exactly as before KDE 3.0.\n" }, 
 { "SessSaveFile",	0, 0, 
 "# Where (relatively to the user's home directory) to store the last\n"
-"# selected session. Default is .wmrc\n" },
+"# selected session. Default is .kdmrc\n" },
 { "ServerAttempts",	0, 0, 
 "# How often to try to run the X-server. Running includes executing it and\n"
 "# waiting for it to come up. Default is 1\n" },
@@ -1887,11 +1889,11 @@
 "# currently is quite greyish. Default is \"\"\n" },
 { "LogoArea",		0, upd_logoarea, 
 "# What should be shown righthand of the input lines:\n"
-"# \"Logo\" - the image specified by LogoPixmap (Default)\n"
+"# \"Logo\" - the image specified by LogoPixmap\n"
 "# \"Clock\" - a neat analog clock\n"
-"# \"None\" - nothing\n" },
+"# \"None\" - nothing (Default)\n" },
 { "LogoPixmap",		0, 0, 
-"# The image to show when LogoArea=Logo. Default is kdelogo.png\n" },
+"# The image to show when LogoArea=Logo. Default is pldlogo.png\n" },
 { "GreeterPosFixed",	0, 0, 
 "# Normally, the greeter is centered on the screen. Use this, if you want\n"
 "# it to appear elsewhere on the screen. Default is false\n" },
@@ -1912,13 +1914,13 @@
 "# - %r -> the operating system's version\n"
 "# - %m -> the machine (hardware) type\n"
 "# - %% -> a single %\n"
-"# Default is \"Welcome to %s at %n\"\n" },
+"# Default is \"Welcome to PLD Linux at %n\"\n" },
 { "GreetFont",		0, 0, 
 "# The font for the headline. Default is charter,20,bold\n" },
 { "StdFont",		0, 0, 
-"# The normal font used in the greeter. Default is helvetica,10\n" },
+"# The normal font used in the greeter. Default is helvetica,11\n" },
 { "FailFont",		0, 0, 
-"# The font used for the \"Login Failed\" message. Default is helvetica,10,bold\n" },
+"# The font used for the \"Login Failed\" message. Default is helvetica,11,bold\n" },
 { "AntiAliasing",	0, 0,
 "# Whether the fonts shown in the greeter should be antialiased. Default is false\n" },
 { "NumLock",		0, 0,
@@ -1949,7 +1951,7 @@
 "# If true, they are sorted alphabetically. Default is true\n" },
 { "FaceSource",		0, 0, 
 "# Specify, where the users' pictures should be taken from.\n"
-"# \"AdminOnly\" - from $KDEDIR/share/apps/kdm/pics/users/$USER.png (Default)\n"
+"# \"AdminOnly\" - from /etc/X11/kdm/pics/users/$USER.png (Default)\n"
 "# \"UserOnly\" - from the user's $HOME/.face\n"
 "# \"PreferAdmin\" - prefer $USER.png; fallback on $HOME/.face\n"
 "# \"PreferUser\" - ... and the other way round\n" },
@@ -2125,10 +2127,10 @@
 { "SystemShell",	"/bin/bash", 0 },
 { "UserAuthDir",	"", 0 },
 { "AutoReLogin",	"true", 0 },
-{ "AllowRootLogin",	"false", 1 },
-{ "AllowNullPasswd",	"false", 1 },
-{ "AllowShutdown",	"Root", 1 },
-{ "AllowSdForceNow",	"Root", 0 },
+{ "AllowRootLogin",	"true", 0 },
+{ "AllowNullPasswd",	"true", 0 },
+{ "AllowShutdown",	"All", 0 },
+{ "AllowSdForceNow",	"All", 0 },
 { "DefaultSdMode",	"ForceNow", 0}, 
 { "InteractiveSd",	"false", 0}, 
 { "SessSaveFile",	"", 0 },	/* XXX .dmrc */
@@ -2138,7 +2140,7 @@
 { "SessionTypes",	"default,kde,failsafe", 1 },
 { "GUIStyle",		"Windows", 0 },
 { "ColorScheme",	"Pumpkin", 0 },
-{ "LogoArea",		"None", 0 },
+{ "LogoArea",		"Logo", 0 },
 { "LogoPixmap",		"", 0 },
 { "GreeterPosFixed",	"true", 0 },
 { "GreeterPosX",	"200", 0 },
@@ -2171,8 +2173,8 @@
 { "ServerAttempts",	"", 0 },
 { "ServerTimeout",	"", 0 },
 { "AllowShutdown",	"All", 1 },
-{ "AllowRootLogin",	"true", 1 },
-{ "AllowNullPasswd",	"true", 1 },
+{ "AllowRootLogin",	"true", 0 },
+{ "AllowNullPasswd",	"true", 0 },
 { "NoPassEnable",	"true", 0 },
 { "NoPassUsers",	"fred,ethel", 0 },
 };
--- kdebase-030527/kdm/kfrontend/kdm_config.c~	2003-05-19 00:41:26.000000000 +0200
+++ kdebase-030527/kdm/kfrontend/kdm_config.c	2003-05-27 12:40:41.000000000 +0200
@@ -38,7 +38,7 @@
 #include <netdb.h>
 #include <grp.h>
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 
 
 /*
@@ -498,11 +498,11 @@
 { "AutoLoginPass",	C_autoPass,	(void *)PautoLoginX,	"" },
 { "AutoLoginSession",	C_autoString,	(void *)PautoLoginX,	"" },
 { "AutoReLogin",	C_autoReLogin | C_BOOL,	0,	"false" },
-{ "AllowNullPasswd",	C_allowNullPasswd | C_BOOL, 0,	"true" },
-{ "AllowRootLogin",	C_allowRootLogin | C_BOOL, 0,	"true" },
-{ "SessSaveFile",	C_sessSaveFile,		0,	".wmrc" },
-{ "AllowShutdown",	C_allowShutdown | C_ENUM, sd_who, "All" },
-{ "AllowSdForceNow",	C_allowNuke | C_ENUM, sd_who,	"All" },
+{ "AllowNullPasswd",	C_allowNullPasswd | C_BOOL, 0,	"false" },
+{ "AllowRootLogin",	C_allowRootLogin | C_BOOL, 0,	"false" },
+{ "SessSaveFile",	C_sessSaveFile,		0,	".kdmrc" },
+{ "AllowShutdown",	C_allowShutdown | C_ENUM, sd_who, "Root" },
+{ "AllowSdForceNow",	C_allowNuke | C_ENUM, sd_who,	"Root" },
 { "DefaultSdMode",	C_defSdMode | C_ENUM, sd_mode,	"Schedule" },
 { "InteractiveSd",	C_interactiveSd | C_BOOL, 0,	"true" },
 };
@@ -517,9 +517,9 @@
 { "GreeterPosX",	C_GreeterPosX,		0,	"0" },
 { "GreeterPosY",	C_GreeterPosY,		0,	"0" },
 { "GreeterScreen",	C_GreeterScreen,	0,	"0" },
-{ "StdFont",		C_StdFont,		0,	"helvetica,10,5,0,50,0" },
-{ "FailFont",		C_FailFont,		0,	"helvetica,10,5,0,75,0" },
-{ "GreetString",	C_GreetString,		0,	"Welcome to %s at %n" },
+{ "StdFont",		C_StdFont,		0,	"helvetica,11,5,0,50,0" },
+{ "FailFont",		C_FailFont,		0,	"helvetica,11,5,0,75,0" },
+{ "GreetString",	C_GreetString,		0,	"Welcome to PLD Linux at %n" },
 { "GreetFont",		C_GreetFont,		0,	"charter,20,5,0,50,0" },
 { "AntiAliasing",	C_AntiAliasing | C_BOOL,0,	"false" },
 { "NumLock",		C_NumLock | C_ENUM, numlock,	"Keep" },
--- kdebase-3.1/kdm/kfrontend/kdmconfig.cpp.orig	Mon Oct 28 10:55:24 2002
+++ kdebase-3.1/kdm/kfrontend/kdmconfig.cpp	Fri Nov 29 18:27:33 2002
@@ -153,7 +153,7 @@
 
     _logo = GetCfgQStr (C_LogoPixmap);
     if( _logo.isEmpty())
-	_logo = locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") );
+	_logo = QString::fromLatin1("/etc/X11/kdm/pics/pldlogo.png");
 
     _showUsers = GetCfgInt (C_ShowUsers);
     _users = GetCfgQStrList (C_SelectedUsers);
--- kdebase-030317/kdm/kfrontend/kgreeter.cpp~	Sat Mar 15 00:39:49 2003
+++ kdebase-030317/kdm/kfrontend/kgreeter.cpp	Tue Mar 18 11:02:15 2003
@@ -115,8 +115,7 @@
   , capslocked( -1 )
   , loginfailed( false )
 {
-    user_pic_dir = KGlobal::dirs()->resourceDirs( "data" ).last() +
-				      QString::fromLatin1("kdm/pics/users/");	/* XXX standardize */
+    user_pic_dir = QString::fromLatin1("/etc/X11/kdm/pics/users/");	/* XXX standardize */
 
     QGridLayout* main_grid = new QGridLayout( winFrame, 4, 2, 10 );
     QBoxLayout* hbox1 = new QHBoxLayout( 10 );
@@ -283,7 +282,7 @@
 
     UpdateLock();
 
-    stsfile = new KSimpleConfig( QString::fromLatin1(KDE_CONFDIR "/kdm/kdmsts") ); // XXX get from kdm_config
+    stsfile = new KSimpleConfig( QString::fromLatin1("/etc/X11/kdm/kdmsts") ); // XXX get from kdm_config
     stsfile->setGroup( "PrevUser" );
     enam = QString::fromLocal8Bit( dname );
     if (kdmcfg->_preselUser != PRESEL_PREV)
--- kdebase-3.1/kdm/kfrontend/pics/Makefile.am.orig	Sun Feb 20 17:34:48 2000
+++ kdebase-3.1/kdm/kfrontend/pics/Makefile.am	Fri Nov 29 18:38:41 2002
@@ -1,5 +1,5 @@
 
-picsdir = $(kde_datadir)/kdm/pics
+picsdir = /etc/X11/kdm/pics
 pics_DATA = kdelogo.png kdeflower.png
 
 usersdir = $(picsdir)/users
--- kdebase-3.1/kdm/kfrontend/pics/Makefile.in.orig	Mon Nov 25 04:46:12 2002
+++ kdebase-3.1/kdm/kfrontend/pics/Makefile.in	Fri Nov 29 18:38:16 2002
@@ -286,7 +286,7 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 
-picsdir = $(kde_datadir)/kdm/pics
+picsdir = /etc/X11/kdm/pics
 pics_DATA = kdelogo.png kdeflower.png
 
 usersdir = $(picsdir)/users
