--- kdebase-3.0.3/kate/app/katepluginmanager.cpp--	2002-08-25 15:37:02.000000000 +0200
+++ kdebase-3.0.3/kate/app/katepluginmanager.cpp	2002-08-25 15:37:33.000000000 +0200
@@ -76,6 +76,7 @@ void KatePluginManager::loadConfig ()
     if  (config->readBoolEntry(myPluginList.at(i)->libname, false))
       myPluginList.at(i)->load = true;
   }
+  delete config;
 }
 
 void KatePluginManager::writeConfig ()
@@ -89,6 +90,7 @@ void KatePluginManager::writeConfig ()
   }
 
   config->sync();
+  delete config;
 }
 
 
