--- kdebase/kdesktop/desktop.cc.arrange	Fri Feb  9 10:33:10 2001
+++ kdebase/kdesktop/desktop.cc	Wed Feb 14 12:27:25 2001
@@ -23,6 +23,7 @@
 #include <sys/stat.h>
 #include <time.h>
 #include <unistd.h>
+#include <iostream>
 #include <kcolordrag.h>
 
 #include <qkeycode.h>
@@ -30,6 +31,7 @@
 #include <qdragobject.h>
 #include <qclipboard.h>
 #include <qevent.h>
+#include <qfile.h>
 #include <qrect.h>
 #include <qdir.h>
 
@@ -95,7 +97,6 @@
 
     DCOPObject( "KDesktopIface" )
 {
-
   m_bAutoStart = auto_start;
   m_bWaitForKded = wait_for_kded;
   KGlobal::locale()->insertCatalogue("kdesktop");
@@ -125,6 +126,9 @@
            this, SLOT(slotClipboardDataChanged()) );
 
   setURL( desktopURL() ); // sets m_url and m_dotDirectoryPath
+  need_arrange=QFile::exists(m_dotDirectoryPath);
+  kdDebug() << "***BERO: Checked " << m_dotDirectoryPath << " -> need_arrange = " << need_arrange << endl;
+  
   assert( !m_url.isMalformed() );
 
   // Dont repaint on configuration changes during construction
@@ -302,7 +306,6 @@
    }
 
   connect(kapp, SIGNAL(appearanceChanged()), SLOT(slotConfigure()));
-
 }
 
 // -----------------------------------------------------------------------------
@@ -806,6 +809,12 @@
     }
 
     kdDebug() << "KDesktop::slotCompleted " << m_bNeedSave << " " << m_bNeedRepaint << endl;
+    kdDebug() << "***BERO: KDesktop::slotCompleted " << m_dotDirectoryPath << " -> need_arrange = " << need_arrange << endl;
+    if ( need_arrange ) {
+	need_arrange=false;
+	cerr << "***BERO: calling rearrangeIcons()" << endl;
+	rearrangeIcons();
+    }
     if ( m_bNeedSave )
         slotSaveIconPositions();
     if ( m_bNeedRepaint )
--- kdebase/kdesktop/desktop.h.arrange	Fri Feb  9 10:33:10 2001
+++ kdebase/kdesktop/desktop.h	Wed Feb 14 12:04:21 2001
@@ -249,6 +249,9 @@
   QString m_language;
 
   QStringList m_preview;
+
+private:
+  bool need_arrange;
 };
 
 #endif
