--- kdebase-3.0.3/kicker/extensions/childpanel/Makefile.in~	Sat Oct 12 15:47:17 2002
+++ kdebase-3.0.3/kicker/extensions/childpanel/Makefile.in	Sat Oct 12 15:47:17 2002
@@ -295,7 +295,7 @@
 #>+ 1
 childpanel_panelextension_la_SOURCES=childpanelextension.cpp  childpanelextension_skel.cpp
 #>- childpanel_panelextension_la_METASOURCES = AUTO
-childpanel_panelextension_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module 
+childpanel_panelextension_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
 childpanel_panelextension_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) ../../kicker.la
 
 noinst_HEADERS = childpanelextension.h
--- kdebase-3.0.3/kicker/extensions/childpanel/Makefile.am~	Sat Oct 12 15:47:18 2002
+++ kdebase-3.0.3/kicker/extensions/childpanel/Makefile.am	Sat Oct 12 15:47:18 2002
@@ -4,7 +4,7 @@
 
 childpanel_panelextension_la_SOURCES = childpanelextension.cpp childpanelextension.skel
 childpanel_panelextension_la_METASOURCES = AUTO
-childpanel_panelextension_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module 
+childpanel_panelextension_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
 childpanel_panelextension_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) ../../kicker.la
 
 noinst_HEADERS = childpanelextension.h
--- kdebase-3.0.3/kicker/extensions/kasbar/Makefile.in~	Sat Oct 12 15:50:36 2002
+++ kdebase-3.0.3/kicker/extensions/kasbar/Makefile.in	Sat Oct 12 15:50:36 2002
@@ -299,7 +299,7 @@
 
 
 #>- kasbar_panelextension_la_METASOURCES = AUTO
-kasbar_panelextension_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+kasbar_panelextension_la_LDFLAGS = $(all_libraries) -no-undefined -module -no-undefined
 kasbar_panelextension_la_LIBADD = $(LIB_KDEUI) ../../taskmanager/libtaskmanager.la
 
 noinst_HEADERS = kasbar.h kasitem.h kasbarextension.h \
--- kdebase-3.0.3/kicker/extensions/kasbar/Makefile.am~	Sat Oct 12 15:50:22 2002
+++ kdebase-3.0.3/kicker/extensions/kasbar/Makefile.am	Sat Oct 12 15:50:22 2002
@@ -9,7 +9,7 @@
 				kasgroupitem.cpp kasgrouppopup.cpp
 
 kasbar_panelextension_la_METASOURCES = AUTO
-kasbar_panelextension_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+kasbar_panelextension_la_LDFLAGS = $(all_libraries) -no-undefined -module -no-undefined
 kasbar_panelextension_la_LIBADD =  $(LIB_KDEUI) ../../taskmanager/libtaskmanager.la
 
 noinst_HEADERS = kasbar.h kasitem.h kasbarextension.h \
--- kdebase-3.0.3/kicker/applets/clock/Makefile.in~	Sat Oct 12 15:52:49 2002
+++ kdebase-3.0.3/kicker/applets/clock/Makefile.in	Sat Oct 12 15:52:49 2002
@@ -306,7 +306,7 @@
 
 EXTRA_DIST = $(lnk_DATA) $(pic_DATA)
 
-clock_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+clock_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 clock_panelapplet_la_LIBADD = $(LIB_KDECORE) $(LIB_KDEUI)
 subdir = kicker/applets/clock
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
--- kdebase-3.0.3/kicker/extensions/dockbar/Makefile.in~	Sat Oct 12 15:58:56 2002
+++ kdebase-3.0.3/kicker/extensions/dockbar/Makefile.in	Sat Oct 12 15:58:56 2002
@@ -293,7 +293,7 @@
 
 dockbar_panelextension_la_SOURCES = dockcontainer.cpp dockbarextension.cpp
 #>- dockbar_panelextension_la_METASOURCES = AUTO
-dockbar_panelextension_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+dockbar_panelextension_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 dockbar_panelextension_la_LIBADD = $(LIB_KDEUI)
 
 noinst_HEADERS = dockcontainer.h dockbarextension.h
--- kdebase-3.0.3/kicker/extensions/dockbar/Makefile.am~	Sat Oct 12 15:58:48 2002
+++ kdebase-3.0.3/kicker/extensions/dockbar/Makefile.am	Sat Oct 12 15:58:48 2002
@@ -4,7 +4,7 @@
 
 dockbar_panelextension_la_SOURCES = dockcontainer.cpp dockbarextension.cpp
 dockbar_panelextension_la_METASOURCES = AUTO
-dockbar_panelextension_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+dockbar_panelextension_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 dockbar_panelextension_la_LIBADD = $(LIB_KDEUI)
 
 noinst_HEADERS = dockcontainer.h dockbarextension.h
--- kdebase-3.0.3/kicker/extensions/taskbar/Makefile.in~	Sat Oct 12 16:00:07 2002
+++ kdebase-3.0.3/kicker/extensions/taskbar/Makefile.in	Sat Oct 12 16:00:07 2002
@@ -293,7 +293,7 @@
 
 taskbar_panelextension_la_SOURCES = taskbarextension.cpp
 #>- taskbar_panelextension_la_METASOURCES = AUTO
-taskbar_panelextension_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+taskbar_panelextension_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 taskbar_panelextension_la_LIBADD = $(LIB_KDEUI) ../../taskmanager/libtaskmanager.la ../../taskbar/libtaskbar.la
 
 noinst_HEADERS = taskbarextension.h
--- kdebase-3.0.3/kicker/applets/taskbar/Makefile.in~	Sat Oct 12 16:01:36 2002
+++ kdebase-3.0.3/kicker/applets/taskbar/Makefile.in	Sat Oct 12 16:01:36 2002
@@ -294,7 +294,7 @@
 taskbar_panelapplet_la_SOURCES = taskbarapplet.cpp
 
 #>- taskbar_panelapplet_la_METASOURCES = USE_AUTOMOC
-taskbar_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+taskbar_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 taskbar_panelapplet_la_LIBADD = $(LIB_KDEUI)  ../../taskbar/libtaskbar.la
 
 noinst_HEADERS = taskbarapplet.h
--- kdebase-3.0.3/kicker/applets/systemtray/Makefile.in~	Sat Oct 12 16:02:50 2002
+++ kdebase-3.0.3/kicker/applets/systemtray/Makefile.in	Sat Oct 12 16:02:50 2002
@@ -301,7 +301,7 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-systemtray_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+systemtray_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 systemtray_panelapplet_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA)
 subdir = kicker/applets/systemtray
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
--- kdebase-3.0.3/kicker/applets/swallow/Makefile.in~	Sat Oct 12 16:03:34 2002
+++ kdebase-3.0.3/kicker/applets/swallow/Makefile.in	Sat Oct 12 16:03:34 2002
@@ -291,7 +291,7 @@
 
 kde_module_LTLIBRARIES = swallow_panelapplet.la
 
-swallow_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+swallow_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 swallow_panelapplet_la_LIBADD = $(LIB_KDEUI) $(LIB_KFILE)
 
 #>- swallow_panelapplet_la_SOURCES = swallow.cpp prefwidgetbase.ui prefwidget.cpp
--- kdebase-3.0.3/kicker/applets/run/Makefile.in~	Sat Oct 12 16:04:20 2002
+++ kdebase-3.0.3/kicker/applets/run/Makefile.in	Sat Oct 12 16:04:20 2002
@@ -301,7 +301,7 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-run_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+run_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 run_panelapplet_la_LIBADD = $(LIB_KSYCOCA) $(LIB_KDEUI)
 subdir = kicker/applets/run
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
--- kdebase-3.0.3/kicker/applets/naughty/Makefile.in~	Sat Oct 12 16:05:00 2002
+++ kdebase-3.0.3/kicker/applets/naughty/Makefile.in	Sat Oct 12 16:05:00 2002
@@ -313,7 +313,7 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-naughty_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+naughty_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 naughty_panelapplet_la_LIBADD = $(LIB_KDEUI)
 subdir = kicker/applets/naughty
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
--- kdebase-3.0.3/kicker/applets/naughty/Makefile.am~	Sat Oct 12 16:04:53 2002
+++ kdebase-3.0.3/kicker/applets/naughty/Makefile.am	Sat Oct 12 16:04:53 2002
@@ -22,7 +22,7 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-naughty_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+naughty_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 naughty_panelapplet_la_LIBADD = $(LIB_KDEUI)
 
 messages: rc.cpp
--- kdebase-3.0.3/kicker/applets/minipager/Makefile.in~	Sat Oct 12 16:05:56 2002
+++ kdebase-3.0.3/kicker/applets/minipager/Makefile.in	Sat Oct 12 16:05:56 2002
@@ -303,7 +303,7 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-minipager_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+minipager_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module -no-undefined
 minipager_panelapplet_la_LIBADD = $(LIB_KDEUI)
 subdir = kicker/applets/minipager
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
--- kdebase-3.0.3/kicker/applets/launcher/Makefile.in~	Sat Oct 12 16:06:49 2002
+++ kdebase-3.0.3/kicker/applets/launcher/Makefile.in	Sat Oct 12 16:06:49 2002
@@ -301,7 +301,7 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-launcher_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module 
+launcher_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module 
 launcher_panelapplet_la_LIBADD = ../../kicker.la $(LIB_KSYCOCA) $(LIB_KDEUI) 
 subdir = kicker/applets/launcher
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
--- kdebase-3.0.3/kicker/applets/launcher/Makefile.am~	Sat Oct 12 16:06:42 2002
+++ kdebase-3.0.3/kicker/applets/launcher/Makefile.am	Sat Oct 12 16:06:42 2002
@@ -13,7 +13,7 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-launcher_panelapplet_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module 
+launcher_panelapplet_la_LDFLAGS = $(all_libraries) -avoid-version -module 
 launcher_panelapplet_la_LIBADD = ../../kicker.la $(LIB_KSYCOCA) $(LIB_KDEUI) 
 
 messages:
