--- kdebase-030317/kappfinder/kappfinder.desktop~	Mon Mar  3 00:43:32 2003
+++ kdebase-030317/kappfinder/kappfinder.desktop	Wed Mar 19 19:12:37 2003
@@ -83,4 +83,4 @@
 GenericName[zh_TW]=選單更新工具
 GenericName[zu]=Ithuluzi Lokulungisa ngokusha Imenu
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;System
+Categories=Qt;KDE;Settings
--- kdebase-030317/kmenuedit/kmenuedit.desktop~	Wed Jan 22 00:36:26 2003
+++ kdebase-030317/kmenuedit/kmenuedit.desktop	Wed Mar 19 19:13:50 2003
@@ -64,4 +64,4 @@
 Name[zu]=Umlungisi wemenu
 
 X-DCOP-ServiceType=Unique
-Categories=Qt;KDE;System
+Categories=Qt;KDE;Settings
--- kdebase-030317/konqueror/konquerorsu.desktop~	Fri Feb 28 00:41:58 2003
+++ kdebase-030317/konqueror/konquerorsu.desktop	Wed Mar 19 19:06:14 2003
@@ -56,4 +56,4 @@
 Name[zu]=Imenenja yamafayela - Indlela Yomsebenzisi Omkhulu
 
 
-Categories=Qt;KDE;System;X-KDE-More
+Categories=Qt;KDE;System;Administration
--- kdebase-030317/konsole/konsolesu.desktop~	Tue Mar 18 18:35:21 2003
+++ kdebase-030317/konsole/konsolesu.desktop	Wed Mar 19 19:11:09 2003
@@ -106,4 +106,4 @@
 GenericName[zh_TW]=終端機程式
 GenericName[zu]=Uhlelo lwemisebenzi lwangaohandle
 X-KDE-AuthorizeAction=shell_access
-Categories=Qt;KDE;System;TerminalEmulator;X-KDE-More
+Categories=Qt;KDE;System;TerminalEmulator;Administration
--- kdebase-030317/kpersonalizer/kpersonalizer.desktop~	Wed Jan 22 00:36:30 2003
+++ kdebase-030317/kpersonalizer/kpersonalizer.desktop	Wed Mar 19 19:15:10 2003
@@ -58,4 +58,4 @@
 X-KDE-autostart-after=kdesktop
 X-KDE-autostart-condition=kpersonalizerrc:General:FirstLogin:true
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;System
+Categories=Qt;KDE;Settings
