--- kdebase-3.0.3/kcontrol/background/bgrender.cc--	2002-08-24 10:39:59.000000000 +0200
+++ kdebase-3.0.3/kcontrol/background/bgrender.cc	2002-08-24 10:40:40.000000000 +0200
@@ -49,24 +49,6 @@ KBackgroundRenderer::KBackgroundRenderer
     : KBackgroundSettings(desk, config)
 {
     m_State = 0;
-
-    if (! config) {
-	int screen_number = 0;
-	if (qt_xdisplay())
-	    screen_number = DefaultScreen(qt_xdisplay());
-	QCString configname;
-	if (screen_number == 0)
-	    configname = "kdesktoprc";
-	else
-	    configname.sprintf("kdesktop-screen-%drc", screen_number);
-
-	m_pConfig = new KConfig(configname, false, true);
-        m_bDeleteConfig = true;
-    } else {
-	m_pConfig = config;
-        m_bDeleteConfig = false;
-    }
-
     m_pDirs = KGlobal::dirs();
     m_rSize = m_Size = QApplication::desktop()->size();
     m_pBackground = 0L; m_pImage = 0L; m_pPixmap = 0L;
@@ -84,8 +66,6 @@ KBackgroundRenderer::~KBackgroundRendere
 {
     cleanup();
     delete m_Tempfile; m_Tempfile = 0;
-    if( m_bDeleteConfig )
-        delete m_pConfig;
 }
 
 
--- kdebase-3.0.3/kcontrol/background/bgrender.h--	2002-08-24 10:40:02.000000000 +0200
+++ kdebase-3.0.3/kcontrol/background/bgrender.h	2002-08-24 10:40:57.000000000 +0200
@@ -95,8 +95,6 @@ private:
     QPixmap *m_pPixmap;
     QTimer *m_pTimer;
 
-    KConfig *m_pConfig;
-    bool m_bDeleteConfig;
     KStandardDirs *m_pDirs;
     KShellProcess *m_pProc;
 };
--- kdebase-3.0.3/kcontrol/background/bgsettings.cc--	2002-08-24 10:40:07.000000000 +0200
+++ kdebase-3.0.3/kcontrol/background/bgsettings.cc	2002-08-24 10:42:18.000000000 +0200
@@ -503,6 +503,9 @@ KBackgroundSettings::KBackgroundSettings
 
 KBackgroundSettings::~KBackgroundSettings()
 {
+  if( m_bDeleteConfig )
+    delete m_pConfig;
+
 }
 
 
--- kdebase-3.0.3/kcontrol/background/bgsettings.h--	2002-08-24 10:40:11.000000000 +0200
+++ kdebase-3.0.3/kcontrol/background/bgsettings.h	2002-08-24 10:41:52.000000000 +0200
@@ -255,6 +255,7 @@ private:
 
     KConfig *m_pConfig;
     KStandardDirs *m_pDirs;
+    bool m_bDeleteConfig;
 
 public:
     QMap<QString,int> m_BMMap;
