--- kdebase-050210/kicker/menuext/prefmenu/prefmenu.cpp.orig	2005-02-08 21:51:37.000000000 +0100
+++ kdebase-050210/kicker/menuext/prefmenu/prefmenu.cpp	2005-02-11 10:35:21.736082784 +0100
@@ -291,7 +291,7 @@
     }
 
     // We ask KSycoca to give us all services under Settings/
-    KServiceGroup::Ptr root = KServiceGroup::group(m_root.isEmpty() ? "Settings/" : m_root);
+    KServiceGroup::Ptr root = KServiceGroup::group(m_root.isEmpty() ? "KDE-Settings/" : m_root);
 
     if (!root || !root->isValid())
     {
