--- kdmrc.3.0.9	Fri Nov  1 23:48:32 2002
+++ kdmrc	Sat Nov  2 00:01:04 2002
@@ -52,7 +52,7 @@
 # Default is ":0 local@tty1 /usr/X11R6/bin/X vt7"
 # XXX i'm planning to absorb this file into kdmrc, but i'm not sure how to
 # do this best.
-Xservers=/usr/X11R6/share/config/kdm/Xservers
+Xservers=/etc/X11/kdm/Xservers
 # Where KDM should store its PID. Default is "" (don't store)
 PidFile=/var/run/kdm.pid
 # Whether KDM should lock the pid file to prevent having multiple KDM
@@ -81,12 +81,12 @@
 #Port=177
 # File with the private keys of X-terminals. Required for XDM authentication.
 # Default is ""
-#KeyFile=/usr/X11R6/share/config/kdm/kdmkeys
+#KeyFile=/etc/X11/kdm/kdmkeys
 # XDMCP access control file in the usual XDM-Xaccess format.
 # Default is /usr/X11R6/share/config/kdm/Xaccess
 # XXX i'm planning to absorb this file into kdmrc, but i'm not sure how to
 # do this best.
-#Xaccess=
+Xaccess=/etc/X11/kdm/Xaccess
 # Number of seconds to wait for display to respond after the user has
 # selected a host from the chooser. Default is 15.
 #ChoiceTimeout=10
@@ -99,7 +99,7 @@
 # The program which is invoked to dynamically generate replies to XDMCP
 # BroadcastQuery requests.
 # By default no program is invoked and "Willing to manage" is sent.
-Willing=/usr/X11R6/share/config/kdm/Xwilling
+Willing=/etc/X11/kdm/Xwilling
 
 [Shutdown]
 # The command to run to halt the system. Default is /sbin/halt
@@ -174,20 +174,20 @@
 # A program to run before the greeter is shown. You should start kdmdesktop
 # there. Also, xconsole can be started by this script.
 # Default is ""
-Setup=/usr/X11R6/share/config/kdm/Xsetup
+Setup=/etc/X11/kdm/Xsetup
 # A program to run before a user session starts. You should invoke sessreg
 # there and optionally change the ownership of the console, etc.
 # Default is ""
-Startup=/usr/X11R6/share/config/kdm/Xstartup
+Startup=/etc/X11/kdm/Xstartup
 # A program to run after a user session exits. You should invoke sessreg
 # there and optionally change the ownership of the console, etc.
 # Default is ""
-Reset=/usr/X11R6/share/config/kdm/Xreset
+Reset=/etc/X11/kdm/Xreset
 # The program which is run as the user which logs in. It is supposed to
 # interpret the session argument (see SessionTypes) and start an appropriate
 # session according to it.
 # Default is /usr/X11R6/bin/xterm -ls -T
-Session=/usr/X11R6/share/config/kdm/Xsession
+Session=/etc/X11/kdm/Xsession
 # The program to run if Session fails.
 # Default is /usr/X11R6/bin/xterm
 #FailsafeClient=
@@ -223,7 +223,7 @@
 AllowShutdown=Root
 # Who is allowed to abort all still running sessions when shutting down.
 # Same options as for AllowShutdown. Default is All
-#AllowSdForceNow=Root
+AllowSdForceNow=Root
 # The default choice for the shutdown condition/timing.
 # "Schedule" - shutdown after all sessions exit (possibly at once) (Default)
 # "TryNow" - shutdown, if no sessions are open, otherwise do nothing
@@ -238,14 +238,14 @@
 #InteractiveSd=false
 # Where (relatively to the user's home directory) to store the last
 # selected session. Default is .wmrc
-#SessSaveFile=
+SessSaveFile=.kdmrc
 
 # Greeter config for all displays
 [X-*-Greeter]
 # Session types the users can select. It is advisable to have "default" and
 # "failsafe" listed herein, which is also the default.
 # Note, that the meaning of this value is entirely up to your Session program.
-SessionTypes=default,kde,failsafe
+SessionTypes=default,kde,gnome,twm,failsafe
 # Widget Style of the greeter:
 # Default, Windows, Platinum, Motif, MotifPlus, CDE, SGI, and any styles you
 # may have installed, e.g., HighColor; Default is "Default"
@@ -276,6 +276,7 @@
 # - %% -> a single %
 # Default is "Welcome to %s at %n"
 #GreetString=K Desktop Environment (%n)
+GreetString=Welcome to PLD Linux at %n
 # The font for the headline. Default is charter,24,bold
 #GreetFont=charter,24,5,0,50,0
 # The normal font used in the greeter. Default is helvetica,12
@@ -339,7 +340,7 @@
 #UseBackground=false
 # The configuration file to be used by krootimage.
 # Default is /usr/X11R6/share/config/kdm/backgroundrc
-#BackgroundCfg=
+BackgroundCfg=/etc/X11/kdm/backgroundrc
 # Hold the X-server grabbed the whole time the greeter is visible. This
 # may be more secure, but it will disable any background and other
 # X-clients started from the Setup script. Default is false
@@ -357,9 +358,9 @@
 # See above
 AllowShutdown=All
 # See above
-AllowRootLogin=true
+AllowRootLogin=false
 # See above
-AllowNullPasswd=true
+AllowNullPasswd=false
 # Enable password-less logins on this display. USE WITH EXTREME CARE!
 # Default is false
 #NoPassEnable=true
