diff -urbNB kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/Makefile.am kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/Makefile.am
--- kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/Makefile.am	2005-10-10 17:03:45.000000000 +0200
+++ kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/Makefile.am	2005-11-13 21:03:50.979231000 +0100
@@ -74,6 +74,10 @@
 		rpmfind.desktop \
 		ruby_application_archive.desktop \
 		seek.desktop \
+		seesar.desktop \
+		seesar_film.desktop \
+		seesar_music.desktop \
+		seesar_video.desktop \
 		sourceforge.desktop \
 		technorati.desktop \
 		technoratitags.desktop \
diff -urbNB kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar.desktop kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar.desktop
--- kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar.desktop	1970-01-01 01:00:00.000000000 +0100
+++ kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar.desktop	2005-11-13 21:03:22.325440250 +0100
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Encoding=UTF-8
+Type=Service
+Name=Seesar Search (PW dormitory search engine)
+Name[pl]=Seesar Search (wyszukiwarka akademików PW)
+ServiceTypes=SearchProvider
+Keys=ss
+Query=http://exor.mikrus/cgi-bin/ss.cgi?q=\\{@}&tplik=0&eftep=1&rozmiar=1&kategoria=all
+Charset=iso8859-2
diff -urbNB kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_film.desktop kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_film.desktop
--- kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_film.desktop	1970-01-01 01:00:00.000000000 +0100
+++ kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_film.desktop	2005-11-13 21:03:22.325440250 +0100
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Encoding=UTF-8
+Type=Service
+Name=Seesar Search: movies (PW dormitory search engine)
+Name[pl]=Seesar Search: filmy (wyszukiwarka akademików PW)
+ServiceTypes=SearchProvider
+Keys=ssf
+Query=http://exor.mikrus/cgi-bin/ss.cgi?q=\\{@}&tplik=0&eftep=1&rozmiar=6&kategoria=filmy
+Charset=iso8859-2
diff -urbNB kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_music.desktop kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_music.desktop
--- kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_music.desktop	1970-01-01 01:00:00.000000000 +0100
+++ kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_music.desktop	2005-11-13 21:03:22.325440250 +0100
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Encoding=UTF-8
+Type=Service
+Name=Seesar Search: music (PW dormitory search engine)
+Name[pl]=Seesar Search: muzyka (wyszukiwarka akademików PW)
+ServiceTypes=SearchProvider
+Keys=ssm
+Query=http://exor.mikrus/cgi-bin/ss.cgi?q=\\{@}&tplik=0&eftep=1&rozmiar=1&kategoria=muzyka
+Charset=iso8859-2
diff -urbNB kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_video.desktop kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_video.desktop
--- kdebase-3.5.0.org/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_video.desktop	1970-01-01 01:00:00.000000000 +0100
+++ kdebase-3.5.0/kcontrol/ebrowsing/plugins/ikws/searchproviders/seesar_video.desktop	2005-11-13 21:03:22.325440250 +0100
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Encoding=UTF-8
+Type=Service
+Name=Seesar Search: videos (PW dormitory search engine)
+Name[pl]=Seesar Search: filmiki (wyszukiwarka akademików PW)
+ServiceTypes=SearchProvider
+Keys=ssv
+Query=http://exor.mikrus/cgi-bin/ss.cgi?q=\\{@}&tplik=0&eftep=1&rozmiar=1&kategoria=filmy
+Charset=iso8859-2

