diff -ur kdebase-3.0.4.orig/applnk/internet-terminal.desktop kdebase-3.0.4/applnk/internet-terminal.desktop
--- kdebase-3.0.4.orig/applnk/internet-terminal.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/internet-terminal.desktop	Fri Nov  1 01:53:23 2002
@@ -47,5 +47,5 @@
 Name[zh_CN]=终端应用程序
 Name[zh_TW]=終端機程式
 Name[zu]=Izicelo Zokuphela kwendlela
-Icon=konsole
+Icon=konsole.png
 Version=1.0
diff -ur kdebase-3.0.4.orig/applnk/settings-filebrowsing.desktop kdebase-3.0.4/applnk/settings-filebrowsing.desktop
--- kdebase-3.0.4.orig/applnk/settings-filebrowsing.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-filebrowsing.desktop	Fri Nov  1 01:59:25 2002
@@ -54,4 +54,4 @@
 Name[zh_CN]=文件浏览
 Name[zh_TW]=檔案瀏覽
 Name[zu]=Ukubheka Ifayela
-Icon=kfm
+Icon=kfm.png
diff -ur kdebase-3.0.4.orig/applnk/settings-help.desktop kdebase-3.0.4/applnk/settings-help.desktop
--- kdebase-3.0.4.orig/applnk/settings-help.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-help.desktop	Sun Nov  3 11:50:47 2002
@@ -56,4 +56,4 @@
 Name[zh_CN]=帮助
 Name[zh_TW]=求助
 Name[zu]=Usizo
-Icon=help
+Icon=help.png
diff -ur kdebase-3.0.4.orig/applnk/settings-information.desktop kdebase-3.0.4/applnk/settings-information.desktop
--- kdebase-3.0.4.orig/applnk/settings-information.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-information.desktop	Fri Nov  1 02:03:25 2002
@@ -55,4 +55,4 @@
 Name[zh_CN]=信息
 Name[zh_TW]=資訊
 Name[zu]=Ulwazi
-Icon=hwinfo
+Icon=hwinfo.png
diff -ur kdebase-3.0.4.orig/applnk/settings-looknfeel.desktop kdebase-3.0.4/applnk/settings-looknfeel.desktop
--- kdebase-3.0.4.orig/applnk/settings-looknfeel.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-looknfeel.desktop	Fri Nov  1 02:04:45 2002
@@ -57,4 +57,4 @@
 Name[zh_CN]=观感
 Name[zh_TW]=外觀感覺
 Name[zu]=Buka & Thinta
-Icon=looknfeel
+Icon=looknfeel.png
diff -ur kdebase-3.0.4.orig/applnk/settings-network.desktop kdebase-3.0.4/applnk/settings-network.desktop
--- kdebase-3.0.4.orig/applnk/settings-network.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-network.desktop	Sun Nov  3 11:49:54 2002
@@ -57,4 +57,4 @@
 Name[zh_CN]=网络
 Name[zh_TW]=網路
 Name[zu]=Umsebenzi wokuxhumanisa
-Icon=network
+Icon=network.png
diff -ur kdebase-3.0.4.orig/applnk/settings-peripherals.desktop kdebase-3.0.4/applnk/settings-peripherals.desktop
--- kdebase-3.0.4.orig/applnk/settings-peripherals.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-peripherals.desktop	Fri Nov  1 02:04:04 2002
@@ -57,4 +57,4 @@
 Name[zh_CN]=外设
 Name[zh_TW]=週邊設備
 Name[zu]=Olayini abazibophelayo
-Icon=input_devices_settings
+Icon=input_devices_settings.png
diff -ur kdebase-3.0.4.orig/applnk/settings-person.desktop kdebase-3.0.4/applnk/settings-person.desktop
--- kdebase-3.0.4.orig/applnk/settings-person.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-person.desktop	Fri Nov  1 02:02:52 2002
@@ -57,4 +57,4 @@
 Name[zh_CN]=个人化
 Name[zh_TW]=個人化
 Name[zu]=Ukufaka ubuntu
-Icon=personal
+Icon=personal.png
diff -ur kdebase-3.0.4.orig/applnk/settings-power.desktop kdebase-3.0.4/applnk/settings-power.desktop
--- kdebase-3.0.4.orig/applnk/settings-power.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-power.desktop	Fri Nov  1 02:04:03 2002
@@ -56,4 +56,4 @@
 Name[zh_CN]=电源控制
 Name[zh_TW]=電源控制
 Name[zu]=Umkhosi Wamandla
-Icon=energy
+Icon=energy.png
diff -ur kdebase-3.0.4.orig/applnk/settings-sound.desktop kdebase-3.0.4/applnk/settings-sound.desktop
--- kdebase-3.0.4.orig/applnk/settings-sound.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-sound.desktop	Fri Nov  1 02:02:52 2002
@@ -58,4 +58,4 @@
 Name[zh_CN]=音效
 Name[zh_TW]=音效
 Name[zu]=Umsindo
-Icon=multimedia
+Icon=multimedia.png
diff -ur kdebase-3.0.4.orig/applnk/settings-system.desktop kdebase-3.0.4/applnk/settings-system.desktop
--- kdebase-3.0.4.orig/applnk/settings-system.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-system.desktop	Fri Nov  1 02:03:45 2002
@@ -52,4 +52,4 @@
 Name[zh_CN]=系统
 Name[zh_TW]=系統
 Name[zu]=Indlela esetshenziswayo
-Icon=kcmsystem
+Icon=kcmsystem.png
diff -ur kdebase-3.0.4.orig/applnk/settings-webbrowsing.desktop kdebase-3.0.4/applnk/settings-webbrowsing.desktop
--- kdebase-3.0.4.orig/applnk/settings-webbrowsing.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/settings-webbrowsing.desktop	Fri Nov  1 01:59:18 2002
@@ -55,4 +55,4 @@
 Name[zh_CN]=Web 浏览
 Name[zh_TW]=Web 瀏覽
 Name[zu]=Ukubheka izincwadi kwi Web
-Icon=konqueror
+Icon=konqueror.png
diff -ur kdebase-3.0.4.orig/applnk/system-screensavers.desktop kdebase-3.0.4/applnk/system-screensavers.desktop
--- kdebase-3.0.4.orig/applnk/system-screensavers.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/system-screensavers.desktop	Sun Nov  3 11:54:12 2002
@@ -55,7 +55,7 @@
 Name[zh_CN]=屏幕保护程序
 Name[zh_TW]=螢幕保護程式
 Name[zu]=Abagcini Besikrini
-Icon=kscreensaver
+Icon=kscreensaver.png
 X-KDE-BaseGroup=screensavers
 
 Comment=Screen saver demos
diff -ur kdebase-3.0.4.orig/applnk/system-terminal.desktop kdebase-3.0.4/applnk/system-terminal.desktop
--- kdebase-3.0.4.orig/applnk/system-terminal.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/applnk/system-terminal.desktop	Fri Nov  1 01:52:55 2002
@@ -47,5 +47,5 @@
 Name[zh_CN]=终端应用程序
 Name[zh_TW]=終端機程式
 Name[zu]=Izicelo Zokuphela kwendlela
-Icon=konsole
+Icon=konsole.png
 Version=1.0
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-databases.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-databases.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-databases.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-databases.desktop	Sun Nov  3 13:33:01 2002
@@ -46,5 +46,5 @@
 Name[zh_CN]=数据库
 Name[zh_TW]=資料庫
 Name[zu]=Iziseko zedata
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-debian.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-debian.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-debian.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-debian.desktop	Sun Nov  3 12:32:09 2002
@@ -7,5 +7,5 @@
 Name[th]=เดเบียน
 Name[ven]=Debiene
 Name[xx]=xx
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-emulators.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-emulators.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-emulators.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-emulators.desktop	Sun Nov  3 13:32:57 2002
@@ -47,5 +47,5 @@
 Name[zh_CN]=模拟器
 Name[zh_TW]=模擬器
 Name[zu]=Abazami bokulinganayo
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-hamradio.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-hamradio.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-hamradio.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-hamradio.desktop	Sun Nov  3 12:32:06 2002
@@ -25,5 +25,5 @@
 Name[zh_CN]=业余无线电
 Name[zh_TW]=業餘無線電
 Name[zu]=Umsakazo weHam
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-help.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-help.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-help.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-help.desktop	Sun Nov  3 13:32:54 2002
@@ -49,5 +49,5 @@
 Name[zh_CN]=Debian 帮助
 Name[zh_TW]=Debian 求助
 Name[zu]=Usizo lwe Debian
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-lock.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-lock.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-lock.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-lock.desktop	Sun Nov  3 12:31:45 2002
@@ -45,5 +45,5 @@
 Name[zh_CN]=锁
 Name[zh_TW]=鎖住
 Name[zu]=Ukukhiya
-Icon=lock
+Icon=lock.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-screen.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-screen.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-screen.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-screen.desktop	Sun Nov  3 18:51:16 2002
@@ -47,5 +47,5 @@
 Name[zh_CN]=屏幕
 Name[zh_TW]=螢幕
 Name[zu]=Isikrini
-Icon=window_list
+Icon=window_list.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-shells.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-shells.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-shells.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-shells.desktop	Sun Nov  3 13:32:11 2002
@@ -39,5 +39,5 @@
 Name[xx]=xx
 Name[zh_TW]=命令列
 Name[zu]=Amagobolondo
-Icon=shellscript
+Icon=shellscript.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-technical.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-technical.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-technical.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-technical.desktop	Sun Nov  3 18:50:40 2002
@@ -43,5 +43,5 @@
 Name[zh_CN]=工艺
 Name[zh_TW]=工藝
 Name[zu]=Ezobuchwepheshe bonjiniyela
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-text.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-text.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-text.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-text.desktop	Sun Nov  3 18:50:36 2002
@@ -45,5 +45,5 @@
 Name[zh_CN]=文本
 Name[zh_TW]=文字
 Name[zu]=Okubhaliweyo
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-viewers.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-viewers.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-viewers.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-viewers.desktop	Sun Nov  3 18:50:33 2002
@@ -48,5 +48,5 @@
 Name[zh_CN]=查看器
 Name[zh_TW]=查看器
 Name[zu]=Ababonisi
-Icon=video
+Icon=video.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-window-managers.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-window-managers.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-window-managers.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-window-managers.desktop	Sun Nov  3 13:29:24 2002
@@ -47,5 +47,5 @@
 Name[zh_CN]=窗口管理器
 Name[zh_TW]=視窗管理程式
 Name[zu]=Abaphathi be Windi
-Icon=deb
+Icon=deb.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-xshells.desktop kdebase-3.0.4/debian/patches/kdebase4-data/d-xshells.desktop
--- kdebase-3.0.4.orig/debian/patches/kdebase4-data/d-xshells.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/debian/patches/kdebase4-data/d-xshells.desktop	Sun Nov  3 18:50:28 2002
@@ -37,5 +37,5 @@
 Name[xh]=Amaqokombe e X
 Name[xx]=xx
 Name[zu]=XAmagobolondo
-Icon=shellscript
+Icon=shellscript.png
 
diff -ur kdebase-3.0.4.orig/debian/patches/kdesktop/debian.desktop kdebase-3.0.4/debian/patches/kdesktop/debian.desktop
--- kdebase-3.0.4.orig/debian/patches/kdesktop/debian.desktop	Wed Apr 17 06:31:32 2002
+++ kdebase-3.0.4/debian/patches/kdesktop/debian.desktop	Sun Nov  3 13:29:21 2002
@@ -1,4 +1,4 @@
 [Desktop Entry]
-Icon=deb
+Icon=deb.png
 Type=Link
 URL=file:/etc/kde3/debian/debian.html
diff -ur kdebase-3.0.4.orig/kaddressbook/kaddressbook.desktop kdebase-3.0.4/kaddressbook/kaddressbook.desktop
--- kdebase-3.0.4.orig/kaddressbook/kaddressbook.desktop	Sat Sep 28 20:21:11 2002
+++ kdebase-3.0.4/kaddressbook/kaddressbook.desktop	Fri Nov  1 02:02:27 2002
@@ -70,7 +70,7 @@
 GenericName[zh_TW]=地址簿管理員
 GenericName[zu]=Umphathi we Kheli
 Exec=kaddressbook
-Icon=kaddressbook
+Icon=kaddressbook.png
 Type=Application
 DocPath=kaddressbook/index.html
 Terminal=0
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Development/designer.desktop kdebase-3.0.4/kappfinder/apps/Development/designer.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Development/designer.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Development/designer.desktop	Sun Nov  3 18:50:06 2002
@@ -78,7 +78,7 @@
 GenericName[zu]=Umdizanisis woxhumaniso olubhekeneyo
 X-KDE-StartupNotify=true
 MimeType=application/x-designer
-Icon=designer
+Icon=designer.png
 Terminal=false
 Type=Application
 
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Editors/emacs.desktop kdebase-3.0.4/kappfinder/apps/Editors/emacs.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Editors/emacs.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Editors/emacs.desktop	Sun Nov  3 14:02:28 2002
@@ -53,7 +53,7 @@
 BinaryPattern=
 MimeType=text/english;text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/x-c;text/x-c++;
 Exec=emacs %f
-Icon=emacs
+Icon=emacs.png
 TerminalOptions=
 Path=
 Type=Application
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Editors/gvim.desktop kdebase-3.0.4/kappfinder/apps/Editors/gvim.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Editors/gvim.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Editors/gvim.desktop	Sun Nov  3 14:02:16 2002
@@ -73,7 +73,7 @@
 GenericName[zu]=Umhleli Wokubhaliweyo 
 MimeType=text/english;text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/x-c;text/x-c++;
 Exec=gvim -f %f
-Icon=gvim
+Icon=gvim.png
 Type=Application
 Terminal=0
 X-KDE-StartupNotify=true
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Editors/lemacs.desktop kdebase-3.0.4/kappfinder/apps/Editors/lemacs.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Editors/lemacs.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Editors/lemacs.desktop	Sun Nov  3 13:28:20 2002
@@ -54,7 +54,7 @@
 BinaryPattern=
 MimeType=text/english;text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/x-c;text/x-c++;
 Exec=lemacs %f
-Icon=emacs
+Icon=emacs.png
 Path=
 Type=Application
 Terminal=0
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Editors/nedit.desktop kdebase-3.0.4/kappfinder/apps/Editors/nedit.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Editors/nedit.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Editors/nedit.desktop	Sun Nov  3 14:02:05 2002
@@ -57,7 +57,7 @@
 GenericName[zh_TW]=文字編輯器
 GenericName[zu]=Umhleli Wokubhaliweyo 
 Exec=nedit %f
-Icon=kedit
+Icon=kedit.png
 TerminalOptions=
 Path=
 Type=Application
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Editors/xedit.desktop kdebase-3.0.4/kappfinder/apps/Editors/xedit.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Editors/xedit.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Editors/xedit.desktop	Sun Nov  3 14:01:50 2002
@@ -84,7 +84,7 @@
 BinaryPattern=
 MimeType=text/english;text/plain;
 Exec=xedit %f
-Icon=xedit
+Icon=xedit.png
 TerminalOptions=
 Path=
 Type=Application
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Editors/xemacs.desktop kdebase-3.0.4/kappfinder/apps/Editors/xemacs.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Editors/xemacs.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Editors/xemacs.desktop	Sun Nov  3 13:28:09 2002
@@ -58,7 +58,7 @@
 BinaryPattern=
 MimeType=text/english;text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/x-c;text/x-c++;
 Exec=xemacs %f
-Icon=emacs
+Icon=emacs.png
 TerminalOptions=
 Path=
 Type=Application
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Graphics/gimp.desktop kdebase-3.0.4/kappfinder/apps/Graphics/gimp.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Graphics/gimp.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Graphics/gimp.desktop	Sun Nov  3 14:01:07 2002
@@ -10,7 +10,7 @@
 Name[wa]=Li GIMP
 Name[xx]=xx
 Exec=gimp %F
-Icon=gimp
+Icon=gimp.png
 Type=Application
 GenericName=Image Manipulation Program
 GenericName[af]=Beeld Manipulasie Program
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Graphics/xv.desktop kdebase-3.0.4/kappfinder/apps/Graphics/xv.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Graphics/xv.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Graphics/xv.desktop	Sun Nov  3 13:56:43 2002
@@ -52,7 +52,7 @@
 GenericName[zh_TW]=圖片瀏覽程式
 GenericName[zu]=Umboniseli Womfanekiso
 Exec=xv %F
-Icon=xv
+Icon=xv.png
 TerminalOptions=
 Path=
 Type=Application
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/coolmail.desktop kdebase-3.0.4/kappfinder/apps/Internet/coolmail.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/coolmail.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/coolmail.desktop	Sun Nov  3 13:18:15 2002
@@ -58,7 +58,7 @@
 GenericName[zh_TW]=郵件提示
 GenericName[zu]=Umposi Ohlomile
 Exec=coolmail
-Icon=email
+Icon=email.png
 TerminalOptions=
 Type=Application
 Terminal=0
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/ethereal.desktop kdebase-3.0.4/kappfinder/apps/Internet/ethereal.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/ethereal.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/ethereal.desktop	Sun Nov  3 13:55:57 2002
@@ -63,7 +63,7 @@
 Type=Application
 BinaryPattern=
 MimeType=
-Icon=xmag
+Icon=xmag.png
 Terminal=0
 TerminalOptions=
 Path=
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/licq.desktop kdebase-3.0.4/kappfinder/apps/Internet/licq.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/licq.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/licq.desktop	Sun Nov  3 13:18:01 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=licq
-Icon=licq
+Icon=licq.png
 Name=Licq
 Name[eo]=LICQ
 Name[xx]=xx
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/mozilla.desktop kdebase-3.0.4/kappfinder/apps/Internet/mozilla.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/mozilla.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/mozilla.desktop	Sun Nov  3 13:55:40 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=mozilla %u
-Icon=mozilla
+Icon=mozilla.png
 Type=Application
 Name=Mozilla 
 GenericName=Web Browser
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/netscape.desktop kdebase-3.0.4/kappfinder/apps/Internet/netscape.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/netscape.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/netscape.desktop	Sun Nov  3 13:13:49 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=netscape %u
-Icon=netscape
+Icon=netscape.png
 Type=Application
 Name=Netscape
 Name[ar]=نيتسكيب
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/netscape6.desktop kdebase-3.0.4/kappfinder/apps/Internet/netscape6.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/netscape6.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/netscape6.desktop	Sun Nov  3 13:55:26 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=netscape6 %u
-Icon=netscape
+Icon=netscape.png
 Type=Application
 Name=Netscape6
 Name[af]=Netscape-6
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/opera.desktop kdebase-3.0.4/kappfinder/apps/Internet/opera.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/opera.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/opera.desktop	Sun Nov  3 13:13:37 2002
@@ -51,6 +51,6 @@
 GenericName[zu]=Umkhangeli zincwadi we Web
 Exec=opera %u
 Terminal=false
-Icon=opera
+Icon=opera.png
 Type=Application
 MimeType=text/html
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Internet/xbiff.desktop kdebase-3.0.4/kappfinder/apps/Internet/xbiff.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Internet/xbiff.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/apps/Internet/xbiff.desktop	Sun Nov  3 13:55:11 2002
@@ -58,7 +58,7 @@
 GenericName[zh_CN]=邮件提示程序
 GenericName[zh_TW]=郵件提示
 GenericName[zu]=Umposi Ohlomile
-Icon=email
+Icon=email.png
 TerminalOptions=
 Type=Application
 Terminal=0
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Multimedia/aviplay.desktop kdebase-3.0.4/kappfinder/apps/Multimedia/aviplay.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Multimedia/aviplay.desktop	Sat Sep 28 20:21:24 2002
+++ kdebase-3.0.4/kappfinder/apps/Multimedia/aviplay.desktop	Sun Nov  3 13:54:43 2002
@@ -46,7 +46,7 @@
 GenericName[zh_TW]=AVI 視訊播放程式
 GenericName[zu]=Umdlali we Vidiyo ye AVI
 Exec=aviplay %f
-Icon=video
+Icon=video.png
 MimeType=video/mpeg;video/quicktime;video/x-flic;video/x-msvideo
 Name=Aviplay
 Name[eo]=AVI-ludilo
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Multimedia/gtv.desktop kdebase-3.0.4/kappfinder/apps/Multimedia/gtv.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Multimedia/gtv.desktop	Sat Sep 28 20:21:24 2002
+++ kdebase-3.0.4/kappfinder/apps/Multimedia/gtv.desktop	Sun Nov  3 13:12:23 2002
@@ -45,7 +45,7 @@
 GenericName[zh_CN]=MPEG 播放器
 GenericName[zh_TW]=MPEG 播放程式
 Exec=gtv
-Icon=video
+Icon=video.png
 MimeType=video/mpeg
 Name=GTV
 Name[af]=Gtv
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Multimedia/mplayer.desktop kdebase-3.0.4/kappfinder/apps/Multimedia/mplayer.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Multimedia/mplayer.desktop	Sat Sep 28 20:21:24 2002
+++ kdebase-3.0.4/kappfinder/apps/Multimedia/mplayer.desktop	Sun Nov  3 13:12:18 2002
@@ -58,7 +58,7 @@
 GenericName[zh_TW]=視像檔播放程式
 GenericName[zu]=Umdlali we Vidiyo
 Exec=mplayer %f
-Icon=video
+Icon=video.png
 MimeType=video/mpeg;video/quicktime;video/x-flic;video/x-msvideo
 Name=MPlayer
 Name[af]=M-speler
diff -ur kdebase-3.0.4.orig/kappfinder/apps/Multimedia/mtv.desktop kdebase-3.0.4/kappfinder/apps/Multimedia/mtv.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/Multimedia/mtv.desktop	Sat Sep 28 20:21:24 2002
+++ kdebase-3.0.4/kappfinder/apps/Multimedia/mtv.desktop	Sun Nov  3 13:54:35 2002
@@ -58,7 +58,7 @@
 GenericName[zh_TW]=視像檔播放程式
 GenericName[zu]=Umdlali we Vidiyo
 Exec=mtv
-Icon=video
+Icon=video.png
 Name=MpegTV
 Name[af]=Mpegtv
 Name[eo]=MPEG-TV-ludilo
diff -ur kdebase-3.0.4.orig/kappfinder/apps/System/sam.desktop kdebase-3.0.4/kappfinder/apps/System/sam.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/System/sam.desktop	Sat Sep 28 20:21:24 2002
+++ kdebase-3.0.4/kappfinder/apps/System/sam.desktop	Sun Nov  3 18:52:13 2002
@@ -58,7 +58,7 @@
 Name[zu]=Indlela yokusebenza ye SAM-Umphathi
 MimeType=
 Exec=sam
-Icon=log
+Icon=log.png
 TerminalOptions=
 Path=
 Type=Application
diff -ur kdebase-3.0.4.orig/kappfinder/apps/System/xosview.desktop kdebase-3.0.4/kappfinder/apps/System/xosview.desktop
--- kdebase-3.0.4.orig/kappfinder/apps/System/xosview.desktop	Sat Sep 28 20:21:24 2002
+++ kdebase-3.0.4/kappfinder/apps/System/xosview.desktop	Sun Nov  3 13:53:58 2002
@@ -74,7 +74,7 @@
 GenericName[zh_TW]=系統監視器
 GenericName[zu]=Umphendli wendlela esetshenziswayo
 Exec=xosview
-Icon=log
+Icon=log.png
 TerminalOptions=
 Path=
 Type=Application
diff -ur kdebase-3.0.4.orig/kappfinder/kappfinder.desktop kdebase-3.0.4/kappfinder/kappfinder.desktop
--- kdebase-3.0.4.orig/kappfinder/kappfinder.desktop	Sat Sep 28 20:21:23 2002
+++ kdebase-3.0.4/kappfinder/kappfinder.desktop	Fri Nov  1 01:59:57 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kappfinder
-Icon=kappfinder
+Icon=kappfinder.png
 Type=Application
 Terminal=0
 X-KDE-StartupNotify=true
diff -ur kdebase-3.0.4.orig/kate/data/kate.desktop kdebase-3.0.4/kate/data/kate.desktop
--- kdebase-3.0.4.orig/kate/data/kate.desktop	Sat Sep 28 20:21:31 2002
+++ kdebase-3.0.4/kate/data/kate.desktop	Sun Nov  3 12:17:39 2002
@@ -57,7 +57,7 @@
 MimeType=text/english;text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/x-c;text/x-c++;
 Exec=kate %U
 TerminalOptions=
-Icon=kate
+Icon=kate.png
 Path=
 DocPath=kate/index.html
 Type=Application
diff -ur kdebase-3.0.4.orig/kate/data/kwrite.desktop kdebase-3.0.4/kate/data/kwrite.desktop
--- kdebase-3.0.4.orig/kate/data/kwrite.desktop	Sat Sep 28 20:21:31 2002
+++ kdebase-3.0.4/kate/data/kwrite.desktop	Sun Nov  3 12:15:06 2002
@@ -60,7 +60,7 @@
 Exec=kwrite -caption "%c" %i %m %U
 X-KDE-StartupNotify=true
 TerminalOptions=
-Icon=kwrite
+Icon=kwrite.png
 Path=
 DocPath=kwrite/index.html
 Type=Application
diff -ur kdebase-3.0.4.orig/kcontrol/access/kcmaccess.desktop kdebase-3.0.4/kcontrol/access/kcmaccess.desktop
--- kdebase-3.0.4.orig/kcontrol/access/kcmaccess.desktop	Sat Sep 28 20:21:31 2002
+++ kdebase-3.0.4/kcontrol/access/kcmaccess.desktop	Sun Nov  3 12:14:54 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=access
+Icon=access.png
 Type=Application
 Exec=kcmshell kcmaccess
 DocPath=kcontrol/accessibility.html
diff -ur kdebase-3.0.4.orig/kcontrol/bell/bell.desktop kdebase-3.0.4/kcontrol/bell/bell.desktop
--- kdebase-3.0.4.orig/kcontrol/bell/bell.desktop	Sat Sep 28 20:21:56 2002
+++ kdebase-3.0.4/kcontrol/bell/bell.desktop	Sun Nov  3 12:18:39 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell bell
-Icon=bell
+Icon=bell.png
 Type=Application
 DocPath=kcontrol/bell.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/clock/clock.desktop kdebase-3.0.4/kcontrol/clock/clock.desktop
--- kdebase-3.0.4.orig/kcontrol/clock/clock.desktop	Sat Sep 28 20:21:56 2002
+++ kdebase-3.0.4/kcontrol/clock/clock.desktop	Sun Nov  3 12:14:25 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kdesu kcmshell clock
-Icon=date
+Icon=date.png
 DocPath=kcontrol/datetime.html
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/colors/colors.desktop kdebase-3.0.4/kcontrol/colors/colors.desktop
--- kdebase-3.0.4.orig/kcontrol/colors/colors.desktop	Sat Sep 28 20:21:56 2002
+++ kdebase-3.0.4/kcontrol/colors/colors.desktop	Sun Nov  3 12:19:13 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell colors
-Icon=colors
+Icon=colors.png
 Type=Application
 DocPath=kcontrol/color.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/css/kcmcss.desktop kdebase-3.0.4/kcontrol/css/kcmcss.desktop
--- kdebase-3.0.4.orig/kcontrol/css/kcmcss.desktop	Sat Sep 28 20:21:56 2002
+++ kdebase-3.0.4/kcontrol/css/kcmcss.desktop	Sun Nov  3 12:19:10 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=stylesheet
+Icon=stylesheet.png
 Type=Application
 Exec=kcmshell WebBrowsing/kcmcss
 DocPath=kcontrol/stylesheets.html
diff -ur kdebase-3.0.4.orig/kcontrol/ebrowsing/ebrowsing.desktop kdebase-3.0.4/kcontrol/ebrowsing/ebrowsing.desktop
--- kdebase-3.0.4.orig/kcontrol/ebrowsing/ebrowsing.desktop	Sat Sep 28 20:21:56 2002
+++ kdebase-3.0.4/kcontrol/ebrowsing/ebrowsing.desktop	Sun Nov  3 12:13:36 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 DocPath=kcontrol/ebrowse.html
-Icon=enhanced_browsing
+Icon=enhanced_browsing.png
 Exec=kcmshell ebrowsing
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/email/email.desktop kdebase-3.0.4/kcontrol/email/email.desktop
--- kdebase-3.0.4.orig/kcontrol/email/email.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/email/email.desktop	Sun Nov  3 12:19:25 2002
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Icon=email
+Icon=email.png
 Type=Application
 Exec=kcmshell email
 DocPath=kcontrol/e-mail.html
diff -ur kdebase-3.0.4.orig/kcontrol/energy/energy.desktop kdebase-3.0.4/kcontrol/energy/energy.desktop
--- kdebase-3.0.4.orig/kcontrol/energy/energy.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/energy/energy.desktop	Sun Nov  3 12:13:24 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell energy
-Icon=energy_star
+Icon=energy_star.png
 DocPath=kcontrol/energy.html
 Type=Application
 Terminal=0
diff -ur kdebase-3.0.4.orig/kcontrol/filetypes/filetypes.desktop kdebase-3.0.4/kcontrol/filetypes/filetypes.desktop
--- kdebase-3.0.4.orig/kcontrol/filetypes/filetypes.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/filetypes/filetypes.desktop	Sun Nov  3 12:21:04 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=filetypes
+Icon=filetypes.png
 Type=Application
 Exec=kcmshell filetypes
 DocPath=kcontrol/modules.html
diff -ur kdebase-3.0.4.orig/kcontrol/fonts/fonts.desktop kdebase-3.0.4/kcontrol/fonts/fonts.desktop
--- kdebase-3.0.4.orig/kcontrol/fonts/fonts.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/fonts/fonts.desktop	Sun Nov  3 12:10:38 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell fonts
-Icon=fonts
+Icon=fonts.png
 Type=Application
 DocPath=kcontrol/fonts.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/icons/icons.desktop kdebase-3.0.4/kcontrol/icons/icons.desktop
--- kdebase-3.0.4.orig/kcontrol/icons/icons.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/icons/icons.desktop	Sun Nov  3 12:25:21 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 DocPath=kcontrol/icons.html
-Icon=icons
+Icon=icons.png
 Exec=kcmshell icons
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/info/devices.desktop kdebase-3.0.4/kcontrol/info/devices.desktop
--- kdebase-3.0.4.orig/kcontrol/info/devices.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/devices.desktop	Sun Nov  3 12:25:10 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell devices
-Icon=kcmdevices
+Icon=kcmdevices.png
 Type=Application
 DocPath=kcontrol/devices.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/dma.desktop kdebase-3.0.4/kcontrol/info/dma.desktop
--- kdebase-3.0.4.orig/kcontrol/info/dma.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/dma.desktop	Sun Nov  3 12:04:54 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell dma
-Icon=kcmmemory
+Icon=kcmmemory.png
 Type=Application
 DocPath=kcontrol/dma.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/interrupts.desktop kdebase-3.0.4/kcontrol/info/interrupts.desktop
--- kdebase-3.0.4.orig/kcontrol/info/interrupts.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/interrupts.desktop	Sun Nov  3 12:25:03 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell interrupts
-Icon=kcmmemory
+Icon=kcmmemory.png
 Type=Application
 DocPath=kcontrol/interrupts.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/ioports.desktop kdebase-3.0.4/kcontrol/info/ioports.desktop
--- kdebase-3.0.4.orig/kcontrol/info/ioports.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/ioports.desktop	Sun Nov  3 12:04:50 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell ioports
-Icon=kcmmemory
+Icon=kcmmemory.png
 Type=Application
 DocPath=kcontrol/ioports.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/memory.desktop kdebase-3.0.4/kcontrol/info/memory.desktop
--- kdebase-3.0.4.orig/kcontrol/info/memory.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/memory.desktop	Sun Nov  3 12:24:58 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell memory
-Icon=kcmmemory
+Icon=kcmmemory.png
 Type=Application
 DocPath=kcontrol/memory.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/partitions.desktop kdebase-3.0.4/kcontrol/info/partitions.desktop
--- kdebase-3.0.4.orig/kcontrol/info/partitions.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/partitions.desktop	Sun Nov  3 12:05:03 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell partitions
-Icon=kcmpartitions
+Icon=kcmpartitions.png
 Type=Application
 DocPath=kcontrol/partitions.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/pci.desktop kdebase-3.0.4/kcontrol/info/pci.desktop
--- kdebase-3.0.4.orig/kcontrol/info/pci.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/pci.desktop	Sun Nov  3 12:24:52 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell pci
-Icon=kcmpci
+Icon=kcmpci.png
 Type=Application
 DocPath=kcontrol/pci.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/processor.desktop kdebase-3.0.4/kcontrol/info/processor.desktop
--- kdebase-3.0.4.orig/kcontrol/info/processor.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/processor.desktop	Sun Nov  3 12:05:12 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell processor
-Icon=kcmprocessor
+Icon=kcmprocessor.png
 Type=Application
 DocPath=kcontrol/processor.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/info/scsi.desktop kdebase-3.0.4/kcontrol/info/scsi.desktop
--- kdebase-3.0.4.orig/kcontrol/info/scsi.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/info/scsi.desktop	Sun Nov  3 12:24:42 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell scsi
-Icon=kcmscsi
+Icon=kcmscsi.png
 Type=Application
 DocPath=kcontrol/scsi.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/input/mouse.desktop kdebase-3.0.4/kcontrol/input/mouse.desktop
--- kdebase-3.0.4.orig/kcontrol/input/mouse.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/input/mouse.desktop	Sun Nov  3 12:10:00 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell mouse
-Icon=mouse
+Icon=mouse.png
 Type=Application
 DocPath=kcontrol/mouse.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/kcontrol/KControl.desktop kdebase-3.0.4/kcontrol/kcontrol/KControl.desktop
--- kdebase-3.0.4.orig/kcontrol/kcontrol/KControl.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/kcontrol/KControl.desktop	Sun Nov  3 12:23:53 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcontrol -caption "%c" %i %m
-Icon=kcontrol
+Icon=kcontrol.png
 Type=Application
 DocPath=kcontrol/index.html
 X-KDE-StartupNotify=true
diff -ur kdebase-3.0.4.orig/kcontrol/keys/keys.desktop kdebase-3.0.4/kcontrol/keys/keys.desktop
--- kdebase-3.0.4.orig/kcontrol/keys/keys.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/keys/keys.desktop	Sun Nov  3 12:23:45 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell keys
-Icon=key_bindings
+Icon=key_bindings.png
 Type=Application
 DocPath=kcontrol/key-bindings.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/kfontinst/kfontinst/kcmfontinst.desktop kdebase-3.0.4/kcontrol/kfontinst/kfontinst/kcmfontinst.desktop
--- kdebase-3.0.4.orig/kcontrol/kfontinst/kfontinst/kcmfontinst.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/kfontinst/kfontinst/kcmfontinst.desktop	Sun Nov  3 12:57:18 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell kcmfontinst
-Icon=kcmfontinst
+Icon=kcmfontinst.png
 Type=Application
 DocPath=kcontrol/fontinst.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/kio/cookies.desktop kdebase-3.0.4/kcontrol/kio/cookies.desktop
--- kdebase-3.0.4.orig/kcontrol/kio/cookies.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/kio/cookies.desktop	Sun Nov  3 12:09:30 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 DocPath=kcontrol/cookie.html
-Icon=cookie
+Icon=cookie.png
 Exec=kcmshell cookies
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/kio/netpref.desktop kdebase-3.0.4/kcontrol/kio/netpref.desktop
--- kdebase-3.0.4.orig/kcontrol/kio/netpref.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/kio/netpref.desktop	Sun Nov  3 12:23:14 2002
@@ -42,7 +42,7 @@
 Comment[zh_TW]=設定一般網路喜好設定，例如逾時值
 Comment[zu]=Hlanganisa okufanayo koxhumaniso olusakazekile lokuncanyelwayo, njengamanani okuphelelwa isikhathi
 Exec=kcmshell netpref
-Icon=socket
+Icon=socket.png
 Keywords=timeout,iopref,netpref,network preferences
 Keywords[af]=timeout,iopref,netpref,network voorkeure
 Keywords[ar]=الوقت المحدد,iopref,netpref,تفضيلات الشبكة
diff -ur kdebase-3.0.4.orig/kcontrol/kio/proxy.desktop kdebase-3.0.4/kcontrol/kio/proxy.desktop
--- kdebase-3.0.4.orig/kcontrol/kio/proxy.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/kio/proxy.desktop	Sun Nov  3 12:09:21 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 DocPath=kcontrol/proxies.html
-Icon=proxy
+Icon=proxy.png
 Exec=kcmshell proxy
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/kio/smb.desktop kdebase-3.0.4/kcontrol/kio/smb.desktop
--- kdebase-3.0.4.orig/kcontrol/kio/smb.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/kio/smb.desktop	Sun Nov  3 12:23:02 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 DocPath=kcontrol/windows-shares.html
-Icon=samba
+Icon=samba.png
 Exec=kcmshell smb
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/kio/useragent.desktop kdebase-3.0.4/kcontrol/kio/useragent.desktop
--- kdebase-3.0.4.orig/kcontrol/kio/useragent.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/kio/useragent.desktop	Sun Nov  3 12:09:12 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Type=Application
-Icon=agent
+Icon=agent.png
 Exec=kcmshell useragent
 DocPath=kcontrol/user-agent.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/knotify/kcmnotify.desktop kdebase-3.0.4/kcontrol/knotify/kcmnotify.desktop
--- kdebase-3.0.4.orig/kcontrol/knotify/kcmnotify.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/knotify/kcmnotify.desktop	Sun Nov  3 12:22:52 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell kcmnotify
-Icon=knotify
+Icon=knotify.png
 Type=Application
 DocPath=kcontrol/sys-notify.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/konq/desktop.desktop kdebase-3.0.4/kcontrol/konq/desktop.desktop
--- kdebase-3.0.4.orig/kcontrol/konq/desktop.desktop	Sat Sep 28 20:21:57 2002
+++ kdebase-3.0.4/kcontrol/konq/desktop.desktop	Sun Nov  3 12:09:02 2002
@@ -3,7 +3,7 @@
 Type=Application
 DocPath=kcontrol/desktop.html
 # Needs another icon !
-Icon=desktop
+Icon=desktop.png
 Exec=kcmshell desktop
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/konq/kcmkonq.desktop kdebase-3.0.4/kcontrol/konq/kcmkonq.desktop
--- kdebase-3.0.4.orig/kcontrol/konq/kcmkonq.desktop	Sat Sep 28 20:22:21 2002
+++ kdebase-3.0.4/kcontrol/konq/kcmkonq.desktop	Sun Nov  3 12:22:40 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 DocPath=kcontrol/file-manager.html
-Icon=kfm
+Icon=kfm.png
 Exec=kcmshell kcmkonq
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/konq/virtualdesktops.desktop kdebase-3.0.4/kcontrol/konq/virtualdesktops.desktop
--- kdebase-3.0.4.orig/kcontrol/konq/virtualdesktops.desktop	Sat Sep 28 20:22:21 2002
+++ kdebase-3.0.4/kcontrol/konq/virtualdesktops.desktop	Sun Nov  3 12:22:31 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=desktop
+Icon=desktop.png
 Type=Application
 Exec=kcmshell virtualdesktops
 DocPath=kcontrol/desktop.html
diff -ur kdebase-3.0.4.orig/kcontrol/konqhtml/konqhtml.desktop kdebase-3.0.4/kcontrol/konqhtml/konqhtml.desktop
--- kdebase-3.0.4.orig/kcontrol/konqhtml/konqhtml.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/konqhtml/konqhtml.desktop	Sun Nov  3 12:03:36 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 DocPath=kcontrol/konq-browsing.html
-Icon=konqueror
+Icon=konqueror.png
 Exec=kcmshell konqhtml
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/konsole/kcmkonsole.desktop kdebase-3.0.4/kcontrol/konsole/kcmkonsole.desktop
--- kdebase-3.0.4.orig/kcontrol/konsole/kcmkonsole.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/konsole/kcmkonsole.desktop	Sun Nov  3 12:22:23 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=konsole
+Icon=konsole.png
 Type=Application
 Exec=kcmshell kcmkonsole
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/locale/language.desktop kdebase-3.0.4/kcontrol/locale/language.desktop
--- kdebase-3.0.4.orig/kcontrol/locale/language.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/locale/language.desktop	Sun Nov  3 12:22:20 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell language
-Icon=locale
+Icon=locale.png
 Type=Application
 DocPath=kcontrol/locale.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/midi/midi.desktop kdebase-3.0.4/kcontrol/midi/midi.desktop
--- kdebase-3.0.4.orig/kcontrol/midi/midi.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/midi/midi.desktop	Sun Nov  3 12:07:09 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell midi
-Icon=kcmmidi
+Icon=kcmmidi.png
 Type=Application
 DocPath=kcontrol/midi.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/nics/nic.desktop kdebase-3.0.4/kcontrol/nics/nic.desktop
--- kdebase-3.0.4.orig/kcontrol/nics/nic.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/nics/nic.desktop	Sun Nov  3 12:22:02 2002
@@ -2,9 +2,8 @@
 Encoding=UTF-8
 Exec=kcmshell nic
 Type=Application
-Icon=network
+Icon=network.png
 DocPath=kcontrol/net-interfaces.html
-Icon=network
 
 X-KDE-ModuleType=Library
 X-KDE-Library=nic
diff -ur kdebase-3.0.4.orig/kcontrol/passwords/passwords.desktop kdebase-3.0.4/kcontrol/passwords/passwords.desktop
--- kdebase-3.0.4.orig/kcontrol/passwords/passwords.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/passwords/passwords.desktop	Sun Nov  3 12:08:18 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kcmshell passwords
-Icon=password
+Icon=password.png
 DocPath=kcontrol/passwords.html
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kcontrol/samba/smbstatus.desktop kdebase-3.0.4/kcontrol/samba/smbstatus.desktop
--- kdebase-3.0.4.orig/kcontrol/samba/smbstatus.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/samba/smbstatus.desktop	Sun Nov  3 12:21:37 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell smbstatus
-Icon=samba
+Icon=samba.png
 Type=Application
 DocPath=kcontrol/sambastatus.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/screensaver/screensaver.desktop kdebase-3.0.4/kcontrol/screensaver/screensaver.desktop
--- kdebase-3.0.4.orig/kcontrol/screensaver/screensaver.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/screensaver/screensaver.desktop	Sun Nov  3 12:03:13 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell screensaver
-Icon=kscreensaver
+Icon=kscreensaver.png
 Type=Application
 DocPath=kcontrol/screensaver.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/smserver/kcmsmserver.desktop kdebase-3.0.4/kcontrol/smserver/kcmsmserver.desktop
--- kdebase-3.0.4.orig/kcontrol/smserver/kcmsmserver.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/smserver/kcmsmserver.desktop	Sun Nov  3 12:02:54 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=go
+Icon=go.png
 Type=Application
 DocPath=kcontrol/sessions.html
 Exec=kcmshell kcmsmserver
diff -ur kdebase-3.0.4.orig/kcontrol/socks/socks.desktop kdebase-3.0.4/kcontrol/socks/socks.desktop
--- kdebase-3.0.4.orig/kcontrol/socks/socks.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/socks/socks.desktop	Sun Nov  3 12:21:14 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=proxy
+Icon=proxy.png
 Type=Application
 Exec=kcmshell socks
 DocPath=kcontrol/socks.html
diff -ur kdebase-3.0.4.orig/kcontrol/spellchecking/spellchecking.desktop kdebase-3.0.4/kcontrol/spellchecking/spellchecking.desktop
--- kdebase-3.0.4.orig/kcontrol/spellchecking/spellchecking.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/spellchecking/spellchecking.desktop	Sun Nov  3 12:08:08 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell spellchecking
-Icon=spellcheck
+Icon=spellcheck.png
 Type=Application
 DocPath=kcontrol/spell-checking.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/style/style.desktop kdebase-3.0.4/kcontrol/style/style.desktop
--- kdebase-3.0.4.orig/kcontrol/style/style.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/style/style.desktop	Sun Nov  3 12:20:54 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell style
-Icon=style
+Icon=style.png
 Type=Application
 DocPath=kcontrol/style.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/taskbar/kcmtaskbar.desktop kdebase-3.0.4/kcontrol/taskbar/kcmtaskbar.desktop
--- kdebase-3.0.4.orig/kcontrol/taskbar/kcmtaskbar.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/taskbar/kcmtaskbar.desktop	Sun Nov  3 12:03:02 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=go
+Icon=go.png
 Type=Application
 DocPath=kcontrol/taskbar.html
 Exec=kcmshell kcmtaskbar
diff -ur kdebase-3.0.4.orig/kcontrol/thememgr/kthememgr.desktop kdebase-3.0.4/kcontrol/thememgr/kthememgr.desktop
--- kdebase-3.0.4.orig/kcontrol/thememgr/kthememgr.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/thememgr/kthememgr.desktop	Sun Nov  3 12:20:37 2002
@@ -1,7 +1,7 @@
 # KDE Config File
 [KDE Desktop Entry]
 Encoding=UTF-8
-Icon=kthememgr
+Icon=kthememgr.png
 Exec=kcmshell kthememgr
 DocPath=kcontrol/theme-manager.html
 
diff -ur kdebase-3.0.4.orig/kcontrol/thememgr/x-ktheme.desktop kdebase-3.0.4/kcontrol/thememgr/x-ktheme.desktop
--- kdebase-3.0.4.orig/kcontrol/thememgr/x-ktheme.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/thememgr/x-ktheme.desktop	Sun Nov  3 12:02:31 2002
@@ -49,7 +49,7 @@
 Comment[xx]=xx
 Comment[zh_CN]=KDE 主题
 Comment[zh_TW]=KDE 主題
-Icon=ktheme
+Icon=ktheme.png
 Type=MimeType
 Patterns=*.ktheme;
 X-KDE-AutoEmbed=false
diff -ur kdebase-3.0.4.orig/kcontrol/usbview/kcmusb.desktop kdebase-3.0.4/kcontrol/usbview/kcmusb.desktop
--- kdebase-3.0.4.orig/kcontrol/usbview/kcmusb.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kcontrol/usbview/kcmusb.desktop	Sun Nov  3 12:20:24 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=usb
+Icon=usb.png
 Type=Application
 DocPath=kcontrol/usb.html
 Exec=kcmshell kcmusb
diff -ur kdebase-3.0.4.orig/kdesktop/init/Home.desktop kdebase-3.0.4/kdesktop/init/Home.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Home.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Home.desktop	Sun Nov  3 12:19:35 2002
@@ -110,5 +110,5 @@
 GenericName[zh_TW]=個人檔案
 GenericName[zu]=Amafayela Ayimfihlo
 Exec=kfmclient openProfile filemanagement
-Icon=kfm_home
+Icon=kfm_home.png
 Type=Application
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/CDROM-Device.desktop kdebase-3.0.4/kdesktop/init/Templates/CDROM-Device.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/CDROM-Device.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/CDROM-Device.desktop	Sun Nov  3 13:36:40 2002
@@ -5,7 +5,7 @@
 ReadOnly=0
 FSType=Default
 Type=FSDevice
-Icon=cdrom_mount
+Icon=cdrom_mount.png
 UnmountIcon=cdrom_unmount
 Actions=Eject
 [Desktop Action Eject]
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/Directory.desktop kdebase-3.0.4/kdesktop/init/Templates/Directory.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/Directory.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/Directory.desktop	Sun Nov  3 12:40:15 2002
@@ -112,4 +112,4 @@
 Comment[zu]=Ulawulo Olusha
 Type=Link
 URL=.source/emptydir
-Icon=folder
+Icon=folder.png
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/Floppy.desktop kdebase-3.0.4/kdesktop/init/Templates/Floppy.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/Floppy.desktop	Fri Feb  1 01:39:57 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/Floppy.desktop	Sun Nov  3 13:36:37 2002
@@ -5,5 +5,5 @@
 ReadOnly=0
 FSType=Default
 Type=FSDevice
-Icon=3floppy_mount
+Icon=3floppy_mount.png
 UnmountIcon=3floppy_unmount
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/HD.desktop kdebase-3.0.4/kdesktop/init/Templates/HD.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/HD.desktop	Fri Feb  1 01:39:57 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/HD.desktop	Sun Nov  3 12:39:43 2002
@@ -5,5 +5,5 @@
 ReadOnly=0
 FSType=Default
 Type=FSDevice
-Icon=hdd_mount
+Icon=hdd_mount.png
 UnmountIcon=hdd_unmount
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/HTMLFile.desktop kdebase-3.0.4/kdesktop/init/Templates/HTMLFile.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/HTMLFile.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/HTMLFile.desktop	Sun Nov  3 13:36:33 2002
@@ -113,4 +113,4 @@
 Comment[zu]=Ifayela Entsha ye HTML
 Type=Link
 URL=.source/HTMLFile.html
-Icon=html
+Icon=html.png
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/Program.desktop kdebase-3.0.4/kdesktop/init/Templates/Program.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/Program.desktop	Fri Feb  1 01:39:57 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/Program.desktop	Sun Nov  3 12:39:22 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=
-Icon=exec
+Icon=exec.png
 TerminalOptions=
 Path=
 Terminal=0
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/TextFile.desktop kdebase-3.0.4/kdesktop/init/Templates/TextFile.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/TextFile.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/TextFile.desktop	Sun Nov  3 13:36:22 2002
@@ -113,4 +113,4 @@
 Comment[zu]=Ifayela Entsha Yokubhaliweyo
 Type=Link
 URL=.source/TextFile.txt
-Icon=txt
+Icon=txt.png
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/linkCDROM.desktop kdebase-3.0.4/kdesktop/init/Templates/linkCDROM.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/linkCDROM.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/linkCDROM.desktop	Sun Nov  3 13:35:45 2002
@@ -90,4 +90,4 @@
 Comment[zu]=Elisha ithuluzi le CD/DVD-ROM
 Type=Link
 URL=.source/CDROM-Device.desktop
-Icon=cdrom_unmount
+Icon=cdrom_unmount.png
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/linkFloppy.desktop kdebase-3.0.4/kdesktop/init/Templates/linkFloppy.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/linkFloppy.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/linkFloppy.desktop	Sun Nov  3 12:39:08 2002
@@ -112,4 +112,4 @@
 Comment[zu]=Icebo Elisha Lefloppy
 Type=Link
 URL=.source/Floppy.desktop
-Icon=3floppy_unmount
+Icon=3floppy_unmount.png
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/linkHD.desktop kdebase-3.0.4/kdesktop/init/Templates/linkHD.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/linkHD.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/linkHD.desktop	Sun Nov  3 13:35:42 2002
@@ -99,4 +99,4 @@
 Comment[zu]=Idiski Elisha Eqinileyo
 Type=Link
 URL=.source/HD.desktop
-Icon=hdd_unmount
+Icon=hdd_unmount.png
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/linkProgram.desktop kdebase-3.0.4/kdesktop/init/Templates/linkProgram.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/linkProgram.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/linkProgram.desktop	Sun Nov  3 12:36:44 2002
@@ -95,6 +95,6 @@
 Comment[zh_CN]=新建应用程序链接
 Comment[zh_TW]=新建應用程式連結
 Comment[zu]=Uxhumaniso Olutsha Kwisicelo
-Icon=exec
+Icon=exec.png
 Type=Link
 URL=.source/Program.desktop
diff -ur kdebase-3.0.4.orig/kdesktop/init/Templates/linkURL.desktop kdebase-3.0.4/kdesktop/init/Templates/linkURL.desktop
--- kdebase-3.0.4.orig/kdesktop/init/Templates/linkURL.desktop	Sat Sep 28 20:22:22 2002
+++ kdebase-3.0.4/kdesktop/init/Templates/linkURL.desktop	Sun Nov  3 13:35:28 2002
@@ -97,4 +97,4 @@
 Comment[zu]=Uxhumaniso Olutsha kwindawo ekuyo (URL)
 URL=.source/URL.desktop
 Type=Link
-Icon=www
+Icon=www.png
diff -ur kdebase-3.0.4.orig/kdesktop/init/directory.desktop kdebase-3.0.4/kdesktop/init/directory.desktop
--- kdebase-3.0.4.orig/kdesktop/init/directory.desktop	Fri Feb  1 01:39:57 2002
+++ kdebase-3.0.4/kdesktop/init/directory.desktop	Sun Nov  3 12:12:54 2002
@@ -2,5 +2,5 @@
 Encoding=UTF-8
 Type=Directory
 BgImage=
-Icon=desktop
+Icon=desktop.png
 
diff -ur kdebase-3.0.4.orig/kfind/Kfind.desktop kdebase-3.0.4/kfind/Kfind.desktop
--- kdebase-3.0.4.orig/kfind/Kfind.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/kfind/Kfind.desktop	Fri Nov  1 02:02:06 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kfind -caption "%c" %i %m %f
-Icon=kfind
+Icon=kfind.png
 DocPath=kfind/index.html
 TerminalOptions=
 Path=
diff -ur kdebase-3.0.4.orig/khelpcenter/Help.desktop kdebase-3.0.4/khelpcenter/Help.desktop
--- kdebase-3.0.4.orig/khelpcenter/Help.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/khelpcenter/Help.desktop	Sun Nov  3 11:52:15 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=khelpcenter
-Icon=khelpcenter
+Icon=khelpcenter.png
 DocPath=khelpcenter/index.html
 Type=Application
 Terminal=0
diff -ur kdebase-3.0.4.orig/khelpcenter/htmlsearch/htmlsearch.desktop kdebase-3.0.4/khelpcenter/htmlsearch/htmlsearch.desktop
--- kdebase-3.0.4.orig/khelpcenter/htmlsearch/htmlsearch.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/khelpcenter/htmlsearch/htmlsearch.desktop	Sun Nov  3 12:13:06 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell htmlsearch
-Icon=help_index
+Icon=help_index.png
 Type=Application
 DocPath=kcontrol/help-index.html
 
diff -ur kdebase-3.0.4.orig/khelpcenter/khelpcenter.desktop kdebase-3.0.4/khelpcenter/khelpcenter.desktop
--- kdebase-3.0.4.orig/khelpcenter/khelpcenter.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/khelpcenter/khelpcenter.desktop	Fri Nov  1 02:02:11 2002
@@ -91,7 +91,7 @@
 Comment[zh_CN]=KDE 帮助中心
 Comment[zh_TW]=KDE 求助中心
 Comment[zu]=Isikhungo Soziso se KDE
-Icon=khelpcenter
+Icon=khelpcenter.png
 DocPath=khelpcenter/index.html
 Type=Service
 Terminal=0
diff -ur kdebase-3.0.4.orig/kicker/applets/minipager/minipagerapplet.desktop kdebase-3.0.4/kicker/applets/minipager/minipagerapplet.desktop
--- kdebase-3.0.4.orig/kicker/applets/minipager/minipagerapplet.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/kicker/applets/minipager/minipagerapplet.desktop	Sun Nov  3 13:35:00 2002
@@ -98,7 +98,7 @@
 Comment[zh_TW]=一個基本的桌面切換面板小程式
 Comment[zu]=Elula i desktop yokushintshela iphanela ye applet
 
-Icon=kpager
+Icon=kpager.png
 
 X-KDE-Library=minipager_panelapplet
 X-KDE-UniqueApplet=false
diff -ur kdebase-3.0.4.orig/kioslave/audiocd/kcmaudiocd/audiocd.desktop kdebase-3.0.4/kioslave/audiocd/kcmaudiocd/audiocd.desktop
--- kdebase-3.0.4.orig/kioslave/audiocd/kcmaudiocd/audiocd.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/kioslave/audiocd/kcmaudiocd/audiocd.desktop	Sun Nov  3 12:33:47 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell audiocd
-Icon=cdaudio_unmount
+Icon=cdaudio_unmount.png
 Type=Application
 
 X-KDE-ModuleType=Library
diff -ur kdebase-3.0.4.orig/kioslave/smb/smb-network.desktop kdebase-3.0.4/kioslave/smb/smb-network.desktop
--- kdebase-3.0.4.orig/kioslave/smb/smb-network.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/kioslave/smb/smb-network.desktop	Sun Nov  3 12:01:13 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=network
+Icon=network.png
 Name=Samba Shares
 Name[af]=Samba Gedeelde bronne
 Name[ar]=مشاركة Samba
diff -ur kdebase-3.0.4.orig/klipper/klipper.desktop kdebase-3.0.4/klipper/klipper.desktop
--- kdebase-3.0.4.orig/klipper/klipper.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/klipper/klipper.desktop	Fri Nov  1 02:01:49 2002
@@ -52,7 +52,7 @@
 GenericName[zh_TW]=剪貼簿工具
 GenericName[zu]=Ithuluzi Lebhodi lokuciphiza
 Exec=klipper %i %m
-Icon=klipper
+Icon=klipper.png
 Type=Application
 DocPath=klipper/index.html
 Terminal=0
diff -ur kdebase-3.0.4.orig/kmenuedit/kmenuedit.desktop kdebase-3.0.4/kmenuedit/kmenuedit.desktop
--- kdebase-3.0.4.orig/kmenuedit/kmenuedit.desktop	Fri Nov  1 01:41:25 2002
+++ kdebase-3.0.4/kmenuedit/kmenuedit.desktop	Fri Nov  1 02:05:43 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kmenuedit  -caption "%c" %i %m %u
-Icon=kmenuedit
+Icon=kmenuedit.png
 DocPath=kmenuedit/index.html
 Type=Application
 X-KDE-StartupNotify=true
diff -ur kdebase-3.0.4.orig/konqueror/Home.desktop kdebase-3.0.4/konqueror/Home.desktop
--- kdebase-3.0.4.orig/konqueror/Home.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/konqueror/Home.desktop	Sun Nov  3 11:55:12 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kfmclient openProfile filemanagement
-Icon=folder_home
+Icon=folder_home.png
 Terminal=0
 
 Name=Home
diff -ur kdebase-3.0.4.orig/konqueror/kfmclient.desktop kdebase-3.0.4/konqueror/kfmclient.desktop
--- kdebase-3.0.4.orig/konqueror/kfmclient.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/konqueror/kfmclient.desktop	Fri Nov  1 01:59:02 2002
@@ -3,7 +3,7 @@
 # This desktop file is the one used by KApplication::invokeBrowser
 Type=Application
 Exec=kfmclient openURL %u
-Icon=konqueror
+Icon=konqueror.png
 Terminal=0
 InitialPreference=9
 NoDisplay=true
diff -ur kdebase-3.0.4.orig/konqueror/kfmclient_dir.desktop kdebase-3.0.4/konqueror/kfmclient_dir.desktop
--- kdebase-3.0.4.orig/konqueror/kfmclient_dir.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/konqueror/kfmclient_dir.desktop	Fri Nov  1 01:57:48 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kfmclient openURL %u inode/directory
-Icon=konqueror
+Icon=konqueror.png
 Terminal=0
 MimeType=inode/directory
 InitialPreference=9
diff -ur kdebase-3.0.4.orig/konqueror/kfmclient_html.desktop kdebase-3.0.4/konqueror/kfmclient_html.desktop
--- kdebase-3.0.4.orig/konqueror/kfmclient_html.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/konqueror/kfmclient_html.desktop	Fri Nov  1 01:59:00 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kfmclient openURL %u text/html
-Icon=konqueror
+Icon=konqueror.png
 Terminal=0
 MimeType=text/html
 InitialPreference=9
diff -ur kdebase-3.0.4.orig/konqueror/konqbrowser.desktop kdebase-3.0.4/konqueror/konqbrowser.desktop
--- kdebase-3.0.4.orig/konqueror/konqbrowser.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/konqueror/konqbrowser.desktop	Fri Nov  1 01:57:46 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kfmclient openProfile webbrowsing
-Icon=konqueror
+Icon=konqueror.png
 DocPath=konqueror/index.html
 
 Name=Konqueror Web Browser
diff -ur kdebase-3.0.4.orig/konqueror/konqfilemgr.desktop kdebase-3.0.4/konqueror/konqfilemgr.desktop
--- kdebase-3.0.4.orig/konqueror/konqfilemgr.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/konqueror/konqfilemgr.desktop	Fri Nov  1 01:58:57 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kfmclient openProfile filemanagement
-Icon=kfm
+Icon=kfm.png
 DocPath=konqueror/index.html
 
 Name=File Manager
diff -ur kdebase-3.0.4.orig/konqueror/konqueror.desktop kdebase-3.0.4/konqueror/konqueror.desktop
--- kdebase-3.0.4.orig/konqueror/konqueror.desktop	Sat Sep 28 20:23:13 2002
+++ kdebase-3.0.4/konqueror/konqueror.desktop	Fri Nov  1 01:57:43 2002
@@ -65,7 +65,7 @@
 Comment[zh_TW]=KDE 檔案管理程式和 Web 瀏覽器
 Comment[zu]=Umphathi Wefayela ye KDE & Umbheki zincwadi we Web
 X-KDE-Library=libkonqueror
-Icon=konqueror
+Icon=konqueror.png
 ServiceTypes=Konqueror
 X-DCOP-ServiceType=multi
 X-KDE-StartupNotify=true
diff -ur kdebase-3.0.4.orig/konqueror/konqueror_config.desktop kdebase-3.0.4/konqueror/konqueror_config.desktop
--- kdebase-3.0.4.orig/konqueror/konqueror_config.desktop	Sat Sep 28 20:23:14 2002
+++ kdebase-3.0.4/konqueror/konqueror_config.desktop	Fri Nov  1 01:58:53 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Exec=kcmshell kcmkonq filetypes konqhtml ebrowsing cookies cache proxy kcmcss crypto useragent nsplugin
-Icon=konqueror
+Icon=konqueror.png
 DocPath=konqueror/index.html
 Name=Configure Konqueror
 Name[af]=Konfigureer Konqueror
diff -ur kdebase-3.0.4.orig/konqueror/konquerorsu.desktop kdebase-3.0.4/konqueror/konquerorsu.desktop
--- kdebase-3.0.4.orig/konqueror/konquerorsu.desktop	Sat Sep 28 20:23:14 2002
+++ kdebase-3.0.4/konqueror/konquerorsu.desktop	Fri Nov  1 01:57:40 2002
@@ -3,7 +3,7 @@
 Type=Application
 Exec=kdesu konqueror
 DocPath=konqueror/index.html
-Icon=kfm
+Icon=kfm.png
 
 Name=File Manager - Super User Mode
 Name[af]=Lêer Bestuurder - Super Gebruiker Modus
diff -ur kdebase-3.0.4.orig/konsole/kcmkonsole.desktop kdebase-3.0.4/konsole/kcmkonsole.desktop
--- kdebase-3.0.4.orig/konsole/kcmkonsole.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/konsole/kcmkonsole.desktop	Fri Nov  1 01:56:09 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmkonsole
-Icon=konsole
+Icon=konsole.png
 Type=Application
 
 X-KDE-ControlModuleType=User
diff -ur kdebase-3.0.4.orig/konsole/konsole/x-konsole.desktop kdebase-3.0.4/konsole/konsole/x-konsole.desktop
--- kdebase-3.0.4.orig/konsole/konsole/x-konsole.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/konsole/konsole/x-konsole.desktop	Sun Nov  3 12:00:48 2002
@@ -3,7 +3,7 @@
 Type=MimeType
 MimeType=application/x-konsole
 X-KDE-AutoEmbed=true
-Icon=konsole
+Icon=konsole.png
 Patterns=*.shell;
 Comment=File to open a shell
 Comment[af]=Lêer na open 'n tolk
diff -ur kdebase-3.0.4.orig/konsole/konsole.desktop kdebase-3.0.4/konsole/konsole.desktop
--- kdebase-3.0.4.orig/konsole/konsole.desktop	Fri Nov  1 01:41:25 2002
+++ kdebase-3.0.4/konsole/konsole.desktop	Fri Nov  1 01:50:26 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=konsole %i %m
-Icon=konsole
+Icon=konsole.png
 DocPath=konsole/index.html
 Terminal=0
 X-KDE-StartupNotify=true
diff -ur kdebase-3.0.4.orig/konsole/konsolepart.desktop kdebase-3.0.4/konsole/konsolepart.desktop
--- kdebase-3.0.4.orig/konsole/konsolepart.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/konsole/konsolepart.desktop	Fri Nov  1 01:55:54 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Exec=konsole %i %m -caption "%c"
-Icon=konsole
+Icon=konsole.png
 ServiceTypes=KParts/ReadOnlyPart,Browser/View,inode/directory
 
 X-KDE-Library=libkonsolepart
diff -ur kdebase-3.0.4.orig/konsole/konsolesu.desktop kdebase-3.0.4/konsole/konsolesu.desktop
--- kdebase-3.0.4.orig/konsole/konsolesu.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/konsole/konsolesu.desktop	Fri Nov  1 01:50:21 2002
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=konsole --type su
-Icon=konsole
+Icon=konsole.png
 DocPath=konsole/index.html
 MimeType=application/x-konsole
 Terminal=0
diff -ur kdebase-3.0.4.orig/kpager/kpager.desktop kdebase-3.0.4/kpager/kpager.desktop
--- kdebase-3.0.4.orig/kpager/kpager.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/kpager/kpager.desktop	Fri Nov  1 01:49:42 2002
@@ -52,7 +52,7 @@
 GenericName[zh_TW]=桌面呼叫器
 GenericName[zu]=Umphenyi we Desktop
 
-Icon=kpager
+Icon=kpager.png
 Type=Application
 Exec=kpager
 DocPath=kpager/index.html
diff -ur kdebase-3.0.4.orig/kscreensaver/KBlankscreen.desktop kdebase-3.0.4/kscreensaver/KBlankscreen.desktop
--- kdebase-3.0.4.orig/kscreensaver/KBlankscreen.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/kscreensaver/KBlankscreen.desktop	Sun Nov  3 11:54:01 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kblankscrn.kss
-Icon=kscreensaver
+Icon=kscreensaver.png
 Type=Application
 Actions=Setup;InWindow;Root
 Name=Blank Screen
@@ -119,7 +119,7 @@
 Name[zh_CN]=设置...
 Name[zh_TW]=設定...
 Name[zu]=Iyalungiselela...
-Icon=kscreensaver
+Icon=kscreensaver.png
  
 [Desktop Action InWindow]
 Exec=kblankscrn.kss -window-id %w
diff -ur kdebase-3.0.4.orig/kscreensaver/KRandom.desktop kdebase-3.0.4/kscreensaver/KRandom.desktop
--- kdebase-3.0.4.orig/kscreensaver/KRandom.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/kscreensaver/KRandom.desktop	Sun Nov  3 11:54:06 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=krandom.kss
-Icon=kscreensaver
+Icon=kscreensaver.png
 Type=Application
 Actions=Setup;InWindow;Root
 Name=Random
@@ -118,7 +118,7 @@
 Name[zh_CN]=设置...
 Name[zh_TW]=設定...
 Name[zu]=Iyalungiselela...
-Icon=kscreensaver
+Icon=kscreensaver.png
  
 [Desktop Action InWindow]
 Exec=krandom.kss -window-id %w
diff -ur kdebase-3.0.4.orig/ksysguard/gui/ksysguard.desktop kdebase-3.0.4/ksysguard/gui/ksysguard.desktop
--- kdebase-3.0.4.orig/ksysguard/gui/ksysguard.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/ksysguard/gui/ksysguard.desktop	Sun Nov  3 12:11:52 2002
@@ -55,7 +55,7 @@
 Name[zh_TW]=KDE 系統守護神
 Name[zu]=Unogada Wendlela ye KDE
 Exec=ksysguard %U
-Icon=ksysguard
+Icon=ksysguard.png
 Type=Application
 MimeType=application/x-ksysguard
 DocPath=ksysguard/index.html
diff -ur kdebase-3.0.4.orig/ksysguard/gui/x-ksysguard.desktop kdebase-3.0.4/ksysguard/gui/x-ksysguard.desktop
--- kdebase-3.0.4.orig/ksysguard/gui/x-ksysguard.desktop	Sat Sep 28 20:26:17 2002
+++ kdebase-3.0.4/ksysguard/gui/x-ksysguard.desktop	Sun Nov  3 12:11:56 2002
@@ -50,7 +50,7 @@
 Comment[zh_CN]=KDE 系统卫士
 Comment[zh_TW]=KDE 系統守護神
 Comment[zu]=Unogada Wendlela ye KDE
-Icon=ksysguard
+Icon=ksysguard.png
 Type=MimeType
 Patterns=*.sgrd;
 X-KDE-AutoEmbed=false
diff -ur kdebase-3.0.4.orig/ktip/ktip.desktop kdebase-3.0.4/ktip/ktip.desktop
--- kdebase-3.0.4.orig/ktip/ktip.desktop	Mon Jun 24 13:57:29 2002
+++ kdebase-3.0.4/ktip/ktip.desktop	Fri Nov  1 01:45:01 2002
@@ -48,7 +48,7 @@
 Name[zh_TW]=Kandalf 的小技巧
 Name[zu]=Amaphuzu ze Kandalf
 Exec=ktip
-Icon=ktip
+Icon=ktip.png
 Type=Application
 Terminal=0
 X-KDE-StartupNotify=true
diff -ur kdebase-3.0.4.orig/kwin/kcmkwin/kwindecoration/kwindecoration.desktop kdebase-3.0.4/kwin/kcmkwin/kwindecoration/kwindecoration.desktop
--- kdebase-3.0.4.orig/kwin/kcmkwin/kwindecoration/kwindecoration.desktop	Sat Sep 28 20:26:18 2002
+++ kdebase-3.0.4/kwin/kcmkwin/kwindecoration/kwindecoration.desktop	Sun Nov  3 12:33:06 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell kwindecoration
-Icon=kcmkwm
+Icon=kcmkwm.png
 Type=Application
 DocPath=kcontrol/window-deco.html
 
diff -ur kdebase-3.0.4.orig/kwin/kcmkwin/kwinoptions/kwinoptions.desktop kdebase-3.0.4/kwin/kcmkwin/kwinoptions/kwinoptions.desktop
--- kdebase-3.0.4.orig/kwin/kcmkwin/kwinoptions/kwinoptions.desktop	Sat Sep 28 20:26:18 2002
+++ kdebase-3.0.4/kwin/kcmkwin/kwinoptions/kwinoptions.desktop	Sun Nov  3 13:33:20 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=kcmkwm
+Icon=kcmkwm.png
 Type=Application
 Exec=kcmshell kwinoptions 
 DocPath=kcontrol/window-behavior.html
diff -ur kdebase-3.0.4.orig/kxkb/keyboard.desktop kdebase-3.0.4/kxkb/keyboard.desktop
--- kdebase-3.0.4.orig/kxkb/keyboard.desktop	Sat Sep 28 20:26:18 2002
+++ kdebase-3.0.4/kxkb/keyboard.desktop	Fri Nov  1 02:05:06 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=kcmshell keyboard
-Icon=keyboard
+Icon=keyboard.png
 Type=Application
 DocPath=kcontrol/kbd.html
 
diff -ur kdebase-3.0.4.orig/legacyimport/klegacyimport.desktop kdebase-3.0.4/legacyimport/klegacyimport.desktop
--- kdebase-3.0.4.orig/legacyimport/klegacyimport.desktop	Sat Sep 28 20:26:18 2002
+++ kdebase-3.0.4/legacyimport/klegacyimport.desktop	Sun Nov  3 11:51:26 2002
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=klegacyimport
-Icon=style
+Icon=style.png
 Type=Application
 
 Name=GTK Theme Importer
diff -ur kdebase-3.0.4.orig/libkonq/directory_bookmarkbar.desktop kdebase-3.0.4/libkonq/directory_bookmarkbar.desktop
--- kdebase-3.0.4.orig/libkonq/directory_bookmarkbar.desktop	Sat Sep 28 20:26:18 2002
+++ kdebase-3.0.4/libkonq/directory_bookmarkbar.desktop	Sun Nov  3 11:51:10 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=bookmark_toolbar
+Icon=bookmark_toolbar.png
 Name=Bookmark Toolbar
 Name[af]=Boekmerk Nutsbalk
 Name[ar]=شريط أدوات علامات المواقع
diff -ur kdebase-3.0.4.orig/nsplugins/control/nsplugin.desktop kdebase-3.0.4/nsplugins/control/nsplugin.desktop
--- kdebase-3.0.4.orig/nsplugins/control/nsplugin.desktop	Sat Sep 28 20:26:18 2002
+++ kdebase-3.0.4/nsplugins/control/nsplugin.desktop	Sun Nov  3 12:17:55 2002
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Icon=netscape
+Icon=netscape.png
 Type=Application
 Exec=kcmshell nsplugin
 DocPath=kcontrol/nsplugins.html
