--- kdebase-031021/startkde.orig	2003-10-14 22:48:48.000000000 +0200
+++ kdebase-031021/startkde	2003-10-21 13:58:14.000000000 +0200
@@ -181,6 +181,7 @@
 
 # run KPersonalizer before the session, if this is the first login
 if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
+  if test `/usr/bin/which kpersonalizer 2>/dev/null`; then
     # start only dcopserver, don't start whole kdeinit (takes too long)
     echo 'startkde: Running kpersonalizer...'  1>&2
     dcopserver
@@ -192,6 +193,7 @@
     done
     dcopquit kwin
     dcopserver_shutdown --wait
+  fi
 fi
 
 # the splashscreen and progress indicator

