diff -urN kdebase-2.2.1.orig/kioslave/info/info.cc kdebase-2.2.1/kioslave/info/info.cc
--- kdebase-2.2.1.orig/kioslave/info/info.cc	Sun Dec  3 22:01:55 2000
+++ kdebase-2.2.1/kioslave/info/info.cc	Sun Sep 23 14:39:46 2001
@@ -225,7 +225,7 @@
     // "* infopage:" at the start of a line
     QRegExp regex( "^\\*  *[^: ][^:]*:", false );
  
-    QFile f( "/usr/info/dir" );
+    QFile f( "/usr/share/info/dir" );
  
     if ( f.open(IO_ReadOnly) ) {
         QTextStream t( &f );
@@ -266,7 +266,7 @@
         finished();
     }
     else {
-        kdError(7108) << "cannot open file '/usr/info/dir'" << endl;
+        kdError(7108) << "cannot open file '/usr/share/info/dir'" << endl;
     }
     kdDebug( 7108 ) << "InfoProtocol::listDir - done" << endl;
 }
