--- kdebase-050210/kdm/backend/dm.h.orig	2005-02-08 21:51:08.000000000 +0100
+++ kdebase-050210/kdm/backend/dm.h	2005-02-11 12:00:00.277539336 +0100
@@ -196,15 +196,7 @@
 #   define WTMP_FILE "/usr/adm/wtmp"
 #  endif
 # endif
-# ifndef UTMP_FILE
-#  ifdef _PATH_UTMPX
-#   define UTMP_FILE _PATH_UTMPX
-#  elif defined(_PATH_UTMP)
-#   define UTMP_FILE _PATH_UTMP
-#  else
-#   define UTMP_FILE "/etc/utmp"
-#  endif
-# endif
+# define UTMP_FILE "/var/run/utmpx"
 #endif /* NEED_UTMP */
 
 typedef struct GPipe {
