--- kdebase-3.0.3/kpersonalizer/kstylepage.cpp--	2002-08-25 16:03:24.000000000 +0200
+++ kdebase-3.0.3/kpersonalizer/kstylepage.cpp	2002-08-25 16:04:06.000000000 +0200
@@ -234,6 +234,7 @@ void KStylePage::setDefaults(){
 /** No descriptions */
 void KStylePage::getColors(colorSet *set, bool colorfile ){
     KConfig* config;
+    bool deleteConfig=false;
     // get the color scheme file and go to the color scheme group
     if(colorfile){
        KGlobal::dirs()->addResourceType("colors", KStandardDirs::kde_default("data")+"kdisplay/color-schemes");
@@ -269,6 +270,7 @@ void KStylePage::getColors(colorSet *set
       set->contrast=7;
       config = new KSimpleConfig(set->colorFile, true);
       config->setGroup("Color Scheme");
+      deleteConfig=true;
     }
     else{  //
       KConfig kdesktop("kdesktoprc");
@@ -312,6 +314,7 @@ void KStylePage::getColors(colorSet *set
     set->inactiveBlend=config->readColorEntry("inactiveBlend", &widget);
     set->activeTitleBtnBg=config->readColorEntry("activeTitleBtnBg", &widget);
     set->inactiveTitleBtnBg=config->readColorEntry("inactiveTitleBtnBg", &widget);
-
+    if(deleteConfig)
+      delete config;
 }
 #include "kstylepage.moc"
