diff -urN kdebase-2.2.1.orig/konsole/src/konsole.C kdebase-2.2.1/konsole/src/konsole.C
--- kdebase-2.2.1.orig/konsole/src/konsole.C	Wed Sep  5 01:16:57 2001
+++ kdebase-2.2.1/konsole/src/konsole.C	Wed Sep 19 17:21:11 2001
@@ -1352,7 +1352,7 @@
 
 TESession *Konsole::newSession(KSimpleConfig *co, QString program, const QStrList &args, const QString &_term)
 {
-  QString emu = "xterm";
+  QString emu = "xterm-color";
   QString key; 
   QString sch = s_kconfigSchema;
   QString txt = s_title;
diff -urN kdebase-2.2.1.orig/konsole/src/konsole_part.C kdebase-2.2.1/konsole/src/konsole_part.C
--- kdebase-2.2.1.orig/konsole/src/konsole_part.C	Wed Sep  5 01:16:57 2001
+++ kdebase-2.2.1/konsole/src/konsole_part.C	Wed Sep 19 17:22:03 2001
@@ -118,7 +118,7 @@
   setWidget(te);
   // faking a KMainwindow - TESession assumes that (wrong design!)
   initial = new TESession((KMainWindow*)parentWidget,
-                          te,shell,eargs,"xterm");
+                          te,shell,eargs,"xterm-color");
   connect( initial,SIGNAL(done(TESession*,int)),
            this,SLOT(doneSession(TESession*,int)) );
   connect( te,SIGNAL(configureRequest(TEWidget*,int,int,int)),
