diff -urN -x CVS kdebase.orig/kdesktop/kdiconview.cc kdebase/kdesktop/kdiconview.cc
--- kdebase.orig/kdesktop/kdiconview.cc	Mon Jul 12 23:33:10 2004
+++ kdebase/kdesktop/kdiconview.cc	Tue Aug 17 18:00:53 2004
@@ -392,7 +392,7 @@
         KAction *pasteTo = KStdAction::paste( this, SLOT( slotPopupPasteTo() ), &m_actionCollection, "pasteto" );
         pasteTo->setEnabled( false ); // only enabled during popupMenu()
 
-        (void) new KAction( i18n( "&Rename" ), /*"editrename",*/ Key_F2, this, SLOT( renameCurrentItem() ), &m_actionCollection, "rename" );
+        (void) new KAction( i18n( "&Rename" ), /*"editrename",*/ Key_F2, this, SLOT( renameSelectedItem() ), &m_actionCollection, "rename" );
         (void) new KAction( i18n( "&Move to Trash" ), "edittrash", Key_Delete, this, SLOT( slotTrash() ), &m_actionCollection, "trash" );
 
         KConfig config("konquerorrc", true, false);
diff -urN -x CVS kdebase.orig/kdesktop/krootwm.cc kdebase/kdesktop/krootwm.cc
--- kdebase.orig/kdesktop/krootwm.cc	Fri Jun 25 22:49:09 2004
+++ kdebase/kdesktop/krootwm.cc	Fri Aug 13 21:43:16 2004
@@ -739,13 +739,13 @@
 }
 
 void KRootWm::slotNewSession() {
-   KDialogBase *dialog= new KDialogBase(
-                       i18n("Warning - New Session"),
+   KDialogBase *dialog= new KDialogBase("",
                        KDialogBase::Yes | KDialogBase::No,
                        KDialogBase::Yes, KDialogBase::No,
                        m_pDesktop,
                        "warningYesNo", true, true,
                        KGuiItem(i18n("&Start New Session"), "fork"), KStdGuiItem::cancel() );
+    dialog->setPlainCaption(i18n("Warning - New Session"));
 
     bool checkboxResult = kapp->authorize("lock_screen");
     const QString text = i18n("<p>You have chosen to open another desktop session.<br>"
diff -urN -x CVS kdebase.orig/kdesktop/kwebdesktop/kwebdesktop.cpp kdebase/kdesktop/kwebdesktop/kwebdesktop.cpp
--- kdebase.orig/kdesktop/kwebdesktop/kwebdesktop.cpp	Sun Jun 27 18:18:51 2004
+++ kdebase/kdesktop/kwebdesktop/kwebdesktop.cpp	Tue Aug 17 12:59:40 2004
@@ -119,7 +119,7 @@
     KWebDesktop *webDesktop = new KWebDesktop( part, imageFile );
     QObject::connect( part, SIGNAL( canceled(const QString &) ),
                       webDesktop, SLOT( slotCompleted() ) );
-    QObject::connect( part, SIGNAL( completed() ),
+    QObject::connect( part->widget(), SIGNAL( finishedLayout() ),
                       webDesktop, SLOT( slotCompleted() ) );
 
     KConfig config( "kwebdesktoprc" );
diff -urN -x CVS kdebase.orig/kdesktop/lock/lockprocess.cc kdebase/kdesktop/lock/lockprocess.cc
--- kdebase.orig/kdesktop/lock/lockprocess.cc	Fri May  7 06:12:39 2004
+++ kdebase/kdesktop/lock/lockprocess.cc	Mon Aug 16 15:56:08 2004
@@ -787,7 +787,11 @@
 void LockProcess::suspend()
 {
     if(!mSuspended)
+    {
         mHackProc.kill(SIGSTOP);
+        QApplication::syncX();
+        mSavedScreen = QPixmap::grabWindow( winId());
+    }
     mSuspended = true;
 }
 
@@ -798,7 +802,11 @@
     if(!mVisibility)
         return; // no need to resume, not visible
     if(mSuspended)
+    {
+        bitBlt( this, 0, 0, &mSavedScreen );
+        QApplication::syncX();
         mHackProc.kill(SIGCONT);
+    }
     mSuspended = false;
 }
 
diff -urN -x CVS kdebase.orig/kdesktop/lock/lockprocess.h kdebase/kdesktop/lock/lockprocess.h
--- kdebase.orig/kdesktop/lock/lockprocess.h	Sun Feb  8 15:16:22 2004
+++ kdebase/kdesktop/lock/lockprocess.h	Mon Aug 16 15:56:08 2004
@@ -17,6 +17,7 @@
 #include <qtimer.h>
 #include <qvaluestack.h>
 #include <qmessagebox.h>
+#include <qpixmap.h>
 
 #include <X11/Xlib.h>
 
@@ -119,6 +120,7 @@
     QStringList mPlugins, mPluginOptions;
     QString     mMethod;
     GreeterPluginHandle greetPlugin;
+    QPixmap     mSavedScreen;
 };
 
 #endif
diff -urN -x CVS kdebase.orig/kdesktop/minicli.cpp kdebase/kdesktop/minicli.cpp
--- kdebase.orig/kdesktop/minicli.cpp	Mon Jul 19 12:19:57 2004
+++ kdebase/kdesktop/minicli.cpp	Fri Aug 13 21:43:16 2004
@@ -73,7 +73,7 @@
         :KDialog( parent, name ),
          m_autoCheckedRunInTerm(false)
 {
-  setCaption( i18n("Run Command") );
+  setPlainCaption( i18n("Run Command") );
   KWin::setIcons( winId(), DesktopIcon("run"), SmallIcon("run") );
 
   QVBoxLayout* mainLayout = new QVBoxLayout( this, 0, KDialog::spacingHint() );
@@ -81,7 +81,7 @@
   mainLayout->addWidget(m_dlg);
 
   m_dlg->lbRunIcon->setPixmap(DesktopIcon("kmenu"));
-  m_dlg->lbComment->setAlignment( Qt::WordBreak );  
+  m_dlg->lbComment->setAlignment( Qt::WordBreak );
 
   m_dlg->cbCommand->setDuplicatesEnabled( false );
   m_dlg->cbCommand->setTrapReturnKey( true );
@@ -234,15 +234,15 @@
   config->writePathEntry( "History", m_dlg->cbCommand->historyItems() );
   config->writePathEntry( "TerminalApps", m_terminalAppList );
   config->writePathEntry( "CompletionItems", m_dlg->cbCommand->completionObject()->items() );
-  
-  // Ensure that the current default completion mode will be used the next 
+
+  // Ensure that the current default completion mode will be used the next
   // time we read from the config file. That means do not store the default
   // completion mode...
   if( m_dlg->cbCommand->completionMode() == KGlobalSettings::completionMode() )
     config->deleteEntry("CompletionMode");
   else
     config->writeEntry( "CompletionMode", m_dlg->cbCommand->completionMode() );
-    
+
   config->sync();
 }
 
@@ -552,7 +552,7 @@
             KRun::run(*service, KURL::List());
             return 0;
           }
-          
+
           service = KService::serviceByName(cmd);
           if (service)
           {
diff -urN -x CVS kdebase.orig/kicker/buttons/servicebutton.cpp kdebase/kicker/buttons/servicebutton.cpp
--- kdebase.orig/kicker/buttons/servicebutton.cpp	Sat Aug  7 12:59:45 2004
+++ kdebase/kicker/buttons/servicebutton.cpp	Wed Aug 18 11:50:53 2004
@@ -121,6 +121,7 @@
         _valid = false;
         return;
     }
+    QToolTip::remove( this );
     if ( !_service->genericName().isEmpty() ) {
         QToolTip::add(  this, _service->genericName() );
     }
diff -urN -x CVS kdebase.orig/kicker/buttons/urlbutton.cpp kdebase/kicker/buttons/urlbutton.cpp
--- kdebase.orig/kicker/buttons/urlbutton.cpp	Wed May 19 16:06:34 2004
+++ kdebase/kicker/buttons/urlbutton.cpp	Wed Aug 18 11:51:56 2004
@@ -102,6 +102,7 @@
 
 void URLButton::setToolTip()
 {
+	QToolTip::remove(this);
     if (fileItem->isLocalFile() 
      && KDesktopFile::isDesktopFile(fileItem->url().path()))
     {
diff -urN -x CVS kdebase.orig/kicker/ui/k_mnu.cpp kdebase/kicker/ui/k_mnu.cpp
--- kdebase.orig/kicker/ui/k_mnu.cpp	Sat Jul 17 06:37:09 2004
+++ kdebase/kicker/ui/k_mnu.cpp	Fri Aug 13 21:43:17 2004
@@ -360,13 +360,13 @@
 
 void PanelKMenu::slotNewSession()
 {
-    KDialogBase *dialog= new KDialogBase(
-                       i18n("Warning - New Session"),
+    KDialogBase *dialog= new KDialogBase("",
                        KDialogBase::Yes | KDialogBase::No,
                        KDialogBase::Yes, KDialogBase::No,
                        kapp->desktop()->screen(kapp->desktop()->screenNumber(this)),
                        "warningYesNo", true, true,
                        KGuiItem(i18n("&Start New Session"), "fork"), KStdGuiItem::cancel() );
+    dialog->setPlainCaption(i18n("Warning - New Session"));
 
     bool checkboxResult = kapp->authorize("lock_screen");
     const QString text = i18n("<p>You have chosen to open another desktop session.<br>"
diff -urN -x CVS kdebase.orig/konqueror/iconview/konq_iconview.cc kdebase/konqueror/iconview/konq_iconview.cc
--- kdebase.orig/konqueror/iconview/konq_iconview.cc	Sun Jul 18 15:15:25 2004
+++ kdebase/konqueror/iconview/konq_iconview.cc	Sat Aug 21 03:44:33 2004
@@ -883,10 +883,14 @@
     // Root item ? Store root item in konqiconviewwidget (whether 0L or not)
     m_pIconView->setRootItem( m_dirLister->rootItem() );
 
-    if ( m_bNeedSetCurrentItem ) // only after initial listing, not after updates
+    // only after initial listing, not after updates
         // If we don't set a current item, the iconview has none (one more keypress needed)
         // but it appears on focusin... qiconview bug, Reggie acknowledged it LONG ago (07-2000).
+    if ( m_bNeedSetCurrentItem )
+    {
         m_pIconView->setCurrentItem( m_pIconView->firstItem() );
+        m_bNeedSetCurrentItem = false;
+    }
 
     if ( m_bUpdateContentsPosAfterListing ) {
          m_pIconView->setContentsPos( extension()->urlArgs().xOffset,
diff -urN -x CVS kdebase.orig/konqueror/iconview/konq_iconview.h kdebase/konqueror/iconview/konq_iconview.h
--- kdebase.orig/konqueror/iconview/konq_iconview.h	Sun Jul 18 15:15:25 2004
+++ kdebase/konqueror/iconview/konq_iconview.h	Tue Aug 17 18:00:56 2004
@@ -245,7 +245,7 @@
 
   void refreshMimeTypes() { m_iconView->iconViewWidget()->refreshMimeTypes(); }
 
-  void rename() { m_iconView->iconViewWidget()->renameCurrentItem(); }
+  void rename() { m_iconView->iconViewWidget()->renameSelectedItem(); }
   void cut() { m_iconView->iconViewWidget()->cutSelection(); }
   void copy() { m_iconView->iconViewWidget()->copySelection(); }
   void paste() { m_iconView->iconViewWidget()->pasteSelection(); }
diff -urN -x CVS kdebase.orig/konqueror/listview/konq_treeviewwidget.cc kdebase/konqueror/listview/konq_treeviewwidget.cc
--- kdebase.orig/konqueror/listview/konq_treeviewwidget.cc	Tue May 18 15:40:33 2004
+++ kdebase/konqueror/listview/konq_treeviewwidget.cc	Mon Aug 16 16:32:04 2004
@@ -139,6 +139,11 @@
             m_urlsToReload.remove( it.currentKey() );
          }
       }
+      
+      // Delete all child items, their file-items are no longer valid
+      QListViewItem* child;
+      while((child = item->firstChild()))
+         delete child;
    }
 }
 
diff -urN -x CVS kdebase.orig/kwin/activation.cpp kdebase/kwin/activation.cpp
--- kdebase.orig/kwin/activation.cpp	Mon Jun 21 14:40:43 2004
+++ kdebase/kwin/activation.cpp	Mon Aug 16 15:50:49 2004
@@ -696,7 +696,7 @@
       // use the _KDE_NET_WM_USER_CREATION_TIME trick.
       // Otherwise, refuse activation of a window
       // from already running application if this application
-      // is not the active one.
+      // is not the active one (unless focus stealing prevention is turned off).
         Client* act = workspace()->mostRecentlyActivatedClient();
         if( act != NULL && !belongToSameApplication( act, this, true ))
             {
@@ -717,7 +717,8 @@
                 if( workspace()->findClient( SameApplicationActiveHackPredicate( this )))
                     first_window = false;
                 }
-            if( !first_window )
+            // don't refuse if focus stealing prevention is turned off
+            if( !first_window && rules()->checkFSP( options->focusStealingPreventionLevel ) > 0 )
                 {
                 kdDebug( 1212 ) << "User timestamp, already exists:" << 0 << endl;
                 return 0; // refuse activation
diff -urN -x CVS kdebase.orig/kwin/geometry.cpp kdebase/kwin/geometry.cpp
--- kdebase.orig/kwin/geometry.cpp	Tue Aug  3 11:21:18 2004
+++ kdebase/kwin/geometry.cpp	Mon Aug 16 16:00:13 2004
@@ -1035,6 +1035,8 @@
     w = QMAX( min_size.width(), w );
     h = QMAX( min_size.height(), h );
 
+    int w1 = w;
+    int h1 = h;
     int width_inc = xSizeHint.width_inc;
     int height_inc = xSizeHint.height_inc;
     int basew_inc = xSizeHint.min_width; // see getWmNormalHints()
@@ -1154,6 +1156,11 @@
         w += xSizeHint.base_width;
         h += xSizeHint.base_height;
         }
+    // disobey increments and aspect when maximized
+    if( maximizeMode() & MaximizeHorizontal )
+        w = w1;
+    if( maximizeMode() & MaximizeVertical )
+        h = h1;
 
     w += border_left + border_right;
     h += border_top + border_bottom;
@@ -1223,7 +1230,7 @@
     if( isManaged())
         { // update to match restrictions
         QSize new_size = adjustedSize( size());
-        if( new_size != size() && !isShade()) // SHADE
+        if( new_size != size() && !isShade() && !isFullScreen()) // SHADE
             resizeWithChecks( new_size );
         }
     updateAllowedActions(); // affects isResizeable()
@@ -1904,7 +1911,7 @@
         if( maximizeMode() != MaximizeRestore )
             changeMaximize( false, false, true ); // adjust size
         else if( !geom_fs_restore.isNull())
-            setGeometry( geom_fs_restore );
+            setGeometry( QRect( geom_fs_restore.topLeft(), adjustedSize( geom_fs_restore.size())));
         // TODO isShaded() ?
         else
             { // does this ever happen?
diff -urN -x CVS kdebase.orig/kwin/workspace.cpp kdebase/kwin/workspace.cpp
--- kdebase.orig/kwin/workspace.cpp	Wed Aug  4 11:05:53 2004
+++ kdebase/kwin/workspace.cpp	Mon Aug 16 15:52:56 2004
@@ -99,6 +99,11 @@
     layoutY(2),
     workarea(NULL),
     screenarea(NULL),
+    topmenu_selection( NULL ),
+    topmenu_watcher( NULL ),
+    topmenu_height( 0 ),
+    managing_topmenus( false ),
+    topmenu_space( NULL ),
     set_active_client_recursion( 0 ),
     block_stacking_updates( 0 ),
     forced_global_mouse_grab( false )
@@ -309,9 +314,6 @@
     Atom topmenu_atom = XInternAtom( qt_xdisplay(), nm, False );
     topmenu_selection = new KSelectionOwner( topmenu_atom );
     topmenu_watcher = new KSelectionWatcher( topmenu_atom );
-    topmenu_height = 0;
-    managing_topmenus = false;
-    topmenu_space = NULL;
 // TODO grabXServer(); - where exactly put this? topmenu selection claiming down belong must be before
 
         { // begin updates blocker block
@@ -1111,6 +1113,8 @@
             requestFocus( findDesktop( true, currentDesktop()));
         }
 
+    updateCurrentTopMenu();
+
     // Update focus chain:
     //  If input: chain = { 1, 2, 3, 4 } and current_desktop = 3,
     //   Output: chain = { 3, 1, 2, 4 }.
diff -urN -x CVS kdebase.orig/libkonq/knewmenu.cc kdebase/libkonq/knewmenu.cc
--- kdebase.orig/libkonq/knewmenu.cc	Wed May 19 14:25:43 2004
+++ kdebase/libkonq/knewmenu.cc	Sun Aug 15 11:32:21 2004
@@ -347,6 +347,48 @@
     KonqOperations::newDir(d->m_parentWidget, popupFiles.first());
 }
 
+// copied from kdelibs/kio/kio/renamedlg.cpp
+static QString suggestName(const KURL& baseURL, const QString& oldName)
+{
+  QString dotSuffix, suggestedName;
+  QString basename = oldName;
+
+  int index = basename.find( '.' );
+  if ( index != -1 ) {
+    dotSuffix = basename.mid( index );
+    basename.truncate( index );
+  }
+
+  int pos = basename.findRev( '_' );
+  if(pos != -1 ){
+    QString tmp = basename.mid( pos+1 );
+    bool ok;
+    int number = tmp.toInt( &ok );
+    if ( !ok ) {// ok there is no number
+      suggestedName = basename + "1" + dotSuffix;
+    }
+    else {
+     // yes there's already a number behind the _ so increment it by one
+      basename.replace( pos+1, tmp.length(), QString::number(number+1) );
+      suggestedName = basename + dotSuffix;
+    }
+  }
+  else // no underscore yet
+    suggestedName = basename + "_1" + dotSuffix ;
+
+  // Check if suggested name already exists
+  bool exists = false;
+  // TODO: network transparency. However, using NetAccess from a modal dialog
+  // could be a problem, no? (given that it uses a modal widget itself....)
+  if ( baseURL.isLocalFile() )
+     exists = QFileInfo( baseURL.path(+1) + suggestedName ).exists();
+
+  if ( !exists )
+    return suggestedName;
+  else // already exists -> recurse
+    return suggestName( baseURL, suggestedName );
+}
+
 void KNewMenu::slotNewFile()
 {
     int id = QString( sender()->name() + 7 ).toInt(); // skip "newmenu"
@@ -401,7 +443,13 @@
                 //kdDebug(1203) << "third arg=" << entry.text << endl;
                 QString text = entry.text;
                 text.replace( "...", QString::null ); // the ... is fine for the menu item but not for the default filename
-                KURL templateURL;
+                
+		KURL defaultFile( *it );
+		defaultFile.addPath( KIO::encodeFileName( text ) );
+		if ( defaultFile.isLocalFile() && QFile::exists( defaultFile.path() ) )
+		    text = suggestName( *it, text);
+
+		KURL templateURL;
                 templateURL.setPath( entry.templatePath );
                 (void) new KPropertiesDialog( templateURL, *it, text, d->m_parentWidget );
     	    }
@@ -415,7 +463,13 @@
         bool ok;
         QString text = entry.text;
         text.replace( "...", QString::null ); // the ... is fine for the menu item but not for the default filename
-        name = KInputDialog::getText( QString::null, entry.comment,
+        
+	KURL defaultFile( *(popupFiles.begin()) );
+	defaultFile.addPath( KIO::encodeFileName( text ) );
+	if ( defaultFile.isLocalFile() && QFile::exists( defaultFile.path() ) )
+	    text = suggestName( *(popupFiles.begin()), text);
+
+	name = KInputDialog::getText( QString::null, entry.comment,
     	text, &ok, d->m_parentWidget );
         if ( !ok )
 	    return;
diff -urN -x CVS kdebase.orig/libkonq/konq_iconviewwidget.cc kdebase/libkonq/konq_iconviewwidget.cc
--- kdebase.orig/libkonq/konq_iconviewwidget.cc	Sat Jul 17 17:17:27 2004
+++ kdebase/libkonq/konq_iconviewwidget.cc	Tue Aug 17 18:00:55 2004
@@ -1337,7 +1337,7 @@
     emit enableAction( "del", canDel > 0 );
     emit enableAction( "properties", iCount > 0 && KPropertiesDialog::canDisplay( selectedFileItems() ) );
     emit enableAction( "editMimeType", ( iCount == 1 ) );
-    emit enableAction( "rename", currentItem() && !bInTrash );
+    emit enableAction( "rename", ( iCount == 1 ) && !bInTrash );
 }
 
 void KonqIconViewWidget::renameCurrentItem()
