diff -ur kdebase-2.2.orig/konsole/src/konsole.C kdebase-2.2/konsole/src/konsole.C
--- kdebase-2.2.orig/konsole/src/konsole.C	Sun Aug  5 02:41:24 2001
+++ kdebase-2.2/konsole/src/konsole.C	Wed Aug 15 16:07:41 2001
@@ -1325,7 +1325,7 @@
 
 TESession *Konsole::newSession(KSimpleConfig *co, QString program, const QStrList &args)
 {
-  QCString emu = "xterm";
+  QCString emu = "xterm-color";
   QString sch = s_kconfigSchema;
   QString txt = s_title;
   unsigned int     fno = n_defaultFont;
diff -ur kdebase-2.2.orig/konsole/src/konsole_part.C kdebase-2.2/konsole/src/konsole_part.C
--- kdebase-2.2.orig/konsole/src/konsole_part.C	Sun Jul 29 06:56:43 2001
+++ kdebase-2.2/konsole/src/konsole_part.C	Wed Aug 15 16:08:05 2001
@@ -118,7 +118,7 @@
   setWidget(te);
   // faking a KMainwindow - TESession assumes that (wrong design!)
   initial = new TESession((KMainWindow*)parentWidget,
-                          te,shell,eargs,"xterm");
+                          te,shell,eargs,"xterm-color");
   connect( initial,SIGNAL(done(TESession*,int)),
            this,SLOT(doneSession(TESession*,int)) );
   connect( te,SIGNAL(configureRequest(TEWidget*,int,int,int)),
