--- kdebase-trinity-3.5.13.2/konsole/fonts/CMakeLists.txt~	2012-05-24 04:35:16.000000000 +0300
+++ kdebase-trinity-3.5.13.2/konsole/fonts/CMakeLists.txt	2014-06-04 10:58:16.675576558 +0300
@@ -11,8 +11,7 @@
 
 install( FILES
     ${CMAKE_CURRENT_BINARY_DIR}/console8x16.pcf.gz
-    ${CMAKE_CURRENT_BINARY_DIR}/9x15.pcf.gz
-  DESTINATION ${DATA_INSTALL_DIR}/konsole/fonts )
+  DESTINATION ${SHARE_INSTALL_PREFIX}/fonts/misc )
 
 # FIXME this should be improved, to guarantee that mkfontdir is executed after fonts are installed
 install( CODE "execute_process( COMMAND mkfontdir \$ENV{DESTDIR}${DATA_INSTALL_DIR}/konsole/fonts )" )
