--- kdmrc.orig	Sat Dec  1 18:39:40 2001
+++ kdmrc	Sat Dec  1 20:14:10 2001
@@ -38,7 +38,7 @@
 
 # XXX do this on a per-display basis: should be [X-*-Desktop]
 # HELPME: I need help with kbackgroundrenderer!  -- ossi@kde.org
-[Desktop0]
+[X-*-Desktop]
 BackgroundMode=VerticalGradient
 BlendBalance=100
 BlendMode=NoBlending
@@ -64,7 +64,7 @@
 # Default is /usr/X11R6/share/config/kdm/Xservers
 # XXX i'm planning to absorb this file into kdmrc, but i'm not sure how to
 # do this best.
-#Xservers=
+Xservers=/etc/X11/kdm/Xservers
 # Where KDM should store its PID. Default is /var/run/xdm.pid - this
 # is an intentional conflict with "plain" XDM.
 #PidFile=/var/run/kdm.pid
@@ -89,12 +89,12 @@
 #Port=177
 # File with the private keys of X-terminals. Required for XDM authentication.
 # Default is /usr/X11R6/share/config/kdm/kdmkeys
-#KeyFile=
+KeyFile=/etc/X11/kdm/kdmkeys
 # XDMCP access control file in the usual xdm-Xaccess format.
 # Default is /usr/X11R6/share/config/kdm/Xaccess
 # XXX i'm planning to absorb this file into kdmrc, but i'm not sure how to
 # do this best.
-#Xaccess=
+Xaccess=/etc/X11/kdm/Xaccess
 # Number of seconds to wait for display to respond after the user has
 # selected a host from the chooser. Default is 15.
 #ChoiceTimeout=10
@@ -107,11 +107,11 @@
 # The program which is invoked to dynamically generate replies to XDMCP
 # BroadcastQuery requests.
 # By default no program is invoked and "Willing to manage" is sent.
-Willing=/usr/X11R6/share/config/kdm/Xwilling
+Willing=/etc/X11/kdm/Xwilling
 
 [Shutdown]
 # The command to run to halt the system. Default is /sbin/halt
-HaltCmd=/sbin/poweroff
+#HaltCmd=/sbin/poweroff
 # The command to run to reboot the system. Default is /sbin/reboot
 RebootCmd=/sbin/reboot
 # Offer LiLo boot options in shutdown dialog. Default is false
@@ -175,20 +175,20 @@
 # A program to run before the greeter is shown. You should start kdmdesktop
 # there. Also, xconsole can be started by this script.
 # Default is /usr/X11R6/share/config/kdm/Xsetup
-#Setup=
+Setup=/etc/X11/kdm/Xsetup
 # A program to run before a user session starts. You should invoke sessreg
 # there and optionally change the ownership of the console, etc.
 # Default is /usr/X11R6/share/config/kdm/Xstartup
-#Startup=
+Startup=/etc/X11/kdm/Xstartup
 # A program to run after a user session exits. You should invoke sessreg
 # there and optionally change the ownership of the console, etc.
 # Default is /usr/X11R6/share/config/kdm/Xreset
-#Reset=
+Reset=/etc/X11/kdm/Xreset
 # The program which is run as the user which logs in. It is supposed to
 # interpret the session argument and start an appropriate session according
 # to it. See SessionTypes.
 # Default is /usr/X11R6/share/config/kdm/Xsession
-#Session=
+Session=/etc/X11/kdm/Xsession
 # The program to run if Session fails.
 # Default is /usr/X11R6/bin/xterm
 #FailsafeClient=
@@ -218,7 +218,7 @@
 AllowNullPasswd=false
 # Where (relatively to the user's home directory) to store the last
 # selected session. Default is .wmrc
-#SessSaveFile=.wmrc
+SessSaveFile=.kdmrc
 # Command FiFo options.
 # XXX these options will probably change ...
 # Default is false
@@ -258,7 +258,7 @@
 # - %m -> the machine (hardware) type
 # - %% -> a single %
 # Default is "Welcome to %s at %n"
-GreetString=Polish(ed) Linux Distribution (%n)
+GreetString=Polish(ed) Linux Distribution on %n
 # The font for the headline. Default is charter,24,bold
 GreetFont=heveltica,12,5,0,75,0
 # The normal font used in the greeter. Default is helvetica,12
@@ -280,7 +280,7 @@
 NoUsers=adm,apache,bin,bind,daemon,exim,falken,ftp,games,gdm,gopher,halt,httpd,ingres,kmem,lp,mail,mailnull,man,mta,news,nobody,nscd,operator,pop,postfix,reboot,rpc,rpcuser,sendmail,shutdown,sync,tty,uucp,xfs,xten,xdm
 # Special case of NoUsers: users with a UID less than this number (except root)
 # will not be shown as well. Default is 0
-#MinShowUID=1000
+MinShowUID=100
 # Complement to MinShowUID: users with a UID greater than this number will
 # not be shown as well. Default is 65535
 #MaxShowUID=29999
