--- kdebase-030317/kcontrol/kdm/background.cpp~	Sun Mar 16 00:35:44 2003
+++ kdebase-030317/kcontrol/kdm/background.cpp	Tue Mar 18 19:31:12 2003
@@ -53,7 +53,7 @@
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
     config->setGroup( "X-*-Greeter" );
-    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg",KDE_CONFDIR "/kdm/backgroundrc" ) );
+    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg","/etc/X11/kdm/backgroundrc" ) );
     m_background = new BGDialog( this, m_simpleConf, false );
 
     connect(m_background, SIGNAL(changed(bool)), SIGNAL(changed(bool)));
@@ -105,7 +105,7 @@
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
-    m_background->defaults();
+    m_background->kdmdefaults();
     emit changed(true);
 }
 
-void KBackground::defaults()
+void KBackground::kdmdefaults()
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
--- kdebase-031010/kcontrol/kdm/kdm-appear.cpp.orig	2003-09-23 21:46:11.000000000 +0200
+++ kdebase-031010/kcontrol/kdm/kdm-appear.cpp	2003-10-10 19:41:03.000000000 +0200
@@ -385,7 +385,7 @@
 bool KDMAppearanceWidget::setLogo(QString logo)
 {
     QString flogo = logo.isEmpty() ?
-                    locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") ) :
+                    locate("data", QString::fromLatin1("kdm/pics/pldlogo.png") ) :
                     logo;
     QPixmap p(flogo);
     if (p.isNull())
@@ -516,7 +516,7 @@
   config->setGroup("X-*-Greeter");
 
   // Read the greeting string
-  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to %s at %n")));
+  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to PLD Linux at %n")));
 
   // Regular logo or clock
   QString logoArea = config->readEntry("LogoArea", "Logo" );
@@ -566,7 +566,7 @@
 
 void KDMAppearanceWidget::defaults()
 {
-  greetstr_lined->setText( i18n("Welcome to %s at %n") );
+  greetstr_lined->setText( i18n("Welcome to PLD Linux at %n") );
   logoRadio->setChecked( true );
   slotAreaRadioClicked( KdmLogo );
   posCenterRadio->setChecked( true );
--- kdebase-040116/kcontrol/kdm/kdm-font.cpp.orig	2004-01-16 23:07:36.000000000 +0100
+++ kdebase-040116/kcontrol/kdm/kdm-font.cpp	2004-01-16 23:25:36.000000000 +0100
@@ -87,8 +87,8 @@
 
 void KDMFontWidget::set_def()
 {
-  stdFontChooser->setFont(QFont("helvetica", 12));
-  failFontChooser->setFont(QFont("helvetica", 12, QFont::Bold));
+  stdFontChooser->setFont(QFont("helvetica", 11));
+  failFontChooser->setFont(QFont("helvetica", 11, QFont::Bold));
   greetingFontChooser->setFont(QFont("charter", 24));
 }
 
#--- kdebase-030418/kcontrol/kdm/kdm-sess.cpp~	2003-04-04 00:37:58.000000000 +0200
#+++ kdebase-030418/kcontrol/kdm/kdm-sess.cpp	2003-04-19 19:26:08.000000000 +0200
#@@ -364,7 +364,7 @@
#   QString str;
# 
#   config->setGroup("X-:*-Core");
#-  readSD(sdlcombo, "All");
#+  readSD(sdlcombo, "Root");
# 
#   config->setGroup("X-*-Core");
#   readSD(sdrcombo, "Root");
--- kdebase-030924/kcontrol/kdm/kdm-users.cpp~	2003-09-23 21:46:11.000000000 +0200
+++ kdebase-030924/kcontrol/kdm/kdm-users.cpp	2003-09-24 16:39:25.000000000 +0200
@@ -65,7 +65,7 @@
     // We assume that $kde_datadir/kdm exists, but better check for pics/ and pics/users,
     // and create them if necessary.
     config->setGroup( "X-*-Greeter" );
-    m_userPixDir = config->readEntry( "FaceDir", KGlobal::dirs()->resourceDirs("data").last() + "kdm/faces" ) + '/';
+    m_userPixDir = config->readEntry( "FaceDir", "/etc/X11/kdm/faces" ) + '/';
     m_notFirst = false;
     QDir testDir( m_userPixDir );
     if ( !testDir.exists() && !testDir.mkdir( testDir.absPath() ))
--- kdebase-3.1/kcontrol/kdm/main.cpp.orig	Sun Oct 27 00:05:07 2002
+++ kdebase-3.1/kcontrol/kdm/main.cpp	Thu Nov 21 02:51:19 2002
@@ -92,7 +92,7 @@
     usermap.insert( QString::fromLocal8Bit( ps->pw_name ), ps->pw_uid );
   endpwent();
 
-  config = new KSimpleConfig( QString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));
+  config = new KSimpleConfig( QString::fromLatin1( "/etc/X11/kdm/kdmrc" ));
 
   QVBoxLayout *top = new QVBoxLayout(this);
   tab = new QTabWidget(this);
--- kdebase-030930/kdm/kfrontend/Makefile.am~	2003-09-30 00:47:50.000000000 +0200
+++ kdebase-030930/kdm/kfrontend/Makefile.am	2003-09-30 10:06:29.000000000 +0200
@@ -38,7 +38,7 @@
 genkdmconf_LDADD = $(LIB_X11)
 
 install-data-local: genkdmconf
-	./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm --face-src $(srcdir)/pics $(GENKDMCONF_FLAGS)
+	./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm --face-src $(srcdir)/pics $(GENKDMCONF_FLAGS)
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase/kdm/kfrontend/genkdmconf.c.orig	2004-03-23 14:55:38.000000000 +0100
+++ kdebase/kdm/kfrontend/genkdmconf.c	2004-03-25 21:55:08.236691776 +0100
@@ -793,14 +791,15 @@
 "esac\n"
 "exec xmessage -center -buttons OK:0 -default OK \"Sorry, cannot execute $1. Check $DESKTOP_SESSION.desktop.\"\n";
 
+/* Keep in sync with bgdefaults.h & BGDialog::kdmdefaults() */
 static const char def_background[] =
 "[Desktop0]\n"
-"BackgroundMode=Flat\n"
+"BackgroundMode=VerticalGradient\n"
 "BlendBalance=100\n"
 "BlendMode=NoBlending\n"
 "ChangeInterval=60\n"
-"Color1=0,0,200\n"
-"Color2=192,192,192\n"
+"Color1=57,55,75\n"
+"Color2=66,82,143\n"
 "CurrentWallpaper=0\n"
 "LastChange=0\n"
 "MinOptimizationDepth=1\n"
@@ -809,7 +808,7 @@
 "Program=\n"
 "ReverseBlending=false\n"
 "UseSHM=false\n"
-"Wallpaper=default_blue.jpg\n"
+"Wallpaper=kdm_pld.png\n"
 "WallpaperList=\n"
 "WallpaperMode=Scaled\n";
 
--- kdebase-040724/kdm/config.def.orig	2004-07-24 12:50:04.000000000 +0200
+++ kdebase-040724/kdm/config.def	2004-07-24 13:09:38.487484896 +0200
@@ -3,7 +3,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 2
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #ifdef _AIX
@@ -1062,7 +1062,7 @@
 
 Key: Willing
 Type: string
-Default: ""
+Default: KDMCONF "/Xwilling"
 User: core
 # will be overwritten
 Instance: #
@@ -1430,7 +1430,7 @@
 
 Key: Setup
 Type: string
-Default: ""
+Default: KDMCONF "/Xsetup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1450,7 +1450,7 @@
 
 Key: Startup
 Type: string
-Default: ""
+Default: KDMCONF "/Xstartup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1468,7 +1468,7 @@
 
 Key: Reset
 Type: string
-Default: ""
+Default: KDMCONF "/Xreset"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1486,7 +1486,7 @@
 
 Key: Session
 Type: string
-Default: XBINDIR "/xterm -ls -T"
+Default: KDMCONF "/Xsession"
 #Merge: xdm - incompatible!
 User: core
 # will be overwritten
@@ -1617,7 +1617,7 @@
 
 Key: AllowRootLogin
 Type: bool
-Default: true
+Default: false
 User: core
 User: greeter(showRoot)
 Instance: */false
@@ -1608,10 +1608,10 @@
 
 Key: AllowNullPasswd
 Type: bool
-Default: true
+Default: false
 User: core
 # sensible?
-Instance: */false
+Instance: */true
 Instance: :*/true
 Merge: xdm
 Comment:
@@ -1742,10 +1742,10 @@
 
 Key: SessionsDirs
 Type: list
-Default: KDMDATA "/sessions"
+Default: "/usr/share/xsessions"
 User: core
 User: greeter-c
-Instance: #*/"/etc/X11/sessions,/usr/share/xsessions"
+Instance: #*/"/etc/X11/sessions"
 Comment:
  The directories containing session type definitions in .desktop format.
 Description:
@@ -1754,7 +1754,7 @@
 
 Key: ClientLogFile
 Type: string
-Default: ".xsession-errors"
+Default: ""
 User: core
 Instance: */".xsession-errors-%s"
 Instance: :0/".xsession-errors"
@@ -1804,9 +1804,9 @@
  None/LOGO_NONE: nothing
  Logo/LOGO_LOGO: the image specified by <option>LogoPixmap</option>
  Clock/LOGO_CLOCK: a neat analog clock
-Default: Clock
+Default: Logo
 User: greeter
-Instance: */Logo
+Instance: */Clock
 Comment:
  What should be shown in the greeter's logo are:
 Description:
@@ -1816,7 +1816,7 @@
 
 Key: LogoPixmap
 Type: string
-Default: ""
+Default: KDMDATA "/pics/pldlogo.png"
 User: greeter(logo)
 Instance: */KDMDATA "pics/kdelogo.png"
 Comment:
@@ -1851,7 +1851,7 @@
 
 Key: GreetString
 Type: string
-Default: "Welcome to %s at %n"
+Default: "Welcome to PLD Linux at %n"
 User: greeter
 Instance: #*/"K Desktop Environment (%n)"
 Comment:
@@ -1912,8 +1912,8 @@
 
 Key: StdFont
 Type: string
-Default: "helvetica,12,5,0,50,0"
-CDefault: "helvetica,12"
+Default: "helvetica,11,5,0,50,0"
+CDefault: "helvetica,11"
 User: greeter(normalFont):font
 Instance: #*/"helvetica,10,5,0,50,0"
 Comment: &
@@ -1922,8 +1922,8 @@
 
 Key: FailFont
 Type: string
-Default: "helvetica,12,5,0,75,0"
-CDefault: "helvetica,12,bold"
+Default: "helvetica,11,5,0,75,0"
+CDefault: "helvetica,11,bold"
 User: greeter:font
 Instance: #*/"helvetica,10,5,0,75,0"
 Comment: &
@@ -2033,7 +2033,7 @@
 
 Key: MinShowUID
 Type: int
-Default: 0
+Default: 500
 User: greeter(lowUserId)
 # will be overwritten
 Instance: #*/
@@ -2094,7 +2094,7 @@
 
 Key: FaceDir
 Type: string
-Default: *KDMDATA "/faces"
+Default: *KDMCONF "/faces"
 User: greeter
 Instance: #*/"/usr/share/faces"
 Update: upd_facedir
