--- kdebase-3.0.99/konsole/fonts/Makefile.am.orig	Fri Apr 28 22:08:08 2000
+++ kdebase-3.0.99/konsole/fonts/Makefile.am	Tue Nov 19 04:17:58 2002
@@ -1,4 +1,4 @@
-fontdir = $(prefix)/share/fonts
+fontdir = /usr/share/fonts/misc
 font_DATA =console8x16.pcf.gz console8x8.pcf.gz 9x15.pcf.gz
 
 EXTRA_DIST = $(font_DATA)
--- kdebase-3.0.99/konsole/fonts/Makefile.in.orig	Mon Nov 11 21:16:35 2002
+++ kdebase-3.0.99/konsole/fonts/Makefile.in	Tue Nov 19 04:18:17 2002
@@ -283,7 +283,7 @@
 qt_libraries = @qt_libraries@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
-fontdir = $(prefix)/share/fonts
+fontdir = /usr/share/fonts/misc
 font_DATA = console8x16.pcf.gz console8x8.pcf.gz 9x15.pcf.gz
 
 EXTRA_DIST = $(font_DATA)
