--- kdebase-030317/kcontrol/konsole/sessioneditor.cpp~	Sun Mar 16 00:35:44 2003
+++ kdebase-030317/kcontrol/konsole/sessioneditor.cpp	Tue Mar 18 10:16:38 2003
@@ -201,7 +201,7 @@
         i = co->readUnsignedNumEntry("Font",(unsigned int)-1);
         fontCombo->setCurrentItem(i+1);
 
-        str = co->readEntry("Term","xterm");
+        str = co->readEntry("Term","xterm-color");
         termLine->setText(str);
 
         str = co->readEntry("KeyTab","");
--- kdebase-050428/konsole/konsole/konsole.cpp.orig	2005-04-28 19:38:19.000000000 +0200
+++ kdebase-050428/konsole/konsole/konsole.cpp	2005-04-30 14:46:13.743872128 +0200
@@ -1013,7 +1013,7 @@
                               SLOT(slotSaveHistory()), m_shortcuts, "save_history");
   m_saveHistory->setEnabled(b_histEnabled );
 
-  m_clearHistory = new KAction(i18n("Clear &History"), "history_clear", 0, this,
+  m_clearHistory = new KAction(i18n("Clear &History"), "history_clear", Qt::CTRL+Qt::ALT+Qt::Key_W, this,
                                SLOT(slotClearHistory()), m_shortcuts, "clear_history");
   m_clearHistory->setEnabled(b_histEnabled);
 
@@ -2564,7 +2564,7 @@
                             const QString &_term,const QString &_icon,
                             const QString &_title, const QString &_cwd)
 {
-  QString emu = "xterm";
+  QString emu = "xterm-color";
   QString icon = "konsole";
   QString key;
   QString sch = s_kconfigSchema;
--- kdebase-3.3.92/konsole/konsole/main.cpp.orig	2005-02-04 00:19:05.000000000 +0100
+++ kdebase-3.3.92/konsole/konsole/main.cpp	2005-02-10 21:19:03.141540912 +0100
@@ -57,7 +57,7 @@
    { "name <name>",     I18N_NOOP("Set window class"), 0 },
    { "ls",              I18N_NOOP("Start login shell"), 0 },
    { "T <title>",       I18N_NOOP("Set the window title"), 0 },
-   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm" },
+   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm-color" },
    { "noclose",         I18N_NOOP("Do not close Konsole when command exits"), 0 },
    { "nohist",          I18N_NOOP("Do not save lines in history"), 0 },
    { "nomenubar",       I18N_NOOP("Do not display menubar"), 0 },
--- kdebase/konsole/konsole/konsole_part.cpp.orig	2005-09-25 12:47:42.000000000 +0200
+++ kdebase/konsole/konsole/konsole_part.cpp	2005-09-25 16:54:12.000000000 +0200
@@ -1059,7 +1059,7 @@
 void konsolePart::newSession()
 {
   if ( se ) delete se;
-  se = new TESession(te, "xterm", parentWidget->winId());
+  se = new TESession(te, "xterm-color", parentWidget->winId());
   connect( se,SIGNAL(done(TESession*)),
            this,SLOT(doneSession(TESession*)) );
   connect( se,SIGNAL(openURLRequest(const QString &)),
