--- kdebase-040531/konsole/konsole/default.keytab.h.orig	2004-05-31 18:25:21.000000000 +0200
+++ kdebase-040531/konsole/konsole/default.keytab.h	2004-05-31 18:39:11.481231816 +0200
@@ -8,7 +8,7 @@
   "key Return-Shift-NewLine : \"\\r\"\n"
   "key Return-Shift+NewLine : \"\\r\\n\"\n"
   "key Return+Shift         : \"\\EOM\"\n"
-  "key Backspace      : \"\\x7f\"\n"
+  "key Backspace      : \"\\x08\"\n"
   "key Up   -Shift-Ansi : \"\\EA\"\n"
   "key Down -Shift-Ansi : \"\\EB\"\n"
   "key Right-Shift-Ansi : \"\\EC\"\n"
