diff -urN kdebase-3.0beta2.orig/konsole/konsole/konsole.cpp kdebase-3.0beta2/konsole/konsole/konsole.cpp
--- kdebase-3.0beta2.orig/konsole/konsole/konsole.cpp	Sat Jan 26 17:04:54 2002
+++ kdebase-3.0beta2/konsole/konsole/konsole.cpp	Sun Feb 17 21:21:29 2002
@@ -1538,7 +1538,7 @@
 
 QString Konsole::newSession(KSimpleConfig *co, QString program, const QStrList &args, const QString &_term,const QString &_icon,const QString &_title)
 {
-  QString emu = "xterm";
+  QString emu = "xterm-color";
   QString icon = "openterm";
   QString key;
   QString sch = s_kconfigSchema;
diff -urN kdebase-3.0beta2.orig/konsole/konsole/konsole_part.cpp kdebase-3.0beta2/konsole/konsole/konsole_part.cpp
--- kdebase-3.0beta2.orig/konsole/konsole/konsole_part.cpp	Sat Jan 26 17:04:54 2002
+++ kdebase-3.0beta2/konsole/konsole/konsole_part.cpp	Sun Feb 17 21:21:29 2002
@@ -165,7 +165,7 @@
 
   setWidget(te);
   // faking a KMainwindow - TESession assumes that (wrong design!)
-  se = new TESession((KMainWindow*)parentWidget,te,shell,eargs,"xterm");
+  se = new TESession((KMainWindow*)parentWidget,te,shell,eargs,"xterm-color");
   connect( se,SIGNAL(done(TESession*,int)),
            this,SLOT(doneSession(TESession*,int)) );
   connect( te,SIGNAL(configureRequest(TEWidget*,int,int,int)),
diff -urN kdebase-3.0beta2.orig/konsole/konsole/main.cpp kdebase-3.0beta2/konsole/konsole/main.cpp
--- kdebase-3.0beta2.orig/konsole/konsole/main.cpp	Sat Jan 26 17:04:54 2002
+++ kdebase-3.0beta2/konsole/konsole/main.cpp	Sun Feb 17 21:22:42 2002
@@ -36,7 +36,7 @@
    { "name <name>",  I18N_NOOP("Set window class"), 0 },
    { "ls",    I18N_NOOP("Start login shell"), 0 },
    { "T <title>",       I18N_NOOP("Set the window title"), 0 },
-   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm" },
+   { "tn <terminal>",   I18N_NOOP("Specify terminal type as set in the TERM\nenvironment variable"), "xterm-color" },
    { "xwin",            I18N_NOOP("ignored"), 0 },
    { "nohist",          I18N_NOOP("Do not save lines in history"), 0 },
    { "nomenubar",       I18N_NOOP("Do not display menubar"), 0 },
