diff -urN kdebase-2.2beta1.orig/kdesktop/desktop.cc kdebase-2.2beta1/kdesktop/desktop.cc
--- kdebase-2.2beta1.orig/kdesktop/desktop.cc	Sun Jun 10 23:37:39 2001
+++ kdebase-2.2beta1/kdesktop/desktop.cc	Tue Jul 10 15:01:50 2001
@@ -23,6 +23,7 @@
 #include <sys/stat.h>
 #include <time.h>
 #include <unistd.h>
+#include <iostream>
 #include <kcolordrag.h>
 
 #include <qkeycode.h>
@@ -30,6 +31,7 @@
 #include <qdragobject.h>
 #include <qclipboard.h>
 #include <qevent.h>
+#include <qfile.h>
 #include <qrect.h>
 #include <qdir.h>
 
@@ -126,6 +128,9 @@
            this, SLOT(slotClipboardDataChanged()) );
 
   setURL( desktopURL() ); // sets m_url and m_dotDirectoryPath
+  need_arrange=QFile::exists(m_dotDirectoryPath);
+  kdDebug() << "***BERO: Checked " << m_dotDirectoryPath << " -> need_arrange = " << need_arrange << endl;
+  
   assert( !m_url.isMalformed() );
 
   // Dont repaint on configuration changes during construction
@@ -865,6 +870,12 @@
     }
 
     kdDebug() << "KDesktop::slotCompleted " << m_bNeedSave << " " << m_bNeedRepaint << endl;
+    kdDebug() << "***BERO: KDesktop::slotCompleted " << m_dotDirectoryPath << " -> need_arrange = " << need_arrange << endl;
+    if ( need_arrange ) {
+	need_arrange=false;
+	cerr << "***BERO: calling rearrangeIcons()" << endl;
+	rearrangeIcons();
+    }
     if ( m_bNeedSave )
         slotSaveIconPositions();
     if ( m_bNeedRepaint )
diff -urN kdebase-2.2beta1.orig/kdesktop/desktop.h kdebase-2.2beta1/kdesktop/desktop.h
--- kdebase-2.2beta1.orig/kdesktop/desktop.h	Sun Jun 10 23:37:39 2001
+++ kdebase-2.2beta1/kdesktop/desktop.h	Tue Jul 10 15:05:41 2001
@@ -270,6 +270,9 @@
   QStringList itemsAlwaysFirst;
   bool set_vroot;
   StartupId* startup_id;
+
+private:
+   bool need_arrange;
 };
 
 #endif
