Index: kcontrol/ebrowsing/plugins/ikws/searchproviders/fsd.desktop
===================================================================
--- kcontrol/ebrowsing/plugins/ikws/searchproviders/fsd.desktop	(.../tags/KDE/3.5.9/kdebase)	(wersja 780364)
+++ kcontrol/ebrowsing/plugins/ikws/searchproviders/fsd.desktop	(.../branches/KDE/3.5/kdebase)	(wersja 780364)
@@ -60,6 +60,7 @@
 Name[zh_TW]=FSF/UNESCO 自由軟體目錄
 ServiceTypes=SearchProvider
 Keys=fsd
-Query=http://directory.fsf.org/search/fsd-search.py?q=\\{@}
+Query=http://directory.fsf.org/search/?query=\\{@}
+Query[nds]=http://directory.fsf.org/search/fsd-search.py?q=\\{@}
 
 
Index: konqueror/konquerorsu.desktop
===================================================================
--- konqueror/konquerorsu.desktop	(.../tags/KDE/3.5.9/kdebase)	(wersja 780364)
+++ konqueror/konquerorsu.desktop	(.../branches/KDE/3.5/kdebase)	(wersja 780364)
@@ -60,7 +60,7 @@
 Name[pt]=Gestor de Ficheiros - Modo de Super Utilizador
 Name[pt_BR]=Gerenciador de Arquivos (Modo Super-Usuário)
 Name[ro]=Manager de fișiere - mod superutilizator
-Name[ru]=Менеджер файлов (с root-привилегиями)
+Name[ru]=Менеджер файлов (в режиме администратора)
 Name[rw]=Mucungadosiye - Uburyo Ukoresha Mukuru
 Name[se]=Fiilagieđahalli – Supergeavaheaddjemodus
 Name[sk]=Správca súborov - Super-používateľský mód
Index: konsole/other/sumc.desktop
===================================================================
--- konsole/other/sumc.desktop	(.../tags/KDE/3.5.9/kdebase)	(wersja 780364)
+++ konsole/other/sumc.desktop	(.../branches/KDE/3.5/kdebase)	(wersja 780364)
@@ -45,6 +45,7 @@
 Name[pt]=Midnight Commander como Root
 Name[pt_BR]=Midnight Commander - Root
 Name[ro]=Midnight Commander "root"
+Name[ru]=Midnight Commander c правами root
 Name[rw]=Mutangabwiriza Hagati-Ijoro y'Umuzi
 Name[se]=Midnight Commander root geavaheaddji várás
 Name[sl]=Midnight Commander za root
@@ -115,7 +116,7 @@
 Comment[pt]=Novo Midnight Commander como Root
 Comment[pt_BR]=Nova Janela do Midnight Commander- Root
 Comment[ro]=Consolă Midnight Commander "root" nouă
-Comment[ru]=Новое окно MC с правами root
+Comment[ru]=Новое окно Midnight Commander с правами root
 Comment[rw]=Mutangabwiriza Hagati-Ijoro y'Umuzi Nshya
 Comment[se]=Ođđa Midnight Commander root geavadeaddji várás
 Comment[sk]=Nový root Midnight Commander
Index: konsole/other/su.desktop
===================================================================
--- konsole/other/su.desktop	(.../tags/KDE/3.5.9/kdebase)	(wersja 780364)
+++ konsole/other/su.desktop	(.../branches/KDE/3.5/kdebase)	(wersja 780364)
@@ -43,6 +43,7 @@
 Name[pt]=Linha de Comandos Root
 Name[pt_BR]=Shell do Root
 Name[ro]=Interpretor "root"
+Name[ru]=Консоль с правами root
 Name[rw]=Sheli Umuzi
 Name[se]=Ruohtasskálžu
 Name[sl]=Korenska lupina
@@ -103,7 +104,7 @@
 Comment[pt]=Nova Linha de Comandos Root
 Comment[pt_BR]=Novo Shell do Root
 Comment[ro]=Interpretor "roo" nou
-Comment[ru]=Новая оболочка с правами root
+Comment[ru]=Новая консоль с правами root
 Comment[rw]=Sheli Umuzi Nshya
 Comment[se]=Ođđa ruohtasskálžu
 Comment[sk]=Nový Root shell
Index: konsole/other/shell.desktop
===================================================================
--- konsole/other/shell.desktop	(.../tags/KDE/3.5.9/kdebase)	(wersja 780364)
+++ konsole/other/shell.desktop	(.../branches/KDE/3.5/kdebase)	(wersja 780364)
@@ -44,6 +44,7 @@
 Name[pl]=Powłoka
 Name[pt]=Linha de Comandos
 Name[ro]=Interpretor
+Name[ru]=Консоль
 Name[rw]=Sheli
 Name[se]=Skálžu
 Name[sl]=Ukazna lupina
@@ -114,7 +115,7 @@
 Comment[pt]=Nova Linha de Comandos
 Comment[pt_BR]=Novo Shell
 Comment[ro]=Interpretor nou
-Comment[ru]=Новое окно Shell
+Comment[ru]=Новая консоль
 Comment[rw]=Sheli Nshya
 Comment[se]=Ođđa skálžu
 Comment[sk]=Nový shell
Index: konsole/other/linux.desktop
===================================================================
--- konsole/other/linux.desktop	(.../tags/KDE/3.5.9/kdebase)	(wersja 780364)
+++ konsole/other/linux.desktop	(.../branches/KDE/3.5/kdebase)	(wersja 780364)
@@ -125,7 +125,7 @@
 Comment[pt]=Nova Consola de Linux
 Comment[pt_BR]=Novo Console do Linux
 Comment[ro]=Consolă Linux nouă
-Comment[ru]=Новая консоль
+Comment[ru]=Новая консоль Linux
 Comment[rw]=Konsole Linux Nshya
 Comment[se]=Ođđa Linux-konsolla
 Comment[sk]=Nová linuxová konzola
Index: ksysguard/ksysguardd/Linux/lmsensors.c
===================================================================
--- ksysguard/ksysguardd/Linux/lmsensors.c	(.../tags/KDE/3.5.9/kdebase)	(wersja 780364)
+++ ksysguard/ksysguardd/Linux/lmsensors.c	(.../branches/KDE/3.5/kdebase)	(wersja 780364)
@@ -29,7 +29,7 @@
 
 #include "lmsensors.h"
 
-#ifdef HAVE_LMSENSORS
+#ifdef HAVE_SENSORS_SENSORS_H
 #include <sensors/sensors.h>
 
 #ifndef SENSORS_API_VERSION
@@ -293,7 +293,7 @@
 #endif
 }
 
-#else /* HAVE_LMSENSORS */
+#else /* HAVE_SENSORS_SENSORS_H */
 
 /* dummy version for systems that have no lmsensors support */
 

Zmiany atrybutów dla: .
___________________________________________________________________
Nazwa: svn:mergeinfo
   - 
Nazwa: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


