--- kdebase-3.5.5/kicker/applets/minipager/pagerapplet.cpp.org	2006-10-05 06:43:59.241437000 +0000
+++ kdebase-3.5.5/kicker/applets/minipager/pagerapplet.cpp	2006-10-05 06:45:14.378132750 +0000
@@ -124,7 +124,7 @@
     desktopLayoutY = -1;
 
     QSize s(m_kwin->numberOfViewports(m_kwin->currentDesktop()));
-    m_useViewports = s.width() + s.height() > 1;
+    m_useViewports = s.width() * s.height() > 1;
 
     drawButtons();
 
@@ -223,7 +223,7 @@
 
 void KMiniPager::slotButtonSelected( int desk )
 {
-    if (m_kwin->numberOfViewports(0).width()+m_kwin->numberOfViewports(0).height() > 1)
+    if (m_kwin->numberOfViewports(0).width() * m_kwin->numberOfViewports(0).height() > 1)
     {
         QPoint p;
 
@@ -435,7 +435,7 @@
 {
     int newDesk;
     int desktops = KWin::numberOfDesktops();
-    if (m_kwin->numberOfViewports(0).width() + m_kwin->numberOfViewports(0).height() > 1 )
+    if (m_kwin->numberOfViewports(0).width() * m_kwin->numberOfViewports(0).height() > 1 )
         desktops = m_kwin->numberOfViewports(0).width() * m_kwin->numberOfViewports(0).height();
     if (e->delta() < 0)
     {
--- kdebase-3.5.5/kicker/applets/minipager/pagerbutton.cpp.org	2006-10-05 06:44:04.849787500 +0000
+++ kdebase-3.5.5/kicker/applets/minipager/pagerbutton.cpp	2006-10-05 06:46:36.743280250 +0000
@@ -116,12 +116,17 @@
       return false;
 
     if (m_useViewports) {
-        QPoint windowSpot = info->frameGeometry().topLeft();
+	QPoint windowTopLeft = info->frameGeometry().topLeft();
+	QPoint windowBottomRight = info->frameGeometry().bottomRight();
 
-        windowSpot += QPoint((m_desktop - m_pager->kwin()->currentViewport(0).x()) *
-                QApplication::desktop()->width(), 0);
-
-        if (!QApplication::desktop()->screenGeometry().contains(windowSpot))
+	windowTopLeft += QPoint((m_desktop - m_pager->kwin()->currentViewport(0).x()) *
+			QApplication::desktop()->width(), 0);
+	windowBottomRight += QPoint((m_desktop - m_pager->kwin()->currentViewport(0).x()) *
+			QApplication::desktop()->width(), 0);
+
+	if (!(info->state() & NET::Sticky) &&
+			!QApplication::desktop()->geometry().contains(windowTopLeft) &&
+			!QApplication::desktop()->geometry().contains(windowBottomRight))
             return false;
     }
 
