--- kdebase-2.1/kcontrol/colors/colorscm.cpp.deush	Tue Mar  6 15:28:27 2001
+++ kdebase-2.1/kcontrol/colors/colorscm.cpp	Tue Mar  6 15:30:37 2001
@@ -63,7 +63,7 @@
 
     KConfig *cfg = new KConfig("kcmdisplayrc");
     cfg->setGroup("X11");
-    useRM = cfg->readBoolEntry("useResourceManager", true);
+    useRM = cfg->readBoolEntry("useResourceManager", false);
     delete cfg;
 
     cs = new WidgetCanvas( this );
--- kdebase-2.1/kcontrol/display/general.cpp.deush	Tue Mar  6 15:32:44 2001
+++ kdebase-2.1/kcontrol/display/general.cpp	Tue Mar  6 15:34:57 2001
@@ -112,7 +112,7 @@
     void init_style() {
         KConfig config("kcmdisplayrc", true, true);
         config.setGroup("X11");
-        if (config.readBoolEntry( "useResourceManager", true ))
+        if (config.readBoolEntry( "useResourceManager", false ))
         {
 	  runRdb();
           applyGtkStyles(true);
@@ -494,7 +494,7 @@
     tbMoveTransparent = config->readBoolEntry( "TransparentMoving", true);
 
     config->setGroup("X11");
-    useRM = config->readBoolEntry( "useResourceManager", true );
+    useRM = config->readBoolEntry( "useResourceManager", false );
 }
 
 
--- kdebase-2.1/kcontrol/fonts/fonts.cpp.deush	Tue Mar  6 15:31:03 2001
+++ kdebase-2.1/kcontrol/fonts/fonts.cpp	Tue Mar  6 15:31:39 2001
@@ -351,7 +351,7 @@
   KConfig * cfg = KGlobal::config();
   cfg->setGroup("X11");
 
-  if (cfg->readBoolEntry("useResourceManager", true)) {
+  if (cfg->readBoolEntry("useResourceManager", false)) {
     QApplication::setOverrideCursor( waitCursor );
     KProcess proc;
     proc.setExecutable("krdb");
