--- kdebase-3.4.0/konqueror/konq_mainwindow.cc.orig	2005-04-13 08:34:19.000000000 +0200
+++ kdebase-3.4.0/konqueror/konq_mainwindow.cc	2005-04-13 08:34:23.000000000 +0200
@@ -3752,7 +3752,7 @@
   m_paActivateNextTab = new KAction( i18n( "Activate Next Tab" ), "tab_next", QApplication::reverseLayout() ? KStdAccel::tabPrev() : KStdAccel::tabNext(), this, SLOT( slotActivateNextTab() ), actionCollection(), "activatenexttab" );
   m_paActivatePrevTab = new KAction( i18n( "Activate Previous Tab" ), "tab_previous", QApplication::reverseLayout() ? KStdAccel::tabNext() : KStdAccel::tabPrev(), this, SLOT( slotActivatePrevTab() ), actionCollection(), "activateprevtab" );
 
-  char actionname[15];
+  char actionname[16];
   for (int i=1;i<13;i++) {
     sprintf(actionname,"activate_tab_%02d", i);
     new KAction(i18n("Activate Tab %1").arg(i), 0, this, SLOT(slotActivateTab()), actionCollection(), actionname);
