--- kdebase-3.1.2/kdm/kfrontend/kgreeter.cpp.orig	2003-05-12 15:48:50.000000000 +0200
+++ kdebase-3.1.2/kdm/kfrontend/kgreeter.cpp	2003-05-17 13:18:55.000000000 +0200
@@ -114,8 +114,7 @@
   , capslocked( -1 )
   , loginfailed( false )
 {
-    user_pic_dir = KGlobal::dirs()->resourceDirs( "data" ).last() +
-				      QString::fromLatin1("kdm/pics/users/");	/* XXX standardize */
+    user_pic_dir = QString::fromLatin1("/etc/X11/kdm/pics/users/");	/* XXX standardize */
 
     QGridLayout* main_grid = new QGridLayout( winFrame, 4, 2, 10 );
     QBoxLayout* hbox1 = new QHBoxLayout( 10 );
@@ -273,7 +272,7 @@
 
     UpdateLock();
 
-    stsfile = new KSimpleConfig( QString::fromLatin1(KDE_CONFDIR "/kdm/kdmsts") ); // XXX get from kdm_config
+    stsfile = new KSimpleConfig( QString::fromLatin1("/etc/X11/kdm/kdmsts") ); // XXX get from kdm_config
     stsfile->setGroup( "PrevUser" );
     enam = QString::fromLocal8Bit( dname );
     if (kdmcfg->_preselUser != PRESEL_PREV)
