diff -uNr kdebase-3.5.4/kioslave/media/configure.in.in kdebase-3.5.4.patched/kioslave/media/configure.in.in
--- kdebase-3.5.4/kioslave/media/configure.in.in	2006-07-22 08:15:45.000000000 +0000
+++ kdebase-3.5.4.patched/kioslave/media/configure.in.in	2006-08-18 14:17:29.489632500 +0000
@@ -106,12 +106,12 @@
   dbusqt_lib=NOTFOUND
   dbusqt=NOTFOUND
 
-  search_incs="$kde_includes $kde_extra_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
-  AC_FIND_FILE(dbus/connection.h, $search_incs, dbusqt_incdir)
+  search_incs="$kde_includes $kde_extra_includes /usr/include /usr/include/dbus-1.0/qt3 /usr/local/include /usr/local/include/dbus-1.0"
+  AC_FIND_FILE(dbus/qdbusconnection.h, $search_incs, dbusqt_incdir)
 
-  if test -r $dbusqt_incdir/dbus/connection.h ; then
+  if test -r $dbusqt_incdir/dbus/qdbusconnection.h ; then
     have_qt_patch=0
-    grep dbus_connection_setup_with_qt_main $dbusqt_incdir/dbus/connection.h \
+    grep dbus_connection_setup_with_qt_main $dbusqt_incdir/dbus/qdbusconnection.h \
     > /dev/null 2>&1 && have_qt_patch=1
     if test $have_qt_patch = 1 ; then
       DBUSQT_INCS="-I$dbusqt_incdir"
@@ -120,20 +120,20 @@
   fi
 
   search_libs="$kde_libraries $kde_extra_libs /usr/lib$kdelibsuff /usr/local/lib$kdelibsuff"
-  AC_FIND_FILE(libdbus-qt-1.so, $search_libs, dbusqt_libdir)
+  AC_FIND_FILE(libdbus-1-qt3.so, $search_libs, dbusqt_libdir)
 
-  if test -r $dbusqt_libdir/libdbus-qt-1.so ; then
-    DBUSQT_LIBS="-L$dbusqt_libdir -ldbus-qt-1"
+  if test -r $dbusqt_libdir/libdbus-1-qt3.so ; then
+    DBUSQT_LIBS="-L$dbusqt_libdir -ldbus-1-qt3"
     dbusqt_lib=FOUND
   fi
 
   if test $dbusqt_inc != FOUND  || test $dbusqt_lib != FOUND ; then 
 
     search_incs="`pkg-config --cflags dbus-1  |sed 's/-I//g'`"
-    AC_FIND_FILE(dbus/connection.h, $search_incs, dbusqt_incdir)
-    if test -r $dbusqt_incdir/dbus/connection.h ; then
+    AC_FIND_FILE(dbus/qdbusconnection.h, $search_incs, dbusqt_incdir)
+    if test -r $dbusqt_incdir/dbus/qdbusconnection.h ; then
       have_qt_patch=0
-      grep dbus_connection_setup_with_qt_main $dbusqt_incdir/dbus/connection.h \
+      grep dbus_connection_setup_with_qt_main $dbusqt_incdir/dbus/qdbusconnection.h \
       > /dev/null 2>&1 && have_qt_patch=1
       if test $have_qt_patch = 1 ; then
         DBUSQT_INCS="-I$dbusqt_incdir"
@@ -142,10 +142,10 @@
     fi
 
     search_libs="`pkg-config --libs dbus-1 --libs-only-L | sed 's/-L//g'`"
-    AC_FIND_FILE(libdbus-qt-1.so, $search_libs, dbusqt_libdir)
+    AC_FIND_FILE(libdbus-1-qt3.so, $search_libs, dbusqt_libdir)
 
-    if test -r $dbusqt_libdir/libdbus-qt-1.so ; then
-      DBUSQT_LIBS="-L$dbusqt_libdir -ldbus-qt-1"
+    if test -r $dbusqt_libdir/libdbus-1-qt3.so ; then
+      DBUSQT_LIBS="-L$dbusqt_libdir -ldbus-1-qt3"
       dbusqt_lib=FOUND
     fi
 
diff -uNr kdebase-3.5.4/kioslave/media/mediamanager/halbackend.cpp kdebase-3.5.4.patched/kioslave/media/mediamanager/halbackend.cpp
--- kdebase-3.5.4/kioslave/media/mediamanager/halbackend.cpp	2006-08-18 14:15:56.115797000 +0000
+++ kdebase-3.5.4.patched/kioslave/media/mediamanager/halbackend.cpp	2006-08-18 14:07:18.865346000 +0000
@@ -346,7 +346,7 @@
 
 void HALBackend::MainLoopIntegration(DBusConnection *dbusConnection)
 {
-    m_dBusQtConnection = new DBusQt::Connection(m_parent);
+    m_dBusQtConnection = new QDBusConnection();
     m_dBusQtConnection->dbus_connection_setup_with_qt_main(dbusConnection);
 }
 
diff -uNr kdebase-3.5.4/kioslave/media/mediamanager/halbackend.h kdebase-3.5.4.patched/kioslave/media/mediamanager/halbackend.h
--- kdebase-3.5.4/kioslave/media/mediamanager/halbackend.h	2006-07-22 08:15:45.000000000 +0000
+++ kdebase-3.5.4.patched/kioslave/media/mediamanager/halbackend.h	2006-08-18 14:04:38.091715250 +0000
@@ -40,7 +40,7 @@
 /* We acknowledge the the dbus API is unstable */
 #define DBUS_API_SUBJECT_TO_CHANGE
 /* DBus-Qt bindings */
-#include <dbus/connection.h>
+#include <dbus/qdbusconnection.h>
 /* HAL libraries */
 #include <libhal.h>
 #include <libhal-storage.h>
@@ -207,7 +207,7 @@
 	/**
 	* The DBus-Qt bindings connection for mainloop integration
 	*/
-	DBusQt::Connection*	m_dBusQtConnection;
+	QDBusConnection*	m_dBusQtConnection;
 
 	/**
 	* Object for the kded module
