--- kdebase-3.2.1/kdm/kfrontend/genkdmconf.c.orig	2004-03-12 11:09:59.000000000 +0100
+++ kdebase-3.2.1/kdm/kfrontend/genkdmconf.c	2004-03-12 11:11:54.000000000 +0100
@@ -532,7 +532,7 @@
 #endif
 
 #ifdef __linux__
-# define DEF_SERVER_LINE ":0 local@tty1 " XBINDIR "/X -nolisten tcp vt7"
+# define DEF_SERVER_LINE ":0 local@tty1 " XBINDIR "/X -nolisten tcp vt9"
 /* ###
 # define DEF_SERVER_LINES DEF_SERVER_LINE "\n" \
 			 ":1 local@tty2 reserve " XBINDIR "/X -nolisten tcp :1 vt8\n" \
@@ -541,10 +541,8 @@
 			 "#:4 local@tty5 reserve " XBINDIR "/X -nolisten tcp :4 vt11\n"
 */
 # define DEF_SERVER_LINES DEF_SERVER_LINE "\n" \
-			 "#:1 local@tty2 reserve " XBINDIR "/X -nolisten tcp :1 vt8\n" \
-			 "#:2 local@tty3 reserve " XBINDIR "/X -nolisten tcp :2 vt9\n" \
-			 "#:3 local@tty4 reserve " XBINDIR "/X -nolisten tcp :3 vt10\n" \
-			 "#:4 local@tty5 reserve " XBINDIR "/X -nolisten tcp :4 vt11\n"
+			 "#:1 local@tty2 reserve " XBINDIR "/X -nolisten tcp :1 vt10\n" \
+			 "#:2 local@tty3 reserve " XBINDIR "/X -nolisten tcp :2 vt11\n" 
 #elif defined(__sun__)
 # define DEF_SERVER_LINE ":0 local@console " XBINDIR "/X -nolisten tcp"
 #elif defined(_AIX)
