--- kdebase-030317/kcontrol/kdm/background.cpp~	Sun Mar 16 00:35:44 2003
+++ kdebase-030317/kcontrol/kdm/background.cpp	Tue Mar 18 19:31:12 2003
@@ -53,7 +53,7 @@
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
     config->setGroup( "X-*-Greeter" );
-    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg",KDE_CONFDIR "/kdm/backgroundrc" ) );
+    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg","/etc/X11/kdm/backgroundrc" ) );
     m_background = new BGDialog( this, m_simpleConf, false );
 
     connect(m_background, SIGNAL(changed(bool)), SIGNAL(changed(bool)));
@@ -105,7 +105,7 @@
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
-    m_background->defaults();
+    m_background->kdmdefaults();
     emit changed(true);
 }
 
-void KBackground::defaults()
+void KBackground::kdmdefaults()
 {
     m_pCBEnable->setChecked( true );
     slotEnableChanged();
--- kdebase-031010/kcontrol/kdm/kdm-appear.cpp.orig	2003-09-23 21:46:11.000000000 +0200
+++ kdebase-031010/kcontrol/kdm/kdm-appear.cpp	2003-10-10 19:41:03.000000000 +0200
@@ -385,7 +385,7 @@
 bool KDMAppearanceWidget::setLogo(QString logo)
 {
     QString flogo = logo.isEmpty() ?
-                    locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") ) :
+                    locate("data", QString::fromLatin1("kdm/pics/pldlogo.png") ) :
                     logo;
     QPixmap p(flogo);
     if (p.isNull())
@@ -516,7 +516,7 @@
   config->setGroup("X-*-Greeter");
 
   // Read the greeting string
-  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to %s at %n")));
+  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to PLD Linux at %n")));
 
   // Regular logo or clock
   QString logoArea = config->readEntry("LogoArea", "Logo" );
@@ -566,7 +566,7 @@
 
 void KDMAppearanceWidget::defaults()
 {
-  greetstr_lined->setText( i18n("Welcome to %s at %n") );
+  greetstr_lined->setText( i18n("Welcome to PLD Linux at %n") );
   logoRadio->setChecked( true );
   slotAreaRadioClicked( KdmLogo );
   posCenterRadio->setChecked( true );
--- kdebase-040116/kcontrol/kdm/kdm-font.cpp.orig	2004-01-16 23:07:36.000000000 +0100
+++ kdebase-040116/kcontrol/kdm/kdm-font.cpp	2004-01-16 23:25:36.000000000 +0100
@@ -87,8 +87,8 @@
 
 void KDMFontWidget::set_def()
 {
-  stdFontChooser->setFont(QFont("helvetica", 12));
-  failFontChooser->setFont(QFont("helvetica", 12, QFont::Bold));
+  stdFontChooser->setFont(QFont("helvetica", 11));
+  failFontChooser->setFont(QFont("helvetica", 11, QFont::Bold));
   greetingFontChooser->setFont(QFont("charter", 24));
 }
 
--- kdebase-030418/kcontrol/kdm/kdm-sess.cpp~	2003-04-04 00:37:58.000000000 +0200
+++ kdebase-030418/kcontrol/kdm/kdm-sess.cpp	2003-04-19 19:26:08.000000000 +0200
@@ -364,7 +364,7 @@
   QString str;
 
   config->setGroup("X-:*-Core");
-  readSD(sdlcombo, "All");
+  readSD(sdlcombo, "Root");
 
   config->setGroup("X-*-Core");
   readSD(sdrcombo, "Root");
--- kdebase-030924/kcontrol/kdm/kdm-users.cpp~	2003-09-23 21:46:11.000000000 +0200
+++ kdebase-030924/kcontrol/kdm/kdm-users.cpp	2003-09-24 16:39:25.000000000 +0200
@@ -65,7 +65,7 @@
     // We assume that $kde_datadir/kdm exists, but better check for pics/ and pics/users,
     // and create them if necessary.
     config->setGroup( "X-*-Greeter" );
-    m_userPixDir = config->readEntry( "FaceDir", KGlobal::dirs()->resourceDirs("data").last() + "kdm/faces" ) + '/';
+    m_userPixDir = config->readEntry( "FaceDir", "/etc/X11/kdm/faces" ) + '/';
     m_notFirst = false;
     QDir testDir( m_userPixDir );
     if ( !testDir.exists() && !testDir.mkdir( testDir.absPath() ))
--- kdebase-3.1/kcontrol/kdm/main.cpp.orig	Sun Oct 27 00:05:07 2002
+++ kdebase-3.1/kcontrol/kdm/main.cpp	Thu Nov 21 02:51:19 2002
@@ -92,7 +92,7 @@
     usermap.insert( QString::fromLocal8Bit( ps->pw_name ), ps->pw_uid );
   endpwent();
 
-  config = new KSimpleConfig( QString::fromLatin1( KDE_CONFDIR "/kdm/kdmrc" ));
+  config = new KSimpleConfig( QString::fromLatin1( "/etc/X11/kdm/kdmrc" ));
 
   QVBoxLayout *top = new QVBoxLayout(this);
   tab = new QTabWidget(this);
--- kdebase-030930/kdm/kfrontend/Makefile.am~	2003-09-30 00:47:50.000000000 +0200
+++ kdebase-030930/kdm/kfrontend/Makefile.am	2003-09-30 10:06:29.000000000 +0200
@@ -38,7 +38,7 @@
 genkdmconf_LDADD = $(LIB_X11)
 
 install-data-local: genkdmconf
-	./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm --face-src $(srcdir)/pics $(GENKDMCONF_FLAGS)
+	./genkdmconf --no-old --in $(DESTDIR)/etc/X11/kdm --face-src $(srcdir)/pics $(GENKDMCONF_FLAGS)
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
--- kdebase/kdm/kfrontend/genkdmconf.c.orig	2004-03-23 14:55:38.000000000 +0100
+++ kdebase/kdm/kfrontend/genkdmconf.c	2004-03-25 21:55:08.236691776 +0100
@@ -63,7 +63,7 @@
 #define __stringify(x) #x
 #define stringify(x) __stringify(x)
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #define RCVERMAJOR 2
@@ -531,7 +531,7 @@
 #endif
 
 #ifdef __linux__
-# define DEF_SERVER_LINE ":0 local@tty1 " XBINDIR "/X -nolisten tcp vt7"
+# define DEF_SERVER_LINE ":0 local@tty1 " XBINDIR "/X -nolisten tcp vt9"
 /* ###
 # define DEF_SERVER_LINES DEF_SERVER_LINE "\n" \
 			 ":1 local@tty2 reserve " XBINDIR "/X -nolisten tcp :1 vt8\n" \
@@ -540,10 +540,8 @@
 			 "#:4 local@tty5 reserve " XBINDIR "/X -nolisten tcp :4 vt11\n"
 */
 # define DEF_SERVER_LINES DEF_SERVER_LINE "\n" \
-			 "#:1 local@tty2 reserve " XBINDIR "/X -nolisten tcp :1 vt8\n" \
-			 "#:2 local@tty3 reserve " XBINDIR "/X -nolisten tcp :2 vt9\n" \
-			 "#:3 local@tty4 reserve " XBINDIR "/X -nolisten tcp :3 vt10\n" \
-			 "#:4 local@tty5 reserve " XBINDIR "/X -nolisten tcp :4 vt11\n"
+			 "#:3 local@tty4 reserve " XBINDIR "/X -nolisten tcp :1 vt10\n" \
+			 "#:4 local@tty5 reserve " XBINDIR "/X -nolisten tcp :2 vt11\n"
 #elif defined(__sun__)
 # define DEF_SERVER_LINE ":0 local@console " XBINDIR "/X -nolisten tcp"
 #elif defined(_AIX)
@@ -793,14 +791,15 @@
 "esac\n"
 "exec xmessage -center -buttons OK:0 -default OK \"Sorry, cannot execute $1. Check $DESKTOP_SESSION.desktop.\"\n";
 
+/* Keep in sync with bgdefaults.h & BGDialog::kdmdefaults() */
 static const char def_background[] =
 "[Desktop0]\n"
-"BackgroundMode=Flat\n"
+"BackgroundMode=VerticalGradient\n"
 "BlendBalance=100\n"
 "BlendMode=NoBlending\n"
 "ChangeInterval=60\n"
-"Color1=0,0,200\n"
-"Color2=192,192,192\n"
+"Color1=57,55,75\n"
+"Color2=66,82,143\n"
 "CurrentWallpaper=0\n"
 "LastChange=0\n"
 "MinOptimizationDepth=1\n"
@@ -809,7 +808,7 @@
 "Program=\n"
 "ReverseBlending=false\n"
 "UseSHM=false\n"
-"Wallpaper=default_blue.jpg\n"
+"Wallpaper=kdm_pld.png\n"
 "WallpaperList=\n"
 "WallpaperMode=Scaled\n";
 
@@ -2019,18 +2018,20 @@
 "# this opens a security hole: a secured display lock can be circumvented\n"
 "# (unless you use KDE's built-in screen lock). Default is false\n" },
 { "AllowRootLogin",	0, 0, 
-"# Allow root logins? Default is true\n" },
+"# Allow root logins? Default is false\n" },
 { "AllowNullPasswd",	0, 0, 
-"# Allow to log in, when user has set an empty password? Default is true\n" },
+"# Allow to log in, when user has set an empty password? Default is false\n" },
 { "AllowShutdown",	0, 0, 
 "# Who is allowed to shut down the system. This applies both to the\n"
-"# greeter and to the command FiFo. Default is All\n"
+"# greeter and to the command FiFo.\n"
 "# \"None\" - no \"Shutdown...\" button is shown at all\n"
 "# \"Root\" - the root password must be entered to shut down\n"
-"# \"All\" - everybody can shut down the machine (Default)\n" },
+"# \"All\" - everybody can shut down the machine\n"
+"# Default value is \"Root\". For local displays we set this option to \"All\"\n"  
+"# (see the \"[X-:*-Core]\" section below).\n" },
 { "AllowSdForceNow",	0, 0, 
 "# Who is allowed to abort all still running sessions when shutting down.\n"
-"# Same options as for AllowShutdown. Default is All\n" },
+"# Same options as for AllowShutdown. Default is Root\n" },
 { "DefaultSdMode",	0, 0, 
 "# The default choice for the shutdown condition/timing.\n"
 "# \"Schedule\" - shutdown after all sessions exit (possibly at once) (Default)\n"
@@ -2065,11 +2066,11 @@
 "# it, you should \"chmod 600 kdmrc\" for obvious reasons. Default is \"\"\n" },
 { "SessionsDirs",	0, 0, 
 "# The directories containing session type definitions in .desktop format.\n"
-"# Default is " KDMDATA "/sessions\n" },
+"# Default is /usr/share/xsessions\n" },
 { "ClientLogFile",	0, 0,
 "# The file (relative to $HOME) to redirect the session output to. This is\n"
 "# a printf format string; one %s will be replaced with the display name.\n"
-"# Default is \".xsession-errors\"\n" },
+"# Default is \"\"\n" },
 };
 
 static Ent entsGreeter[] = {
@@ -2085,7 +2086,7 @@
 "# \"Clock\" - a neat analog clock\n"
 "# \"None\" - nothing\n" },
 { "LogoPixmap",		0, 0, 
-"# The image to show when LogoArea=Logo. Default is kdelogo.png\n" },
+"# The image to show when LogoArea=Logo. Default is pldlogo.png\n" },
 { "GreeterPosFixed",	0, 0, 
 "# Normally, the greeter is centered on the screen. Use this, if you want\n"
 "# it to appear elsewhere on the screen. Default is false\n" },
@@ -2106,13 +2107,13 @@
 "# - %r -> the operating system's version\n"
 "# - %m -> the machine (hardware) type\n"
 "# - %% -> a single %\n"
-"# Default is \"Welcome to %s at %n\"\n" },
+"# Default is \"Welcome to PLD Linux at %n\"\n" },
 { "GreetFont",		0, 0, 
 "# The font for the headline. Default is charter,24,bold\n" },
 { "StdFont",		0, 0, 
-"# The normal font used in the greeter. Default is helvetica,12\n" },
+"# The normal font used in the greeter. Default is helvetica,11\n" },
 { "FailFont",		0, 0, 
-"# The font used for the \"Login Failed\" message. Default is helvetica,12,bold\n" },
+"# The font used for the \"Login Failed\" message. Default is helvetica,11,bold\n" },
 { "AntiAliasing",	0, 0,
 "# Whether the fonts shown in the greeter should be antialiased. Default is false\n" },
 { "NumLock",		0, 0,
@@ -2151,7 +2152,7 @@
 "# \"PreferUser\" - ... and the other way round\n" },
 { "FaceDir",		0, upd_facedir, 
 "# The directory containing the user images if FaceSource is not UserOnly.\n"
-"# Default is " KDE_DATADIR "/kdm/faces\n" },
+"# Default is " KDMCONF "/faces\n" },
 { "PreselectUser",	0, 0, 
 "# Specify, if/which user should be preselected for log in.\n"
 "# Note, that enabling this feature can be considered a security hole,\n"
@@ -2345,14 +2346,14 @@
 { "SystemShell",	"/bin/bash", 0 },
 { "UserAuthDir",	"", 0 },
 { "AutoReLogin",	"true", 0 },
-{ "AllowRootLogin",	"false", 1 },
-{ "AllowNullPasswd",	"false", 1 },
-{ "AllowShutdown",	"Root", 1 },
-{ "AllowSdForceNow",	"Root", 0 },
+{ "AllowRootLogin",	"true", 0 },
+{ "AllowNullPasswd",	"true", 0 },
+{ "AllowShutdown",	"All", 0 },
+{ "AllowSdForceNow",	"All", 0 },
 { "DefaultSdMode",	"ForceNow", 0}, 
 { "InteractiveSd",	"false", 0}, 
 { "SessionsDirs",	"/etc/X11/sessions,/usr/share/xsessions", 0 },
-{ "ClientLogFile",	".xsession-errors-%s", 1 },
+{ "ClientLogFile",	".xsession-errors-%s", 0 },
 };
 
 static DEnt dEntsAnyGreeter[] = {
@@ -2399,8 +2400,8 @@
 { "ServerAttempts",	"", 0 },
 { "ServerTimeout",	"", 0 },
 { "AllowShutdown",	"All", 1 },
-{ "AllowRootLogin",	"true", 1 },
-{ "AllowNullPasswd",	"true", 1 },
+{ "AllowRootLogin",	"true", 0 },
+{ "AllowNullPasswd",	"true", 0 },
 { "NoPassEnable",	"true", 0 },
 { "NoPassUsers",	"fred,ethel", 0 },
 };
@@ -2417,7 +2418,7 @@
 { "AutoLoginEnable",	"true", 0 },
 { "AutoLoginUser",	"fred", 0 },
 { "AutoLoginPass",	"secret!", 0 },
-{ "ClientLogFile",	".xsession-errors", 1 },
+{ "ClientLogFile",	".xsession-errors", 0 },
 };
 
 static DEnt dEnts0Greeter[] = {
--- kdebase/kdm/kfrontend/kdm_config.c.orig	2004-03-23 14:55:38.000000000 +0100
+++ kdebase/kdm/kfrontend/kdm_config.c	2004-03-25 21:51:15.423084824 +0100
@@ -34,7 +34,7 @@
 #include <netdb.h>
 #include <grp.h>
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 
@@ -515,14 +515,14 @@
 { "AutoLoginUser",	C_autoUser,	(void *)PautoLoginX,	"" },
 { "AutoLoginPass",	C_autoPass,	(void *)PautoLoginX,	"" },
 { "AutoReLogin",	C_autoReLogin | C_BOOL,	0,	"false" },
-{ "AllowNullPasswd",	C_allowNullPasswd | C_BOOL, 0,	"true" },
-{ "AllowRootLogin",	C_allowRootLogin | C_BOOL, 0,	"true" },
-{ "AllowShutdown",	C_allowShutdown | C_ENUM, sd_who, "All" },
-{ "AllowSdForceNow",	C_allowNuke | C_ENUM, sd_who,	"All" },
+{ "AllowNullPasswd",	C_allowNullPasswd | C_BOOL, 0,	"false" },
+{ "AllowRootLogin",	C_allowRootLogin | C_BOOL, 0,	"false" },
+{ "AllowShutdown",	C_allowShutdown | C_ENUM, sd_who, "Root" },
+{ "AllowSdForceNow",	C_allowNuke | C_ENUM, sd_who,	"Root" },
 { "DefaultSdMode",	C_defSdMode | C_ENUM, sd_mode,	"Schedule" },
 { "InteractiveSd",	C_interactiveSd | C_BOOL, 0,	"true" },
-{ "SessionsDirs",	C_sessionsDirs,		0,	KDMDATA "/sessions" },
-{ "ClientLogFile",	C_clientLogFile,	0,	".xsession-errors" },
+{ "SessionsDirs",	C_sessionsDirs,		0,	"/usr/share/xsessions" },
+{ "ClientLogFile",	C_clientLogFile,	0,	"" },
 };
 
 Ent entsGreeter[] = {
@@ -534,9 +534,9 @@
 { "GreeterPosX",	C_GreeterPosX,		0,	"0" },
 { "GreeterPosY",	C_GreeterPosY,		0,	"0" },
 { "GreeterScreen",	C_GreeterScreen,	0,	"0" },
-{ "StdFont",		C_StdFont,		0,	"helvetica,12,5,0,50,0" },
-{ "FailFont",		C_FailFont,		0,	"helvetica,12,5,0,75,0" },
-{ "GreetString",	C_GreetString,		0,	"Welcome to %s at %n" },
+{ "StdFont",		C_StdFont,		0,	"helvetica,11,5,0,50,0" },
+{ "FailFont",		C_FailFont,		0,	"helvetica,11,5,0,75,0" },
+{ "GreetString",	C_GreetString,		0,	"Welcome to PLD Linux at %n" },
 { "GreetFont",		C_GreetFont,		0,	"charter,24,5,0,50,0" },
 { "AntiAliasing",	C_AntiAliasing | C_BOOL,0,	"false" },
 { "NumLock",		C_NumLock | C_ENUM, numlock,	"Keep" },
@@ -550,7 +550,7 @@
 { "MaxShowUID",		C_MaxShowUID,		0,	"65535" },
 { "SortUsers",		C_SortUsers | C_BOOL,	0,	"true" },
 { "FaceSource",		C_FaceSource | C_ENUM, facesource, "AdminOnly" },
-{ "FaceDir",		C_FaceDir,		0,	KDMDATA "/faces" },
+{ "FaceDir",		C_FaceDir,		0,	KDMCONF "/faces" },
 { "PreselectUser",	C_PreselectUser | C_ENUM, preseluser, "None" },
 { "DefaultUser",	C_DefaultUser,		0,	"" },
 { "FocusPasswd",	C_FocusPasswd | C_BOOL, 0,	"false" },
--- kdebase-3.1/kdm/kfrontend/kdmconfig.cpp.orig	Mon Oct 28 10:55:24 2002
+++ kdebase-3.1/kdm/kfrontend/kdmconfig.cpp	Fri Nov 29 18:27:33 2002
@@ -153,7 +153,7 @@
 
     _logo = GetCfgQStr (C_LogoPixmap);
     if( _logo.isEmpty())
-	_logo = locate("data", QString::fromLatin1("kdm/pics/kdelogo.png") );
+	_logo = locate("data", QString::fromLatin1("kdm/pics/pldlogo.png") );
 
     _showUsers = GetCfgInt (C_ShowUsers);
     _users = GetCfgQStrList (C_SelectedUsers);
