--- kdebase-031029/nsplugins/pluginscan.cpp.orig	2003-10-29 01:07:30.000000000 +0100
+++ kdebase-031029/nsplugins/pluginscan.cpp	2003-10-29 22:23:26.000000000 +0100
@@ -441,22 +441,7 @@
     // setup default paths
     if ( !config->hasKey("scanPaths") ) {
         QStringList paths;
-        paths.append("$HOME/.netscape/plugins");
-        paths.append("/usr/lib64/browser-plugins");
-        paths.append("/usr/lib/browser-plugins");
-        paths.append("/usr/local/netscape/plugins");
-        paths.append("/opt/mozilla/plugins");
-	paths.append("/opt/mozilla/lib/plugins");
-        paths.append("/opt/netscape/plugins");
-        paths.append("/opt/netscape/communicator/plugins");
-        paths.append("/usr/lib/netscape/plugins");
-        paths.append("/usr/lib/netscape/plugins-libc5");
-        paths.append("/usr/lib/netscape/plugins-libc6");
-        paths.append("/usr/lib/mozilla/plugins");
-	paths.append("/usr/lib/browser-plugins");
-	paths.append("/usr/lib64/netscape/plugins");
-	paths.append("/usr/lib64/mozilla/plugins");
-        paths.append("$MOZILLA_HOME/plugins");
+        paths.append("/usr/lib/kde3/plugins/konqueror");
         config->writeEntry( "scanPaths", paths );
     }
 
--- kdebase-3.2.3/kcontrol/konqhtml/pluginopts.cpp.orig	2004-05-30 15:41:15.000000000 +0200
+++ kdebase-3.2.3/kcontrol/konqhtml/pluginopts.cpp	2004-06-05 21:21:31.983293200 +0200
@@ -352,21 +352,7 @@
     if ( config->hasKey( "scanPaths" ) )
         paths = config->readListEntry( "scanPaths" );
     else {//keep sync with kdebase/nsplugins/pluginscan
-        paths.append("$HOME/.netscape/plugins");
-        paths.append("/usr/lib64/browser-plugins");
-        paths.append("/usr/lib/browser-plugins");
-        paths.append("/usr/local/netscape/plugins");
-        paths.append("/opt/mozilla/plugins");
-	paths.append("/opt/mozilla/lib/plugins");
-        paths.append("/opt/netscape/plugins");
-        paths.append("/opt/netscape/communicator/plugins");
-        paths.append("/usr/lib/netscape/plugins");
-        paths.append("/usr/lib/netscape/plugins-libc5");
-        paths.append("/usr/lib/netscape/plugins-libc6");
-        paths.append("/usr/lib/mozilla/plugins");
-        paths.append("/usr/lib64/netscape/plugins");
-        paths.append("/usr/lib64/mozilla/plugins");
-        paths.append("$MOZILLA_HOME/plugins");
+	paths.append("/usr/ilb/kde3/plugins/konqueror");
     }
 
     // fill list
