--- kdebase-R14.0.0/kcontrol/tdm/background.cpp~	2013-02-16 06:27:24.000000000 +0200
+++ kdebase-R14.0.0/kcontrol/tdm/background.cpp	2015-03-22 22:58:05.883155469 +0200
@@ -51,7 +51,7 @@
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in tdmrc (usually Xsetup).") );
     config->setGroup( "X-*-Greeter" );
-    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg",KDE_CONFDIR "/tdm/backgroundrc" ) );
+    m_simpleConf=new KSimpleConfig(config->readEntry( "BackgroundCfg","/etc/X11/kdm/backgroundrc" ) );
     m_background = new BGDialog( this, m_simpleConf, false );
 
     connect(m_background, TQT_SIGNAL(changed(bool)), TQT_SIGNAL(changed(bool)));
--- kdebase-R14.0.0/kcontrol/tdm/tdm-appear.cpp~	2015-03-22 22:59:16.000000000 +0200
+++ kdebase-R14.0.0/kcontrol/tdm/tdm-appear.cpp	2015-03-22 23:00:16.913255807 +0200
@@ -355,7 +355,7 @@
 bool TDMAppearanceWidget::setLogo(TQString logo)
 {
     TQString flogo = logo.isEmpty() ?
-                    locate("data", TQString::fromLatin1("tdm/pics/kdelogo.png") ) :
+                    locate("data", TQString::fromLatin1("tdm/pics/pldlogo.png") ) :
                     logo;
     TQImage p(flogo);
     if (p.isNull())
@@ -502,7 +502,7 @@
   config->setGroup("X-*-Greeter");
 
   // Read the greeting string
-  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to %n")));
+  greetstr_lined->setText(config->readEntry("GreetString", i18n("Welcome to PLD Linux at %n")));
 
   // Regular logo or clock
   TQString logoArea = config->readEntry("LogoArea", "Logo" );
@@ -556,7 +556,7 @@
 
 void KDMAppearanceWidget::defaults()
 {
-  greetstr_lined->setText( i18n("Welcome to %n") );
+  greetstr_lined->setText( i18n("Welcome to PLD Linux at %n") );
   logoRadio->setChecked( true );
   slotAreaRadioClicked( KdmLogo );
   setLogo( "" );
--- kdebase-R14.0.0/kcontrol/tdm/tdm-users.cpp~	2014-11-13 03:00:24.000000000 +0200
+++ kdebase-R14.0.0/kcontrol/tdm/tdm-users.cpp	2015-03-22 23:01:49.701383180 +0200
@@ -74,7 +74,7 @@
     // We assume that $kde_datadir/tdm exists, but better check for pics/ and pics/users,
     // and create them if necessary.
     config->setGroup( "X-*-Greeter" );
-    m_userPixDir = config->readEntry( "FaceDir", TDEGlobal::dirs()->resourceDirs("data").last() + "tdm/faces" ) + '/';
+    m_userPixDir = config->readEntry( "FaceDir", "/etc/X11/kdm/faces" ) + '/';
     m_notFirst = false;
     TQDir testDir( m_userPixDir );
     if ( !testDir.exists() && !testDir.mkdir( testDir.absPath() ) && !geteuid() )
--- kdebase-R14.0.0/kcontrol/tdm/main.cpp~	2014-11-24 15:00:51.000000000 +0200
+++ kdebase-R14.0.0/kcontrol/tdm/main.cpp	2015-03-22 22:50:45.317002792 +0200
@@ -174,11 +174,11 @@
 	<< "' have unknown GID " << tgmapci.key() << endl;
 
   struct stat st;
-  if( stat( KDE_CONFDIR "/tdm/tdmdistrc" ,&st ) == 0) {
-    config = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/tdm/tdmdistrc" ));
+  if( stat( "/etc/X11/kdm/kdmdistrc" ,&st ) == 0) {
+    config = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmdistrc" ));
   }
   else {
-    config = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/tdm/tdmrc" ));
+    config = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmrc" ));
   }
 
   TQVBoxLayout *top = new TQVBoxLayout(this);
--- kdebase-R14.0.0/kcontrol/screensaver/scrnsave.cpp~	2014-05-05 15:20:20.000000000 +0300
+++ kdebase-R14.0.0/kcontrol/screensaver/scrnsave.cpp	2015-03-22 22:51:52.440535817 +0200
@@ -52,11 +52,11 @@
 #include <fixx11h.h>
 
 #define OPEN_TDMCONFIG_AND_SET_GROUP									\
-if( stat( KDE_CONFDIR "/tdm/tdmdistrc" , &st ) == 0) {							\
-	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/tdm/tdmdistrc" ));		\
+if( stat( "/etc/X11/kdm/kdmdistrc" , &st ) == 0) {							\
+	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmdistrc" ));		\
 }													\
 else {													\
-	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/tdm/tdmrc" ));		\
+	mTDMConfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmrc" ));		\
 }													\
 mTDMConfig->setGroup("X-:*-Greeter");
 
--- kdebase-R14.0.0/kdesktop/lock/main.cc~	2014-12-03 09:00:49.000000000 +0200
+++ kdebase-R14.0.0/kdesktop/lock/main.cc	2015-03-22 22:52:54.543743116 +0200
@@ -55,11 +55,11 @@
 #endif
 
 #define OPEN_TDMCONFIG_AND_SET_GROUP									\
-if( stat( KDE_CONFDIR "/tdm/tdmdistrc" , &st ) == 0) {							\
-	tdmconfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/tdm/tdmdistrc" ));		\
+if( stat( "/etc/X11/kdm/kdmdistrc" , &st ) == 0) {							\
+	tdmconfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmdistrc" ));		\
 }													\
 else {													\
-	tdmconfig = new KSimpleConfig( TQString::fromLatin1( KDE_CONFDIR "/tdm/tdmrc" ));		\
+	tdmconfig = new KSimpleConfig( TQString::fromLatin1( "/etc/X11/kdm/kdmrc" ));		\
 }													\
 tdmconfig->setGroup("X-:*-Greeter");
 
--- kdebase-R14.0.0/tdm/kfrontend/tdmctl.c~	2012-03-08 06:01:31.000000000 +0200
+++ kdebase-R14.0.0/tdm/kfrontend/tdmctl.c	2015-03-22 22:54:40.705892492 +0200
@@ -146,7 +146,7 @@
 {
 	char *dpy = getenv( "DISPLAY" );
 	const char *ctl = getenv( "DM_CONTROL" );
-	const char *cfg = KDE_CONFDIR "/tdm/tdmrc";
+	const char *cfg = "/etc/X11/kdm/kdmrc";
 	char *ptr;
 	int fd;
 
--- kdebase-R14.0.0/tdm/kfrontend/CMakeLists.txt~	2013-08-06 07:00:09.000000000 +0300
+++ kdebase-R14.0.0/tdm/kfrontend/CMakeLists.txt	2015-03-22 22:57:16.357264389 +0200
@@ -36,7 +36,7 @@
   set( GENTDMCONF_FLAGS "--no-old" )
 endif( )
 
-install( CODE "execute_process( COMMAND ${CMAKE_CURRENT_BINARY_DIR}/gentdmconf --in \$ENV{DESTDIR}${CONFIG_INSTALL_DIR}/tdm --no-in-notice --face-src ${CMAKE_CURRENT_SOURCE_DIR}/pics ${GENTDMCONF_FLAGS} )" )
+install( CODE "execute_process( COMMAND ${CMAKE_CURRENT_BINARY_DIR}/gentdmconf --in \$ENV{DESTDIR}/etc/X11/kdm --no-in-notice --face-src ${CMAKE_CURRENT_SOURCE_DIR}/pics ${GENKDMCONF_FLAGS} )" )
 
 
 ##### config.ci (generated) #####################
--- kdebase-R14.0.0/tdm/config.def	2014-12-09 09:10:58.000000000 +0200
+++ kdebase-R14.0.0/tdm/config.def	2015-03-23 06:34:45.452474544 +0200
@@ -30,7 +30,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 3
 
-#define TDMCONF KDE_CONFDIR "/tdm"
+#define TDMCONF "/etc/X11/kdm"
 #define TDMDATA KDE_DATADIR "/tdm"
 
 #ifdef _AIX
@@ -1173,7 +1173,7 @@
 
 Key: Willing
 Type: string
-Default: ""
+Default: TDMCONF "/Xwilling"
 User: core
 # will be overwritten
 Instance: #
@@ -1590,7 +1590,7 @@
 
 Key: Setup
 Type: string
-Default: ""
+Default: TDMCONF "/Xsetup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1611,7 +1611,7 @@
 
 Key: Startup
 Type: string
-Default: ""
+Default: TDMCONF "/Xstartup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1629,7 +1629,7 @@
 
 Key: Reset
 Type: string
-Default: ""
+Default: TDMCONF "/Xreset"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1647,7 +1647,7 @@
 
 Key: Session
 Type: string
-Default: XBINDIR "/xterm -ls -T"
+Default: TDMCONF "/Xsession"
 #Merge: xdm - incompatible!
 User: core
 # will be overwritten
@@ -1755,10 +1755,10 @@
 
 Key: AllowRootLogin
 Type: bool
-Default: true
+Default: false
 User: core
 User: greeter(showRoot)
-Instance: */false
+Instance: #*/true
 Merge: xdm
 Comment:
  Allow root logins?
@@ -1768,11 +1768,11 @@
 
 Key: AllowNullPasswd
 Type: bool
-Default: true
+Default: false
 User: core
 # sensible?
-Instance: */false
-Instance: :*/true
+Instance: #*/true
+Instance: #:*/true
 Merge: xdm
 Comment:
  Allow to log in, when user has set an empty password?
@@ -1933,7 +1933,7 @@
 Default: "/usr/share/xsessions,/var/lib/menu-xdg/xsessions," TDMDATA "/sessions"
 User: core
 User: greeter-c
-Instance: */"/usr/share/xsessions,/var/lib/menu-xdg/xsessions," TDMDATA "/sessions"
+Instance: */"/etc/X11/sessions,/usr/share/xsessions,/var/lib/menu-xdg/xsessions," TDMDATA "/sessions"
 Comment:
  The directories containing session type definitions in .desktop format.
 Description:
@@ -1942,10 +1942,10 @@
 
 Key: ClientLogFile
 Type: string
-Default: ".xsession-errors"
+Default: ""
 User: core
-Instance: */".xsession-errors-%s"
-Instance: :0/".xsession-errors"
+Instance: #*/".xsession-errors-%s"
+Instance: #:0/".xsession-errors"
 Comment:
  The file (relative to $HOME) to redirect the session output to. This is
  a printf format string; one %s will be replaced with the display name.
@@ -2047,7 +2047,7 @@
  None/LOGO_NONE: nothing
  Logo/LOGO_LOGO: the image specified by <option>LogoPixmap</option>
  Clock/LOGO_CLOCK: a neat analog clock
-Default: None
+Default: Logo
 User: greeter
 Instance: */None
 Comment:
@@ -2059,9 +2059,9 @@
 
 Key: LogoPixmap
 Type: string
-Default: ""
+Default: TDMDATA "/pics/pldlogo.png"
 User: greeter(logo)
-Instance: */TDMDATA "/pics/kdelogo.png"
+Instance: */None
 Comment:
  The image to show when LogoArea=Logo.
 Description:
@@ -2094,9 +2094,9 @@
 
 Key: GreetString
 Type: string
-Default: "Welcome to Trinity at %n"
+Default: "Welcome to PLD Linux at %n"
 User: greeter
-Instance: #*/"Welcome to Trinity at %n"
+Instance: #*/"Welcome to PLD Linux at %n"
 Comment:
  The headline in the greeter. The following character pairs are replaced:
  - %d -> current display
@@ -2277,7 +2277,7 @@
 
 Key: MinShowUID
 Type: int
-Default: 0
+Default: 500
 User: greeter(lowUserId)
 # will be overwritten
 Instance: #*/
@@ -2338,7 +2338,7 @@
 
 Key: FaceDir
 Type: string
-Default: *TDMDATA "/faces"
+Default: *TDMCONF "/faces"
 User: greeter
 Instance: #*/"/usr/share/faces"
 Update: upd_facedir
