--- kdebase-R14.0.0/kcontrol/screensaver/scrnsave.cpp~	2015-03-23 06:44:16.000000000 +0200
+++ kdebase-R14.0.0/kcontrol/screensaver/scrnsave.cpp	2015-03-23 06:46:05.480967022 +0200
@@ -96,10 +96,9 @@
     struct stat st;
     OPEN_TDMCONFIG_AND_SET_GROUP
 
-    // Add non-TDE path
     TDEGlobal::dirs()->addResourceType("scrsav",
-                                     TDEGlobal::dirs()->kde_default("apps") +
-                                     "apps/ScreenSavers/");
+                                     TDEKGlobal::dirs()->kde_default("data") +
+                                     "kscreensaver/");
 
     setQuickHelp( i18n("<h1>Screen Saver</h1> This module allows you to enable and"
        " configure a screen saver. Note that you can enable a screen saver"
@@ -113,19 +112,6 @@
 
     setButtons( TDECModule::Help | TDECModule::Default | TDECModule::Apply );
 
-    // Add KDE specific screensaver path
-    TQString relPath="System/ScreenSavers/";
-    KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers" );
-    if (servGroup)
-    {
-      relPath=servGroup->relPath();
-      kdDebug() << "relPath=" << relPath << endl;
-    }
-
-    TDEGlobal::dirs()->addResourceType("scrsav",
-                                     TDEGlobal::dirs()->kde_default("apps") +
-                                     relPath);
-
     readSettings( false );
 
     mSetupProc = new TDEProcess;
--- kdebase-trinity-3.5.13.2/tdescreensaver/CMakeLists.txt~	2012-09-16 15:52:04.000000000 +0300
+++ kdebase-trinity-3.5.13.2/tdescreensaver/CMakeLists.txt	2014-06-04 11:07:10.329823207 +0300
@@ -23,7 +23,7 @@
 
 ##### other data ################################
 
-install( FILES KRandom.desktop KBlankscreen.desktop DESTINATION ${APPS_INSTALL_DIR}/System/ScreenSavers )
+install( FILES KRandom.desktop KBlankscreen.desktop DESTINATION ${DATA_INSTALL_DIR}/kscreensaver)
 
 
 ##### krandom (executable) ######################
--- kdebase-R14.0.0/kdesktop/lock/lockprocess.cc~	2014-11-29 01:19:50.000000000 +0200
+++ kdebase-R14.0.0/kdesktop/lock/lockprocess.cc	2015-03-23 06:47:24.508389362 +0200
@@ -261,22 +261,9 @@
     }
     XSelectInput( tqt_xdisplay(), tqt_xrootwin(), SubstructureNotifyMask | rootAttr.your_event_mask );
 
-    // Add non-TDE path
     TDEGlobal::dirs()->addResourceType("scrsav",
-                                    TDEGlobal::dirs()->kde_default("apps") +
-                                    "System/ScreenSavers/");
-
-    // Add KDE specific screensaver path
-    TQString relPath="System/ScreenSavers/";
-    KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers");
-    if (servGroup)
-    {
-      relPath=servGroup->relPath();
-      kdDebug(1204) << "relPath=" << relPath << endl;
-    }
-    TDEGlobal::dirs()->addResourceType("scrsav",
-                                     TDEGlobal::dirs()->kde_default("apps") +
-                                     relPath);
+                                     TDEGlobal::dirs()->kde_default("data") +
+                                     "kscreensaver/");
 
     // virtual root property
     gXA_VROOT = XInternAtom (tqt_xdisplay(), "__SWM_VROOT", False);
