diff -ur konsole/konsole/TEmulation.cpp konsole/konsole/TEmulation.cpp
--- konsole/konsole/TEmulation.cpp	2002-03-08 02:16:54.000000000 +0100
+++ konsole.bak/konsole/TEmulation.cpp	2002-07-12 16:39:28.000000000 +0200
@@ -424,6 +424,7 @@
     //kdDebug(1211)<<"TEmulation::setConnect() showBulk()"<<endl;
     showBulk();
     //kdDebug(1211)<<"TEmulation::setConnect() done"<<endl;
+    gui->update();
   }
   else
   {
@@ -462,6 +463,8 @@
   if (!connected) return;
   scr->setHistCursor(cursor);
   showBulk();
+  gui->update();
+
 }
 
 void TEmulation::setColumns(int columns)
Only in konsole.bak/konsole: TEmulation.cpp.~1.37.~
diff -ur konsole/konsole/TEWidget.cpp konsole/konsole/TEWidget.cpp
--- konsole/konsole/TEWidget.cpp	2002-07-15 11:51:37.000000000 +0200
+++ konsole.bak/konsole/TEWidget.cpp	2002-07-12 17:39:01.000000000 +0200
@@ -41,7 +41,6 @@
    - evtl. be sensitive to `paletteChange' while using default colors.
    - set different 'rounding' styles? I.e. have a mode to show clipped chars?
 */
-
 #include "config.h"
 #include "TEWidget.h"
 #include "session.h"
@@ -70,6 +69,7 @@
 #include <knotifyclient.h>
 #include <kglobalsettings.h>
 #include <kurldrag.h>
+#include <iostream>
 
 #ifndef HERE
 #define HERE printf("%s(%d): %s\n",__FILE__,__LINE__,__FUNCTION__)
@@ -440,11 +442,17 @@
 void TEWidget::setImage(const ca* const newimg, int lines, int columns)
 { int y,x,len;
   const QPixmap* pm = backgroundPixmap();
+  
   QPainter paint;
   setUpdatesEnabled(FALSE);
   paint.begin( this );
 HCNT("setImage");
 
+  /*std::cerr << "---> setImage " 
+    << std::endl;*/
+
+  paint.setClipRegion( contentsRect() );
+
   QPoint tL  = contentsRect().topLeft();
   int    tLx = tL.x();
   int    tLy = tL.y();
@@ -504,6 +512,7 @@
   drawFrame( &paint );
   paint.end();
   setUpdatesEnabled(TRUE);
+  //repaint(visibleRect(), TRUE);
   if ( hasBlinker && !blinkT->isActive()) blinkT->start(1000); // 1000 ms
   if (!hasBlinker && blinkT->isActive()) { blinkT->stop(); blinking = FALSE; }
   delete [] disstrU;
@@ -565,6 +574,7 @@
 void TEWidget::paintEvent( QPaintEvent* pe )
 {
 
+
 //{ static int cnt = 0; printf("paint %d\n",cnt++); }
   const QPixmap* pm = backgroundPixmap();
   QPainter paint;
@@ -579,8 +589,21 @@
   // can thus be larger than the image, but less then the size
   // of one character.
 
+  /*std::cerr << "---> paintEvent " 
+	    << "(" << pe->rect().left() << "," << pe->rect().top() << ") " 
+	    << "(" << pe->rect().right() << "," << pe->rect().bottom() << ") " 
+	    << std::endl;*/
+
   QRect rect = pe->rect().intersect(contentsRect());
 
+  /*std::cerr << "---> intersect " 
+	    << "(" << rect.left() << "," << rect.top() << ") " 
+	    << "(" << rect.right() << "," << rect.bottom() << ") " 
+	    << std::endl;*/
+
+  paint.save();
+  paint.setClipRegion( rect );
+
   QPoint tL  = contentsRect().topLeft();
   int    tLx = tL.x();
   int    tLy = tL.y();
@@ -632,7 +655,12 @@
     }
   }
   delete [] disstrU;
+  paint.restore();
+
+  paint.save();
+  paint.setClipRegion( pe->region().intersect(frameRect()) );
   drawFrame( &paint );
+  paint.restore();
   paint.end();
   setUpdatesEnabled(TRUE);
 }
